
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Project documentation with Markdown.">
      
      
      
      
        <link rel="prev" href="../aws_env/">
      
      
        <link rel="next" href="../real_life_example_cli/">
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.1.6">
    
    
      
        <title>R for Bioinformatics - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.ded33207.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.a0c5b2b5.min.css">
      
      
  
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
  
  <style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M64 32C28.7 32 0 60.7 0 96v320c0 35.3 28.7 64 64 64h224V368c0-26.5 21.5-48 48-48h112V96c0-35.3-28.7-64-64-64H64zm384 320H336c-8.8 0-16 7.2-16 16v112l32-32 64-64 32-32z"/></svg>');--md-admonition-icon--abstract:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M96 0C43 0 0 43 0 96v320c0 53 43 96 96 96h320c17.7 0 32-14.3 32-32s-14.3-32-32-32v-64c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H96zm0 384h256v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"/></svg>');--md-admonition-icon--info:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-40-176h24v-64h-24c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24h-80c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"/></svg>');--md-admonition-icon--tip:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M480 32c0-12.9-7.8-24.6-19.8-29.6S434.5.2 425.3 9.3L381.7 53c-48 48-113.1 75-181 75H64c-35.3 0-64 28.7-64 64v96c0 35.3 28.7 64 64 64v128c0 17.7 14.3 32 32 32h64c17.7 0 32-14.3 32-32V352h8.7c67.9 0 133 27 181 75l43.6 43.6c9.2 9.2 22.9 11.9 34.9 6.9s19.8-16.6 19.8-29.6V300.4c18.6-8.8 32-32.5 32-60.4s-13.4-51.6-32-60.4V32zm-64 76.7v262.6C357.2 317.8 280.5 288 200.7 288H192v-96h8.7c79.8 0 156.5-29.8 215.3-83.3z"/></svg>');--md-admonition-icon--success:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7l233.4-233.3c12.5-12.5 32.8-12.5 45.3 0z"/></svg>');--md-admonition-icon--question:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-86.2-346.7c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1 0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6-13.3 0-24-10.7-24-24v-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1 0-8.4-6.8-15.1-15.1-15.1h-58.3c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1-64 0z"/></svg>');--md-admonition-icon--warning:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7.2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8.2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24v112c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0-64 0 32 32 0 1 0 64 0z"/></svg>');--md-admonition-icon--failure:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.1 52.4 442.6 6.5c-1.9-3.9-6.1-6.5-10.5-6.5s-8.5 2.6-10.4 6.5l-16.5 45.9-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4 0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2 0-4.6-3-8.9-7.2-10.4l-45.6-16.8zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9c-22-8-45.8-12.3-70.5-12.3C93.1 96 0 189.1 0 304s93.1 208 208 208 208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg>');--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M416 398.9c58.5-41.1 96-104.1 96-174.9C512 100.3 397.4 0 256 0S0 100.3 0 224c0 70.7 37.5 133.8 96 174.9V464c0 26.5 21.5 48 48 48h48v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h64v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h48c26.5 0 48-21.5 48-48v-65.1zM96 256a64 64 0 1 1 128 0 64 64 0 1 1-128 0zm256-64a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"/></svg>');--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M320 0c17.7 0 32 14.3 32 32v64h120c39.8 0 72 32.2 72 72v272c0 39.8-32.2 72-72 72H168c-39.8 0-72-32.2-72-72V168c0-39.8 32.2-72 72-72h120V32c0-17.7 14.3-32 32-32zM208 384c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zM264 256a40 40 0 1 0-80 0 40 40 0 1 0 80 0zm152 40a40 40 0 1 0 0-80 40 40 0 1 0 0 80zM48 224h16v192H48c-26.5 0-48-21.5-48-48v-96c0-26.5 21.5-48 48-48zm544 0c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48h-16V224h16z"/></svg>');--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M288 0H128c-17.7 0-32 14.3-32 32s14.3 32 32 32v132.8c0 11.8-3.3 23.5-9.5 33.5L10.3 406.2C3.6 417.2 0 429.7 0 442.6 0 480.9 31.1 512 69.4 512h309.2c38.3 0 69.4-31.1 69.4-69.4 0-12.8-3.6-25.4-10.3-36.4L329.5 230.4c-6.2-10.1-9.5-21.7-9.5-33.5V64c17.7 0 32-14.3 32-32S337.7 0 320 0h-32zm-96 196.8V64h64v132.8c0 23.7 6.6 46.9 19 67.1l34.5 56.1h-171l34.5-56.1c12.4-20.2 19-43.4 19-67.1z"/></svg>');--md-admonition-icon--quote:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 216C0 149.7 53.7 96 120 96h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V216zm256 0c0-66.3 53.7-120 120-120h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64h-64c-35.3 0-64-28.7-64-64V216z"/></svg>');}</style>


    
    
    
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#programming-in-r-for-bioinformatics" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              R for Bioinformatics
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      Course Details
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../schedule/" class="md-tabs__link">
      Schedule
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../aws/aws_ec2_instances/" class="md-tabs__link">
      AWS Instances Login
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../lectures/lectures/" class="md-tabs__link">
      Lectures
    </a>
  </li>

      
        
  
  
    
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../basic_unix/" class="md-tabs__link md-tabs__link--active">
        Lab Session
      </a>
    </li>
  

  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_1" checked>
      
      
      
        <label class="md-nav__link" for="__nav_5_1" id="__nav_5_1_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_1_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          R for Bioinformatics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        R for Bioinformatics
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#what-is-r" class="md-nav__link">
    What is R?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#some-of-rs-strengths" class="md-nav__link">
    Some of R’s strengths:
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-r-environment" class="md-nav__link">
    The R environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-r-homepage" class="md-nav__link">
    The R Homepage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rstudio-has-many-features" class="md-nav__link">
    RStudio has many features:
  </a>
  
    <nav class="md-nav" aria-label="RStudio has many features:">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-getting-started" class="md-nav__link">
    1. Getting started
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-open-a-new-rscript-file" class="md-nav__link">
    2. Open a new RScript File
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-basics-of-your-environment" class="md-nav__link">
    3. Basics of your environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-writing-and-running-r-commands" class="md-nav__link">
    4. Writing and running R commands
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-the-assignment-operator-vs-equals" class="md-nav__link">
    5. The assignment operator ( &lt;- ) vs equals ( = )
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-the-rstudio-cheat-sheets" class="md-nav__link">
    6. The RStudio Cheat Sheets
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ggplot2/" class="md-nav__link">
        ggplot tutorial
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_4" >
      
      
      
        <label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../structural_variation/" class="md-nav__link">
        Structural Variants
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../analysis_of_cnv/" class="md-nav__link">
        Copy Number Analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
          Day 5
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          Day 5
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../joint_curation/" class="md-nav__link">
        Joint Curation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../genomic_alteration_interpretation_exe/" class="md-nav__link">
        Genomic Alterations Interpretation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#what-is-r" class="md-nav__link">
    What is R?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#some-of-rs-strengths" class="md-nav__link">
    Some of R’s strengths:
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-r-environment" class="md-nav__link">
    The R environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-r-homepage" class="md-nav__link">
    The R Homepage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rstudio-has-many-features" class="md-nav__link">
    RStudio has many features:
  </a>
  
    <nav class="md-nav" aria-label="RStudio has many features:">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-getting-started" class="md-nav__link">
    1. Getting started
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-open-a-new-rscript-file" class="md-nav__link">
    2. Open a new RScript File
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-basics-of-your-environment" class="md-nav__link">
    3. Basics of your environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-writing-and-running-r-commands" class="md-nav__link">
    4. Writing and running R commands
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-the-assignment-operator-vs-equals" class="md-nav__link">
    5. The assignment operator ( &lt;- ) vs equals ( = )
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-the-rstudio-cheat-sheets" class="md-nav__link">
    6. The RStudio Cheat Sheets
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="programming-in-r-for-bioinformatics">Programming in R for Bioinformatics<a class="headerlink" href="#programming-in-r-for-bioinformatics" title="Permanent link">&para;</a></h1>
<blockquote>
<p>Tip: In order to take most out of this tutorial you should not miss reading any lines in this tutorial and follow the flow and write codes on your computer.</p>
</blockquote>
<h3 id="what-is-r">What is R?<a class="headerlink" href="#what-is-r" title="Permanent link">&para;</a></h3>
<p>R is a language and environment for statistical computing and graphics developed in 1993. It provides a wide variety of statistical and graphical techniques (linear and nonlinear modeling, statistical tests, time series analysis, classification, clustering, …), and is highly extensible, meaning that the user community can write new R tools. It is a GNU project (Free and Open Source).</p>
<p>The R language has its roots in the S language and environment which was developed at Bell Laboratories (formerly AT&amp;T, now Lucent Technologies) by John Chambers and colleagues. R was created by Ross Ihaka and Robert Gentleman at the University of Auckland, New Zealand, and now, R is developed by the R Development Core Team, of which Chambers is a member. R is named partly after the first names of the first two R authors (Robert Gentleman and Ross Ihaka), and partly as a play on the name of S. R can be considered as a different implementation of S. There are some important differences, but much code written for S runs unaltered under R.</p>
<h3 id="some-of-rs-strengths">Some of R’s strengths:<a class="headerlink" href="#some-of-rs-strengths" title="Permanent link">&para;</a></h3>
<ol>
<li>The ease with which well-designed publication-quality plots can be produced, including mathematical symbols and formulae where needed. Great care has been taken over the defaults for the minor design choices in graphics, but the user retains full control.</li>
<li>It compiles and runs on a wide variety of UNIX platforms and similar systems (including FreeBSD and Linux), Windows and MacOS.</li>
<li>R can be extended (easily) via packages.</li>
<li>R has its own LaTeX-like documentation format, which is used to supply comprehensive documentation, both on-line in a number of formats and in hardcopy.</li>
<li>It has a vast community both in academia and in business.</li>
<li>It’s FREE!</li>
</ol>
<h3 id="the-r-environment">The R environment<a class="headerlink" href="#the-r-environment" title="Permanent link">&para;</a></h3>
<p>R is an integrated suite of software facilities for data manipulation, calculation and graphical display. It includes</p>
<ul>
<li>an effective data handling and storage facility,</li>
<li>a suite of operators for calculations on arrays, in particular matrices,</li>
<li>a large, coherent, integrated collection of intermediate tools for data analysis,</li>
<li>graphical facilities for data analysis and display either on-screen or on hardcopy, and</li>
<li>a well-developed, and effective programming language which includes conditionals, loops, user-defined recursive functions and input and output facilities.</li>
<li>The term “environment” is intended to characterize it as a fully planned and coherent system, rather than an incremental accretion of very specific and inflexible tools, as is frequently the case with other data analysis software.</li>
</ul>
<p>R, like S, is designed around a true computer language, and it allows users to add additional functionality by defining new functions. Much of the system is itself written in the R dialect of S, which makes it easy for users to follow the algorithmic choices made. For computationally-intensive tasks, C, C++ and Fortran code can be linked and called at run time. Advanced users can write C code to manipulate R objects directly.</p>
<p>Many users think of R as a statistics system. The R group prefers to think of it of an environment within which statistical techniques are implemented.</p>
<h3 id="the-r-homepage">The R Homepage<a class="headerlink" href="#the-r-homepage" title="Permanent link">&para;</a></h3>
<p>The R homepage has a wealth of information on it,</p>
<p><a href="https://www.r-project.org/" target="_blank">R-project.org</a></p>
<p>On the homepage you can learn more about R</p>
<ul>
<li>Download R</li>
<li>Get Documentation (official and user supplied)</li>
<li>Get access to CRAN ‘Comprehensive R archival network’</li>
<li>RStudio</li>
<li>RStudio started in 2010, to offer R a more full featured integrated development environment (IDE) and modeled after matlabs IDE.</li>
</ul>
<h3 id="rstudio-has-many-features">RStudio has many features:<a class="headerlink" href="#rstudio-has-many-features" title="Permanent link">&para;</a></h3>
<ul>
<li>syntax highlighting</li>
<li>code completion</li>
<li>smart indentation</li>
<li>“Projects”</li>
<li>workspace browser and data viewer</li>
<li>embedded plots</li>
<li>Markdown notebooks, Sweave authoring and knitr with one click pdf or html</li>
<li>runs on all platforms and over the web</li>
<li>etc. etc. etc.</li>
</ul>
<p>RStudio and its team have contributed to many R packages. These include:</p>
<p><strong>Tidyverse</strong> – R packages for data science, including ggplot2, dplyr, tidyr, and purrr
<strong>Shiny</strong> – An interactive web technology
<strong>RMarkdown</strong> – Insert R code into markdown documents
<strong>knitr</strong> – Dynamic reports combining R, TeX, Markdown &amp; HTML
<strong>packrat</strong> – Package dependency tool
<strong>devtools</strong> – Package development tool</p>
<h4 id="1-getting-started">1. Getting started<a class="headerlink" href="#1-getting-started" title="Permanent link">&para;</a></h4>
<p>Let’s start RStudio</p>
<p><img alt="" src="https://i.imgur.com/Rg8hId9.jpg" target="_blank" /></p>
<h4 id="2-open-a-new-rscript-file">2. Open a new RScript File<a class="headerlink" href="#2-open-a-new-rscript-file" title="Permanent link">&para;</a></h4>
<p>File -&gt; New File -&gt; R Script</p>
<p>RStudio_newfile</p>
<p>Then save the new empty file as Intro2R.R</p>
<p>File -&gt; Save as -&gt; Intro2R.R</p>
<h4 id="3-basics-of-your-environment">3. Basics of your environment<a class="headerlink" href="#3-basics-of-your-environment" title="Permanent link">&para;</a></h4>
<p>The R prompt is the ‘&gt;’ , when R is expecting more (command is not complete) you see a ‘+’</p>
<p><img alt="" src="https://i.imgur.com/hNyKj3h.png" /></p>
<h4 id="4-writing-and-running-r-commands">4. Writing and running R commands<a class="headerlink" href="#4-writing-and-running-r-commands" title="Permanent link">&para;</a></h4>
<p>In the source editor (top left by default) type</p>
<p><code>getwd()</code>
Then on the line Control + Enter (Linux/Windows), Command + Enter (Mac) to execute the line.</p>
<h4 id="5-the-assignment-operator-vs-equals">5. The assignment operator ( &lt;- ) vs equals ( = )<a class="headerlink" href="#5-the-assignment-operator-vs-equals" title="Permanent link">&para;</a></h4>
<p>The assignment operator is used assign data to a variable</p>
<p><div class="highlight"><pre><span></span><code>x &lt;- 1:10
x
[1]  1  2  3  4  5  6  7  8  9 10
</code></pre></div>
In this case, the equal sign works as well</p>
<p><div class="highlight"><pre><span></span><code>x = 1:10
x
[1]  1  2  3  4  5  6  7  8  9 10
</code></pre></div>
But you should NEVER EVER DO THIS</p>
<p><div class="highlight"><pre><span></span><code>1:10 -&gt; x
x
[1]  1  2  3  4  5  6  7  8  9 10
</code></pre></div>
The two act the same in most cases. The difference in assignment operators is clearer when you use them to set an argument value in a function call. For example:</p>
<p><div class="highlight"><pre><span></span><code>rm(x)  # first remove the previously defined x variable
median(x = 1:10)
x 
Error: object &#39;x&#39; not found
</code></pre></div>
In this case, x is declared within the scope of the function, so it does not exist in the user workspace.</p>
<p><div class="highlight"><pre><span></span><code>median(x &lt;- 1:10)
x
[1]  1  2  3  4  5  6  7  8  9 10
</code></pre></div>
In this case, x is declared in the user workspace, so you can use it after the function call has been completed. There is a general preference among the R community for using &lt;- for assignment (other than in function signatures)</p>
<h4 id="6-the-rstudio-cheat-sheets">6. The RStudio Cheat Sheets<a class="headerlink" href="#6-the-rstudio-cheat-sheets" title="Permanent link">&para;</a></h4>
<p><a href="https://course-cg-5534.s3.amazonaws.com/pdf/rstudio-ide.pdf" target="_blank">rstudio-ide.pdf</a></p>
<p>spend 15m getting to know RStudio a little</p>
<h1 id="import-and-export-data-in-r">Import and export data in R<a class="headerlink" href="#import-and-export-data-in-r" title="Permanent link">&para;</a></h1>
<p>R base function read.table() is a general funciton that can be used to read a file in table format. The data will be imported as a data frame.
To download the data, run the following in your terminal/commnad line if you have a Mac or Linux machine. If you have Windows, you can skip download and instead use the option below to read the file in R directly from the Internet.</p>
<p><div class="highlight"><pre><span></span><code><span class="c1"># Download the raw_counts.txt file into your local machine using wget</span>
<span class="nb">cd</span><span class="w"> </span>~/my_wokingdir<span class="w">  </span><span class="c1"># replace &quot;~/my_workingdir&quot; with the output of &quot;getwd()&quot; in your R session</span>
wget<span class="w"> </span>https://course-cg-5534.s3.amazonaws.com/awk_exercise/raw_counts.txt
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># To read a local file. If you have downloaded the raw_counts.txt file to your local machine, you may use the following command to read it in, by providing the full path for the file location. The way to specify the full path is the same as taught in the command line session. Here we assume raw_counts.txt is in our current working directory</span>
<span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;./raw_counts.txt&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span><span class="w"> </span><span class="n">stringsAsFactors</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>

<span class="c1"># There is a very convenient way to read files from the internet.</span>
<span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;https://course-cg-5534.s3.amazonaws.com/awk_exercise/raw_counts.txt&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span><span class="w"> </span><span class="n">stringsAsFactors</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</code></pre></div></p>
<p>Take a look at the beginning part of the data frame.</p>
<div class="highlight"><pre><span></span><code><span class="nf">head</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##            C61  C62  C63  C64  C91  C92  C93 C94 I561 I562 I563 I564 I591 I592
## AT1G01010  322  346  256  396  372  506  361 342  638  488  440  479  770  430
## AT1G01020  149   87  162  144  189  169  147 108  163  141  119  147  182  156
## AT1G01030   15   32   35   22   24   33   21  35   18    8   54   35   23    8
## AT1G01040  687  469  568  651  885  978  794 862  799  769  725  715  811  567
## AT1G01046    1    1    5    4    5    3    0   2    4    3    1    0    2    8
## AT1G01050 1447 1032 1083 1204 1413 1484 1138 938 1247 1516  984 1044 1374 1355
##           I593 I594 I861 I862 I863 I864 I891 I892 I893 I894
## AT1G01010  656  467  143  453  429  206  567  458  520  474
## AT1G01020  153  177   43  144  114   50  161  195  157  144
## AT1G01030   16   24   42   17   22   39   26   28   39   30
## AT1G01040  831  694  345  575  605  404  735  651  725  591
## AT1G01046    8    1    0    4    0    3    5    7    0    5
## AT1G01050 1437 1577  412 1338 1051  621 1434 1552 1248 1186
</code></pre></div>
<p>Depending on the format of the file, several variants of read.table() are available to make reading a file easier.</p>
<p><strong>read.csv()</strong>: for reading "comma separated value" files (.csv).</p>
<p><strong>read.csv2()</strong>: variant used in countries that use a comma "," as decimal point and a semicolon ";" as field separators.</p>
<p><strong>read.delim()</strong>: for reading "tab separated value" files (".txt"). By default, point(".") is used as decimal point.</p>
<p><strong>read.delim2()</strong>: for reading "tab separated value" files (".txt"). By default, comma (",") is used as decimal point.</p>
<p>Choosing the correct function (or parameters) is important! If we use <code>read.csv()</code> to read our tab-delimited file, it becomes a mess.</p>
<div class="highlight"><pre><span></span><code><span class="n">data2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;./raw_counts.txt&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">stringsAsFactors</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>

<span class="nf">head</span><span class="p">(</span><span class="n">data2</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##                                         C61.C62.C63.C64.C91.C92.C93.C94.I561.I562.I563.I564.I591.I592.I593.I594.I861.I862.I863.I864.I891.I892.I893.I894
## 1                     AT1G01010\t322\t346\t256\t396\t372\t506\t361\t342\t638\t488\t440\t479\t770\t430\t656\t467\t143\t453\t429\t206\t567\t458\t520\t474
## 2                        AT1G01020\t149\t87\t162\t144\t189\t169\t147\t108\t163\t141\t119\t147\t182\t156\t153\t177\t43\t144\t114\t50\t161\t195\t157\t144
## 3                                               AT1G01030\t15\t32\t35\t22\t24\t33\t21\t35\t18\t8\t54\t35\t23\t8\t16\t24\t42\t17\t22\t39\t26\t28\t39\t30
## 4                     AT1G01040\t687\t469\t568\t651\t885\t978\t794\t862\t799\t769\t725\t715\t811\t567\t831\t694\t345\t575\t605\t404\t735\t651\t725\t591
## 5                                                                     AT1G01046\t1\t1\t5\t4\t5\t3\t0\t2\t4\t3\t1\t0\t2\t8\t8\t1\t0\t4\t0\t3\t5\t7\t0\t5
## 6 AT1G01050\t1447\t1032\t1083\t1204\t1413\t1484\t1138\t938\t1247\t1516\t984\t1044\t1374\t1355\t1437\t1577\t412\t1338\t1051\t621\t1434\t1552\t1248\t1186
</code></pre></div>
<p>However, the <code>read.csv()</code> function is appropriate for a comma-delimited file.</p>
<p>Since the data contained in these files is the same, we don't need to keep data2 copy.</p>
<div class="highlight"><pre><span></span><code><span class="nf">rm</span><span class="p">(</span><span class="n">data2</span><span class="p">)</span>
</code></pre></div>
<p>R base function write.table() can be used to export data to a file.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># To write to a file called &quot;output.txt&quot; in your current working directory.</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">data</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">,],</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s">&quot;output.txt&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>
</code></pre></div>
<p>It is also possible to export data to a csv file.</p>
<p>write.csv()</p>
<p>write.csv2()</p>
<hr />
<h2 id="basic-statistics-in-r">Basic statistics in R<a class="headerlink" href="#basic-statistics-in-r" title="Permanent link">&para;</a></h2>
<div class="row">
    <div class="col-12 typeset__table">
      <table>
      <thead>
        <tr>
        <th style="text-align:center;"> Description </th>
        <th style="text-align:center;"> R_function </th>
        </tr>
      </thead>
      <tbody>
        <tr>
        <td class="td-align-left"> Mean </td>
        <td style="text-align:center;"> mean() </td>
        </tr>
        <tr>
        <td class="td-align-left"> Standard deviation </td>
        <td style="text-align:center;"> sd() </td>
        </tr>
        <tr>
        <td class="td-align-left"> Variance </td>
        <td style="text-align:center;"> var() </td>
        </tr>
        <tr>
        <td class="td-align-left"> Minimum </td>
        <td style="text-align:center;"> min() </td>
        </tr>
        <tr>
        <td class="td-align-left"> Maximum </td>
        <td style="text-align:center;"> max() </td>
        </tr>
        <tr>
        <td class="td-align-left"> Median </td>
        <td style="text-align:center;"> median() </td>
        </tr>
        <tr>
        <td class="td-align-left"> Range of values: <br/> minimum and maximum </td>
        <td style="text-align:center;"> range() </td>
        </tr>
        <tr>
        <td class="td-align-left"> Sample quantiles </td>
        <td style="text-align:center;"> quantile() </td>
        </tr>
        <tr>
        <td class="td-align-left"> Generic function </td>
        <td style="text-align:center;"> summary() </td>
        </tr>
        <tr>
        <td class="td-align-left"> Interquartile range </td>
        <td style="text-align:center;"> IQR() </td>
        </tr>
      </tbody>
      </table>
  </div>
</div>

<p>Calculate the mean expression for each sample.</p>
<div class="highlight"><pre><span></span><code><span class="nf">apply</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">mean</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##      C61      C62      C63      C64      C91      C92      C93      C94     I561     I562     I563     I564     I591     I592     I593     I594     I861     I862     I863     I864     I891     I892     I893     I894 
## 391.9998 336.4872 333.7007 380.6545 364.6587 407.0191 361.3672 314.1931 398.8421 380.4970 382.0019 378.7685 387.7994 349.4061 400.9421 385.1493 219.8517 379.0522 341.6387 271.0391 395.3089 426.0254 350.8965 358.8508 
</code></pre></div>
<p>Calculate the range of expression for each sample.</p>
<div class="highlight"><pre><span></span><code><span class="nf">apply</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">range</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##        C61   C62   C63   C64   C91   C92   C93   C94   I561  I562  I563  I564   I591  I592   I593  I594  I861   I862  I863  I864   I891  I892   I893  I894
## [1,]     0     0     0     0     0     0     0     0      0     0     0     0      0     0      0     0     0      0     0     0      0     0      0     0
## [2,] 81764 89072 43781 64539 51516 68279 64407 53799 116414 90133 69623 76426 111873 73071 114566 89630 69853 122114 98449 51835 102672 80998 116025 89270
</code></pre></div>
<p>Calculate the quantiles of each samples.</p>
<div class="highlight"><pre><span></span><code><span class="nf">apply</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">quantile</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##        C61   C62   C63   C64   C91   C92   C93   C94   I561  I562  I563  I564   I591  I592   I593  I594  I861   I862  I863  I864   I891  I892      I893  I894
## 0%       0     0     0     0     0     0     0     0      0     0     0     0      0     0      0     0     0      0     0     0      0     0      0.00     0
## 25%      0     0     0     0     0     0     0     0      0     0     0     0      0     0      0     0     0      0     0     0      0     0      0.00     0
## 50%     43    38    45    47    48    45    47    39     41    45    47    45     48    41     45    43    21     49    33    31     46    49     44.00    41
## 75%    330   270   294   331   326   344   311   266    327   333   314   316    330   298    338   333   149    327   274   211    333   354    300.75   304
## 100% 81764 89072 43781 64539 51516 68279 64407 53799 116414 90133 69623 76426 111873 73071 114566 89630 69853 122114 98449 51835 102672 80998 116025.00 89270
</code></pre></div>
<hr />
<h1 id="simple-data-visualization">Simple data visualization<a class="headerlink" href="#simple-data-visualization" title="Permanent link">&para;</a></h1>
<p>Scatter plot and line plot can be produced using the function plot().</p>
<div class="highlight"><pre><span></span><code><span class="n">x</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">50</span><span class="p">)</span>
<span class="n">y</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">sqrt</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="o">/</span><span class="m">2</span>
<span class="nf">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot1" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_01.png" /></p>
<div class="highlight"><pre><span></span><code><span class="nf">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="o">=</span><span class="s">&quot;l&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot2" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_02.png" /></p>
<div class="highlight"><pre><span></span><code><span class="c1"># plot both the points and lines</span>
<span class="c1">## first plot points</span>
<span class="nf">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">)</span>
<span class="nf">lines</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="o">=</span><span class="s">&quot;l&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot3" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_03.png" /></p>
<div class="highlight"><pre><span></span><code><span class="c1">## lines() can only be used to add information to a graph, while it cannot produce a graph on its own.</span>
</code></pre></div>
<p>boxplot() can be used to summarize data.</p>
<div class="highlight"><pre><span></span><code><span class="nf">boxplot</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">xlab</span><span class="o">=</span><span class="s">&quot;Sample ID&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="o">=</span><span class="s">&quot;Raw Counts&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot4" src="https://course-cg-5534.s3.amazonaws.com/Course_img/Rplot_04.png" /></p>
<p>add more details to the plot.</p>
<div class="highlight"><pre><span></span><code><span class="nf">boxplot</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">xlab</span><span class="o">=</span><span class="s">&quot;Sample ID&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="o">=</span><span class="s">&quot;Raw Counts&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">main</span><span class="o">=</span><span class="s">&quot;Expression levels&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">border</span><span class="o">=</span><span class="s">&quot;black&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot5" src="https://course-cg-5534.s3.amazonaws.com/Course_img/Rplot_05.png" /></p>
<div class="highlight"><pre><span></span><code><span class="n">x</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rnorm</span><span class="p">(</span><span class="m">1000</span><span class="p">)</span>
<span class="nf">boxplot</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot6" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_06.png" /></p>
<p>hist() can be used to create histograms of data.</p>
<div class="highlight"><pre><span></span><code><span class="nf">hist</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot7" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_07.png" /></p>
<div class="highlight"><pre><span></span><code><span class="c1"># use user defined break points</span>
<span class="nf">hist</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">breaks</span><span class="o">=</span><span class="nf">seq</span><span class="p">(</span><span class="nf">range</span><span class="p">(</span><span class="n">x</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span><span class="m">-1</span><span class="p">,</span><span class="w"> </span><span class="nf">range</span><span class="p">(</span><span class="n">x</span><span class="p">)[</span><span class="m">2</span><span class="p">]</span><span class="m">+1</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="m">0.5</span><span class="p">))</span>
</code></pre></div>
<p><img alt="Plot8" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_08.png" /></p>
<div class="highlight"><pre><span></span><code><span class="c1"># clear plotting device/area</span>
<span class="nf">dev.off</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## null device 
##           1
</code></pre></div>
<hr />
<h2 id="r-packages-for-rnaseq-analysis">R - packages for RNAseq Analysis<a class="headerlink" href="#r-packages-for-rnaseq-analysis" title="Permanent link">&para;</a></h2>
<p>The following packages need to be installed in your local machine using Rstudio to process RNAseq data in upcoming lab sessions.</p>
<p>R-packages list:</p>
<ul>
<li><code>dplyr</code></li>
<li><code>cowplot</code></li>
<li><code>reshape2</code></li>
<li><code>ggplot2</code></li>
<li><code>jsonlite</code></li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1"># create a list of packages need to be installed</span>
<span class="n">packages</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;dplyr&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;cowplot&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;reshape2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;ggplot2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;jsonlite&quot;</span><span class="p">)</span>

<span class="nf">install.packages</span><span class="p">(</span><span class="n">packages</span><span class="p">,</span><span class="w"> </span><span class="n">dependencies</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
</code></pre></div>
<p>R - Bioconductor Packages:</p>
<ul>
<li><code>BiocManager</code></li>
<li><code>biomaRt</code></li>
<li><code>Biostrings</code></li>
<li><code>ensembldb</code></li>
<li><code>EnsDb.Hsapiens.v86</code></li>
<li><code>chimeraviz</code> </li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1"># install BiocManager </span>
<span class="nf">if </span><span class="p">(</span><span class="o">!</span><span class="nf">requireNamespace</span><span class="p">(</span><span class="s">&quot;BiocManager&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
<span class="w">    </span><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;BiocManager&quot;</span><span class="p">)</span>

<span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;biomaRt&quot;</span><span class="p">)</span>

<span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;Biostrings&quot;</span><span class="p">)</span>

<span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;ensembldb&quot;</span><span class="p">)</span>

<span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;EnsDb.Hsapiens.v86&quot;</span><span class="p">)</span>

<span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;chimeraviz&quot;</span><span class="p">)</span>
</code></pre></div>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2023, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "content.code.annotate", "content.tabs.link", "search.highlight", "search.share", "search.suggest"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.51198bba.min.js"></script>
      
    
  </body>
</html>