
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.2.3, mkdocs-material-8.1.4">
    
    
      
        <title>R for Bioinformatics - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bb3983ee.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.e6a45f82.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#programming-in-r-for-bioinformatics" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              R for Bioinformatics
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          Pre-course setup
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Pre-course setup" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Pre-course setup
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../installation/" class="md-nav__link">
        Installation
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" type="checkbox" id="__nav_4_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4_2">
          Download Files
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Download Files" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_2">
          <span class="md-nav__icon md-icon"></span>
          Download Files
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../references/" class="md-nav__link">
        Reference
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../annotations/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Lab Session" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_1" type="checkbox" id="__nav_5_1" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5_1">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 1" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          R for Bioinformatics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        R for Bioinformatics
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#what-is-r" class="md-nav__link">
    What is R?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#some-of-rs-strengths" class="md-nav__link">
    Some of R’s strengths:
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-r-environment" class="md-nav__link">
    The R environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-r-homepage" class="md-nav__link">
    The R Homepage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rstudio-has-many-features" class="md-nav__link">
    RStudio has many features:
  </a>
  
    <nav class="md-nav" aria-label="RStudio has many features:">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-getting-started" class="md-nav__link">
    1. Getting started
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-open-a-new-rscript-file" class="md-nav__link">
    2. Open a new RScript File
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-basics-of-your-environment" class="md-nav__link">
    3. Basics of your environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-writing-and-running-r-commands" class="md-nav__link">
    4. Writing and running R commands
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-the-assignment-operator-vs-equals" class="md-nav__link">
    5. The assignment operator ( &lt;- ) vs equals ( = )
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-the-rstudio-cheat-sheets" class="md-nav__link">
    6. The RStudio Cheat Sheets
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_2" type="checkbox" id="__nav_5_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_2">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 2" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_3" type="checkbox" id="__nav_5_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_3">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 3" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_4" type="checkbox" id="__nav_5_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_4">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 4" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#what-is-r" class="md-nav__link">
    What is R?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#some-of-rs-strengths" class="md-nav__link">
    Some of R’s strengths:
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-r-environment" class="md-nav__link">
    The R environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-r-homepage" class="md-nav__link">
    The R Homepage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rstudio-has-many-features" class="md-nav__link">
    RStudio has many features:
  </a>
  
    <nav class="md-nav" aria-label="RStudio has many features:">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-getting-started" class="md-nav__link">
    1. Getting started
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-open-a-new-rscript-file" class="md-nav__link">
    2. Open a new RScript File
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-basics-of-your-environment" class="md-nav__link">
    3. Basics of your environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-writing-and-running-r-commands" class="md-nav__link">
    4. Writing and running R commands
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-the-assignment-operator-vs-equals" class="md-nav__link">
    5. The assignment operator ( &lt;- ) vs equals ( = )
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-the-rstudio-cheat-sheets" class="md-nav__link">
    6. The RStudio Cheat Sheets
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                

<h1 id="programming-in-r-for-bioinformatics">Programming in R for Bioinformatics<a class="headerlink" href="#programming-in-r-for-bioinformatics" title="Permanent link">&para;</a></h1>
<blockquote>
<p>Tip: In order to take most out of this tutorial you should not miss reading any lines in this tutorial and follow the flow and write codes on your computer.</p>
</blockquote>
<h3 id="what-is-r">What is R?<a class="headerlink" href="#what-is-r" title="Permanent link">&para;</a></h3>
<p>R is a language and environment for statistical computing and graphics developed in 1993. It provides a wide variety of statistical and graphical techniques (linear and nonlinear modeling, statistical tests, time series analysis, classification, clustering, …), and is highly extensible, meaning that the user community can write new R tools. It is a GNU project (Free and Open Source).</p>
<p>The R language has its roots in the S language and environment which was developed at Bell Laboratories (formerly AT&amp;T, now Lucent Technologies) by John Chambers and colleagues. R was created by Ross Ihaka and Robert Gentleman at the University of Auckland, New Zealand, and now, R is developed by the R Development Core Team, of which Chambers is a member. R is named partly after the first names of the first two R authors (Robert Gentleman and Ross Ihaka), and partly as a play on the name of S. R can be considered as a different implementation of S. There are some important differences, but much code written for S runs unaltered under R.</p>
<h3 id="some-of-rs-strengths">Some of R’s strengths:<a class="headerlink" href="#some-of-rs-strengths" title="Permanent link">&para;</a></h3>
<ol>
<li>The ease with which well-designed publication-quality plots can be produced, including mathematical symbols and formulae where needed. Great care has been taken over the defaults for the minor design choices in graphics, but the user retains full control.</li>
<li>It compiles and runs on a wide variety of UNIX platforms and similar systems (including FreeBSD and Linux), Windows and MacOS.</li>
<li>R can be extended (easily) via packages.</li>
<li>R has its own LaTeX-like documentation format, which is used to supply comprehensive documentation, both on-line in a number of formats and in hardcopy.</li>
<li>It has a vast community both in academia and in business.</li>
<li>It’s FREE!</li>
</ol>
<h3 id="the-r-environment">The R environment<a class="headerlink" href="#the-r-environment" title="Permanent link">&para;</a></h3>
<p>R is an integrated suite of software facilities for data manipulation, calculation and graphical display. It includes</p>
<ul>
<li>an effective data handling and storage facility,</li>
<li>a suite of operators for calculations on arrays, in particular matrices,</li>
<li>a large, coherent, integrated collection of intermediate tools for data analysis,</li>
<li>graphical facilities for data analysis and display either on-screen or on hardcopy, and</li>
<li>a well-developed, and effective programming language which includes conditionals, loops, user-defined recursive functions and input and output facilities.</li>
<li>The term “environment” is intended to characterize it as a fully planned and coherent system, rather than an incremental accretion of very specific and inflexible tools, as is frequently the case with other data analysis software.</li>
</ul>
<p>R, like S, is designed around a true computer language, and it allows users to add additional functionality by defining new functions. Much of the system is itself written in the R dialect of S, which makes it easy for users to follow the algorithmic choices made. For computationally-intensive tasks, C, C++ and Fortran code can be linked and called at run time. Advanced users can write C code to manipulate R objects directly.</p>
<p>Many users think of R as a statistics system. The R group prefers to think of it of an environment within which statistical techniques are implemented.</p>
<h3 id="the-r-homepage">The R Homepage<a class="headerlink" href="#the-r-homepage" title="Permanent link">&para;</a></h3>
<p>The R homepage has a wealth of information on it,</p>
<p>R-project.org</p>
<p>On the homepage you can:</p>
<p>Learn more about R
* Download R
* Get Documentation (official and user supplied)
* Get access to CRAN ‘Comprehensive R archival network’
* RStudio
* RStudio started in 2010, to offer R a more full featured integrated development environment (IDE) and modeled after matlabs IDE.</p>
<h3 id="rstudio-has-many-features">RStudio has many features:<a class="headerlink" href="#rstudio-has-many-features" title="Permanent link">&para;</a></h3>
<ul>
<li>syntax highlighting</li>
<li>code completion</li>
<li>smart indentation</li>
<li>“Projects”</li>
<li>workspace browser and data viewer</li>
<li>embedded plots</li>
<li>Markdown notebooks, Sweave authoring and knitr with one click pdf or html</li>
<li>runs on all platforms and over the web</li>
<li>etc. etc. etc.</li>
</ul>
<p>RStudio and its team have contributed to many R packages. These include:</p>
<p><strong>Tidyverse</strong> – R packages for data science, including ggplot2, dplyr, tidyr, and purrr
<strong>Shiny</strong> – An interactive web technology
<strong>RMarkdown</strong> – Insert R code into markdown documents
<strong>knitr</strong> – Dynamic reports combining R, TeX, Markdown &amp; HTML
<strong>packrat</strong> – Package dependency tool
<strong>devtools</strong> – Package development tool</p>
<h4 id="1-getting-started">1. Getting started<a class="headerlink" href="#1-getting-started" title="Permanent link">&para;</a></h4>
<p>Let’s start RStudio</p>
<p><img alt="" src="https://i.imgur.com/Rg8hId9.jpg" /></p>
<h4 id="2-open-a-new-rscript-file">2. Open a new RScript File<a class="headerlink" href="#2-open-a-new-rscript-file" title="Permanent link">&para;</a></h4>
<p>File -&gt; New File -&gt; R Script</p>
<p>RStudio_newfile</p>
<p>Then save the new empty file as Intro2R.R</p>
<p>File -&gt; Save as -&gt; Intro2R.R</p>
<h4 id="3-basics-of-your-environment">3. Basics of your environment<a class="headerlink" href="#3-basics-of-your-environment" title="Permanent link">&para;</a></h4>
<p>The R prompt is the ‘&gt;’ , when R is expecting more (command is not complete) you see a ‘+’</p>
<p><img alt="" src="https://i.imgur.com/hNyKj3h.png" /></p>
<h4 id="4-writing-and-running-r-commands">4. Writing and running R commands<a class="headerlink" href="#4-writing-and-running-r-commands" title="Permanent link">&para;</a></h4>
<p>In the source editor (top left by default) type</p>
<p><code>getwd()</code>
Then on the line Control + Enter (Linux/Windows), Command + Enter (Mac) to execute the line.</p>
<h4 id="5-the-assignment-operator-vs-equals">5. The assignment operator ( &lt;- ) vs equals ( = )<a class="headerlink" href="#5-the-assignment-operator-vs-equals" title="Permanent link">&para;</a></h4>
<p>The assignment operator is used assign data to a variable</p>
<p><div class="highlight"><pre><span></span><code>x &lt;- 1:10
x
[1]  1  2  3  4  5  6  7  8  9 10
</code></pre></div>
In this case, the equal sign works as well</p>
<p><div class="highlight"><pre><span></span><code>x = 1:10
x
[1]  1  2  3  4  5  6  7  8  9 10
</code></pre></div>
But you should NEVER EVER DO THIS</p>
<p><div class="highlight"><pre><span></span><code>1:10 -&gt; x
x
[1]  1  2  3  4  5  6  7  8  9 10
</code></pre></div>
The two act the same in most cases. The difference in assignment operators is clearer when you use them to set an argument value in a function call. For example:</p>
<p><div class="highlight"><pre><span></span><code>median(x = 1:10)
x 
Error: object &#39;x&#39; not found
</code></pre></div>
In this case, x is declared within the scope of the function, so it does not exist in the user workspace.</p>
<p><div class="highlight"><pre><span></span><code>median(x &lt;- 1:10)
x
[1]  1  2  3  4  5  6  7  8  9 10
</code></pre></div>
In this case, x is declared in the user workspace, so you can use it after the function call has been completed. There is a general preference among the R community for using &lt;- for assignment (other than in function signatures)</p>
<h4 id="6-the-rstudio-cheat-sheets">6. The RStudio Cheat Sheets<a class="headerlink" href="#6-the-rstudio-cheat-sheets" title="Permanent link">&para;</a></h4>
<p><a href="https://github.com/rstudio/cheatsheets/raw/master/rstudio-ide.pdf">rstudio-ide.pdf</a></p>
<p>spend 15m getting to know RStudio a little</p>
<h1 id="import-and-export-data-in-r">Import and export data in R<a class="headerlink" href="#import-and-export-data-in-r" title="Permanent link">&para;</a></h1>
<p>R base function read.table() is a general funciton that can be used to read a file in table format. The data will be imported as a data frame.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># To read a local file. If you have downloaded the raw_counts.txt file to your local machine, you may use the following command to read it in, by providing the full path for the file location. The way to specify the full path is the same as taught in the command line session. Here we assume raw_counts.txt is in our current working directory</span>
<span class="n">data</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;./raw_counts.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">stringsAsFactors</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>

<span class="c1"># There is a very convenient way to read files from the internet.</span>
<span class="n">data</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;https://raw_counts.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">stringsAsFactors</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</code></pre></div>
<p>Take a look at the beginning part of the data frame.</p>
<div class="highlight"><pre><span></span><code><span class="nf">head</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##            C61  C62  C63  C64  C91  C92  C93 C94 I561 I562 I563 I564 I591 I592
## AT1G01010  322  346  256  396  372  506  361 342  638  488  440  479  770  430
## AT1G01020  149   87  162  144  189  169  147 108  163  141  119  147  182  156
## AT1G01030   15   32   35   22   24   33   21  35   18    8   54   35   23    8
## AT1G01040  687  469  568  651  885  978  794 862  799  769  725  715  811  567
## AT1G01046    1    1    5    4    5    3    0   2    4    3    1    0    2    8
## AT1G01050 1447 1032 1083 1204 1413 1484 1138 938 1247 1516  984 1044 1374 1355
##           I593 I594 I861 I862 I863 I864 I891 I892 I893 I894
## AT1G01010  656  467  143  453  429  206  567  458  520  474
## AT1G01020  153  177   43  144  114   50  161  195  157  144
## AT1G01030   16   24   42   17   22   39   26   28   39   30
## AT1G01040  831  694  345  575  605  404  735  651  725  591
## AT1G01046    8    1    0    4    0    3    5    7    0    5
## AT1G01050 1437 1577  412 1338 1051  621 1434 1552 1248 1186
</code></pre></div>
<p>Depending on the format of the file, several variants of read.table() are available to make reading a file easier.</p>
<p>read.csv(): for reading "comma separated value" files (.csv).</p>
<p>read.csv2(): variant used in countries that use a comma "," as decimal point and a semicolon ";" as field separators.</p>
<p>read.delim(): for reading "tab separated value" files (".txt"). By default, point(".") is used as decimal point.</p>
<p>read.delim2(): for reading "tab separated value" files (".txt"). By default, comma (",") is used as decimal point.</p>
<p>Choosing the correct function (or parameters) is important! If we use <code>read.csv()</code> to read our tab-delimited file, it becomes a mess.</p>
<div class="highlight"><pre><span></span><code><span class="n">data2</span> <span class="o">&lt;-</span> <span class="nf">read.csv</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;https://raw_counts.txt&quot;</span><span class="p">,</span> <span class="n">stringsAsFactors</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>

<span class="nf">head</span><span class="p">(</span><span class="n">data2</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##                                         C61.C62.C63.C64.C91.C92.C93.C94.I561.I562.I563.I564.I591.I592.I593.I594.I861.I862.I863.I864.I891.I892.I893.I894
## 1                     AT1G01010\t322\t346\t256\t396\t372\t506\t361\t342\t638\t488\t440\t479\t770\t430\t656\t467\t143\t453\t429\t206\t567\t458\t520\t474
## 2                        AT1G01020\t149\t87\t162\t144\t189\t169\t147\t108\t163\t141\t119\t147\t182\t156\t153\t177\t43\t144\t114\t50\t161\t195\t157\t144
## 3                                               AT1G01030\t15\t32\t35\t22\t24\t33\t21\t35\t18\t8\t54\t35\t23\t8\t16\t24\t42\t17\t22\t39\t26\t28\t39\t30
## 4                     AT1G01040\t687\t469\t568\t651\t885\t978\t794\t862\t799\t769\t725\t715\t811\t567\t831\t694\t345\t575\t605\t404\t735\t651\t725\t591
## 5                                                                     AT1G01046\t1\t1\t5\t4\t5\t3\t0\t2\t4\t3\t1\t0\t2\t8\t8\t1\t0\t4\t0\t3\t5\t7\t0\t5
## 6 AT1G01050\t1447\t1032\t1083\t1204\t1413\t1484\t1138\t938\t1247\t1516\t984\t1044\t1374\t1355\t1437\t1577\t412\t1338\t1051\t621\t1434\t1552\t1248\t1186
</code></pre></div>
<p>However, the <code>read.csv()</code> function is appropriate for a comma-delimited file.</p>
<div class="highlight"><pre><span></span><code><span class="n">data3</span> <span class="o">&lt;-</span> <span class="nf">read.csv</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;https://raw_counts.csv&quot;</span><span class="p">,</span> <span class="n">stringsAsFactors</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>

<span class="nf">head</span><span class="p">(</span><span class="n">data3</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##            C61  C62  C63  C64  C91  C92  C93 C94 I561 I562 I563 I564 I591 I592
## AT1G01010  322  346  256  396  372  506  361 342  638  488  440  479  770  430
## AT1G01020  149   87  162  144  189  169  147 108  163  141  119  147  182  156
## AT1G01030   15   32   35   22   24   33   21  35   18    8   54   35   23    8
## AT1G01040  687  469  568  651  885  978  794 862  799  769  725  715  811  567
## AT1G01046    1    1    5    4    5    3    0   2    4    3    1    0    2    8
## AT1G01050 1447 1032 1083 1204 1413 1484 1138 938 1247 1516  984 1044 1374 1355
##           I593 I594 I861 I862 I863 I864 I891 I892 I893 I894
## AT1G01010  656  467  143  453  429  206  567  458  520  474
## AT1G01020  153  177   43  144  114   50  161  195  157  144
## AT1G01030   16   24   42   17   22   39   26   28   39   30
## AT1G01040  831  694  345  575  605  404  735  651  725  591
## AT1G01046    8    1    0    4    0    3    5    7    0    5
## AT1G01050 1437 1577  412 1338 1051  621 1434 1552 1248 1186
</code></pre></div>
<p>Since the data contained in these files is the same, we don't need to keep three copies.</p>
<div class="highlight"><pre><span></span><code><span class="nf">identical</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">data3</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] TRUE
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="nf">rm</span><span class="p">(</span><span class="n">data2</span><span class="p">,</span> <span class="n">data3</span><span class="p">)</span>
</code></pre></div>
<p>R base function write.table() can be used to export data to a file.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># To write to a file called &quot;output.txt&quot; in your current working directory.</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">data</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">,],</span> <span class="n">file</span><span class="o">=</span><span class="s">&quot;output.txt&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span> <span class="n">row.names</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">col.names</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>
</code></pre></div>
<p>It is also possible to export data to a csv file.</p>
<p>write.csv()</p>
<p>write.csv2()</p>
<hr />
<h1 id="basic-statistics-in-r">Basic statistics in R<a class="headerlink" href="#basic-statistics-in-r" title="Permanent link">&para;</a></h1>
<table>
<thead>
<tr>
<th>Description</th>
<th>R_function</th>
</tr>
</thead>
<tbody>
<tr>
<td>Mean</td>
<td>mean()</td>
</tr>
<tr>
<td>Standard deviation</td>
<td>sd()</td>
</tr>
<tr>
<td>Variance</td>
<td>var()</td>
</tr>
<tr>
<td>Minimum</td>
<td>min()</td>
</tr>
<tr>
<td>Maximum</td>
<td>max()</td>
</tr>
<tr>
<td>Median</td>
<td>median()</td>
</tr>
<tr>
<td>Range of values: minimum and maximum</td>
<td>range()</td>
</tr>
<tr>
<td>Sample quantiles</td>
<td>quantile()</td>
</tr>
<tr>
<td>Generic function</td>
<td>summary()</td>
</tr>
<tr>
<td>Interquartile range</td>
<td>IQR()</td>
</tr>
</tbody>
</table>
<p>Calculate the mean expression for each sample.</p>
<div class="highlight"><pre><span></span><code><span class="nf">apply</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="m">2</span><span class="p">,</span> <span class="n">mean</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##         V1         V2         V3         V4         V5         V6         V7 
##  0.1800418 -0.2534468 -0.1897400 -0.5426865 -0.1894511 -0.2531511 -0.3651648
</code></pre></div>
<p>Calculate the range of expression for each sample.</p>
<div class="highlight"><pre><span></span><code><span class="nf">apply</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="m">2</span><span class="p">,</span> <span class="n">range</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##              V1        V2         V3         V4        V5         V6        V7
## [1,] -0.6225434 -2.452988 -1.1885474 -1.5352739 -1.311441 -1.3231301 -3.108072
## [2,]  0.9157785  1.558738  0.5844574  0.5020093  1.986243  0.8357217  1.515365
</code></pre></div>
<p>Calculate the quantiles of each samples.</p>
<div class="highlight"><pre><span></span><code><span class="nf">apply</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="m">2</span><span class="p">,</span> <span class="n">quantile</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##               V1         V2         V3          V4          V5         V6
## 0%   -0.62254340 -2.4529883 -1.1885474 -1.53527385 -1.31144098 -1.3231301
## 25%  -0.24319891 -1.4878457 -0.4367767 -1.03930207 -0.83116242 -0.9900499
## 50%   0.08886876  0.2487017 -0.3765026 -0.58285066 -0.50073601 -0.4826310
## 75%   0.68229320  0.9235564  0.2629829 -0.05204306  0.08105049  0.5890407
## 100%  0.91577852  1.5587378  0.5844574  0.50200930  1.98624311  0.8357217
##               V7
## 0%   -3.10807211
## 25%  -0.84236920
## 50%  -0.05402889
## 75%   0.38766034
## 100%  1.51536487
</code></pre></div>
<hr />
<h1 id="simple-data-visualization">Simple data visualization<a class="headerlink" href="#simple-data-visualization" title="Permanent link">&para;</a></h1>
<p>Scatter plot and line plot can be produced using the function plot().</p>
<div class="highlight"><pre><span></span><code><span class="n">x</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">50</span><span class="p">)</span>
<span class="n">y</span> <span class="o">&lt;-</span> <span class="m">1</span> <span class="o">+</span> <span class="nf">sqrt</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="o">/</span><span class="m">2</span>
<span class="nf">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot1" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_01.png" /></p>
<div class="highlight"><pre><span></span><code><span class="nf">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;l&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot2" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_02.png" /></p>
<div class="highlight"><pre><span></span><code><span class="c1"># plot both the points and lines</span>
<span class="c1">## first plot points</span>
<span class="nf">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">)</span>
<span class="nf">lines</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;l&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot2" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_03.png" /></p>
<div class="highlight"><pre><span></span><code><span class="c1">## lines() can only be used to add information to a graph, while it cannot produce a graph on its own.</span>
</code></pre></div>
<p>boxplot() can be used to summarize data.</p>
<div class="highlight"><pre><span></span><code><span class="nf">boxplot</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">xlab</span><span class="o">=</span><span class="s">&quot;Sample ID&quot;</span><span class="p">,</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&quot;Raw Counts&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot4" src="plot_uploaded--HERE" /><!-- --></p>
<p>add more details to the plot.</p>
<div class="highlight"><pre><span></span><code><span class="nf">boxplot</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">xlab</span><span class="o">=</span><span class="s">&quot;Sample ID&quot;</span><span class="p">,</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&quot;Raw Counts&quot;</span><span class="p">,</span> <span class="n">main</span><span class="o">=</span><span class="s">&quot;Expression levels&quot;</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="s">&quot;blue&quot;</span><span class="p">,</span> <span class="n">border</span><span class="o">=</span><span class="s">&quot;black&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot5" src="plot_uploaded--HERE" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="n">x</span> <span class="o">&lt;-</span> <span class="nf">rnorm</span><span class="p">(</span><span class="m">1000</span><span class="p">)</span>
<span class="nf">boxplot</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot6" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_06.png" /></p>
<p>hist() can be used to create histograms of data.</p>
<div class="highlight"><pre><span></span><code><span class="nf">hist</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
</code></pre></div>
<p><img alt="Plot7" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_07.png" /></p>
<div class="highlight"><pre><span></span><code><span class="c1"># use user defined break points</span>
<span class="nf">hist</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">breaks</span><span class="o">=</span><span class="nf">seq</span><span class="p">(</span><span class="nf">range</span><span class="p">(</span><span class="n">x</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span><span class="m">-1</span><span class="p">,</span> <span class="nf">range</span><span class="p">(</span><span class="n">x</span><span class="p">)[</span><span class="m">2</span><span class="p">]</span><span class="m">+1</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="m">0.5</span><span class="p">))</span>
</code></pre></div>
<p><img alt="Plot8" src="https://course-5534.s3.amazonaws.com/Course_img/Rplot_08.png" /></p>
<div class="highlight"><pre><span></span><code><span class="c1"># clear plotting device/area</span>
<span class="nf">dev.off</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## null device 
##           1
</code></pre></div>
<hr />
<h1 id="ggplot2">ggplot2<a class="headerlink" href="#ggplot2" title="Permanent link">&para;</a></h1>
<p>ggplot2 is an R package for making visualizations and is based on the ‘Grammar of Graphics’ . It is often referred as one of the best packages for visualizations . Actually , there are other tools for making graphs in R such as Lattice and base R graphics . However , ggplot2 is straightforwardly easy to make any kind of graphs even complex ones . It is easy to add complexity or take it away which makes ggplot2 superior among these three . Since you can’t go back and forth once you created a plot in base R or in lattice . That’s enough for comparison , let’s start learning it now.</p>
<blockquote>
<p>Tip: In order to take most out of this tutorial you should not miss reading any lines in this tutorial and follow the flow and write codes on your computer.</p>
</blockquote>
<p><strong><em>Grammar of Graphics</em></strong></p>
<p>Just as the grammar of language helps us construct meaningful sentences out of words, the Grammar of Graphics helps us to construct graphical figures out of different visual elements. This grammar gives us a way to talk about parts of a plot: all the circles, lines, arrows, and words that are combined into a diagram for visualizing data. Originally developed by Leland Wilkinson, the Grammar of Graphics was adapted by Hadley Wickham to describe the components of a plot, including</p>
<ul>
<li>the data being plotted</li>
<li>the geometric objects (circles, lines, etc.) that appear on the plot</li>
<li>a set of mappings from variables in the data to the aesthetics (appearance) of the geometric objects</li>
<li>a statistical transformation used to calculate the data values used in the plot</li>
<li>a position adjustment for locating each geometric object on the plot</li>
<li>a scale (e.g., range of values) for each aesthetic mapping used</li>
<li>a coordinate system used to organize the geometric objects</li>
<li>the facets or groups of data shown in different plots</li>
</ul>
<p>Similar to how tidyr and dplyr provide efficient data transformation and manipulation, ggplot2 provides more efficient ways to create specific visual images. </p>
<p>ggplot objects can be highly complex, but the basic order of layers will usually look like this:</p>
<ul>
<li>Begin with the baseline ggplot() command - this “opens” the ggplot and allow subsequent functions to be added with +. Typically the dataset is also specified in this command</li>
<li>Add “geom” layers - these functions visualize the data as geometries (shapes), e.g. as a bar graph, line plot, scatter plot, histogram (or a combination!). These functions all start with geom_ as a prefix.</li>
<li>Add design elements to the plot such as axis labels, title, fonts, sizes, color schemes, legends, or axes rotation</li>
<li>A simple example of skeleton code is as follows. We will explain each component in the sections below.</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1"># plot data from my_data columns as red points</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">my_data</span><span class="p">)</span><span class="o">+</span>                   <span class="c1"># use the dataset &quot;my_data&quot;</span>
  <span class="nf">geom_point</span><span class="p">(</span>                             <span class="c1"># add a layer of points (dots)</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">col1</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">col2</span><span class="p">),</span>    <span class="c1"># &quot;map&quot; data column to axes</span>
    <span class="n">color</span> <span class="o">=</span> <span class="s">&quot;red&quot;</span><span class="p">)</span><span class="o">+</span>                       <span class="c1"># other specification for the geom</span>
  <span class="nf">labs</span><span class="p">()</span><span class="o">+</span>                                 <span class="c1"># here you add titles, axes labels, etc.</span>
  <span class="nf">theme</span><span class="p">()</span>                                 <span class="c1"># here you adjust color, font, size etc of non-data plot elements (axes, title, etc.) </span>
</code></pre></div>
<h2 id="starting-up-with-ggplot">Starting up with ggplot<a class="headerlink" href="#starting-up-with-ggplot" title="Permanent link">&para;</a></h2>
<p>Firstly , we need to install ggplot2 package
<div class="highlight"><pre><span></span><code><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;ggplot2&quot;</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="s">&quot;ggplot2&quot;</span><span class="p">)</span>
</code></pre></div></p>
<h3 id="the-basics-of-plotting-with-ggplot">The Basics of plotting with ggplot<a class="headerlink" href="#the-basics-of-plotting-with-ggplot" title="Permanent link">&para;</a></h3>
<p>In order to create a plot, you:</p>
<ol>
<li>Call the ggplot() function which creates a blank canvas</li>
<li>Specify aesthetic mappings, which specifies how you want to map variables to visual aspects. In this case we are simply mapping the displ and hwy variables to the x- and y-axes.</li>
<li>You then add new layers that are geometric objects which will show up on the plot. In this case we add geom_point to add a layer with points (dot) elements as the geometric shapes to represent the data.</li>
</ol>
<p><div class="highlight"><pre><span></span><code><span class="c1"># create canvas</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>

<span class="c1"># variables of interest mapped</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">displ</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">hwy</span><span class="p">))</span>

<span class="c1"># data plotted</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">displ</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">hwy</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">()</span>
</code></pre></div>
<img alt="" src="https://i.imgur.com/g5rCVvp.png" /></p>
<p>To try ggplot hands-on, we will follow the original ggplot basics chapter from <a href="https://www.epirhandbook.com/en/index.html">The Epidemiologist R Handbook</a> (Batra, Neale, et al. The Epidemiologist R Handbook. 2021).</p>
<h2 id="now-lets-go-to-ggplot-hands-on-exercise">Now let's go to <a href="https://www.epirhandbook.com/en/ggplot-basics.html"><b>ggplot hands-on exercise.</b></a><a class="headerlink" href="#now-lets-go-to-ggplot-hands-on-exercise" title="Permanent link">&para;</a></h2>
<p>(clicking on the link above will take you to the handbook's website, please feel free to bookmark the page for later reference.)</p>

              
            </article>
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" data-md-state="hidden">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../aws_env/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Workspace - AWS" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Workspace - AWS
            </div>
          </div>
        </a>
      
      
        
        <a href="../human_genome/" class="md-footer__link md-footer__link--next" aria-label="Next: Human Genome" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Human Genome
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2021, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "navigation.sections", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest"], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.361d90f1.min.js"}</script>
    
    
      <script src="../../assets/javascripts/bundle.289a2a4b.min.js"></script>
      
    
  </body>
</html>