
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../basics_awk/">
      
      
        <link rel="next" href="../R_for_bioinformatics/">
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.1.6">
    
    
      
        <title>Workspace - AWS - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.ded33207.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.a0c5b2b5.min.css">
      
      
  
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
  
  <style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M64 32C28.7 32 0 60.7 0 96v320c0 35.3 28.7 64 64 64h224V368c0-26.5 21.5-48 48-48h112V96c0-35.3-28.7-64-64-64H64zm384 320H336c-8.8 0-16 7.2-16 16v112l32-32 64-64 32-32z"/></svg>');--md-admonition-icon--abstract:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M96 0C43 0 0 43 0 96v320c0 53 43 96 96 96h320c17.7 0 32-14.3 32-32s-14.3-32-32-32v-64c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H96zm0 384h256v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"/></svg>');--md-admonition-icon--info:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-40-176h24v-64h-24c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24h-80c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"/></svg>');--md-admonition-icon--tip:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M480 32c0-12.9-7.8-24.6-19.8-29.6S434.5.2 425.3 9.3L381.7 53c-48 48-113.1 75-181 75H64c-35.3 0-64 28.7-64 64v96c0 35.3 28.7 64 64 64v128c0 17.7 14.3 32 32 32h64c17.7 0 32-14.3 32-32V352h8.7c67.9 0 133 27 181 75l43.6 43.6c9.2 9.2 22.9 11.9 34.9 6.9s19.8-16.6 19.8-29.6V300.4c18.6-8.8 32-32.5 32-60.4s-13.4-51.6-32-60.4V32zm-64 76.7v262.6C357.2 317.8 280.5 288 200.7 288H192v-96h8.7c79.8 0 156.5-29.8 215.3-83.3z"/></svg>');--md-admonition-icon--success:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7l233.4-233.3c12.5-12.5 32.8-12.5 45.3 0z"/></svg>');--md-admonition-icon--question:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-86.2-346.7c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1 0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6-13.3 0-24-10.7-24-24v-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1 0-8.4-6.8-15.1-15.1-15.1h-58.3c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1-64 0z"/></svg>');--md-admonition-icon--warning:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7.2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8.2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24v112c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0-64 0 32 32 0 1 0 64 0z"/></svg>');--md-admonition-icon--failure:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.1 52.4 442.6 6.5c-1.9-3.9-6.1-6.5-10.5-6.5s-8.5 2.6-10.4 6.5l-16.5 45.9-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4 0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2 0-4.6-3-8.9-7.2-10.4l-45.6-16.8zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9c-22-8-45.8-12.3-70.5-12.3C93.1 96 0 189.1 0 304s93.1 208 208 208 208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg>');--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M416 398.9c58.5-41.1 96-104.1 96-174.9C512 100.3 397.4 0 256 0S0 100.3 0 224c0 70.7 37.5 133.8 96 174.9V464c0 26.5 21.5 48 48 48h48v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h64v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h48c26.5 0 48-21.5 48-48v-65.1zM96 256a64 64 0 1 1 128 0 64 64 0 1 1-128 0zm256-64a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"/></svg>');--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M320 0c17.7 0 32 14.3 32 32v64h120c39.8 0 72 32.2 72 72v272c0 39.8-32.2 72-72 72H168c-39.8 0-72-32.2-72-72V168c0-39.8 32.2-72 72-72h120V32c0-17.7 14.3-32 32-32zM208 384c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zM264 256a40 40 0 1 0-80 0 40 40 0 1 0 80 0zm152 40a40 40 0 1 0 0-80 40 40 0 1 0 0 80zM48 224h16v192H48c-26.5 0-48-21.5-48-48v-96c0-26.5 21.5-48 48-48zm544 0c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48h-16V224h16z"/></svg>');--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M288 0H128c-17.7 0-32 14.3-32 32s14.3 32 32 32v132.8c0 11.8-3.3 23.5-9.5 33.5L10.3 406.2C3.6 417.2 0 429.7 0 442.6 0 480.9 31.1 512 69.4 512h309.2c38.3 0 69.4-31.1 69.4-69.4 0-12.8-3.6-25.4-10.3-36.4L329.5 230.4c-6.2-10.1-9.5-21.7-9.5-33.5V64c17.7 0 32-14.3 32-32S337.7 0 320 0h-32zm-96 196.8V64h64v132.8c0 23.7 6.6 46.9 19 67.1l34.5 56.1h-171l34.5-56.1c12.4-20.2 19-43.4 19-67.1z"/></svg>');--md-admonition-icon--quote:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 216C0 149.7 53.7 96 120 96h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V216zm256 0c0-66.3 53.7-120 120-120h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64h-64c-35.3 0-64-28.7-64-64V216z"/></svg>');}</style>


    
    
    
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#the-workspace-on-aws-and-installations" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Workspace - AWS
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      Course Details
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../schedule/" class="md-tabs__link">
      Schedule
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../lectures/lectures/" class="md-tabs__link">
      Lectures
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../aws/aws_ec2_instances/" class="md-tabs__link">
      AWS Instances Login
    </a>
  </li>

      
        
  
  
    
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../basic_unix/" class="md-tabs__link md-tabs__link--active">
        Lab Session
      </a>
    </li>
  

  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
      
      
        
          
        
          
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_1" checked>
      
      
        
          
        
          
        
          
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5_1" id="__nav_5_1_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_1_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Workspace - AWS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Workspace - AWS
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#r" class="md-nav__link">
    R
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-to-use-the-terminal-download-files-to-your-local-machine" class="md-nav__link">
    How to use the terminal download files to your local machine
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#explore-the-workspace-on-aws" class="md-nav__link">
    Explore the workspace on AWS
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#list-of-preinstalled-bioinformatic-tools" class="md-nav__link">
    List of preinstalled bioinformatic tools
  </a>
  
    <nav class="md-nav" aria-label="List of preinstalled bioinformatic tools">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bam-readcount" class="md-nav__link">
    bam-readcount
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bcftools" class="md-nav__link">
    BCFtools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bwa" class="md-nav__link">
    BWA
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cnvkit" class="md-nav__link">
    CNVkit
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#delly" class="md-nav__link">
    Delly
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fastqc" class="md-nav__link">
    FastQC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gatk-4" class="md-nav__link">
    GATK 4
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gffcompare" class="md-nav__link">
    Gffcompare
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hisat2" class="md-nav__link">
    HISAT2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kallisto" class="md-nav__link">
    Kallisto
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mosdepth" class="md-nav__link">
    mosdepth
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiqc" class="md-nav__link">
    MultiQC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#picard" class="md-nav__link">
    PICARD
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pizzly" class="md-nav__link">
    Pizzly
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sambamba" class="md-nav__link">
    Sambamba
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#samtools" class="md-nav__link">
    Samtools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seqtk" class="md-nav__link">
    seqtk
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#strelka" class="md-nav__link">
    Strelka
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#stringtie" class="md-nav__link">
    StringTie
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#varscan" class="md-nav__link">
    Varscan
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vcf-annotation-tools" class="md-nav__link">
    vcf-annotation-tools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vep" class="md-nav__link">
    VEP
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vt" class="md-nav__link">
    vt
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        ggplot tutorial
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_2" >
      
      
        
          
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_3" >
      
      
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_4" >
      
      
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../structural_variation/" class="md-nav__link">
        Structural Variants
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../analysis_of_cnv/" class="md-nav__link">
        Copy Number Variations
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_5" >
      
      
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
          Day 5
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          Day 5
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../joint_curation/" class="md-nav__link">
        Joint Curation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../genomic_alteration_interpretation_exe/" class="md-nav__link">
        Genomic Alterations Interpretation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#r" class="md-nav__link">
    R
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-to-use-the-terminal-download-files-to-your-local-machine" class="md-nav__link">
    How to use the terminal download files to your local machine
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#explore-the-workspace-on-aws" class="md-nav__link">
    Explore the workspace on AWS
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#list-of-preinstalled-bioinformatic-tools" class="md-nav__link">
    List of preinstalled bioinformatic tools
  </a>
  
    <nav class="md-nav" aria-label="List of preinstalled bioinformatic tools">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bam-readcount" class="md-nav__link">
    bam-readcount
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bcftools" class="md-nav__link">
    BCFtools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bwa" class="md-nav__link">
    BWA
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cnvkit" class="md-nav__link">
    CNVkit
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#delly" class="md-nav__link">
    Delly
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fastqc" class="md-nav__link">
    FastQC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gatk-4" class="md-nav__link">
    GATK 4
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gffcompare" class="md-nav__link">
    Gffcompare
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hisat2" class="md-nav__link">
    HISAT2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kallisto" class="md-nav__link">
    Kallisto
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mosdepth" class="md-nav__link">
    mosdepth
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiqc" class="md-nav__link">
    MultiQC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#picard" class="md-nav__link">
    PICARD
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pizzly" class="md-nav__link">
    Pizzly
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sambamba" class="md-nav__link">
    Sambamba
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#samtools" class="md-nav__link">
    Samtools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#seqtk" class="md-nav__link">
    seqtk
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#strelka" class="md-nav__link">
    Strelka
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#stringtie" class="md-nav__link">
    StringTie
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#varscan" class="md-nav__link">
    Varscan
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vcf-annotation-tools" class="md-nav__link">
    vcf-annotation-tools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vep" class="md-nav__link">
    VEP
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vt" class="md-nav__link">
    vt
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="the-workspace-on-aws-and-installations">The workspace on AWS and installations<a class="headerlink" href="#the-workspace-on-aws-and-installations" title="Permanent link">&para;</a></h1>
<p>Due to the current covid-situation in Sweden we converted the course from on-site to on-line using zoom for lectures and the Amazon Web Services (AWS) for the practical sessions. Each student will get an "instance" of a linux computer in AWS to perform the labwork. During the labwork we will also use zoom for discussions and support.</p>
<p>To be able to connect to AWS on windows you need to be able to download the SSH client PuTTY  (<a href="https://www.putty.org">https://www.putty.org</a>). If you have a mac or a linux/unix computer nothing is needed to preinstall to work on the comman line. </p>
<p>To perform the exercises of the course you also need to install the following software on your local machine: <a href="https://software.broadinstitute.org/software/igv/download">IGV</a>.
Choose IGV that comes bundled with Java for your operating system.</p>
<p>This workshop requires a large number of different bioinformatics tools. Thes have been pre-installed on AWS. For installation instructions of the differnet tools, see the list at the end of this document which links to the website of each software tool. </p>
<h2 id="r">R<a class="headerlink" href="#r" title="Permanent link">&para;</a></h2>
<p><a href="https://www.r-project.org">R</a> is a feature rich interpretive programming language originally released in 1995. It is heavily used in the bioinformatics community largely due to numerous R libraries available on <a href="https://www.bioconductor.org">bioconductor</a>. </p>
<p>If you have not installed R yet:
- Download R to your local machine:
    - <a href="https://www.r-project.org">https://www.r-project.org</a>
- Double-click on the icon and start the installation process. </p>
<p>Install the following R libraries on your local machine:
<div class="highlight"><pre><span></span><code><span class="c1">#Open R and run the following code:</span>
<span class="n">packages</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;devtools&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;BiocManager&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;dplyr&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;tidyr&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;ggplot2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;data.table&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;patchwork&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;stringr&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;jsonlite&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;reshape2&quot;</span><span class="p">,</span><span class="s">&quot;cowplot&quot;</span><span class="p">)</span>
<span class="nf">install.packages</span><span class="p">(</span><span class="n">packages</span><span class="p">,</span><span class="w"> </span><span class="n">dependencies</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;biomaRt&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Biostrings&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;ensembldb&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;IRanges&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;EnsDb.Hsapiens.v86&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;bamsignals&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;BSgenome&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;BSgenome.Hsapiens.NCBI.hg19&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;csaw&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;DNAcopy&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;GenomicRanges&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;org.Hs.eg.db&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Rsamtools&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Repitools&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TxDb.Hsapiens.UCSC.hg19.knownGene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;VariantAnnotation&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;chimeraviz&quot;</span><span class="w"> </span><span class="n">dependencies</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># PSCBS requires DNAcopy </span>
<span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;PSCBS&quot;</span><span class="p">)</span>
</code></pre></div>
Note during the course you will be visualising data using R. If some of the libraries do not work on your local machine, you can run R and plot on AWS instead. However, then the plots will have to be dowloaded as pdf-files instead of being viewed interactively in R-studio.</p>
<p>The above mentioned libraries are installed in R, available in the following path, <code>/home/ubuntu/miniconda3/envs/r-rnaseq/bin/R</code>, to access this R env follow the below instructions.
- Using the command line do:
<div class="highlight"><pre><span></span><code><span class="c1"># Start R terminal without GUI</span>
/home/ubuntu/miniconda3/envs/r-rnaseq/bin/R
</code></pre></div>
- Using R do:
<div class="highlight"><pre><span></span><code><span class="c1"># Load libraries. Note, not all are needed every time,</span>
<span class="c1"># just check here that they load without error. </span>
<span class="c1"># Load each library and wait for it to finish before continuing </span>
<span class="nf">library</span><span class="p">(</span><span class="n">BiocManager</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">IRanges</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">patchwork</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">PSCBS</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">jsonlite</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">reshape2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">cowplot</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">biomaRt</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">Biostrings</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ensembldb</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">EnsDb.Hsapiens.v86</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">bamsignals</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">BSgenome</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">BSgenome.Hsapiens.NCBI.hg19</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">csaw</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">DNAcopy</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">GenomicRanges</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">org.Hs.eg.db</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">Rsamtools</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">Repitools</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">VariantAnnotation</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">chimeraviz</span><span class="p">)</span>

<span class="c1"># Check your current directory</span>
<span class="nf">getwd</span><span class="p">()</span>

<span class="c1"># Set to your home directory or some other directory</span>
<span class="nf">setwd</span><span class="p">(</span><span class="o">~</span><span class="p">)</span>
<span class="c1"># here we can run the R scripts </span>

<span class="c1"># Let us test to plot and save as pdf using pre-loaded data in R</span>

<span class="c1"># Check the data</span>
<span class="n">mtcars</span>

<span class="c1"># Assign a plot to p</span>
<span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">mtcars</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">mpg</span><span class="p">,</span><span class="w"> </span><span class="n">wt</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">()</span>

<span class="c1"># Save the p plot to the mtcars.pdf</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;mtcars.pdf&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">)</span>

<span class="c1"># How to download the pdf-file to your local machine,</span>
<span class="c1"># see the section below, however first quit R. &quot;no&quot;</span>
<span class="c1"># means that the R-workspace will not be saved.</span>
<span class="nf">q</span><span class="p">(</span><span class="s">&quot;no&quot;</span><span class="p">)</span>
</code></pre></div></p>
<h2 id="how-to-use-the-terminal-download-files-to-your-local-machine">How to use the terminal download files to your local machine<a class="headerlink" href="#how-to-use-the-terminal-download-files-to-your-local-machine" title="Permanent link">&para;</a></h2>
<p>This will can be used for both downloading files plotted in R on AWS and other files and file types, such as bam-files that will be visualised in IGV.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>TO_PREFERRED_DIRECTORY
scp<span class="w"> </span>-ri<span class="w"> </span>~/PEM_KEY_ID.pem<span class="w"> </span>ubuntu@AWS_ADDRESS_HERE:~/PATH_TO_PDF/FILENAME<span class="w"> </span>.

<span class="c1"># Before proceeding - test that you can get the &quot;mtcars.pdf&quot; file</span>
<span class="c1">#  downloaded to your local computer. If not, ask for assistance.</span>
</code></pre></div>
<h2 id="explore-the-workspace-on-aws">Explore the workspace on AWS<a class="headerlink" href="#explore-the-workspace-on-aws" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="c1">#Make sure the credentials are set for the &quot;pem&quot; (key) file.</span>
<span class="c1"># If needed run:</span>
<span class="c1"># chmod 400 course_ec2_new.pem</span>

<span class="c1"># Log onto AWS cloud (make sure in are in the same directory as the pem key).</span>
ssh<span class="w"> </span>-i<span class="w"> </span>./PEM_KEY_ID.pem<span class="w"> </span>ubuntu@AWS_ADDRESS_HERE

<span class="c1"># Check where you are</span>
<span class="nb">pwd</span>

<span class="c1"># Note &quot;~&quot;&quot; is a shortcut for the user directory. </span>
<span class="c1"># Pre-load shortcut and settings </span>
<span class="nb">source</span><span class="w"> </span>.bashrc

<span class="c1"># the workspace folder contains the course files and pre-installed bioinformatic tools</span>
<span class="nb">cd</span><span class="w"> </span>workspace/

<span class="c1"># The workspace folder contains the course files and pre-installed bioinformatic tools. Navigate to the folders, list the files and reflecton what is there</span>
<span class="nb">cd</span><span class="w">  </span>~/workspace/inputs
ls<span class="w"> </span>-alF
<span class="c1"># Look around in the folders</span>

<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
ls<span class="w"> </span>-alF
<span class="c1"># Look around in the folders</span>

<span class="c1"># Have a look at the raw exone sequencing data for the germline DNA</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/data/fastq/Exome_Norm

<span class="c1"># What are the files in this folder?</span>

<span class="c1"># Explore the file format (gunzip unpacks zipped files)</span>
gunzip<span class="w"> </span>-c<span class="w"> </span>Exome_Norm_R1.fastq.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>less<span class="w"> </span>-SN

<span class="c1"># The format is nicely described here:</span>
<span class="c1"># https://en.wikipedia.org/wiki/FASTQ_format</span>
<span class="c1"># Read through the fastq format to get an understanding of</span>
<span class="c1"># base qualities from illumina DNA sequencing data.</span>
<span class="c1"># The format will be discussed tomorrow during the lab intro.</span>
</code></pre></div>
<h2 id="list-of-preinstalled-bioinformatic-tools">List of preinstalled bioinformatic tools<a class="headerlink" href="#list-of-preinstalled-bioinformatic-tools" title="Permanent link">&para;</a></h2>
<h3 id="bam-readcount">bam-readcount<a class="headerlink" href="#bam-readcount" title="Permanent link">&para;</a></h3>
<p><a href="https://github.com/genome/bam-readcount">bam-readcount</a> is a program for determing read support for individual variants (SNVs and Indels only). </p>
<h3 id="bcftools">BCFtools<a class="headerlink" href="#bcftools" title="Permanent link">&para;</a></h3>
<p><a href="https://samtools.github.io/bcftools/">BCFtools</a> is an open source program for variant calling and manipulating files in Variant Call Format (VCF) or Binary Variant Call Format (BCF). </p>
<h3 id="bwa">BWA<a class="headerlink" href="#bwa" title="Permanent link">&para;</a></h3>
<p><a href="http://bio-bwa.sourceforge.net">BWA</a> is a popular DNA alignment tool used for mapping sequences to a reference genome. It is available under an open source GPLv3 license.</p>
<h3 id="cnvkit">CNVkit<a class="headerlink" href="#cnvkit" title="Permanent link">&para;</a></h3>
<p><a href="https://cnvkit.readthedocs.io/en/stable/">CNVkit</a> is a python based copy number caller designed for use with hybrid capture. It will not be applied during the course but is a frequently used tool for CNV-analysis of copy-number data.</p>
<h3 id="delly">Delly<a class="headerlink" href="#delly" title="Permanent link">&para;</a></h3>
<p><a href="https://github.com/dellytools/delly">Delly</a> is a structural variant caller developed at EMBL. It uses paired-ends, split-reads and read-depth to sensitively and accurately delineate genomic rearrangements throughout the genome. </p>
<h3 id="fastqc">FastQC<a class="headerlink" href="#fastqc" title="Permanent link">&para;</a></h3>
<p><a href="https://github.com/s-andrews/FastQC">FastQC</a> is a quality control program for raw sequencing data. </p>
<h3 id="gatk-4">GATK 4<a class="headerlink" href="#gatk-4" title="Permanent link">&para;</a></h3>
<p><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036194592-Getting-started-with-GATK4">GATK</a> is a toolkit developed by the broad institute focused primarily on variant discovery and genotyping. It is open source, hosted on github, and available under a BSD 3-clause license. </p>
<h3 id="gffcompare">Gffcompare<a class="headerlink" href="#gffcompare" title="Permanent link">&para;</a></h3>
<p><a href="https://ccb.jhu.edu/software/stringtie/gffcompare.shtml">Gffcompare</a> is a program that is used to perform operations on general feature format (GFF) and general transfer format (GTF) files. </p>
<h3 id="hisat2">HISAT2<a class="headerlink" href="#hisat2" title="Permanent link">&para;</a></h3>
<p><a href="http://daehwankimlab.github.io/hisat2/">HISAT2</a> is a graph based alignment algorithm devoloped at Johns Hopkins University. It is heavily used in the bioinformatics community for RNAseq based alignments. </p>
<h3 id="kallisto">Kallisto<a class="headerlink" href="#kallisto" title="Permanent link">&para;</a></h3>
<p><a href="https://pachterlab.github.io/kallisto/">Kallisto</a> is a kmer-based alignment algorithm used for quantifying transcripts in RNAseq data.</p>
<h3 id="mosdepth">mosdepth<a class="headerlink" href="#mosdepth" title="Permanent link">&para;</a></h3>
<p><a href="https://github.com/brentp/mosdepth">mosdepth</a> is a program for determining depth in sequencing data. </p>
<h3 id="multiqc">MultiQC<a class="headerlink" href="#multiqc" title="Permanent link">&para;</a></h3>
<p><a href="https://github.com/ewels/MultiQC">MultiQC</a> is a tool to create a single report with interactive plots for multiple bioinformatics analyses across many samples.</p>
<h3 id="picard">PICARD<a class="headerlink" href="#picard" title="Permanent link">&para;</a></h3>
<p><a href="http://broadinstitute.github.io/picard/">PICARD</a> is a set of java based tools developed by the Broad institute. It is very useful for manipulating next generation sequencing data and is available under an open source MIT license. </p>
<ul>
<li>Try to answer the following:<ul>
<li>Who is Picard?<ul>
<li>Clue: "TNG"</li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="pizzly">Pizzly<a class="headerlink" href="#pizzly" title="Permanent link">&para;</a></h3>
<p><a href="https://github.com/pmelsted/pizzly">Pizzly</a> is a fusion detection algorithm which uses output from Kallisto. </p>
<h3 id="sambamba">Sambamba<a class="headerlink" href="#sambamba" title="Permanent link">&para;</a></h3>
<p><a href="https://lomereiter.github.io/sambamba/">Sambamba</a> is a high performance alternative to samtools and provides a subset of samtools functionality. </p>
<h3 id="samtools">Samtools<a class="headerlink" href="#samtools" title="Permanent link">&para;</a></h3>
<p><a href="https://samtools.github.io">Samtools</a> is a software package based in C which provies utilities for manipulating alignment files (SAM/BAM/CRAM). It is open source, available on github, and is under an MIT license. </p>
<h3 id="seqtk">seqtk<a class="headerlink" href="#seqtk" title="Permanent link">&para;</a></h3>
<p><a href="https://github.com/lh3/seqtk">Seqtk</a> is a lighweight tool for processing FASTQ and FASTA files. We will use seqtk to subset RNA-seq fastq files to more quickly run the fusion alignment module.</p>
<h3 id="strelka">Strelka<a class="headerlink" href="#strelka" title="Permanent link">&para;</a></h3>
<p><a href="https://github.com/Illumina/strelka">Strelka</a> is a germline and somatic variant caller developed by illumina and available under an open source GPLv3 license. </p>
<h3 id="stringtie">StringTie<a class="headerlink" href="#stringtie" title="Permanent link">&para;</a></h3>
<p><a href="https://ccb.jhu.edu/software/stringtie/">StringTie</a> is a software program to perform transcript assembly and quantification of RNAseq data. </p>
<h3 id="varscan">Varscan<a class="headerlink" href="#varscan" title="Permanent link">&para;</a></h3>
<p><a href="http://dkoboldt.github.io/varscan/">Varscan</a> is a java program designed to call variants in sequencing data. It was developed at the Genome Institute at Washington University and is hosted 
on github. </p>
<h3 id="vcf-annotation-tools">vcf-annotation-tools<a class="headerlink" href="#vcf-annotation-tools" title="Permanent link">&para;</a></h3>
<p><a href="https://github.com/griffithlab/VAtools">VCF Annotation Tools</a> is a python package that includes several tools to annotate VCF files with data from other tools. We will be using this for the purpose of adding bam readcounts to the vcf files.</p>
<h3 id="vep">VEP<a class="headerlink" href="#vep" title="Permanent link">&para;</a></h3>
<p><a href="https://www.ensembl.org/info/docs/tools/vep/index.html">VEP</a> is a variant annotation tool developed by ensembl and written in perl. By default VEP will perform annotations by making web-based API queries however it is much faster to have a local copy of cache and fasta files. The AWS AMI image we’re using already has these files for hg38 in the directory ~/workspace/vep_cache as they can take a bit of time to download. </p>
<h3 id="vt">vt<a class="headerlink" href="#vt" title="Permanent link">&para;</a></h3>
<p><a href="https://github.com/atks/vt">vt</a> is a variant tool set that discovers short variants from Next Generation Sequencing data. We will use this for the purpose of splitting multi-allelic variants.</p>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2023, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.path", "content.code.annotate", "content.tabs.link", "navigation.indexes", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "search.highlight", "search.share", "search.suggest"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.51198bba.min.js"></script>
      
    
  </body>
</html>