
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.2.3, mkdocs-material-8.1.4">
    
    
      
        <title>Processing of DNA sequencing data - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bb3983ee.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.e6a45f82.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bioinformatic-processing-of-dna-data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Processing of DNA sequencing data
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Lab Session" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_1" type="checkbox" id="__nav_5_1" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_1">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 1" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_2" type="checkbox" id="__nav_5_2" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5_2">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 2" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Processing of DNA sequencing data
      </a>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_3" type="checkbox" id="__nav_5_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_3">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 3" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_4" type="checkbox" id="__nav_5_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_4">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 4" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../analysis_of_cnv/" class="md-nav__link">
        Copy Number Variations
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                

<h1 id="bioinformatic-processing-of-dna-data">Bioinformatic processing of DNA data<a class="headerlink" href="#bioinformatic-processing-of-dna-data" title="Permanent link">&para;</a></h1>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/inputs/data/fastq

<span class="c1"># list all files</span>
tree

<span class="c1"># view the exome normal sample data files</span>
tree Exome_Norm

<span class="c1"># view the RNA-seq tumor sample data files.</span>
tree RNAseq_Tumor
</code></pre></div>
<h1 id="investigate-the-raw-data-files-fastq-files">Investigate the raw data files (fastq-files)<a class="headerlink" href="#investigate-the-raw-data-files-fastq-files" title="Permanent link">&para;</a></h1>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/inputs/data/fastq/Exome_Tumor

<span class="c1"># show the first ten lines of the Exome Tumor fastq files</span>
<span class="c1"># note this is how paird-end data is stored. </span>
<span class="c1"># Read 1</span>
zcat Exome_Tumor/Exome_Tumor_R1.fastq.gz <span class="p">|</span> head
<span class="c1"># Read 2</span>
zcat Exome_Tumor/Exome_Tumor_R2.fastq.gz <span class="p">|</span> head

<span class="c1"># This wikipedia file gives a very nice overview of fastq-files and Illumina base qualities</span>
<span class="c1"># https://en.wikipedia.org/wiki/FASTQ_format</span>
zcat Exome_Tumor/Exome_Tumor_R1.fastq.gz <span class="p">|</span> head -n <span class="m">8</span>

<span class="c1"># what do R1 and R2 refer to? What is the length of each read?</span>
zcat Exome_Tumor/Exome_Tumor_R1.fastq.gz <span class="p">|</span> head -n <span class="m">2</span> <span class="p">|</span> tail -n <span class="m">1</span> <span class="p">|</span> wc

<span class="c1"># how many lines are there in the Exome_Tumor file</span>
zcat Exome_Tumor/Exome_Tumor_R1.fastq.gz <span class="p">|</span> wc -l <span class="c1"># There are: 33,326,620</span>

<span class="c1"># how many paired reads or fragments are there then?</span>
expr <span class="m">33326620</span> / <span class="m">4</span> <span class="c1"># There are: 8,331,655 paired end reads</span>

<span class="c1"># how many total bases of data are in the Exome Tumor data set?</span>
<span class="nb">echo</span> <span class="s2">&quot;8331655 * (101 * 2)&quot;</span> <span class="p">|</span> bc <span class="c1"># There are: 1,682,994,310 bases of data</span>

<span class="c1"># how many total bases when expressed as &quot;gigabases&quot; (specify 2 decimal points using `scale`)</span>
<span class="nb">echo</span> <span class="s2">&quot;scale=2; (8331655 * (101 * 2))/1000000000&quot;</span> <span class="p">|</span> bc <span class="c1"># There are: 1.68 Gbp of data</span>

<span class="c1"># what is the average coverage we expect to achieve with this much data for the exome region targeted?</span>

<span class="c1"># first determine the size of our exome regions (answer = 6683920). </span>
cat /workspace/inputs/references/exome/exome_regions.bed <span class="p">|</span> perl -ne <span class="s1">&#39;chomp; @l=split(&quot;\t&quot;, $_); $size += $l[2]-$l[1]; if (eof){print &quot;size = $size\n&quot;}&#39;</span> 

<span class="c1"># now determine the average coverage of these positions by our bases of data</span>
<span class="nb">echo</span> <span class="s2">&quot;scale=2; (8331655 * (101 * 2))/6683920&quot;</span> <span class="p">|</span> bc <span class="c1"># Average covered expected = 251.79x</span>

<span class="c1"># what is the fundamental assumption of this calculation that is at least partially not true? What effect will this have on the observed coverage?</span>
</code></pre></div>
<h2 id="run-fastqc-to-check-the-quality-of-the-fastq-files">Run fastqc to check the quality of the fastq-files<a class="headerlink" href="#run-fastqc-to-check-the-quality-of-the-fastq-files" title="Permanent link">&para;</a></h2>
<p>Have a look <a href="https://www.youtube.com/watch?v=lUk5Ju3vCDM">here</a> for a short tutorial on the fastqc output.
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/inputs/data/fastq

fastqc Exome_Norm/Exome_Norm*.fastq.gz
fastqc Exome_Tumor/Exome_Tumor*.fastq.gz
tree

fastqc RNAseq_Norm/RNAseq_Norm*.fastq.gz
fastqc RNAseq_Tumor/RNAseq_Tumor*.fastq.gz
tree
</code></pre></div></p>
<h2 id="run-multiqc">Run MultiQC<a class="headerlink" href="#run-multiqc" title="Permanent link">&para;</a></h2>
<p>We will run mutliQC to provide a combined report of the fastQC data. </p>
<p>More info on MultiQC is available <a href="https://multiqc.info">here</a>.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/inputs
<span class="c1">#Dont do if directory exist</span>
<span class="c1">#mkdir qc</span>
<span class="nb">cd</span> qc
multiqc ~/workspace/inputs/data/fastq/
tree
<span class="c1"># Download MultiQC output to the local computer, open the .html in you favourite browser.</span>
scp -ri ~/PEM_KEY_ID.pem ubuntu@AWS_ADDRESS_HERE:~/workspace/inputs/qc/multiqc* .
</code></pre></div>
<h2 id="run-bwa">Run BWA<a class="headerlink" href="#run-bwa" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="c1"># Run bwa mem using the above information</span>

<span class="nb">cd</span> ~/workspace/align
<span class="c1"># Runtime: ~ 4min</span>
bwa mem -t <span class="m">2</span> -Y -R <span class="s2">&quot;@RG\tID:Exome_Norm\tPL:ILLUMINA\tPU:C1TD1ACXX-CGATGT.7\tLB:exome_norm_lib1\tSM:HCC1395BL_DNA&quot;</span> -o ~/workspace/align/Exome_Norm.sam ~/workspace/inputs/references/genome/ref_genome.fa ~/workspace/inputs/data/fastq/Exome_Norm/Exome_Norm_R1.fastq.gz ~/workspace/inputs/data/fastq/Exome_Norm/Exome_Norm_R2.fastq.gz
<span class="c1"># Runtime: ~ 4min</span>
bwa mem -t <span class="m">2</span> -Y -R <span class="s2">&quot;@RG\tID:Exome_Tumor\tPL:ILLUMINA\tPU:C1TD1ACXX-ATCACG.7\tLB:exome_tumor_lib1\tSM:HCC1395_DNA&quot;</span> -o ~/workspace/align/Exome_Tumor.sam ~/workspace/inputs/references/genome/ref_genome.fa ~/workspace/inputs/data/fastq/Exome_Tumor/Exome_Tumor_R1.fastq.gz ~/workspace/inputs/data/fastq/Exome_Tumor/Exome_Tumor_R2.fastq.gz
</code></pre></div>
<h2 id="convert-sam-to-bam-format">Convert sam to bam format<a class="headerlink" href="#convert-sam-to-bam-format" title="Permanent link">&para;</a></h2>
<p><div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align
<span class="c1"># Runtime: ~30s</span>
samtools view -@ <span class="m">2</span> -h -b -o Exome_Norm.bam Exome_Norm.sam
<span class="c1"># Runtime: ~45s</span>
samtools view -@ <span class="m">2</span> -h -b -o Exome_Tumor.bam Exome_Tumor.sam

<span class="c1"># Have a look at the bam header and try to figure out what type of information that is in the bam file:</span>
samtools view -H Exome_Tumor.bam <span class="p">|</span> less -SN 
<span class="c1"># Have a look at the bam file data and try to figure out what type of information that is in the bam file:</span>
samtools view Exome_Tumor.bam <span class="p">|</span> less -SN 
</code></pre></div>
Detailed information on the sam format can be found <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">HERE</a> .</p>
<h2 id="sort-bam-files">Sort bam files<a class="headerlink" href="#sort-bam-files" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align
<span class="c1"># Runtime: ~ 4min</span>
java -Xmx12g -jar <span class="nv">$PICARD</span> SortSam <span class="nv">I</span><span class="o">=</span>Exome_Norm.bam <span class="nv">O</span><span class="o">=</span>Exome_Norm_namesorted.bam <span class="nv">SO</span><span class="o">=</span>queryname
java -Xmx12g -jar <span class="nv">$PICARD</span> SortSam <span class="nv">I</span><span class="o">=</span>Exome_Tumor.bam <span class="nv">O</span><span class="o">=</span>Exome_Tumor_namesorted.bam <span class="nv">SO</span><span class="o">=</span>queryname
</code></pre></div>
<h2 id="mark-duplicates">Mark duplicates<a class="headerlink" href="#mark-duplicates" title="Permanent link">&para;</a></h2>
<p>During library preparation, PCR is performed. Removing the PCR duplicates is necessary for substantial removal of PCR artefacts. Picard MarkDuplicates use the read names and alignment positions to identify PCR duplicates. </p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align
java -Xmx12g -jar <span class="nv">$PICARD</span> MarkDuplicates -I Exome_Norm_namesorted.bam -O Exome_Norm_namesorted_mrkdup.bam -ASSUME_SORT_ORDER queryname -METRICS_FILE Exome_Norm_mrkdup_metrics.txt -QUIET <span class="nb">true</span> -COMPRESSION_LEVEL <span class="m">0</span> -VALIDATION_STRINGENCY LENIENT
java -Xmx12g -jar <span class="nv">$PICARD</span> MarkDuplicates -I Exome_Tumor_namesorted.bam -O Exome_Tumor_namesorted_mrkdup.bam -ASSUME_SORT_ORDER queryname -METRICS_FILE Exome_Tumor_mrkdup_metrics.txt -QUIET <span class="nb">true</span> -COMPRESSION_LEVEL <span class="m">0</span> -VALIDATION_STRINGENCY LENIENT
</code></pre></div>
<h2 id="position-sort-bam-file">Position sort bam file<a class="headerlink" href="#position-sort-bam-file" title="Permanent link">&para;</a></h2>
<p>For indexing and subsequent steps a position-sorted bam is required. Therefore, we will sort bam file by coordinate.</p>
<div class="highlight"><pre><span></span><code> <span class="nb">cd</span> ~/workspace/align
java -Xmx12g -jar <span class="nv">$PICARD</span> SortSam -I Exome_Norm_namesorted_mrkdup.bam -O Exome_Norm_sorted_mrkdup.bam -SO coordinate
java -Xmx12g -jar <span class="nv">$PICARD</span> SortSam -I Exome_Tumor_namesorted_mrkdup.bam -O Exome_Tumor_sorted_mrkdup.bam -SO coordinate
</code></pre></div>
<h2 id="create-bam-index-for-use-with-gatk-igv-etc">Create bam index for use with GATK, IGV, etc<a class="headerlink" href="#create-bam-index-for-use-with-gatk-igv-etc" title="Permanent link">&para;</a></h2>
<p>In order to efficiently load and search a bam file, downstream applications typically require an index</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align
java -Xmx12g -jar <span class="nv">$PICARD</span> BuildBamIndex -I Exome_Norm_sorted_mrkdup.bam
java -Xmx12g -jar <span class="nv">$PICARD</span> BuildBamIndex -I Exome_Tumor_sorted_mrkdup.bam
</code></pre></div>
<h2 id="indel-realignment">Indel Realignment<a class="headerlink" href="#indel-realignment" title="Permanent link">&para;</a></h2>
<p>Depending on the downstream appoarch something called Indel realignment may be needed. During alignment of the DNA data, each individual read is aligned to the reference sequence individually. This may lead to misaligned reads in e.g. repetitive regions or due to somatic/germline variation. If needed, add the realignment-step here, see docs <a href="https://software.broadinstitute.org/gatk/documentation/article?id=7156">here</a>. In this course the HaplotypeCaller/MuTect2 will be applied to identify germline/somatic variantion. These two variant callers do realignment internally and therefore it is not needed. See the following <a href="https://software.broadinstitute.org/gatk/blog?id=7847">blog</a> for a detailed discussion of this issue. See <a href="https://drive.google.com/drive/folders/1U6Zm_tYn_3yeEgrD1bdxye4SXf5OseIt">here</a> for latest versions of all GATK tutorials.</p>
<h2 id="perform-base-quality-score-recalibration">Perform Base Quality Score Recalibration<a class="headerlink" href="#perform-base-quality-score-recalibration" title="Permanent link">&para;</a></h2>
<p>BQSR stands for Base Quality Score Recalibration. In a nutshell, it is a data pre-processing step that detects systematic errors made by the sequencing machine when it estimates the accuracy of each base call.</p>
<h3 id="calculate-bqsr-table">Calculate BQSR Table<a class="headerlink" href="#calculate-bqsr-table" title="Permanent link">&para;</a></h3>
<p>First, calculate the BQSR table.</p>
<p>NOTE: For exome data, we should modify the below to use the <code>--intervals</code> (<code>-L</code>) option. This excludes off-target sequences and sequences that may be poorly mapped, which have a higher error rate. Including them could lead to a skewed model and bad recalibration.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> BaseRecalibrator -R ~/workspace/inputs/references/genome/ref_genome.fa -I ~/workspace/align/Exome_Norm_sorted_mrkdup.bam -O ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.table --known-sites ~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.dbsnp138.vcf.gz --known-sites ~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.known_indels.vcf.gz --known-sites ~/workspace/inputs/references/gatk/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz --preserve-qscores-less-than <span class="m">6</span> --disable-bam-index-caching <span class="nv">$GATK_REGIONS</span>

gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> BaseRecalibrator -R ~/workspace/inputs/references/genome/ref_genome.fa -I ~/workspace/align/Exome_Tumor_sorted_mrkdup.bam -O ~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.table --known-sites ~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.dbsnp138.vcf.gz --known-sites ~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.known_indels.vcf.gz --known-sites ~/workspace/inputs/references/gatk/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz --preserve-qscores-less-than <span class="m">6</span> --disable-bam-index-caching <span class="nv">$GATK_REGIONS</span>
</code></pre></div>
<h3 id="apply-bqsr">Apply BQSR<a class="headerlink" href="#apply-bqsr" title="Permanent link">&para;</a></h3>
<p>Now, apply the BQSR table to bam files.
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> ApplyBQSR -R ~/workspace/inputs/references/genome/ref_genome.fa -I ~/workspace/align/Exome_Norm_sorted_mrkdup.bam -O ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam --bqsr-recal-file ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.table --preserve-qscores-less-than <span class="m">6</span> --static-quantized-quals <span class="m">10</span> --static-quantized-quals <span class="m">20</span> --static-quantized-quals <span class="m">30</span>

gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> ApplyBQSR -R ~/workspace/inputs/references/genome/ref_genome.fa -I ~/workspace/align/Exome_Tumor_sorted_mrkdup.bam -O ~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam --bqsr-recal-file ~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.table --preserve-qscores-less-than <span class="m">6</span> --static-quantized-quals <span class="m">10</span> --static-quantized-quals <span class="m">20</span> --static-quantized-quals <span class="m">30</span>
</code></pre></div></p>
<p>Create an index for these new bams</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align
java -Xmx12g -jar <span class="nv">$PICARD</span> BuildBamIndex <span class="nv">I</span><span class="o">=</span>Exome_Norm_sorted_mrkdup_bqsr.bam
java -Xmx12g -jar <span class="nv">$PICARD</span> BuildBamIndex <span class="nv">I</span><span class="o">=</span>Exome_Tumor_sorted_mrkdup_bqsr.bam
</code></pre></div>
<h2 id="clean-up-un-needed-sambam-files">Clean up un-needed sam/bam files<a class="headerlink" href="#clean-up-un-needed-sambam-files" title="Permanent link">&para;</a></h2>
<p>Keep final sorted, duplicated marked, bqsr bam/bai/table files and mrkdup.txt files. Delete everything else.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align
mkdir final
mv *_sorted_mrkdup_bqsr.* final/
mv *.txt final/

rm *.sam
rm *.bam
rm *.bai

mv final/* .
rmdir final/
</code></pre></div>
<h2 id="run-samtools-flagstat">Run Samtools flagstat<a class="headerlink" href="#run-samtools-flagstat" title="Permanent link">&para;</a></h2>
<p>Samtools flagstat provides QC-metrics after aligning the reads, e.g. the fraction of read-pairs that map to different chromosomes (which should be limited).</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align/
samtools flagstat ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam &gt; ~/workspace/align/Exome_Norm_flagstat.txt
samtools flagstat ~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam &gt; ~/workspace/align/Exome_Tumor_flagstat.txt
</code></pre></div>
<h2 id="run-various-picard-collectmetrics-tools">Run various picard CollectMetrics tools<a class="headerlink" href="#run-various-picard-collectmetrics-tools" title="Permanent link">&para;</a></h2>
<p>Picard is a widely used QC tool in genomics and it is named after a famous <a href="https://en.wikipedia.org/wiki/Jean-Luc_Picard">Star Trek Captain</a>.</p>
<p>Description of the output of picard can be found <a href="http://broadinstitute.github.io/picard/">here</a>. </p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align/
java -Xmx12g -jar <span class="nv">$PICARD</span> CollectInsertSizeMetrics -I ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam -O ~/workspace/align/Exome_Norm_insert_size_metrics.txt -H ~/workspace/align/Exome_Norm_insert_size_metrics.pdf
java -Xmx12g -jar <span class="nv">$PICARD</span> CollectInsertSizeMetrics -I ~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam -O ~/workspace/align/Exome_Tumor_insert_size_metrics.txt -H ~/workspace/align/Exome_Tumor_insert_size_metrics.pdf

java -Xmx12g -jar <span class="nv">$PICARD</span> CollectAlignmentSummaryMetrics -I ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam -O ~/workspace/align/Exome_Norm_alignment_metrics.txt -R ~/workspace/inputs/references/genome/ref_genome.fa
java -Xmx12g -jar <span class="nv">$PICARD</span> CollectAlignmentSummaryMetrics -I ~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam -O ~/workspace/align/Exome_Tumor_exome_alignment_metrics.txt -R ~/workspace/inputs/references/genome/ref_genome.fa

<span class="c1">#Picard CollectHsMetrics</span>
<span class="c1">#Exome Only</span>
java -Xmx12g -jar <span class="nv">$PICARD</span> CollectHsMetrics -I ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam -O ~/workspace/align/Exome_Norm_hs_metrics.txt -R ~/workspace/inputs/references/genome/ref_genome.fa -BI ~/workspace/inputs/references/exome/probe_regions.bed.interval_list -TI ~/workspace/inputs/references/exome/exome_regions.bed.interval_list
java -Xmx12g -jar <span class="nv">$PICARD</span> CollectHsMetrics -I ~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam -O ~/workspace/align/Exome_Tumor_hs_metrics.txt -R ~/workspace/inputs/references/genome/ref_genome.fa -BI ~/workspace/inputs/references/exome/probe_regions.bed.interval_list -TI ~/workspace/inputs/references/exome/exome_regions.bed.interval_list
</code></pre></div>
<h2 id="run-fastqc">Run FastQC<a class="headerlink" href="#run-fastqc" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align
fastqc -t <span class="m">8</span> Exome_Norm_sorted_mrkdup_bqsr.bam
fastqc -t <span class="m">8</span> Exome_Tumor_sorted_mrkdup_bqsr.bam
tree
</code></pre></div>
<h2 id="run-multiqc-to-produce-a-final-report">Run MultiQC to produce a final report<a class="headerlink" href="#run-multiqc-to-produce-a-final-report" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/align
<span class="c1">#Dont do if directory exist</span>
<span class="c1">#mkdir post_align_qc</span>
<span class="nb">cd</span> post_align_qc
multiqc ~/workspace/align/
tree

<span class="c1"># Download MultiQC output to the local computer, open the .html in you favourite browser.</span>
<span class="c1">#Discuss the MultiQC output with a fellow student</span>
scp -ri ~/PEM_KEY_ID.pem ubuntu@AWS_ADDRESS_HERE:~/workspace/align/post_align_qc/multiqc* .
</code></pre></div>

              
            </article>
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" data-md-state="hidden">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../annotation_files/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Annotation" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Annotation
            </div>
          </div>
        </a>
      
      
        
        <a href="../intro_igv/" class="md-footer__link md-footer__link--next" aria-label="Next: Introduction to IGV" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Introduction to IGV
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2021, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "navigation.sections", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest"], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.361d90f1.min.js"}</script>
    
    
      <script src="../../assets/javascripts/bundle.289a2a4b.min.js"></script>
      
    
  </body>
</html>