
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Project documentation with Markdown.">
      
      
      
      
        <link rel="prev" href="../annotation_files/">
      
      
        <link rel="next" href="../intro_igv/">
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.1.6">
    
    
      
        <title>Processing of DNA sequencing data - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.ded33207.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.a0c5b2b5.min.css">
      
      
  
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
  
  <style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M64 32C28.7 32 0 60.7 0 96v320c0 35.3 28.7 64 64 64h224V368c0-26.5 21.5-48 48-48h112V96c0-35.3-28.7-64-64-64H64zm384 320H336c-8.8 0-16 7.2-16 16v112l32-32 64-64 32-32z"/></svg>');--md-admonition-icon--abstract:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M96 0C43 0 0 43 0 96v320c0 53 43 96 96 96h320c17.7 0 32-14.3 32-32s-14.3-32-32-32v-64c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H96zm0 384h256v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"/></svg>');--md-admonition-icon--info:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-40-176h24v-64h-24c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24h-80c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"/></svg>');--md-admonition-icon--tip:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M480 32c0-12.9-7.8-24.6-19.8-29.6S434.5.2 425.3 9.3L381.7 53c-48 48-113.1 75-181 75H64c-35.3 0-64 28.7-64 64v96c0 35.3 28.7 64 64 64v128c0 17.7 14.3 32 32 32h64c17.7 0 32-14.3 32-32V352h8.7c67.9 0 133 27 181 75l43.6 43.6c9.2 9.2 22.9 11.9 34.9 6.9s19.8-16.6 19.8-29.6V300.4c18.6-8.8 32-32.5 32-60.4s-13.4-51.6-32-60.4V32zm-64 76.7v262.6C357.2 317.8 280.5 288 200.7 288H192v-96h8.7c79.8 0 156.5-29.8 215.3-83.3z"/></svg>');--md-admonition-icon--success:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7l233.4-233.3c12.5-12.5 32.8-12.5 45.3 0z"/></svg>');--md-admonition-icon--question:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-86.2-346.7c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1 0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6-13.3 0-24-10.7-24-24v-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1 0-8.4-6.8-15.1-15.1-15.1h-58.3c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1-64 0z"/></svg>');--md-admonition-icon--warning:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7.2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8.2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24v112c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0-64 0 32 32 0 1 0 64 0z"/></svg>');--md-admonition-icon--failure:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.1 52.4 442.6 6.5c-1.9-3.9-6.1-6.5-10.5-6.5s-8.5 2.6-10.4 6.5l-16.5 45.9-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4 0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2 0-4.6-3-8.9-7.2-10.4l-45.6-16.8zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9c-22-8-45.8-12.3-70.5-12.3C93.1 96 0 189.1 0 304s93.1 208 208 208 208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg>');--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M416 398.9c58.5-41.1 96-104.1 96-174.9C512 100.3 397.4 0 256 0S0 100.3 0 224c0 70.7 37.5 133.8 96 174.9V464c0 26.5 21.5 48 48 48h48v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h64v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h48c26.5 0 48-21.5 48-48v-65.1zM96 256a64 64 0 1 1 128 0 64 64 0 1 1-128 0zm256-64a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"/></svg>');--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M320 0c17.7 0 32 14.3 32 32v64h120c39.8 0 72 32.2 72 72v272c0 39.8-32.2 72-72 72H168c-39.8 0-72-32.2-72-72V168c0-39.8 32.2-72 72-72h120V32c0-17.7 14.3-32 32-32zM208 384c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zM264 256a40 40 0 1 0-80 0 40 40 0 1 0 80 0zm152 40a40 40 0 1 0 0-80 40 40 0 1 0 0 80zM48 224h16v192H48c-26.5 0-48-21.5-48-48v-96c0-26.5 21.5-48 48-48zm544 0c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48h-16V224h16z"/></svg>');--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M288 0H128c-17.7 0-32 14.3-32 32s14.3 32 32 32v132.8c0 11.8-3.3 23.5-9.5 33.5L10.3 406.2C3.6 417.2 0 429.7 0 442.6 0 480.9 31.1 512 69.4 512h309.2c38.3 0 69.4-31.1 69.4-69.4 0-12.8-3.6-25.4-10.3-36.4L329.5 230.4c-6.2-10.1-9.5-21.7-9.5-33.5V64c17.7 0 32-14.3 32-32S337.7 0 320 0h-32zm-96 196.8V64h64v132.8c0 23.7 6.6 46.9 19 67.1l34.5 56.1h-171l34.5-56.1c12.4-20.2 19-43.4 19-67.1z"/></svg>');--md-admonition-icon--quote:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 216C0 149.7 53.7 96 120 96h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V216zm256 0c0-66.3 53.7-120 120-120h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64h-64c-35.3 0-64-28.7-64-64V216z"/></svg>');}</style>


    
    
    
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bioinformatic-processing-of-dna-data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Processing of DNA sequencing data
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      Course Details
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../schedule/" class="md-tabs__link">
      Schedule
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../aws/aws_ec2_instances/" class="md-tabs__link">
      AWS Instances Login
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../lectures/lectures/" class="md-tabs__link">
      Lectures
    </a>
  </li>

      
        
  
  
    
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../basic_unix/" class="md-tabs__link md-tabs__link--active">
        Lab Session
      </a>
    </li>
  

  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_5_1" id="__nav_5_1_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ggplot2/" class="md-nav__link">
        ggplot tutorial
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_2" checked>
      
      
      
        <label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_2_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Processing of DNA sequencing data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Processing of DNA sequencing data
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#investigate-the-raw-data-files-fastq-files" class="md-nav__link">
    Investigate the raw data files (fastq-files)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-fastqc-to-check-the-quality-of-the-fastq-files" class="md-nav__link">
    Run fastqc to check the quality of the fastq-files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-multiqc" class="md-nav__link">
    Run MultiQC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-bwa" class="md-nav__link">
    Run BWA
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#convert-sam-to-bam-format" class="md-nav__link">
    Convert sam to bam format
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sort-bam-files" class="md-nav__link">
    Sort bam files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mark-duplicates" class="md-nav__link">
    Mark duplicates
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#position-sort-bam-file" class="md-nav__link">
    Position sort bam file
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-bam-index-for-use-with-gatk-igv-etc" class="md-nav__link">
    Create bam index for use with GATK, IGV, etc
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#indel-realignment" class="md-nav__link">
    Indel Realignment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#perform-base-quality-score-recalibration" class="md-nav__link">
    Perform Base Quality Score Recalibration
  </a>
  
    <nav class="md-nav" aria-label="Perform Base Quality Score Recalibration">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#calculate-bqsr-table" class="md-nav__link">
    Calculate BQSR Table
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#apply-bqsr" class="md-nav__link">
    Apply BQSR
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#clean-up-un-needed-sambam-files" class="md-nav__link">
    Clean up un-needed sam/bam files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-samtools-flagstat" class="md-nav__link">
    Run Samtools flagstat
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-various-picard-collectmetrics-tools" class="md-nav__link">
    Run various picard CollectMetrics tools
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-fastqc" class="md-nav__link">
    Run FastQC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-multiqc-to-produce-a-final-report" class="md-nav__link">
    Run MultiQC to produce a final report
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_4" >
      
      
      
        <label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../structural_variation/" class="md-nav__link">
        Structural Variants
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../analysis_of_cnv/" class="md-nav__link">
        Copy Number Analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
          Day 5
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          Day 5
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../joint_curation/" class="md-nav__link">
        Joint Curation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../genomic_alteration_interpretation_exe/" class="md-nav__link">
        Genomic Alterations Interpretation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#investigate-the-raw-data-files-fastq-files" class="md-nav__link">
    Investigate the raw data files (fastq-files)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-fastqc-to-check-the-quality-of-the-fastq-files" class="md-nav__link">
    Run fastqc to check the quality of the fastq-files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-multiqc" class="md-nav__link">
    Run MultiQC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-bwa" class="md-nav__link">
    Run BWA
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#convert-sam-to-bam-format" class="md-nav__link">
    Convert sam to bam format
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sort-bam-files" class="md-nav__link">
    Sort bam files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mark-duplicates" class="md-nav__link">
    Mark duplicates
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#position-sort-bam-file" class="md-nav__link">
    Position sort bam file
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-bam-index-for-use-with-gatk-igv-etc" class="md-nav__link">
    Create bam index for use with GATK, IGV, etc
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#indel-realignment" class="md-nav__link">
    Indel Realignment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#perform-base-quality-score-recalibration" class="md-nav__link">
    Perform Base Quality Score Recalibration
  </a>
  
    <nav class="md-nav" aria-label="Perform Base Quality Score Recalibration">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#calculate-bqsr-table" class="md-nav__link">
    Calculate BQSR Table
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#apply-bqsr" class="md-nav__link">
    Apply BQSR
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#clean-up-un-needed-sambam-files" class="md-nav__link">
    Clean up un-needed sam/bam files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-samtools-flagstat" class="md-nav__link">
    Run Samtools flagstat
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-various-picard-collectmetrics-tools" class="md-nav__link">
    Run various picard CollectMetrics tools
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-fastqc" class="md-nav__link">
    Run FastQC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-multiqc-to-produce-a-final-report" class="md-nav__link">
    Run MultiQC to produce a final report
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="bioinformatic-processing-of-dna-data">Bioinformatic processing of DNA data<a class="headerlink" href="#bioinformatic-processing-of-dna-data" title="Permanent link">&para;</a></h1>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/data/fastq

<span class="c1"># list all files</span>
tree

<span class="c1"># view the exome normal sample data files</span>
tree<span class="w"> </span>Exome_Norm

<span class="c1"># view the RNA-seq tumor sample data files.</span>
tree<span class="w"> </span>RNAseq_Tumor
</code></pre></div>
<h2 id="investigate-the-raw-data-files-fastq-files">Investigate the raw data files (fastq-files)<a class="headerlink" href="#investigate-the-raw-data-files-fastq-files" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/data/fastq/Exome_Tumor

<span class="c1"># show the first ten lines of the Exome Tumor fastq files</span>
<span class="c1"># note this is how paird-end data is stored. </span>
<span class="c1"># Read 1</span>
zcat<span class="w"> </span>Exome_Tumor/Exome_Tumor_R1.fastq.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>head
<span class="c1"># Read 2</span>
zcat<span class="w"> </span>Exome_Tumor/Exome_Tumor_R2.fastq.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>head

<span class="c1"># This wikipedia file gives a very nice overview of fastq-files and Illumina base qualities</span>
<span class="c1"># https://en.wikipedia.org/wiki/FASTQ_format</span>
zcat<span class="w"> </span>Exome_Tumor/Exome_Tumor_R1.fastq.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">8</span>

<span class="c1"># what do R1 and R2 refer to? What is the length of each read?</span>
zcat<span class="w"> </span>Exome_Tumor/Exome_Tumor_R1.fastq.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>tail<span class="w"> </span>-n<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>wc

<span class="c1"># how many lines are there in the Exome_Tumor file</span>
zcat<span class="w"> </span>Exome_Tumor/Exome_Tumor_R1.fastq.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>wc<span class="w"> </span>-l<span class="w"> </span><span class="c1"># There are: 33,326,620</span>

<span class="c1"># how many paired reads or fragments are there then?</span>
expr<span class="w"> </span><span class="m">33326620</span><span class="w"> </span>/<span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="c1"># There are: 8,331,655 paired end reads</span>

<span class="c1"># how many total bases of data are in the Exome Tumor data set?</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;8331655 * (101 * 2)&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>bc<span class="w"> </span><span class="c1"># There are: 1,682,994,310 bases of data</span>

<span class="c1"># how many total bases when expressed as &quot;gigabases&quot; (specify 2 decimal points using `scale`)</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;scale=2; (8331655 * (101 * 2))/1000000000&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>bc<span class="w"> </span><span class="c1"># There are: 1.68 Gbp of data</span>

<span class="c1"># what is the average coverage we expect to achieve with this much data for the exome region targeted?</span>

<span class="c1"># first determine the size of our exome regions (answer = 6683920). </span>
cat<span class="w"> </span>/workspace/inputs/references/exome/exome_regions.bed<span class="w"> </span><span class="p">|</span><span class="w"> </span>perl<span class="w"> </span>-ne<span class="w"> </span><span class="s1">&#39;chomp; @l=split(&quot;\t&quot;, $_); $size += $l[2]-$l[1]; if (eof){print &quot;size = $size\n&quot;}&#39;</span><span class="w"> </span>

<span class="c1"># now determine the average coverage of these positions by our bases of data</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;scale=2; (8331655 * (101 * 2))/6683920&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>bc<span class="w"> </span><span class="c1"># Average covered expected = 251.79x</span>

<span class="c1"># what is the fundamental assumption of this calculation that is at least partially not true? What effect will this have on the observed coverage?</span>
</code></pre></div>
<h2 id="run-fastqc-to-check-the-quality-of-the-fastq-files">Run fastqc to check the quality of the fastq-files<a class="headerlink" href="#run-fastqc-to-check-the-quality-of-the-fastq-files" title="Permanent link">&para;</a></h2>
<p>Have a look <a href="https://www.youtube.com/watch?v=lUk5Ju3vCDM">here</a> for a short tutorial on the fastqc output.
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/data/fastq

fastqc<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>Exome_Norm/Exome_Norm*.fastq.gz
fastqc<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>Exome_Tumor/Exome_Tumor*.fastq.gz
tree

fastqc<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>RNAseq_Norm/RNAseq_Norm*.fastq.gz
fastqc<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>RNAseq_Tumor/RNAseq_Tumor*.fastq.gz
tree
</code></pre></div></p>
<h2 id="run-multiqc">Run MultiQC<a class="headerlink" href="#run-multiqc" title="Permanent link">&para;</a></h2>
<p>We will run mutliQC to provide a combined report of the fastQC data. </p>
<p>More info on MultiQC is available <a href="https://multiqc.info">here</a>.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/inputs
<span class="c1">#Dont do if directory exist</span>
<span class="c1">#mkdir qc</span>
<span class="nb">cd</span><span class="w"> </span>qc
multiqc<span class="w"> </span>~/workspace/inputs/data/fastq/
tree
<span class="c1"># Download MultiQC output to the local computer, open the .html in you favourite browser.</span>
scp<span class="w"> </span>-ri<span class="w"> </span>~/PEM_KEY_ID.pem<span class="w"> </span>ubuntu@AWS_ADDRESS_HERE:~/workspace/inputs/qc/multiqc*<span class="w"> </span>.
</code></pre></div>
<h2 id="run-bwa">Run BWA<a class="headerlink" href="#run-bwa" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="c1"># Run bwa mem using the above information</span>

<span class="nb">cd</span><span class="w"> </span>~/workspace/align
<span class="c1"># Runtime: ~ 4min</span>
bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">2</span><span class="w"> </span>-Y<span class="w"> </span>-R<span class="w"> </span><span class="s2">&quot;@RG\tID:Exome_Norm\tPL:ILLUMINA\tPU:C1TD1ACXX-CGATGT.7\tLB:exome_norm_lib1\tSM:HCC1395BL_DNA&quot;</span><span class="w"> </span>-o<span class="w"> </span>~/workspace/align/Exome_Norm.sam<span class="w"> </span>~/workspace/inputs/references/genome/ref_genome.fa<span class="w"> </span>~/workspace/inputs/data/fastq/Exome_Norm/Exome_Norm_R1.fastq.gz<span class="w"> </span>~/workspace/inputs/data/fastq/Exome_Norm/Exome_Norm_R2.fastq.gz
<span class="c1"># Runtime: ~ 4min</span>
bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">2</span><span class="w"> </span>-Y<span class="w"> </span>-R<span class="w"> </span><span class="s2">&quot;@RG\tID:Exome_Tumor\tPL:ILLUMINA\tPU:C1TD1ACXX-ATCACG.7\tLB:exome_tumor_lib1\tSM:HCC1395_DNA&quot;</span><span class="w"> </span>-o<span class="w"> </span>~/workspace/align/Exome_Tumor.sam<span class="w"> </span>~/workspace/inputs/references/genome/ref_genome.fa<span class="w"> </span>~/workspace/inputs/data/fastq/Exome_Tumor/Exome_Tumor_R1.fastq.gz<span class="w"> </span>~/workspace/inputs/data/fastq/Exome_Tumor/Exome_Tumor_R2.fastq.gz
</code></pre></div>
<h2 id="convert-sam-to-bam-format">Convert sam to bam format<a class="headerlink" href="#convert-sam-to-bam-format" title="Permanent link">&para;</a></h2>
<p><div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align
<span class="c1"># Runtime: ~30s</span>
samtools<span class="w"> </span>view<span class="w"> </span>-@<span class="w"> </span><span class="m">2</span><span class="w"> </span>-h<span class="w"> </span>-b<span class="w"> </span>-o<span class="w"> </span>Exome_Norm.bam<span class="w"> </span>Exome_Norm.sam
<span class="c1"># Runtime: ~45s</span>
samtools<span class="w"> </span>view<span class="w"> </span>-@<span class="w"> </span><span class="m">2</span><span class="w"> </span>-h<span class="w"> </span>-b<span class="w"> </span>-o<span class="w"> </span>Exome_Tumor.bam<span class="w"> </span>Exome_Tumor.sam

<span class="c1"># Have a look at the bam header and try to figure out what type of information that is in the bam file:</span>
samtools<span class="w"> </span>view<span class="w"> </span>-H<span class="w"> </span>Exome_Tumor.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>less<span class="w"> </span>-SN<span class="w"> </span>
<span class="c1"># Have a look at the bam file data and try to figure out what type of information that is in the bam file:</span>
samtools<span class="w"> </span>view<span class="w"> </span>Exome_Tumor.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>less<span class="w"> </span>-SN<span class="w"> </span>
</code></pre></div>
Detailed information on the sam format can be found <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">HERE</a> .</p>
<h2 id="sort-bam-files">Sort bam files<a class="headerlink" href="#sort-bam-files" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align
<span class="c1"># Runtime: ~ 4min</span>
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>SortSam<span class="w"> </span><span class="nv">I</span><span class="o">=</span>Exome_Norm.bam<span class="w"> </span><span class="nv">O</span><span class="o">=</span>Exome_Norm_namesorted.bam<span class="w"> </span><span class="nv">SO</span><span class="o">=</span>queryname
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>SortSam<span class="w"> </span><span class="nv">I</span><span class="o">=</span>Exome_Tumor.bam<span class="w"> </span><span class="nv">O</span><span class="o">=</span>Exome_Tumor_namesorted.bam<span class="w"> </span><span class="nv">SO</span><span class="o">=</span>queryname
</code></pre></div>
<h2 id="mark-duplicates">Mark duplicates<a class="headerlink" href="#mark-duplicates" title="Permanent link">&para;</a></h2>
<p>During library preparation, PCR is performed. Removing the PCR duplicates is necessary for substantial removal of PCR artefacts. Picard MarkDuplicates use the read names and alignment positions to identify PCR duplicates. </p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>MarkDuplicates<span class="w"> </span>-I<span class="w"> </span>Exome_Norm_namesorted.bam<span class="w"> </span>-O<span class="w"> </span>Exome_Norm_namesorted_mrkdup.bam<span class="w"> </span>-ASSUME_SORT_ORDER<span class="w"> </span>queryname<span class="w"> </span>-METRICS_FILE<span class="w"> </span>Exome_Norm_mrkdup_metrics.txt<span class="w"> </span>-QUIET<span class="w"> </span><span class="nb">true</span><span class="w"> </span>-COMPRESSION_LEVEL<span class="w"> </span><span class="m">0</span><span class="w"> </span>-VALIDATION_STRINGENCY<span class="w"> </span>LENIENT
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>MarkDuplicates<span class="w"> </span>-I<span class="w"> </span>Exome_Tumor_namesorted.bam<span class="w"> </span>-O<span class="w"> </span>Exome_Tumor_namesorted_mrkdup.bam<span class="w"> </span>-ASSUME_SORT_ORDER<span class="w"> </span>queryname<span class="w"> </span>-METRICS_FILE<span class="w"> </span>Exome_Tumor_mrkdup_metrics.txt<span class="w"> </span>-QUIET<span class="w"> </span><span class="nb">true</span><span class="w"> </span>-COMPRESSION_LEVEL<span class="w"> </span><span class="m">0</span><span class="w"> </span>-VALIDATION_STRINGENCY<span class="w"> </span>LENIENT
</code></pre></div>
<h2 id="position-sort-bam-file">Position sort bam file<a class="headerlink" href="#position-sort-bam-file" title="Permanent link">&para;</a></h2>
<p>For indexing and subsequent steps a position-sorted bam is required. Therefore, we will sort bam file by coordinate.</p>
<div class="highlight"><pre><span></span><code><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>~/workspace/align
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>SortSam<span class="w"> </span>-I<span class="w"> </span>Exome_Norm_namesorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>Exome_Norm_sorted_mrkdup.bam<span class="w"> </span>-SO<span class="w"> </span>coordinate
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>SortSam<span class="w"> </span>-I<span class="w"> </span>Exome_Tumor_namesorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>Exome_Tumor_sorted_mrkdup.bam<span class="w"> </span>-SO<span class="w"> </span>coordinate
</code></pre></div>
<h2 id="create-bam-index-for-use-with-gatk-igv-etc">Create bam index for use with GATK, IGV, etc<a class="headerlink" href="#create-bam-index-for-use-with-gatk-igv-etc" title="Permanent link">&para;</a></h2>
<p>In order to efficiently load and search a bam file, downstream applications typically require an index</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>BuildBamIndex<span class="w"> </span>-I<span class="w"> </span>Exome_Norm_sorted_mrkdup.bam
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>BuildBamIndex<span class="w"> </span>-I<span class="w"> </span>Exome_Tumor_sorted_mrkdup.bam
</code></pre></div>
<h2 id="indel-realignment">Indel Realignment<a class="headerlink" href="#indel-realignment" title="Permanent link">&para;</a></h2>
<p>Depending on the downstream appoarch something called Indel realignment may be needed. During alignment of the DNA data, each individual read is aligned to the reference sequence individually. This may lead to misaligned reads in e.g. repetitive regions or due to somatic/germline variation. If needed, add the realignment-step here, see docs <a href="https://software.broadinstitute.org/gatk/documentation/article?id=7156">here</a>. In this course the HaplotypeCaller/MuTect2 will be applied to identify germline/somatic variantion. These two variant callers do realignment internally and therefore it is not needed. See the following <a href="https://software.broadinstitute.org/gatk/blog?id=7847">blog</a> for a detailed discussion of this issue. See <a href="https://drive.google.com/drive/folders/1U6Zm_tYn_3yeEgrD1bdxye4SXf5OseIt">here</a> for latest versions of all GATK tutorials.</p>
<h2 id="perform-base-quality-score-recalibration">Perform Base Quality Score Recalibration<a class="headerlink" href="#perform-base-quality-score-recalibration" title="Permanent link">&para;</a></h2>
<p>BQSR stands for Base Quality Score Recalibration. In a nutshell, it is a data pre-processing step that detects systematic errors made by the sequencing machine when it estimates the accuracy of each base call.</p>
<h3 id="calculate-bqsr-table">Calculate BQSR Table<a class="headerlink" href="#calculate-bqsr-table" title="Permanent link">&para;</a></h3>
<p>First, calculate the BQSR table.</p>
<p>NOTE: For exome data, we should modify the below to use the <code>--intervals</code> (<code>-L</code>) option. This excludes off-target sequences and sequences that may be poorly mapped, which have a higher error rate. Including them could lead to a skewed model and bad recalibration.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align
gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>BaseRecalibrator<span class="w"> </span>-R<span class="w"> </span>~/workspace/inputs/references/genome/ref_genome.fa<span class="w"> </span>-I<span class="w"> </span>~/workspace/align/Exome_Norm_sorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.table<span class="w"> </span>--known-sites<span class="w"> </span>~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.dbsnp138.vcf.gz<span class="w"> </span>--known-sites<span class="w"> </span>~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.known_indels.vcf.gz<span class="w"> </span>--known-sites<span class="w"> </span>~/workspace/inputs/references/gatk/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz<span class="w"> </span>--preserve-qscores-less-than<span class="w"> </span><span class="m">6</span><span class="w"> </span>--disable-bam-index-caching<span class="w"> </span><span class="nv">$GATK_REGIONS</span>

gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>BaseRecalibrator<span class="w"> </span>-R<span class="w"> </span>~/workspace/inputs/references/genome/ref_genome.fa<span class="w"> </span>-I<span class="w"> </span>~/workspace/align/Exome_Tumor_sorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.table<span class="w"> </span>--known-sites<span class="w"> </span>~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.dbsnp138.vcf.gz<span class="w"> </span>--known-sites<span class="w"> </span>~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.known_indels.vcf.gz<span class="w"> </span>--known-sites<span class="w"> </span>~/workspace/inputs/references/gatk/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz<span class="w"> </span>--preserve-qscores-less-than<span class="w"> </span><span class="m">6</span><span class="w"> </span>--disable-bam-index-caching<span class="w"> </span><span class="nv">$GATK_REGIONS</span>
</code></pre></div>
<h3 id="apply-bqsr">Apply BQSR<a class="headerlink" href="#apply-bqsr" title="Permanent link">&para;</a></h3>
<p>Now, apply the BQSR table to bam files.
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align
gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>ApplyBQSR<span class="w"> </span>-R<span class="w"> </span>~/workspace/inputs/references/genome/ref_genome.fa<span class="w"> </span>-I<span class="w"> </span>~/workspace/align/Exome_Norm_sorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam<span class="w"> </span>--bqsr-recal-file<span class="w"> </span>~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.table<span class="w"> </span>--preserve-qscores-less-than<span class="w"> </span><span class="m">6</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">10</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">20</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">30</span>

gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>ApplyBQSR<span class="w"> </span>-R<span class="w"> </span>~/workspace/inputs/references/genome/ref_genome.fa<span class="w"> </span>-I<span class="w"> </span>~/workspace/align/Exome_Tumor_sorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="w"> </span>--bqsr-recal-file<span class="w"> </span>~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.table<span class="w"> </span>--preserve-qscores-less-than<span class="w"> </span><span class="m">6</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">10</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">20</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">30</span>
</code></pre></div></p>
<p>Create an index for these new bams</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>BuildBamIndex<span class="w"> </span><span class="nv">I</span><span class="o">=</span>Exome_Norm_sorted_mrkdup_bqsr.bam
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>BuildBamIndex<span class="w"> </span><span class="nv">I</span><span class="o">=</span>Exome_Tumor_sorted_mrkdup_bqsr.bam
</code></pre></div>
<h2 id="clean-up-un-needed-sambam-files">Clean up un-needed sam/bam files<a class="headerlink" href="#clean-up-un-needed-sambam-files" title="Permanent link">&para;</a></h2>
<p>Keep final sorted, duplicated marked, bqsr bam/bai/table files and mrkdup.txt files. Delete everything else.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align
mkdir<span class="w"> </span>final
mv<span class="w"> </span>*_sorted_mrkdup_bqsr.*<span class="w"> </span>final/
mv<span class="w"> </span>*.txt<span class="w"> </span>final/

rm<span class="w"> </span>*.sam
rm<span class="w"> </span>*.bam
rm<span class="w"> </span>*.bai

mv<span class="w"> </span>final/*<span class="w"> </span>.
rmdir<span class="w"> </span>final/
</code></pre></div>
<h2 id="run-samtools-flagstat">Run Samtools flagstat<a class="headerlink" href="#run-samtools-flagstat" title="Permanent link">&para;</a></h2>
<p>Samtools flagstat provides QC-metrics after aligning the reads, e.g. the fraction of read-pairs that map to different chromosomes (which should be limited).</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align/
samtools<span class="w"> </span>flagstat<span class="w"> </span>~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam<span class="w"> </span>&gt;<span class="w"> </span>~/workspace/align/Exome_Norm_flagstat.txt
samtools<span class="w"> </span>flagstat<span class="w"> </span>~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="w"> </span>&gt;<span class="w"> </span>~/workspace/align/Exome_Tumor_flagstat.txt
</code></pre></div>
<h2 id="run-various-picard-collectmetrics-tools">Run various picard CollectMetrics tools<a class="headerlink" href="#run-various-picard-collectmetrics-tools" title="Permanent link">&para;</a></h2>
<p>Picard is a widely used QC tool in genomics and it is named after a famous <a href="https://en.wikipedia.org/wiki/Jean-Luc_Picard">Star Trek Captain</a>.</p>
<p>Description of the output of picard can be found <a href="http://broadinstitute.github.io/picard/">here</a>. </p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align/
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>CollectInsertSizeMetrics<span class="w"> </span>-I<span class="w"> </span>~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>~/workspace/align/Exome_Norm_insert_size_metrics.txt<span class="w"> </span>-H<span class="w"> </span>~/workspace/align/Exome_Norm_insert_size_metrics.pdf
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>CollectInsertSizeMetrics<span class="w"> </span>-I<span class="w"> </span>~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>~/workspace/align/Exome_Tumor_insert_size_metrics.txt<span class="w"> </span>-H<span class="w"> </span>~/workspace/align/Exome_Tumor_insert_size_metrics.pdf

java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>CollectAlignmentSummaryMetrics<span class="w"> </span>-I<span class="w"> </span>~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>~/workspace/align/Exome_Norm_alignment_metrics.txt<span class="w"> </span>-R<span class="w"> </span>~/workspace/inputs/references/genome/ref_genome.fa
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>CollectAlignmentSummaryMetrics<span class="w"> </span>-I<span class="w"> </span>~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>~/workspace/align/Exome_Tumor_exome_alignment_metrics.txt<span class="w"> </span>-R<span class="w"> </span>~/workspace/inputs/references/genome/ref_genome.fa

<span class="c1">#Picard CollectHsMetrics</span>
<span class="c1">#Exome Only</span>
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>CollectHsMetrics<span class="w"> </span>-I<span class="w"> </span>~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>~/workspace/align/Exome_Norm_hs_metrics.txt<span class="w"> </span>-R<span class="w"> </span>~/workspace/inputs/references/genome/ref_genome.fa<span class="w"> </span>-BI<span class="w"> </span>~/workspace/inputs/references/exome/probe_regions.bed.interval_list<span class="w"> </span>-TI<span class="w"> </span>~/workspace/inputs/references/exome/exome_regions.bed.interval_list
java<span class="w"> </span>-Xmx12g<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>CollectHsMetrics<span class="w"> </span>-I<span class="w"> </span>~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>~/workspace/align/Exome_Tumor_hs_metrics.txt<span class="w"> </span>-R<span class="w"> </span>~/workspace/inputs/references/genome/ref_genome.fa<span class="w"> </span>-BI<span class="w"> </span>~/workspace/inputs/references/exome/probe_regions.bed.interval_list<span class="w"> </span>-TI<span class="w"> </span>~/workspace/inputs/references/exome/exome_regions.bed.interval_list
</code></pre></div>
<h2 id="run-fastqc">Run FastQC<a class="headerlink" href="#run-fastqc" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align
fastqc<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>Exome_Norm_sorted_mrkdup_bqsr.bam
fastqc<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>Exome_Tumor_sorted_mrkdup_bqsr.bam
tree
</code></pre></div>
<h2 id="run-multiqc-to-produce-a-final-report">Run MultiQC to produce a final report<a class="headerlink" href="#run-multiqc-to-produce-a-final-report" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/align
<span class="c1">#Dont do if directory exist</span>
<span class="c1">#mkdir post_align_qc</span>
<span class="nb">cd</span><span class="w"> </span>post_align_qc
multiqc<span class="w"> </span>~/workspace/align/
tree

<span class="c1"># Download MultiQC output to the local computer, open the .html in you favourite browser.</span>

<span class="c1"># Discuss the MultiQC output with a fellow student</span>
scp<span class="w"> </span>-ri<span class="w"> </span>~/PEM_KEY_ID.pem<span class="w"> </span>ubuntu@AWS_ADDRESS_HERE:~/workspace/align/post_align_qc/multiqc*<span class="w"> </span>.
</code></pre></div>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2023, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "content.code.annotate", "content.tabs.link", "search.highlight", "search.share", "search.suggest"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.51198bba.min.js"></script>
      
    
  </body>
</html>