
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Project documentation with Markdown.">
      
      
      
      
        <link rel="prev" href="../structural_variation/">
      
      
        <link rel="next" href="../processing_of_rnaseq/">
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.1.6">
    
    
      
        <title>Copy number analysis - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.ded33207.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.a0c5b2b5.min.css">
      
      
  
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
  
  <style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M64 32C28.7 32 0 60.7 0 96v320c0 35.3 28.7 64 64 64h224V368c0-26.5 21.5-48 48-48h112V96c0-35.3-28.7-64-64-64H64zm384 320H336c-8.8 0-16 7.2-16 16v112l32-32 64-64 32-32z"/></svg>');--md-admonition-icon--abstract:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M96 0C43 0 0 43 0 96v320c0 53 43 96 96 96h320c17.7 0 32-14.3 32-32s-14.3-32-32-32v-64c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H96zm0 384h256v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"/></svg>');--md-admonition-icon--info:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-40-176h24v-64h-24c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24h-80c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"/></svg>');--md-admonition-icon--tip:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M480 32c0-12.9-7.8-24.6-19.8-29.6S434.5.2 425.3 9.3L381.7 53c-48 48-113.1 75-181 75H64c-35.3 0-64 28.7-64 64v96c0 35.3 28.7 64 64 64v128c0 17.7 14.3 32 32 32h64c17.7 0 32-14.3 32-32V352h8.7c67.9 0 133 27 181 75l43.6 43.6c9.2 9.2 22.9 11.9 34.9 6.9s19.8-16.6 19.8-29.6V300.4c18.6-8.8 32-32.5 32-60.4s-13.4-51.6-32-60.4V32zm-64 76.7v262.6C357.2 317.8 280.5 288 200.7 288H192v-96h8.7c79.8 0 156.5-29.8 215.3-83.3z"/></svg>');--md-admonition-icon--success:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7l233.4-233.3c12.5-12.5 32.8-12.5 45.3 0z"/></svg>');--md-admonition-icon--question:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-86.2-346.7c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1 0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6-13.3 0-24-10.7-24-24v-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1 0-8.4-6.8-15.1-15.1-15.1h-58.3c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1-64 0z"/></svg>');--md-admonition-icon--warning:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7.2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8.2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24v112c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0-64 0 32 32 0 1 0 64 0z"/></svg>');--md-admonition-icon--failure:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.1 52.4 442.6 6.5c-1.9-3.9-6.1-6.5-10.5-6.5s-8.5 2.6-10.4 6.5l-16.5 45.9-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4 0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2 0-4.6-3-8.9-7.2-10.4l-45.6-16.8zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9c-22-8-45.8-12.3-70.5-12.3C93.1 96 0 189.1 0 304s93.1 208 208 208 208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg>');--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M416 398.9c58.5-41.1 96-104.1 96-174.9C512 100.3 397.4 0 256 0S0 100.3 0 224c0 70.7 37.5 133.8 96 174.9V464c0 26.5 21.5 48 48 48h48v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h64v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h48c26.5 0 48-21.5 48-48v-65.1zM96 256a64 64 0 1 1 128 0 64 64 0 1 1-128 0zm256-64a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"/></svg>');--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M320 0c17.7 0 32 14.3 32 32v64h120c39.8 0 72 32.2 72 72v272c0 39.8-32.2 72-72 72H168c-39.8 0-72-32.2-72-72V168c0-39.8 32.2-72 72-72h120V32c0-17.7 14.3-32 32-32zM208 384c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zM264 256a40 40 0 1 0-80 0 40 40 0 1 0 80 0zm152 40a40 40 0 1 0 0-80 40 40 0 1 0 0 80zM48 224h16v192H48c-26.5 0-48-21.5-48-48v-96c0-26.5 21.5-48 48-48zm544 0c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48h-16V224h16z"/></svg>');--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M288 0H128c-17.7 0-32 14.3-32 32s14.3 32 32 32v132.8c0 11.8-3.3 23.5-9.5 33.5L10.3 406.2C3.6 417.2 0 429.7 0 442.6 0 480.9 31.1 512 69.4 512h309.2c38.3 0 69.4-31.1 69.4-69.4 0-12.8-3.6-25.4-10.3-36.4L329.5 230.4c-6.2-10.1-9.5-21.7-9.5-33.5V64c17.7 0 32-14.3 32-32S337.7 0 320 0h-32zm-96 196.8V64h64v132.8c0 23.7 6.6 46.9 19 67.1l34.5 56.1h-171l34.5-56.1c12.4-20.2 19-43.4 19-67.1z"/></svg>');--md-admonition-icon--quote:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 216C0 149.7 53.7 96 120 96h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V216zm256 0c0-66.3 53.7-120 120-120h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64h-64c-35.3 0-64-28.7-64-64V216z"/></svg>');}</style>


    
    
    
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#introduction" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Copy number analysis
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      Course Details
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../schedule/" class="md-tabs__link">
      Schedule
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../aws/aws_ec2_instances/" class="md-tabs__link">
      AWS Instances Login
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../lectures/lectures/" class="md-tabs__link">
      Lectures
    </a>
  </li>

      
        
  
  
    
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../basic_unix/" class="md-tabs__link md-tabs__link--active">
        Lab Session
      </a>
    </li>
  

  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_5_1" id="__nav_5_1_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ggplot2/" class="md-nav__link">
        ggplot tutorial
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_4" checked>
      
      
      
        <label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_4_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../structural_variation/" class="md-nav__link">
        Structural Variants
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Copy Number Analysis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Copy Number Analysis
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#download-data-and-prepare-r" class="md-nav__link">
    Download data and prepare R
  </a>
  
    <nav class="md-nav" aria-label="Download data and prepare R">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#from-cran" class="md-nav__link">
    From CRAN
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#from-bioconductor" class="md-nav__link">
    From Bioconductor
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-sequence-target-definition" class="md-nav__link">
    Parse sequence target definition
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotate-with-gene-symbol" class="md-nav__link">
    Annotate with gene symbol
  </a>
  
    <nav class="md-nav" aria-label="Annotate with gene symbol">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-gene-coordinates" class="md-nav__link">
    Download gene coordinates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-targets-and-genes" class="md-nav__link">
    Overlap targets and genes
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-sequence-read-count" class="md-nav__link">
    Parse sequence read count
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plot-sequence-read-count" class="md-nav__link">
    Plot sequence read count
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#model-and-correct-for-gc-content-bias" class="md-nav__link">
    Model and correct for GC content bias
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#segment-the-data" class="md-nav__link">
    Segment the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-snp-allele-ratio" class="md-nav__link">
    Parse SNP allele ratio
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#estimating-genome-wide-copy-number" class="md-nav__link">
    Estimating genome-wide copy number
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#final-questions" class="md-nav__link">
    Final questions
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
          Day 5
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          Day 5
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../joint_curation/" class="md-nav__link">
        Joint Curation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../genomic_alteration_interpretation_exe/" class="md-nav__link">
        Genomic Alterations Interpretation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#download-data-and-prepare-r" class="md-nav__link">
    Download data and prepare R
  </a>
  
    <nav class="md-nav" aria-label="Download data and prepare R">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#from-cran" class="md-nav__link">
    From CRAN
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#from-bioconductor" class="md-nav__link">
    From Bioconductor
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-sequence-target-definition" class="md-nav__link">
    Parse sequence target definition
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotate-with-gene-symbol" class="md-nav__link">
    Annotate with gene symbol
  </a>
  
    <nav class="md-nav" aria-label="Annotate with gene symbol">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-gene-coordinates" class="md-nav__link">
    Download gene coordinates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-targets-and-genes" class="md-nav__link">
    Overlap targets and genes
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-sequence-read-count" class="md-nav__link">
    Parse sequence read count
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plot-sequence-read-count" class="md-nav__link">
    Plot sequence read count
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#model-and-correct-for-gc-content-bias" class="md-nav__link">
    Model and correct for GC content bias
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#segment-the-data" class="md-nav__link">
    Segment the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-snp-allele-ratio" class="md-nav__link">
    Parse SNP allele ratio
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#estimating-genome-wide-copy-number" class="md-nav__link">
    Estimating genome-wide copy number
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#final-questions" class="md-nav__link">
    Final questions
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  <h1>Copy Number Analysis</h1>

<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>There are plenty of methods and code packages available for copy number
analysis, and you will likely be able to find one for free that suits
your particular application. Although tailored for different types of
sequence data (or microarrays), general processing steps of all copy
number analysis tools include:</p>
<ol>
<li>Quantification of sequence read depth (or array intensity)
    throughout the reference genome as a measure of DNA abundance in the
    sample</li>
<li>Removal of systematic noise using features such as GC content and
    mapability</li>
<li>Removal of systematic noise using normal (typicaly non-cancer)
    reference samples</li>
<li>Segmentation - partitioning of the reference genome so that each
    segment can be assigned one copy number</li>
<li>Copy number calling, assigning each segment some estimate of the
    number of copies per cell</li>
<li>Combine the measure of DNA abundance with SNP allele ratios to
    support copy number estimates</li>
</ol>
<p>In this exercise you will perform the above steps using R, with a BAM
file of aligned reads and a VCF file of SNPs as input.</p>
<h2 id="download-data-and-prepare-r">Download data and prepare R<a class="headerlink" href="#download-data-and-prepare-r" title="Permanent link">&para;</a></h2>
<p>The data used in this exercise are available for download
<a href="https://course-cg-5534.s3.amazonaws.com/cnvs/copy_number_files.tar.bz2" target="_blank">here</a>.
Extract the content on your computer. Before starting, make sure you
have a recent version of R, RStudio and the following packages
installed:</p>
<h3 id="from-cran">From CRAN<a class="headerlink" href="#from-cran" title="Permanent link">&para;</a></h3>
<ul>
<li>data.table</li>
<li>ggplot2</li>
<li>patchwork</li>
<li>PSCBS</li>
<li>stringr</li>
</ul>
<p>Just open RStudio and type in the R console, for each one:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;package_name&quot;</span><span class="p">)</span>
</code></pre></div>
<h3 id="from-bioconductor">From Bioconductor<a class="headerlink" href="#from-bioconductor" title="Permanent link">&para;</a></h3>
<ul>
<li>bamsignals</li>
<li>biomaRt</li>
<li>bioStrings</li>
<li>BSgenome</li>
<li>BSgenome.Hsapiens.UCSC.hg19</li>
<li>DNAcopy</li>
<li>GenomicRanges</li>
<li>Rsamtools</li>
<li>VariantAnnotation</li>
</ul>
<p>Type once:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">if </span><span class="p">(</span><span class="o">!</span><span class="nf">requireNamespace</span><span class="p">(</span><span class="s">&quot;BiocManager&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
<span class="w">    </span><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;BiocManager&quot;</span><span class="p">)</span>
</code></pre></div>
<p>Then, for each package:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;package_name&quot;</span><span class="p">)</span>
</code></pre></div>
<blockquote>
<p>Several packages from the
<a href="https://www.bioconductor.org/" target="_blank">Bioconductor</a> repository are used in
this exercise, for example
<a href="http://bioconductor.org/packages/devel/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesHOWTOs.pdf" target="_blank">GenomicRanges</a>
which facilitates working with anything located on a genome.</p>
</blockquote>
<h2 id="parse-sequence-target-definition">Parse sequence target definition<a class="headerlink" href="#parse-sequence-target-definition" title="Permanent link">&para;</a></h2>
<p>We use targeted sequencing of a prostate cancer sample in this exercise,
as the resulting data files are of a more convenient size than whole
genome or exome. This hybrid-capture panel covers a few hundred cancer
genes and a few thousands of additional targets intended to improve copy
number and structural variant detection.</p>
<p>In RStudio, navigate to the folder containing the exercise files:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">setwd</span><span class="p">(</span><span class="s">&#39;your/path/to/exercise_folder&#39;</span><span class="p">)</span>
</code></pre></div>
<blockquote>
<p>Create a new Quarto Document in the exercise folder. Keep your code as
shown in the template, and write your comments and answers to
questions between code chunks. Make sure you run the code, investigate
the results and answer the questions. You can click <strong>Render</strong> to run
all code and generate a html report. This way, when you are through
this exercise, your report is done too.</p>
</blockquote>
<p>The <a href="https://en.wikipedia.org/wiki/BED_(file_format)" target="_blank">BED</a> (Browser
Extensible Data) format is a text file format commonly used to store
genomic regions as coordinates and associated annotations. The data are
presented in the form of columns separated by spaces or tabs. This
format was developed during the Human Genome Project and then adopted by
other sequencing projects.</p>
<p>We begin by parsing and investigating the target definition BED file.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>

<span class="c1"># Input file</span>
<span class="n">bed_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;targets.bed&#39;</span>

<span class="c1"># Check that it exists</span>
<span class="nf">file.exists</span><span class="p">(</span><span class="n">bed_file</span><span class="p">)</span>

<span class="c1"># Read it </span>
<span class="n">targets</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">bed_file</span><span class="p">)</span>

<span class="c1"># Check content</span>
<span class="n">targets</span>

<span class="c1"># Let the first column be the target number</span>
<span class="n">targets</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">cbind</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="nf">nrow</span><span class="p">(</span><span class="n">targets</span><span class="p">),</span><span class="n">targets</span><span class="p">)</span>

<span class="c1"># Set useful column names</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">targets</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;target&#39;</span><span class="p">,</span><span class="s">&#39;chromosome&#39;</span><span class="p">,</span><span class="s">&#39;start&#39;</span><span class="p">,</span><span class="s">&#39;end&#39;</span><span class="p">)</span>

<span class="c1"># Calculate target length</span>
<span class="n">targets</span><span class="p">[,</span><span class="n">length</span><span class="o">:=</span><span class="n">end</span><span class="o">-</span><span class="n">start</span><span class="p">]</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">targets</span><span class="p">)</span>

<span class="c1"># Check target distribution over chromosomes</span>
<span class="n">targets</span><span class="p">[,</span><span class="nf">table</span><span class="p">(</span><span class="n">chromosome</span><span class="p">)]</span>
</code></pre></div>
<p>The R code in this exercise features <code>data.table</code> syntax, for example
using <code>:=</code> to modify the content of a table. If you are more familiar
with base R or tidyverse syntax, a comparison between them can be found
<a href="https://mgimond.github.io/rug_2019_12/Index.html" target="_blank">here</a>.</p>
<blockquote>
<p>Feel free to complete the exercise using the coding style you prefer.</p>
</blockquote>
<h2 id="annotate-with-gene-symbol">Annotate with gene symbol<a class="headerlink" href="#annotate-with-gene-symbol" title="Permanent link">&para;</a></h2>
<p>Before parsing the sequence data, let's assign gene symbols to targets
that overlap a gene. We'll take advantage of databases accessible
through Bioconductor packages. First, we need to create a GenomicRanges
object representing our targets:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">GenomicRanges</span><span class="p">)</span>

<span class="c1"># Make a GRanges object</span>
<span class="n">target_ranges</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">makeGRangesFromDataFrame</span><span class="p">(</span><span class="n">targets</span><span class="p">)</span>
<span class="n">target_ranges</span>

<span class="c1"># Check the sequence naming style</span>
<span class="nf">seqlevelsStyle</span><span class="p">(</span><span class="n">target_ranges</span><span class="p">)</span>
</code></pre></div>
<p>As you may recognize, our targets have the NCBI/EMBL sequence naming
style (<em>1</em> and not <em>chr1</em>, etc). This matches the reference genome used
with our sequence data.</p>
<h3 id="download-gene-coordinates">Download gene coordinates<a class="headerlink" href="#download-gene-coordinates" title="Permanent link">&para;</a></h3>
<p>Let's use the <code>biomaRt</code> package to download Ensembl gene coordinates for
reference genome b37/Hg19.</p>
<blockquote>
<p>To learn more about a function, just type ?function_name to open its
documentation in RStudio's bottom right window.</p>
</blockquote>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">biomaRt</span><span class="p">)</span>

<span class="o">?</span><span class="n">useEnsembl</span>

<span class="c1"># Open a connection</span>
<span class="n">biomart_connection</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">useEnsembl</span><span class="p">(</span><span class="n">biomart</span><span class="o">=</span><span class="s">&quot;ensembl&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">dataset</span><span class="o">=</span><span class="s">&quot;hsapiens_gene_ensembl&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GRCh</span><span class="o">=</span><span class="m">37</span><span class="p">)</span>

<span class="o">?</span><span class="n">getBM</span>

<span class="c1"># Get gene coordinates</span>
<span class="n">genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getBM</span><span class="p">(</span>
<span class="w">    </span><span class="n">attributes</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;ensembl_gene_id&#39;</span><span class="p">,</span><span class="s">&#39;hgnc_symbol&#39;</span><span class="p">,</span><span class="s">&#39;chromosome_name&#39;</span><span class="p">,</span><span class="s">&#39;start_position&#39;</span><span class="p">,</span><span class="s">&#39;end_position&#39;</span><span class="p">),</span>
<span class="w">    </span><span class="n">mart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">biomart_connection</span><span class="p">)</span>

<span class="c1"># Convert to data.table</span>
<span class="n">genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.table</span><span class="p">(</span><span class="n">genes</span><span class="p">)</span>

<span class="n">genes</span>
</code></pre></div>
<p>We can now make a GenomicRanges object for the genes:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">colnames</span><span class="p">(</span><span class="n">genes</span><span class="p">)[</span><span class="m">3</span><span class="o">:</span><span class="m">5</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;chromosome&#39;</span><span class="p">,</span><span class="s">&#39;start&#39;</span><span class="p">,</span><span class="s">&#39;end&#39;</span><span class="p">)</span>

<span class="n">gene_ranges</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">makeGRangesFromDataFrame</span><span class="p">(</span><span class="n">genes</span><span class="p">)</span>

<span class="nf">seqlevelsStyle</span><span class="p">(</span><span class="n">gene_ranges</span><span class="p">)</span>
</code></pre></div>
<h3 id="overlap-targets-and-genes">Overlap targets and genes<a class="headerlink" href="#overlap-targets-and-genes" title="Permanent link">&para;</a></h3>
<p>The chromosome naming style of the genes is also NCBI/EMBL. We can now
go ahead and match them up to targets using the findOverlaps function:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># Overlap the two GRanges sets</span>
<span class="n">overlap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">findOverlaps</span><span class="p">(</span><span class="n">target_ranges</span><span class="p">,</span><span class="n">gene_ranges</span><span class="p">)</span>

<span class="n">overlap</span>
</code></pre></div>
<p>Overlaps are presented as pairs of query (first argument) and subject
(second argument) rows for which there is any overlap. We can simply
assign targets their corresponding gene name for each overlap, and check
the result:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="n">targets</span><span class="o">$</span><span class="n">gene</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="w"> </span><span class="c1"># default if no overlap</span>

<span class="c1"># Put a gene symbol on each target</span>
<span class="n">targets</span><span class="p">[</span><span class="nf">queryHits</span><span class="p">(</span><span class="n">overlap</span><span class="p">),</span><span class="n">gene</span><span class="o">:=</span><span class="n">genes</span><span class="p">[</span><span class="nf">subjectHits</span><span class="p">(</span><span class="n">overlap</span><span class="p">)]</span><span class="o">$</span><span class="n">hgnc_symbol</span><span class="p">]</span>

<span class="c1"># Count targets per gene</span>
<span class="nf">as.data.table</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">targets</span><span class="o">$</span><span class="n">gene</span><span class="p">))</span>

<span class="nf">as.data.table</span><span class="p">(</span><span class="nf">sort</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">targets</span><span class="o">$</span><span class="n">gene</span><span class="p">),</span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">))[</span><span class="n">N</span><span class="o">&gt;</span><span class="m">25</span><span class="p">]</span>
</code></pre></div>
<blockquote>
<p>There are many ways to achieve the same thing. You should focus mostly
on <em>what</em> we do, and not worry too much about exactly how. Google,
ChatGTP, etc, are there for you as long as you know what you want to
achieve.</p>
</blockquote>
<h2 id="parse-sequence-read-count">Parse sequence read count<a class="headerlink" href="#parse-sequence-read-count" title="Permanent link">&para;</a></h2>
<p>We are now ready to access the BAM file and count the number of reads
(actually read pairs, corresponding to DNA fragments) that map to each
target. For this we use the <code>bamsignals</code> package and the <code>GenomicRanges</code>
object representing the targets. To count a read pair, we require its
midpoint to fall within a target, and that it is not flagged as a
<a href="http://samtools.github.io/hts-specs/SAMv1.pdf" target="_blank">duplicate</a>.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">bamsignals</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">Rsamtools</span><span class="p">)</span>

<span class="c1"># Imput bam file</span>
<span class="n">bamfile</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;sample1.bam&#39;</span>
<span class="nf">file.exists</span><span class="p">(</span><span class="n">bamfile</span><span class="p">)</span>
<span class="nf">file.exists</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">bamfile</span><span class="p">,</span><span class="s">&#39;.bai&#39;</span><span class="p">))</span>

<span class="c1"># Count reads in targets</span>
<span class="n">targets</span><span class="o">$</span><span class="n">count</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">bamCount</span><span class="p">(</span><span class="n">bamfile</span><span class="p">,</span><span class="w"> </span><span class="n">target_ranges</span><span class="p">,</span><span class="w"> </span><span class="n">paired.end</span><span class="o">=</span><span class="s">&quot;midpoint&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">filteredFlag</span><span class="o">=</span><span class="m">1024</span><span class="p">)</span>
<span class="n">targets</span>

<span class="nf">summary</span><span class="p">(</span><span class="n">targets</span><span class="p">)</span>
</code></pre></div>
<h2 id="plot-sequence-read-count">Plot sequence read count<a class="headerlink" href="#plot-sequence-read-count" title="Permanent link">&para;</a></h2>
<p>Let's investigate the raw sequence read count across targets. You are
probably already familiar with
<a href="https://ggplot2.tidyverse.org/" target="_blank">ggplot2</a>.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">theme_set</span><span class="p">(</span><span class="nf">theme_bw</span><span class="p">())</span>

<span class="c1"># Simple ggplot</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">target</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count</span><span class="p">))</span>

<span class="c1"># Some adjustments</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">target</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count</span><span class="p">),</span><span class="n">size</span><span class="o">=</span><span class="n">.</span><span class="m">5</span><span class="p">)</span>
</code></pre></div>
<p>Targets are just shown in order. To plot genomic positions without
mixing up chromosomes, we can separate the plot by chromosome:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># Plot start pos by chromosome</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count</span><span class="p">),</span><span class="n">size</span><span class="o">=</span><span class="n">.</span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">    </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="n">facets</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">vars</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">)</span>
</code></pre></div>
<p>The figure may now not look a bit squished in RStudio's bottom-right
plots tab. Click the <strong>Zoom</strong> button and enlarge the new window.</p>
<p>Let's choose a few genes to highlight in the plot.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># Genes of interest</span>
<span class="n">my_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;AR&#39;</span><span class="p">,</span><span class="s">&#39;ATM&#39;</span><span class="p">,</span><span class="s">&#39;BRCA1&#39;</span><span class="p">,</span><span class="s">&#39;BRCA2&#39;</span><span class="p">,</span><span class="s">&#39;PTEN&#39;</span><span class="p">,</span><span class="s">&#39;TMPRSS2&#39;</span><span class="p">,</span><span class="s">&#39;ERG&#39;</span><span class="p">)</span>
<span class="n">targets</span><span class="p">[</span><span class="n">gene</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">my_genes</span><span class="p">,</span><span class="n">label</span><span class="o">:=</span><span class="n">gene</span><span class="p">]</span>

<span class="nf">ggplot</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">,</span><span class="w"> </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">target</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">label</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="n">.</span><span class="m">5</span><span class="p">)</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">label</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="n">.</span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="n">facets</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">vars</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">)</span>
</code></pre></div>
<h2 id="model-and-correct-for-gc-content-bias">Model and correct for GC content bias<a class="headerlink" href="#model-and-correct-for-gc-content-bias" title="Permanent link">&para;</a></h2>
<p>Sequence GC content is known to affect PCR amplification and sequence
coverage. Let's retrieve the GC content for the targets and investigate
if there is a bias in our data set. <code>BSgenome</code> is a Bioconductor package
that provides a framework for working with genomic sequences in R. The
package provides a set of objects, functions, and methods for working
with different reference genomes and their annotations, such as gene
locations, exons, and transcription start sites.</p>
<p>First we check whether a matching reference genome is available:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">BSgenome</span><span class="p">)</span>

<span class="c1"># Check which genomes are available</span>
<span class="nf">available.genomes</span><span class="p">()</span>
</code></pre></div>
<p>The best match is <code>BSgenome.Hsapiens.UCSC.hg19</code>. But that means our
target <code>GenomicRanges</code> object needs to have UCSC-style sequence naming:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="n">ucsc_ranges</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">target_ranges</span>

<span class="c1"># Swap sequence naming style</span>
<span class="nf">seqlevelsStyle</span><span class="p">(</span><span class="n">ucsc_ranges</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;UCSC&quot;</span>
<span class="n">ucsc_ranges</span>

<span class="nf">seqlevelsStyle</span><span class="p">(</span><span class="n">ucsc_ranges</span><span class="p">)</span>
</code></pre></div>
<p>We can now assign GC content to targets:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">BSgenome.Hsapiens.UCSC.hg19</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>

<span class="c1"># Get the target sequences</span>
<span class="n">dna</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getSeq</span><span class="p">(</span><span class="n">Hsapiens</span><span class="p">,</span><span class="w"> </span><span class="n">ucsc_ranges</span><span class="p">)</span>

<span class="c1"># Calculate GC content for the target sequences</span>
<span class="n">targets</span><span class="o">$</span><span class="n">gc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_count</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="n">dna</span><span class="p">),</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;[GC]&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="nf">width</span><span class="p">(</span><span class="n">dna</span><span class="p">)</span>

<span class="c1"># Plot the GC content and read count</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gc</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count</span><span class="p">),</span><span class="n">alpha</span><span class="o">=</span><span class="n">.</span><span class="m">1</span><span class="p">)</span>
</code></pre></div>
<p>There appears to be a modest effect of target GC content on read count.
Let's build a <a href="https://en.wikipedia.org/wiki/Local_regression" target="_blank">loess</a>
model of the effect.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># Make a loess model</span>
<span class="n">loess_model</span><span class="o">=</span><span class="nf">loess</span><span class="p">(</span><span class="n">count</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">gc</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">)</span>
</code></pre></div>
<p>To plot the model, we predict the read count from GC:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># Add predicted count (model y value) to targets</span>
<span class="n">targets</span><span class="p">[,</span><span class="n">gc_predicted_count</span><span class="o">:=</span><span class="nf">predict</span><span class="p">(</span><span class="n">loess_model</span><span class="p">,</span><span class="n">gc</span><span class="p">)]</span>

<span class="c1"># Plot with the model prediction</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">targets</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gc</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count</span><span class="p">),</span><span class="n">alpha</span><span class="o">=</span><span class="n">.</span><span class="m">1</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_line</span><span class="p">(</span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gc</span><span class="p">,</span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gc_predicted_count</span><span class="p">),</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;blue&#39;</span><span class="p">)</span>
</code></pre></div>
<p>To remove the GC content bias, we divide the observed count with the
predicted. This new count <em>ratio</em> should be a slightly better measure of
DNA abundance in our sample.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># Correct for the effect of GC content</span>
<span class="n">targets</span><span class="p">[,</span><span class="n">count_ratio</span><span class="o">:=</span><span class="n">count</span><span class="o">/</span><span class="n">gc_predicted_count</span><span class="p">]</span>
<span class="n">targets</span>
</code></pre></div>
<p>To compare the new metric with the previous, we can plot them
side-by-side. With <a href="https://patchwork.data-imaginist.com/" target="_blank">patchwork</a> we
can use arithmetic operators to combine and align multiple plots.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">patchwork</span><span class="p">)</span>

<span class="c1"># Plot and align</span>
<span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gc</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count</span><span class="p">),</span><span class="n">alpha</span><span class="o">=</span><span class="n">.</span><span class="m">1</span><span class="p">)</span>

<span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gc</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count_ratio</span><span class="p">),</span><span class="n">alpha</span><span class="o">=</span><span class="n">.</span><span class="m">1</span><span class="p">)</span>

<span class="n">p1</span><span class="o">+</span><span class="n">p2</span>
</code></pre></div>
<p>The GC content correction seems to have had a small but positive effect.
It is common to also use either a matched normal or a pool of normal
reference samples (sequenced similarly) to remove a little bit more
systematic noise. We have omitted that step here.</p>
<blockquote>
<p><strong>Question 1:</strong> An important reason to perform copy number analysis of
cancer samples is to find homozygous deletions of tumor suppressor
genes. Let's assume a patient has a small hemizygous germline deletion
(loss of one copy) affecting one tumor suppressor gene only. In the
tumor cell population there is also a somatic deletion of the other,
homologous, chromosome, including the remaining copy of the gene.
Would the resulting homozygous deletion still be visible if the read
count ratio of the tumor sample were to be divided by that of the
normal sample?</p>
<p>Write answers to all questions as text in your Quarto document, so
that they are stored in the report with your data analysis. When done,
render the report (including all analysis steps you have included) by
clicking the <strong>Render</strong> button. <a href="mailto:markus.mayrhofer@ki.se">Email</a>
the result to the course teacher.</p>
</blockquote>
<h2 id="segment-the-data">Segment the data<a class="headerlink" href="#segment-the-data" title="Permanent link">&para;</a></h2>
<p>Although we can already spot some apparent gains and losses, statistical
tools can help us better define copy number segments, for which we can
then calculate the most likely copy number given the observation.
<a href="https://pubmed.ncbi.nlm.nih.gov/15475419/" target="_blank">Circular binary
segmentation</a> (<strong>CBS</strong>) is
probably the most commonly used segmentation method. Here we use the
<code>PSCBS</code> ("Parent specific" CBS, as it can also use SNP allele data) R
package as a wrapper to perform basic CBS on the GC content-adjusted
count ratio.</p>
<p>CBS requires a DNA abundance measure to be log-transformed, making its
distribution more normal-like. If you have worked with copy number
analysis before, you may recognize the term <strong>log ratio</strong>.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">PSCBS</span><span class="p">)</span>

<span class="c1"># Log transform the read count</span>
<span class="n">targets</span><span class="p">[,</span><span class="n">log_ratio</span><span class="o">:=</span><span class="nf">log2</span><span class="p">(</span><span class="n">count_ratio</span><span class="p">)]</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">targets</span><span class="p">)</span>

<span class="c1"># Segmentation requires values to be finite</span>
<span class="n">targets</span><span class="p">[</span><span class="nf">is.infinite</span><span class="p">(</span><span class="n">log_ratio</span><span class="p">),</span><span class="n">log_ratio</span><span class="o">:=</span><span class="kc">NA</span><span class="p">]</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">targets</span><span class="p">)</span>

<span class="c1"># Segment the log ratio</span>
<span class="n">segments</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">segmentByCBS</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">targets</span><span class="o">$</span><span class="n">log_ratio</span><span class="p">)</span>
<span class="n">segments</span>
</code></pre></div>
<p>After segmentation we can transform the segment mean values back and
plot the segments with the targets.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># Linear space segment means</span>
<span class="n">segments</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.table</span><span class="p">(</span><span class="n">segments</span><span class="p">)[,</span><span class="n">count_ratio</span><span class="o">:=</span><span class="m">2</span><span class="o">^</span><span class="n">mean</span><span class="p">]</span>
<span class="n">segments</span>

<span class="nf">ggplot</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">,</span><span class="w"> </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">target</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">chromosome</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">    </span><span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span>
<span class="w">                 </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">))</span>
</code></pre></div>
<p>Note that the segmentation is based only on the target log ratio and
order, and that segment start and end refer to target number. To avoid
segments spanning across chromosomes, and breakpoints just missing the
chromosome boundary, we can add chromosomes to the segmentation call.
The chromosome vector is required to be numeric.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># Segment with chromosomes specified</span>
<span class="n">segments</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">segmentByCBS</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">targets</span><span class="o">$</span><span class="n">log_ratio</span><span class="p">,</span>
<span class="w">                         </span><span class="n">chromosome</span><span class="o">=</span><span class="nf">as.numeric</span><span class="p">(</span><span class="nf">str_replace</span><span class="p">(</span><span class="n">targets</span><span class="o">$</span><span class="n">chromosome</span><span class="p">,</span><span class="s">&#39;X&#39;</span><span class="p">,</span><span class="s">&#39;23&#39;</span><span class="p">)))</span>
<span class="n">segments</span>

<span class="c1"># Tidy up</span>
<span class="n">segments</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.table</span><span class="p">(</span><span class="n">segments</span><span class="p">)[,</span><span class="n">count_ratio</span><span class="o">:=</span><span class="m">2</span><span class="o">^</span><span class="n">mean</span><span class="p">][</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="m">-1</span><span class="p">]</span>
<span class="n">segments</span>

<span class="nf">ggplot</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">,</span><span class="w"> </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">target</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">chromosome</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">    </span><span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span>
<span class="w">                 </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">))</span>
</code></pre></div>
<p>CBS can also take a vector of chromosomal positions as input, in which
case the resulting segment start and end positions are based on them. We
use the target midpoints:</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># Segment with position and chromosome positions</span>
<span class="n">segments_pos</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">segmentByCBS</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">targets</span><span class="o">$</span><span class="n">log_ratio</span><span class="p">,</span>
<span class="w">                             </span><span class="n">chromosome</span><span class="o">=</span><span class="nf">as.numeric</span><span class="p">(</span><span class="nf">str_replace</span><span class="p">(</span><span class="n">targets</span><span class="o">$</span><span class="n">chromosome</span><span class="p">,</span><span class="s">&#39;X&#39;</span><span class="p">,</span><span class="s">&#39;23&#39;</span><span class="p">)),</span>
<span class="w">                             </span><span class="n">x</span><span class="o">=</span><span class="n">targets</span><span class="o">$</span><span class="n">start</span><span class="m">+60</span><span class="p">)</span>

<span class="n">segments_pos</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.table</span><span class="p">(</span><span class="n">segments_pos</span><span class="p">)[,</span><span class="n">count_ratio</span><span class="o">:=</span><span class="m">2</span><span class="o">^</span><span class="n">mean</span><span class="p">][</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="m">-1</span><span class="p">]</span>
<span class="n">segments_pos</span>


<span class="c1"># convert chromosomes back to NCBI</span>
<span class="n">segments_pos</span><span class="p">[,</span><span class="n">chromosome</span><span class="o">:=</span><span class="nf">str_replace</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="s">&#39;23&#39;</span><span class="p">,</span><span class="s">&#39;X&#39;</span><span class="p">)]</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count_ratio</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments_pos</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span>
<span class="w">                 </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="n">facets</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">vars</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme</span><span class="p">(</span><span class="n">panel.spacing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unit</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;lines&quot;</span><span class="p">),</span><span class="n">strip.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">))</span>
</code></pre></div>
<p>Let's now take another look at segmented targets plotted in order, as
they are very unevenly distributed over the genome. We can spot some
apparent copy number losses and gains of genes in our selection. Take a
closer look at BRCA2 and part of PTEN.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">,</span><span class="w"> </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">target</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">chromosome</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">    </span><span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span>
<span class="w">                 </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">))</span>

<span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">,</span><span class="w"> </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">target</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">    </span><span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span>
<span class="w">                 </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">))</span>

<span class="n">p1</span><span class="o">/</span><span class="n">p2</span>
</code></pre></div>
<blockquote>
<p><strong>Question 2:</strong> Target density is only about one per megabase, with
much higher density in some cancer genes. How many targets do you
think a deletion would have to cover for us to be able to find it?
What else might influence sensitivity?</p>
</blockquote>
<p>For BRCA2 and (part of) PTEN, count ratio (remember, it is the observed
read count (DNA abundance) relative to the expected, given target GC
content), is near 0.5, or 50%. For a largely diploid genome this is
exactly what to expect if these genomic segments were hemizygously
deleted, i.e., one out of the two homologous copies would have been
lost. This would also fit reasonably well with gain of 1 copy of 8q
(resulting in a count ratio of 1.5) and 3 extra copies of AR (being on
the X chromosome, tumor cells would have quadrupled its single original
AR copy, resulting in the count ratio going from about 0.5 to near 2.).</p>
<p>But this solution would require the tumor cell content ("<strong>purity</strong>") in
the sample to be near 100%, which is rare unless the sample comes from a
cell line. Note also that several other segments have a mean count ratio
that would not fit this solution well. There are actually multiple other
combinations of copy number and purity that would fit the observation
reasonably well.</p>
<p>Our measure of relative DNA abundance - count ratio - is automatically
centered around 1, regardless of the average copy number ("<strong>ploidy</strong>")
of the sequenced genome.</p>
<blockquote>
<p>Let's assume the cancer cells sampled had an average of 4 instead of 2
copies. This would neither make us sequence to twice the amount, nor
observe twice the sequence count relative to our GC content model for
most of the genome.</p>
<p>The average DNA abundance measured throughout the genome, whether it
is raw sequence count, count ratio, or a logarithm of count ratio,
does not increase in samples with a higher average copy number (more
DNA per cell). Instead the average measured DNA abundance will be
observed for the average copy number, whatever that is.</p>
</blockquote>
<p>We are also likely to have some normal DNA content in the sample. If
both the tumor genome and normal genome are near diploid on average, and
the tumor cell content is near 50%, homozygous deletion (loss of both
homologous copies) would be observed at a coverge ratio near 0.5, as for
every tumor cell that would contribute zero copies to the DNA
extraction, one normal cell would contribute its normal two. As loss of
one copy in the tumor cell fraction would then appear at about 0.75 and
gains would appear near 1.25, 1.5, ..., this solution also appears to
fit our data reasonably well.</p>
<blockquote>
<p><strong>Question 3:</strong> If a cancer genome has an average of 4 copies, and the
tumor cell content is near 100%, at what count ratio should we ovserve
a copy number of 3?</p>
</blockquote>
<p>Fortunately there is another tool we can use to inform the copy number
estimates. We can use SNPs from a small variant caller and investigate
their variant allele ratios.</p>
<h2 id="parse-snp-allele-ratio">Parse SNP allele ratio<a class="headerlink" href="#parse-snp-allele-ratio" title="Permanent link">&para;</a></h2>
<p>Let's use the <code>VariantAnnotation</code> package to parse a VCF file containing
SNPs.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">VariantAnnotation</span><span class="p">)</span>

<span class="c1"># Parse the vcf file</span>
<span class="n">vcf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readVcf</span><span class="p">(</span><span class="s">&#39;sample1.vcf&#39;</span><span class="p">)</span>
<span class="n">vcf</span>

<span class="c1"># Genotype column</span>
<span class="n">g</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">geno</span><span class="p">(</span><span class="n">vcf</span><span class="p">)</span>
<span class="n">g</span>

<span class="c1"># Allelic and total read depth </span>
<span class="nf">as.data.table</span><span class="p">(</span><span class="n">g</span><span class="o">$</span><span class="n">AD</span><span class="p">)</span>
<span class="nf">as.data.table</span><span class="p">(</span><span class="n">g</span><span class="o">$</span><span class="n">DP</span><span class="p">)</span>
</code></pre></div>
<p>The AD column of the VCF file contains the number of reads supporting
the reference and alternative allele. The DP column contains the total
read depth. Let's make a table of SNPs containing the alt-allele ratio.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># A table of SNPs</span>
<span class="n">snp_table</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.table</span><span class="p">(</span><span class="n">id</span><span class="o">=</span><span class="nf">names</span><span class="p">(</span><span class="n">vcf</span><span class="p">),</span>
<span class="w">                        </span><span class="n">AD</span><span class="o">=</span><span class="nf">sapply</span><span class="p">(</span><span class="n">g</span><span class="o">$</span><span class="n">AD</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;[[&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">),</span>
<span class="w">                        </span><span class="n">DP</span><span class="o">=</span><span class="nf">unlist</span><span class="p">(</span><span class="n">g</span><span class="o">$</span><span class="n">DP</span><span class="p">[,</span><span class="m">1</span><span class="p">]))</span>
<span class="n">snp_table</span>

<span class="c1"># Compute allele ratio</span>
<span class="n">snp_table</span><span class="p">[,</span><span class="n">allele_ratio</span><span class="o">:=</span><span class="nf">round</span><span class="p">(</span><span class="n">AD</span><span class="o">/</span><span class="n">DP</span><span class="p">,</span><span class="m">3</span><span class="p">)]</span>
<span class="n">snp_table</span>
</code></pre></div>
<p>A GenomicRanges object defining the SNPs' positions on the reference
genome can be accessed with the <code>rowRanges</code> function. We can now overlap
that with the GenomicRanges representing our targets and use the
resulting <code>Hits</code> object to assign allele ratio to targets.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="c1"># Match targets to SNPs</span>
<span class="n">overlaps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">findOverlaps</span><span class="p">(</span><span class="n">target_ranges</span><span class="p">,</span><span class="nf">rowRanges</span><span class="p">(</span><span class="n">vcf</span><span class="p">))</span>
<span class="n">overlaps</span>

<span class="c1"># Set SNP allele ratio for targets</span>
<span class="n">targets</span><span class="p">[</span><span class="nf">queryHits</span><span class="p">(</span><span class="n">overlaps</span><span class="p">),</span><span class="n">allele_ratio</span><span class="o">:=</span><span class="n">snp_table</span><span class="o">$</span><span class="n">allele_ratio</span><span class="p">[</span><span class="nf">subjectHits</span><span class="p">(</span><span class="n">overlaps</span><span class="p">)]]</span>
<span class="n">targets</span>
</code></pre></div>
<p>Let's plot the SNP allele ratio with the count ratio and see if the copy
number status becomes more clear. Note that many targets contain no SNP,
and that many SNPs are homozygous. The allele ratio of a homozygous SNP
is either 0 or 1, depending on whether the alleles match the reference
genome or not. Only heterozygous SNP allele ratio is affected by copy
number alteration.</p>
<div class="cell-code highlight"><pre><span></span><code><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">,</span><span class="w"> </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">target</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">    </span><span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span>
<span class="w">                 </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">count_ratio</span><span class="p">))</span>

<span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">,</span><span class="w"> </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">target</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">allele_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span>

<span class="n">p1</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">p2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">plot_layout</span><span class="p">(</span><span class="n">guides</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;collect&#39;</span><span class="p">)</span>
</code></pre></div>
<p>We can see that for most segments, particularly segments with a count
ratio near 1, heterozygous SNPs have allele ratios near 0.5. This
indicates that the average copy number may be 2.</p>
<blockquote>
<p><strong>Question 4:</strong> What do you think is the tumor cell fraction, and what
has most likely happened to PTEN and BRCA2?</p>
</blockquote>
<h2 id="estimating-genome-wide-copy-number">Estimating genome-wide copy number<a class="headerlink" href="#estimating-genome-wide-copy-number" title="Permanent link">&para;</a></h2>
<p>Although the ploidy and purity of this example are relatively clear,
that is not always the case. There are several
<a href="https://academic.oup.com/nar/article/44/16/e131/2460163" target="_blank">methods</a>
available that fits your data to potential solutions of ploidy and
purity, and computes the copy numbers corresponding to the best fit.
Beware, this is somewhat prone to error, especially when the purity is
below 50%, if there are few copy number alterations, and if there is
some tumor cell heterogeneity. Also, reporting the total copy number of
every segment does not make sense in most settings. You should always
review the result and curate if necessary.</p>
<h2 id="final-questions">Final questions<a class="headerlink" href="#final-questions" title="Permanent link">&para;</a></h2>
<blockquote>
<p>Some questions require a bit of exploration and perhaps more code and
visualisation. Motivate your answers to the extent you think is
reasonable, and feel free to collaborate with the other students.</p>
<p><strong>Question 5:</strong> Homozygous deletions are rarely larger than one,
occationally a few, megabases. What appears to be the sizes of the
deletions affecting PTEN and BRCA2?</p>
<p><strong>Question 6:</strong> Let's assume you suspect that this patient may have a
TMPRSS2-ERG fusion. Is that supported at all by copy number data?</p>
<p><strong>Question 7:</strong> Is there a MYC amplification?</p>
<p><strong>Question 8:</strong> How many copies of AR would you say there are in each
tumor cell?</p>
<p><strong>Question 9:</strong> Let's assume there is a genomic segment with somatic
copy-neutral loss of heterozogosity, i.e. one of the two homologous
copies was lost and the other duplicated. What count ratio would we
observe in this sample, and at what allele ratio(s) would the
(germline-heterozygous) SNPs likely appear?</p>
</blockquote>
<p>Once you have completed the exercise, make sure the resulting report
(generated by clicking <strong>Render</strong>) is a self-contained HTML file by
specifying <a href="https://quarto.org/docs/reference/formats/html.html" target="_blank">HTML format
options</a> at the top
of your Quarto document:</p>
<div class="highlight"><pre><span></span><code>author: &quot;Your Name&quot;
format:
  html:
    self-contained: true
</code></pre></div>
<p>You can add more options to make the result look better. Also check that
the report is readable, for example that it does not include full
print-outs of large data structures. When done,
<a href="mailto:markus.mayrhofer@ki.se">email</a> the rendered report (or a link if
it is too large) to the course teacher.</p>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2023, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "content.code.annotate", "content.tabs.link", "search.highlight", "search.share", "search.suggest"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.51198bba.min.js"></script>
      
    
  </body>
</html>