
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.2.3, mkdocs-material-8.1.4">
    
    
      
        <title>Copy Number Variations - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bb3983ee.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.e6a45f82.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#analysis-of-copy-number-variation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Copy Number Variations
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Lab Session" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_1" type="checkbox" id="__nav_5_1" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_1">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 1" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_2" type="checkbox" id="__nav_5_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_2">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 2" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_3" type="checkbox" id="__nav_5_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_3">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 3" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_4" type="checkbox" id="__nav_5_4" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5_4">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 4" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../structural_variation/" class="md-nav__link">
        Structural Variants
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Copy Number Variations
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Copy Number Variations
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#copy-number-analysis-tools" class="md-nav__link">
    Copy number analysis tools
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#download-data-and-prepare-r" class="md-nav__link">
    Download data and prepare R
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-sequence-target-definition" class="md-nav__link">
    Parse sequence target definition
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotate-with-gene-symbol" class="md-nav__link">
    Annotate with gene symbol
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-sequence-read-counts" class="md-nav__link">
    Parse sequence read counts
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plot-sequence-coverage" class="md-nav__link">
    Plot sequence coverage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#model-and-correct-for-gc-content-bias" class="md-nav__link">
    Model and correct for GC content bias
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#segment-the-data" class="md-nav__link">
    Segment the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-snp-allele-ratio" class="md-nav__link">
    Parse SNP allele ratio
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#estimating-genome-wide-copy-number" class="md-nav__link">
    Estimating genome-wide copy number
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#investigate-further" class="md-nav__link">
    Investigate further
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#copy-number-analysis-tools" class="md-nav__link">
    Copy number analysis tools
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#download-data-and-prepare-r" class="md-nav__link">
    Download data and prepare R
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-sequence-target-definition" class="md-nav__link">
    Parse sequence target definition
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotate-with-gene-symbol" class="md-nav__link">
    Annotate with gene symbol
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-sequence-read-counts" class="md-nav__link">
    Parse sequence read counts
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plot-sequence-coverage" class="md-nav__link">
    Plot sequence coverage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#model-and-correct-for-gc-content-bias" class="md-nav__link">
    Model and correct for GC content bias
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#segment-the-data" class="md-nav__link">
    Segment the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parse-snp-allele-ratio" class="md-nav__link">
    Parse SNP allele ratio
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#estimating-genome-wide-copy-number" class="md-nav__link">
    Estimating genome-wide copy number
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#investigate-further" class="md-nav__link">
    Investigate further
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                

<h1 id="analysis-of-copy-number-variation">Analysis of copy number variation<a class="headerlink" href="#analysis-of-copy-number-variation" title="Permanent link">&para;</a></h1>
<h3 id="copy-number-analysis-tools">Copy number analysis tools<a class="headerlink" href="#copy-number-analysis-tools" title="Permanent link">&para;</a></h3>
<p>There are plenty of tools available for copy number analysis today, and you will likely be able to find one for free that suits your particular application. Although tailored for different types of sequence data, general processing steps of all copy number analysis tools include:
1. Quantification of sequence read depth throughout the reference genome as a measure of DNA abundance in the sample
1. Removal of sample-specific systematic noise using features such as GC content and mappability
1. Removal of assay-specific systematic noise using normal (non-cancer) reference samples
1. Segmentation - partitioning of the reference genome so that each segment can be assigned one copy number
1. Copy number calling, assigning each segment some estimate of the number of copies per cell
1. Combine normalized coverage with SNP allele frequencies to support copy number estimates</p>
<p>In this exercise we will perform the above steps using R, and a BAM file of aligned reads and a VCF file of SNPs as input.</p>
<h3 id="download-data-and-prepare-r">Download data and prepare R<a class="headerlink" href="#download-data-and-prepare-r" title="Permanent link">&para;</a></h3>
<p>The data used in this exercise is available for download here: <a href="https://course-cg-5534.s3.amazonaws.com/cnvs/copy_number_files.tar.bz2">copy_number_files.tar.bz2</a></p>
<p>Extract the content into a local folder on your computer.</p>
<div class="highlight"><pre><span></span><code>tar xjvf copy_number_files.tar.bz2
</code></pre></div>
<p>Several packages from the <a href="https://www.bioconductor.org/">Bioconductor</a> repository are used in this exercise, most importantly <strong>GenomicRanges</strong> which facilitates working with anything located on the reference genome. <a href="http://bioconductor.org/packages/devel/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesHOWTOs.pdf">Here</a> is a good overview.</p>
<h3 id="parse-sequence-target-definition">Parse sequence target definition<a class="headerlink" href="#parse-sequence-target-definition" title="Permanent link">&para;</a></h3>
<p>We use targeted sequencing of a prostate cancer sample in this exercise, as the resulting data files are of a more convenient size than whole genome or exome. This hybrid-capture panel covers a few hundred cancer genes and a few thousands of additional targets intended to improve copy number and structural variant detection.</p>
<p>In RStudio, navigate to the folder containing the exercise files:
<div class="highlight"><pre><span></span><code><span class="nf">setwd</span><span class="p">(</span><span class="s">&#39;your/path/to/exercise_folder&#39;</span><span class="p">)</span>
</code></pre></div></p>
<blockquote>
<p>Tip: Open a new Rmarkdown file and save it in the exercise data folder. Keep your code as shown in the template, and write your comments and answers to questions between code chunks. Click <strong>knit</strong> any time to run all code and generate a pdf or html report. This way, when you are through this exercise, your report is done too.</p>
</blockquote>
<p>The <strong>BED</strong> (Browser Extensible Data) format is a text file format used to store genomic regions as coordinates and associated annotations. The data are presented in the form of columns separated by spaces or tabs. This format was developed during the Human Genome Project and then adopted by other sequencing projects.</p>
<p>Parse and investigate the target definition BED file.
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>

<span class="n">bed_file</span> <span class="o">&lt;-</span> <span class="s">&#39;targets.bed&#39;</span>
<span class="nf">file.exists</span><span class="p">(</span><span class="n">bed_file</span><span class="p">)</span>

<span class="n">targets</span> <span class="o">&lt;-</span> <span class="nf">fread</span><span class="p">(</span><span class="n">bed_file</span><span class="p">)</span>
<span class="n">targets</span>

<span class="c1"># Let the first column be the target number</span>
<span class="n">targets</span> <span class="o">&lt;-</span> <span class="nf">cbind</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="nf">nrow</span><span class="p">(</span><span class="n">targets</span><span class="p">),</span><span class="n">targets</span><span class="p">)</span>

<span class="c1"># Set useful column names</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">targets</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;target&#39;</span><span class="p">,</span><span class="s">&#39;chromosome&#39;</span><span class="p">,</span><span class="s">&#39;start&#39;</span><span class="p">,</span><span class="s">&#39;end&#39;</span><span class="p">)</span>

<span class="c1"># Check target length</span>
<span class="n">targets</span><span class="p">[,</span><span class="n">length</span><span class="o">:=</span><span class="n">end</span><span class="o">-</span><span class="n">start</span><span class="p">]</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">targets</span><span class="p">)</span>

<span class="c1"># Check target distribution over chromosomes</span>
<span class="n">targets</span><span class="p">[,</span><span class="nf">table</span><span class="p">(</span><span class="n">chromosome</span><span class="p">)]</span>
</code></pre></div></p>
<blockquote>
<p>The R code in this exercise features some <strong>data.table</strong> syntax. If you are more familiar with base R or tidyverse syntax, a comparison between them can be found <a href="https://mgimond.github.io/rug_2019_12/Index.html">here</a>.</p>
</blockquote>
<h3 id="annotate-with-gene-symbol">Annotate with gene symbol<a class="headerlink" href="#annotate-with-gene-symbol" title="Permanent link">&para;</a></h3>
<p>Before parsing the sequence data, let's assign gene symbols to targets that overlap a gene. We'll take advantage of databases accessible through Bioconductor packages. First, we need to create a GenomicRanges object representing our targets:</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">GenomicRanges</span><span class="p">)</span>

<span class="n">target_ranges</span> <span class="o">&lt;-</span> <span class="nf">makeGRangesFromDataFrame</span><span class="p">(</span><span class="n">targets</span><span class="p">)</span>
<span class="n">target_ranges</span>

<span class="nf">seqlevelsStyle</span><span class="p">(</span><span class="n">target_ranges</span><span class="p">)</span>
</code></pre></div>
<p>As you may recognize, chromosomes have the NCBI/EMBL sequence naming style (1,...). This matches the reference genome used with our sequence data. Let's create a second GenomicRanges object featuring the UCSC names (chr1,...) for use with some database queries.</p>
<div class="highlight"><pre><span></span><code><span class="n">ucsc_ranges</span> <span class="o">&lt;-</span> <span class="n">target_ranges</span>
<span class="nf">seqlevelsStyle</span><span class="p">(</span><span class="n">ucsc_ranges</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="s">&quot;UCSC&quot;</span>

<span class="n">ucsc_ranges</span>
</code></pre></div>
<p>We then use the <strong>detailRanges</strong> function from the <strong>csaw</strong> package to overlap our target ranges with known genes.</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">org.Hs.eg.db</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">csaw</span><span class="p">)</span>

<span class="n">d</span> <span class="o">&lt;-</span> <span class="nf">detailRanges</span><span class="p">(</span><span class="n">ucsc_ranges</span><span class="p">,</span> <span class="n">orgdb</span><span class="o">=</span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
    <span class="n">txdb</span><span class="o">=</span><span class="n">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="p">)</span>
<span class="n">d</span>

<span class="nf">as.data.table</span><span class="p">(</span><span class="n">d</span><span class="p">)</span>
</code></pre></div>
<blockquote>
<p>There are many ways to achieve the same thing. You should focus mostly on <strong>what</strong> we do, and not worry too much about exactly how. Google is your friend and if you know what you want do do, you can always find a way.</p>
</blockquote>
<p>We only care about some level of overlap at this point, let's just retrieve the gene symbols and add as a column to the table of targets. We can then see how targets allocate to genes.</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>

<span class="n">targets</span><span class="o">$</span><span class="n">gene</span> <span class="o">&lt;-</span> <span class="nf">str_remove</span><span class="p">(</span><span class="n">string</span> <span class="o">=</span> <span class="n">d</span><span class="o">$</span><span class="n">overlap</span><span class="p">,</span><span class="n">pattern</span> <span class="o">=</span> <span class="s">&#39;:.*&#39;</span><span class="p">)</span>
<span class="n">targets</span>

<span class="nf">table</span><span class="p">(</span><span class="n">targets</span><span class="o">$</span><span class="n">gene</span><span class="p">)</span>

<span class="nf">as.data.table</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">targets</span><span class="o">$</span><span class="n">gene</span><span class="p">))</span>

<span class="nf">as.data.table</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">targets</span><span class="o">$</span><span class="n">gene</span><span class="p">))[</span><span class="n">N</span><span class="o">&gt;</span><span class="m">25</span><span class="p">]</span>
</code></pre></div>
<h3 id="parse-sequence-read-counts">Parse sequence read counts<a class="headerlink" href="#parse-sequence-read-counts" title="Permanent link">&para;</a></h3>
<p>We are now ready to access the BAM file and count the number of read pairs (i.e. fragments) that map to each target. For this we use the <a href="https://bioconductor.org/packages/release/bioc/vignettes/bamsignals/inst/doc/bamsignals.html">bamsignals</a> package and the GenomicRanges object with NCBI-style sequence names. To count a read pair, we require it's midpoint to fall within a target, and that it is not flagged as a <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">duplicate</a>.</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">bamsignals</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">Rsamtools</span><span class="p">)</span>

<span class="n">tumor_bam</span> <span class="o">&lt;-</span> <span class="s">&#39;Sample1.bam&#39;</span>
<span class="nf">file.exists</span><span class="p">(</span><span class="n">tumor_bam</span><span class="p">)</span>
<span class="nf">file.exists</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">tumor_bam</span><span class="p">,</span><span class="s">&#39;.bai&#39;</span><span class="p">))</span>

<span class="n">targets</span><span class="o">$</span><span class="n">coverage</span> <span class="o">&lt;-</span> <span class="nf">bamCount</span><span class="p">(</span><span class="n">tumor_bam</span><span class="p">,</span> <span class="n">target_ranges</span><span class="p">,</span> <span class="n">paired.end</span><span class="o">=</span><span class="s">&quot;midpoint&quot;</span><span class="p">,</span> <span class="n">filteredFlag</span><span class="o">=</span><span class="m">1024</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
<span class="n">targets</span>

<span class="nf">summary</span><span class="p">(</span><span class="n">targets</span><span class="p">)</span>
</code></pre></div>
<blockquote>
<p>Strictly speaking, read count is not equal to sequence coverage. But given the fragment length and target size in this example, we can safely call the result sequence coverage.</p>
</blockquote>
<p>Zero coverage will be a problem later as it log-transforms to -Inf. We set those to 1:</p>
<div class="highlight"><pre><span></span><code><span class="n">targets</span><span class="p">[</span><span class="n">coverage</span><span class="o">==</span><span class="m">0</span><span class="p">,</span> <span class="n">coverage</span><span class="o">:=</span><span class="m">1</span><span class="p">]</span>
</code></pre></div>
<h3 id="plot-sequence-coverage">Plot sequence coverage<a class="headerlink" href="#plot-sequence-coverage" title="Permanent link">&para;</a></h3>
<p>Let's investigate the raw sequence coverage across targets. You are probably already familiar with <a href="https://ggplot2.tidyverse.org/">ggplot2</a>.
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">);</span> <span class="nf">theme_set</span><span class="p">(</span><span class="nf">theme_bw</span><span class="p">())</span>

<span class="c1"># Simple ggplot</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">))</span>

<span class="c1"># Some adjustments</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">),</span><span class="n">size</span><span class="o">=</span><span class="n">.</span><span class="m">2</span><span class="p">)</span>

<span class="c1"># To use start position, we can separate the plot by chromosome.</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">),</span><span class="n">size</span><span class="o">=</span><span class="n">.</span><span class="m">5</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">facet_wrap</span><span class="p">(</span><span class="n">facets</span> <span class="o">=</span> <span class="nf">vars</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span>
</code></pre></div></p>
<p>The figure may now look squished in RStudio's bottom-right plots pane. Click the <strong>Zoom</strong> button and maximize the new window.</p>
<p>Choose a few genes to highlight in the plot.</p>
<div class="highlight"><pre><span></span><code><span class="n">myGenes</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;AR&#39;</span><span class="p">,</span><span class="s">&#39;ATM&#39;</span><span class="p">,</span><span class="s">&#39;BRCA1&#39;</span><span class="p">,</span><span class="s">&#39;BRCA2&#39;</span><span class="p">,</span><span class="s">&#39;PTEN&#39;</span><span class="p">,</span><span class="s">&#39;TMPRSS2&#39;</span><span class="p">,</span><span class="s">&#39;ERG&#39;</span><span class="p">)</span>

<span class="n">targets</span><span class="o">$</span><span class="n">label</span><span class="o">=</span><span class="s">&#39;&#39;</span>
<span class="n">targets</span><span class="p">[</span><span class="n">gene</span> <span class="o">%in%</span> <span class="n">myGenes</span><span class="p">,</span><span class="n">label</span><span class="o">:=</span><span class="n">gene</span><span class="p">][</span><span class="n">label</span><span class="o">==</span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="n">label</span><span class="o">:=</span><span class="kc">NA</span><span class="p">]</span>


<span class="nf">ggplot</span><span class="p">()</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="n">label</span><span class="p">),</span> <span class="n">size</span><span class="o">=</span><span class="n">.</span><span class="m">2</span><span class="p">)</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="n">label</span><span class="p">),</span> <span class="n">size</span><span class="o">=</span><span class="n">.</span><span class="m">5</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">facet_wrap</span><span class="p">(</span><span class="n">facets</span> <span class="o">=</span> <span class="nf">vars</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span>
</code></pre></div>
<h3 id="model-and-correct-for-gc-content-bias">Model and correct for GC content bias<a class="headerlink" href="#model-and-correct-for-gc-content-bias" title="Permanent link">&para;</a></h3>
<p>Sequence GC content is known to affect PCR amplification and sequence coverage. Let's retrieve the GC content for the targets and investigate if there is a bias in our data set.</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">BSgenome.Hsapiens.UCSC.hg19</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">BSgenome</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">Repitools</span><span class="p">)</span>

<span class="n">targets</span><span class="o">$</span><span class="n">gc</span> <span class="o">&lt;-</span> <span class="nf">gcContentCalc</span><span class="p">(</span><span class="n">ucsc_ranges</span> <span class="p">,</span> <span class="n">organism</span><span class="o">=</span><span class="n">Hsapiens</span><span class="p">)</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">gc</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">),</span><span class="n">alpha</span><span class="o">=</span><span class="n">.</span><span class="m">2</span><span class="p">)</span>
</code></pre></div>
<p>There seems to be a modest effect of target GC content on sequence coverage. Let's build a <a href="https://en.wikipedia.org/wiki/Local_regression">loess</a> model of the effect. We leave the X chromosome out.</p>
<div class="highlight"><pre><span></span><code><span class="n">loess_tumor</span> <span class="o">&lt;-</span> <span class="nf">loess</span><span class="p">(</span><span class="n">coverage</span> <span class="o">~</span> <span class="n">gc</span><span class="p">,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">,</span> 
                    <span class="n">subset</span> <span class="o">=</span> <span class="n">chromosome</span><span class="o">!=</span><span class="s">&#39;X&#39;</span><span class="p">,</span> <span class="n">family</span><span class="o">=</span><span class="s">&quot;symmetric&quot;</span><span class="p">,</span>
                    <span class="n">control</span> <span class="o">=</span> <span class="nf">loess.control</span><span class="p">(</span><span class="n">surface</span> <span class="o">=</span> <span class="s">&quot;direct&quot;</span><span class="p">))</span>
</code></pre></div>
<p>To plot the model, we predict the sequence coverage for GC content ranging from 1% to 100%:</p>
<div class="highlight"><pre><span></span><code><span class="n">tumor_gc_line</span> <span class="o">&lt;-</span> <span class="nf">data.table</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="m">100</span><span class="o">/</span><span class="m">100</span><span class="p">,</span>
                        <span class="n">y</span><span class="o">=</span><span class="nf">predict</span><span class="p">(</span><span class="n">loess_tumor</span><span class="p">,</span><span class="nf">data.table</span><span class="p">(</span><span class="n">gc</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="m">100</span><span class="o">/</span><span class="m">100</span><span class="p">)))</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">targets</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">gc</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">),</span><span class="n">alpha</span><span class="o">=</span><span class="n">.</span><span class="m">2</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">geom_line</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">tumor_gc_line</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">),</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;blue&#39;</span><span class="p">)</span>
</code></pre></div>
<p>We seem to predict negative coverage at some high GC content. We'll try the same model on log2 of coverage. It will now return the predicted log2 coverage, which we can either keep using, or transform back to coverage. We'll keep using coverage for now.</p>
<div class="highlight"><pre><span></span><code><span class="n">loess_tumor</span> <span class="o">&lt;-</span> <span class="nf">loess</span><span class="p">(</span><span class="nf">log2</span><span class="p">(</span><span class="n">coverage</span><span class="p">)</span> <span class="o">~</span> <span class="n">gc</span><span class="p">,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">,</span> 
                    <span class="n">subset</span> <span class="o">=</span> <span class="n">chromosome</span><span class="o">!=</span><span class="s">&#39;X&#39;</span><span class="p">,</span> <span class="n">family</span><span class="o">=</span><span class="s">&quot;symmetric&quot;</span><span class="p">,</span>
                    <span class="n">control</span> <span class="o">=</span> <span class="nf">loess.control</span><span class="p">(</span><span class="n">surface</span> <span class="o">=</span> <span class="s">&quot;direct&quot;</span><span class="p">))</span>

<span class="c1"># prediction is log2(coverage), 2^prediction equals predicted coverage</span>
<span class="n">tumor_gc_line</span> <span class="o">&lt;-</span> <span class="nf">data.table</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="m">100</span><span class="o">/</span><span class="m">100</span><span class="p">,</span>
                        <span class="n">y</span><span class="o">=</span><span class="m">2</span><span class="o">^</span><span class="nf">predict</span><span class="p">(</span><span class="n">loess_tumor</span><span class="p">,</span><span class="nf">data.table</span><span class="p">(</span><span class="n">gc</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="m">100</span><span class="o">/</span><span class="m">100</span><span class="p">)))</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">targets</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">gc</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">),</span><span class="n">alpha</span><span class="o">=</span><span class="n">.</span><span class="m">2</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">geom_line</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">tumor_gc_line</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">),</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;blue&#39;</span><span class="p">)</span>
</code></pre></div>
<p>To adjust for GC content bias, we divide the observed coverage with the predicted (using targets' GC content). This new <strong>coverage ratio</strong> should be a slightly better measure of DNA abundance.</p>
<div class="highlight"><pre><span></span><code><span class="n">targets</span><span class="p">[,</span><span class="n">coverage_ratio</span><span class="o">:=</span><span class="n">coverage</span><span class="o">/</span><span class="m">2</span><span class="o">^</span><span class="nf">predict</span><span class="p">(</span><span class="n">loess_tumor</span><span class="p">,</span><span class="n">gc</span><span class="p">)]</span>
<span class="n">targets</span>
</code></pre></div>
<p>To compare the new metric with the previous, we can plot them side-by-side. With <a href="https://patchwork.data-imaginist.com/">patchwork</a> we can use arithmetic operators to combine and align multiple plots.</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">patchwork</span><span class="p">)</span>

<span class="n">p1</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> 
    <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">gc</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">),</span><span class="n">alpha</span><span class="o">=</span><span class="n">.</span><span class="m">05</span><span class="p">)</span>

<span class="n">p2</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> 
    <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">gc</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage_ratio</span><span class="p">),</span><span class="n">alpha</span><span class="o">=</span><span class="n">.</span><span class="m">05</span><span class="p">)</span>

<span class="n">p1</span><span class="o">+</span><span class="n">p2</span>

<span class="n">p1</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span> <span class="o">+</span>
    <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span>

<span class="n">p2</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span> <span class="o">+</span>
    <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage_ratio</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span>

<span class="n">p1</span><span class="o">/</span><span class="n">p2</span> <span class="o">+</span> <span class="nf">plot_layout</span><span class="p">(</span><span class="n">guides</span> <span class="o">=</span> <span class="s">&#39;collect&#39;</span><span class="p">)</span>

<span class="n">p1</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3500</span><span class="p">))</span> <span class="o">+</span>
    <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">facet_wrap</span><span class="p">(</span><span class="n">facets</span> <span class="o">=</span> <span class="nf">vars</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span>
    <span class="nf">theme</span><span class="p">(</span><span class="n">panel.spacing</span> <span class="o">=</span> <span class="nf">unit</span><span class="p">(</span><span class="m">0</span><span class="p">,</span> <span class="s">&quot;lines&quot;</span><span class="p">),</span><span class="n">strip.text.x</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">4</span><span class="p">))</span>

<span class="n">p2</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span> <span class="o">+</span>
      <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage_ratio</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span> <span class="o">+</span>
      <span class="nf">facet_wrap</span><span class="p">(</span><span class="n">facets</span> <span class="o">=</span> <span class="nf">vars</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span>
      <span class="nf">theme</span><span class="p">(</span><span class="n">panel.spacing</span> <span class="o">=</span> <span class="nf">unit</span><span class="p">(</span><span class="m">0</span><span class="p">,</span> <span class="s">&quot;lines&quot;</span><span class="p">),</span><span class="n">strip.text.x</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">4</span><span class="p">))</span>

<span class="n">p1</span><span class="o">+</span><span class="n">p2</span> <span class="o">+</span> <span class="nf">plot_layout</span><span class="p">(</span><span class="n">guides</span> <span class="o">=</span> <span class="s">&#39;collect&#39;</span><span class="p">)</span>
</code></pre></div>
<p>The GC content correction seems to have had a small but positive effect. It is common to also use either a matched normal or a pool of normal reference samples (sequenced similarly) to remove some assay-specific noise. We have omitted that step here.</p>
<blockquote>
<p>One important reason to perform copy number analysis of cancer samples is to find homozygous deletions of tumor suppressor genes. If a patient has a hemizygous germline deletion (loss of one copy) affecting a tumor suppressor gene, and somatic deletion of the healthy copy in their tumor, would the resulting homozygous deletion still be visible if the coverage ratio of the tumor sample was divided by that of the normal sample?</p>
</blockquote>
<h3 id="segment-the-data">Segment the data<a class="headerlink" href="#segment-the-data" title="Permanent link">&para;</a></h3>
<p>Although we can already spot some apparent gains and losses, statistical tools can help us better estimate copy number segments, for which we can then calculate the most likely copy number given the observation. <strong>Circular binary segmentation</strong> (CBS) is probably the most commonly used segmentation method. Here we use the <strong>PSCBS</strong> ("Parent specific" CBS, as it can also use SNP allele data) R package as a wrapper to perform basic CBS on the GC content-adjusted coverage ratio. </p>
<p>CBS requires the DNA abundance ratio to be log-transformed, making its distribution more normal-like. Conveniently we have no zeroes in the data (that would become -Inf). </p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">PSCBS</span><span class="p">)</span>
<span class="n">targets</span><span class="p">[,</span><span class="n">log_ratio</span><span class="o">:=</span><span class="nf">log2</span><span class="p">(</span><span class="n">coverage_ratio</span><span class="p">)]</span>
<span class="n">targets</span>

<span class="n">segments</span> <span class="o">&lt;-</span> <span class="nf">segmentByCBS</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">targets</span><span class="o">$</span><span class="n">log_ratio</span><span class="p">)</span>
<span class="n">segments</span>
</code></pre></div>
<p>After segmentation we can transform the segment mean values back and plot the segments with the targets.</p>
<div class="highlight"><pre><span></span><code><span class="n">segments</span> <span class="o">&lt;-</span> <span class="nf">as.data.table</span><span class="p">(</span><span class="n">segments</span><span class="p">)[,</span><span class="n">coverage_ratio</span><span class="o">:=</span><span class="m">2</span><span class="o">^</span><span class="n">mean</span><span class="p">]</span>
<span class="n">segments</span>

<span class="nf">ggplot</span><span class="p">()</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">chromosome</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span>
               <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">))</span>
</code></pre></div>
<p>Note that the segmentation is based only on the target log-ratio and order, and that segment start and end refer to target number. To avoid segments spanning across chromosomes, and breakpoints just missing the chromosome boundary, we can add chromosomes to the segmentation call. The chromosome vector is required to be numeric.</p>
<div class="highlight"><pre><span></span><code><span class="n">segments</span> <span class="o">&lt;-</span> <span class="nf">segmentByCBS</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">targets</span><span class="o">$</span><span class="n">log_ratio</span><span class="p">,</span>
                      <span class="n">chromosome</span><span class="o">=</span><span class="nf">as.numeric</span><span class="p">(</span><span class="nf">str_replace</span><span class="p">(</span><span class="n">targets</span><span class="o">$</span><span class="n">chromosome</span><span class="p">,</span><span class="s">&#39;X&#39;</span><span class="p">,</span><span class="s">&#39;23&#39;</span><span class="p">)))</span>
<span class="n">segments</span>

<span class="n">segments</span> <span class="o">&lt;-</span> <span class="nf">as.data.table</span><span class="p">(</span><span class="n">segments</span><span class="p">)[,</span><span class="n">coverage_ratio</span><span class="o">:=</span><span class="m">2</span><span class="o">^</span><span class="n">mean</span><span class="p">][</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="m">-1</span><span class="p">]</span>
<span class="n">segments</span>

<span class="nf">ggplot</span><span class="p">()</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">chromosome</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span>
               <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">))</span>
</code></pre></div>
<p>CBS can also take a vector of chromosomal positions as input, in which case the resulting segment start and end positions are based on them. We use the target midpoints:</p>
<div class="highlight"><pre><span></span><code><span class="n">segments_pos</span> <span class="o">&lt;-</span> <span class="nf">segmentByCBS</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">targets</span><span class="o">$</span><span class="n">log_ratio</span><span class="p">,</span>
                         <span class="n">chromosome</span><span class="o">=</span><span class="nf">as.numeric</span><span class="p">(</span><span class="nf">str_replace</span><span class="p">(</span><span class="n">targets</span><span class="o">$</span><span class="n">chromosome</span><span class="p">,</span><span class="s">&#39;X&#39;</span><span class="p">,</span><span class="s">&#39;23&#39;</span><span class="p">)),</span>
                         <span class="n">x</span><span class="o">=</span><span class="n">targets</span><span class="o">$</span><span class="n">start</span><span class="m">+60</span><span class="p">)</span>

<span class="n">segments_pos</span> <span class="o">&lt;-</span> <span class="nf">as.data.table</span><span class="p">(</span><span class="n">segments_pos</span><span class="p">)[,</span><span class="n">coverage_ratio</span><span class="o">:=</span><span class="m">2</span><span class="o">^</span><span class="n">mean</span><span class="p">][</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="m">-1</span><span class="p">]</span>
<span class="n">segments_pos</span>


<span class="c1"># convert chromosomes back to NCBI</span>
<span class="n">segments_pos</span><span class="p">[,</span><span class="n">chromosome</span><span class="o">:=</span><span class="nf">str_replace</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="s">&#39;23&#39;</span><span class="p">,</span><span class="s">&#39;X&#39;</span><span class="p">)]</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage_ratio</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments_pos</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span>
               <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">facet_wrap</span><span class="p">(</span><span class="n">facets</span> <span class="o">=</span> <span class="nf">vars</span><span class="p">(</span><span class="n">chromosome</span><span class="p">),</span><span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">theme</span><span class="p">(</span><span class="n">panel.spacing</span> <span class="o">=</span> <span class="nf">unit</span><span class="p">(</span><span class="m">0</span><span class="p">,</span> <span class="s">&quot;lines&quot;</span><span class="p">),</span><span class="n">strip.text.x</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">6</span><span class="p">))</span>
</code></pre></div>
<blockquote>
<p>Target density is only about one per megabase, with much higher density in some cancer genes. How many targets do you think a deletion would have to cover for us to be able to find it? What else might influence sensitivity?</p>
</blockquote>
<p>Let's now take another look at segmented targets plotted in order, as they are very unevenly distributed over the genome. We can easily spot some deletions affecting 2-3 genes in our selection, as well as amplification of another. Take a closer look at BRCA2 and part of PTEN. </p>
<div class="highlight"><pre><span></span><code><span class="n">p1</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">()</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span> <span class="o">+</span> 
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">chromosome</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span>
               <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">))</span>

<span class="n">p2</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">()</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span> <span class="o">+</span> 
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span>
               <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">))</span>

<span class="n">p1</span><span class="o">/</span><span class="n">p2</span>
</code></pre></div>
<p>Coverage ratio (observed sequence coverage relative to the expected, given target GC content), is near 0.5, or 50%. For a largely diploid genome this is exactly what to expect if these segments were hemizygously deleted, i.e., one out of the two homologous copies would have been lost. This would also fit reasonably well with gain of 1 copy of 8q (resulting in a coverage ratio of 1.5) and 3 extra copies of AR (being on the X chromosome, tumor cells would have quadrupled its single original AR copy, resulting in the coverage ratio going from about 0.5 to near 2.).</p>
<p>But this solution would require the tumor cell content ("<strong>purity</strong>") in the sample to be near 100%, which is rare unless the sample comes from a cell line. Note also that several other segments have a mean coverage ratio that would not fit this solution well. There are actually multiple other combinations of copy number and purity that would fit the observation reasonably well. </p>
<p>Our measure of relative DNA abundance - coverage ratio - is automatically centered around 1, regardless of the average copy number ("<strong>ploidy</strong>") of the sequenced genome. Let's assume the cancer cells had twice the number of copies of all chromosomes. This would not double the average sequence coverage of the sample, and we would also not observe twice the sequence coverage relative to our GC content model for most of the genome. </p>
<blockquote>
<p>We have analyzed a certain amount of DNA, not a certain number of cells. The average DNA abundance measured throughout the genome, whether it is raw sequence coverage, coverage ratio, or log ratio, does not increase in samples with a higher average copy number (more DNA per cell). Instead the average measured DNA abundance will be observed at the average copy number, whatever that is. Does that make sense?</p>
</blockquote>
<p>We are also likely to have some normal DNA content in the sample. If both the tumor genome and normal genome are near diploid on average, and the tumor cell content is near 50%, homozygous deletion (loss of both homologous copies) would be observed at a coverge ratio near 0.5, as for every tumor cell that would contribute zero copies to the DNA extraction, one normal cell would contribute its normal two. As loss of one copy in the tumor cell fraction would then appear at about 0.75 and gains would appear near 1.25, 1.5, ..., this solution also appears to fit our data reasonably well.</p>
<p>Fortunately there is another tool we can use to inform our copy number estimates. We can use SNPs from a small variant caller and investigate their variant allele ratios.</p>
<h3 id="parse-snp-allele-ratio">Parse SNP allele ratio<a class="headerlink" href="#parse-snp-allele-ratio" title="Permanent link">&para;</a></h3>
<p>Let's use the VariantAnnotation package to parse a VCF file containing SNPs.</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">VariantAnnotation</span><span class="p">)</span>

<span class="n">vcf</span> <span class="o">&lt;-</span> <span class="nf">readVcf</span><span class="p">(</span><span class="s">&#39;Sample1.vcf&#39;</span><span class="p">)</span>
<span class="n">vcf</span>

<span class="n">g</span> <span class="o">&lt;-</span> <span class="nf">geno</span><span class="p">(</span><span class="n">vcf</span><span class="p">)</span>
<span class="n">g</span>

<span class="n">g</span><span class="o">$</span><span class="n">AD</span>

<span class="nf">as.data.table</span><span class="p">(</span><span class="n">g</span><span class="o">$</span><span class="n">AD</span><span class="p">)</span>

<span class="nf">as.data.table</span><span class="p">(</span><span class="n">g</span><span class="o">$</span><span class="n">DP</span><span class="p">)</span>
</code></pre></div>
<p>The AD column of the VCF file contains the number of reads supporting the reference and alternative allele. The DP column contains the total read depth. Let's make a table of SNPs containing the alt-allele coverage ratio.</p>
<div class="highlight"><pre><span></span><code><span class="n">snp_table</span> <span class="o">&lt;-</span> <span class="nf">data.table</span><span class="p">(</span><span class="n">id</span><span class="o">=</span><span class="nf">names</span><span class="p">(</span><span class="n">vcf</span><span class="p">),</span>
                        <span class="n">AD</span><span class="o">=</span><span class="nf">sapply</span><span class="p">(</span><span class="n">g</span><span class="o">$</span><span class="n">AD</span><span class="p">,</span> <span class="s">&quot;[[&quot;</span><span class="p">,</span> <span class="m">2</span><span class="p">),</span>
                        <span class="n">DP</span><span class="o">=</span><span class="nf">unlist</span><span class="p">(</span><span class="n">g</span><span class="o">$</span><span class="n">DP</span><span class="p">[,</span><span class="m">1</span><span class="p">]))</span>
<span class="n">snp_table</span>

<span class="n">snp_table</span><span class="p">[,</span><span class="n">allele_ratio</span><span class="o">:=</span><span class="nf">round</span><span class="p">(</span><span class="n">AD</span><span class="o">/</span><span class="n">DP</span><span class="p">,</span><span class="m">3</span><span class="p">)]</span>
<span class="n">snp_table</span>
</code></pre></div>
<p>A GenomicRanges object defining the SNPs' positions on the reference genome can be accessed with the rowRanges() function. We can now overlap that with the GenomicRanges representing our targets and use the resulting Hits object to assign allele ratio to targets.</p>
<div class="highlight"><pre><span></span><code><span class="n">overlaps</span> <span class="o">&lt;-</span> <span class="nf">findOverlaps</span><span class="p">(</span><span class="n">target_ranges</span><span class="p">,</span><span class="nf">rowRanges</span><span class="p">(</span><span class="n">vcf</span><span class="p">))</span>
<span class="n">overlaps</span>

<span class="n">targets</span><span class="o">$</span><span class="n">allele_ratio</span> <span class="o">&lt;-</span> <span class="kc">NA</span>
<span class="n">targets</span><span class="o">$</span><span class="n">allele_ratio</span><span class="p">[</span><span class="nf">queryHits</span><span class="p">(</span><span class="n">overlaps</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="n">snp_table</span><span class="o">$</span><span class="n">allele_ratio</span><span class="p">[</span><span class="nf">subjectHits</span><span class="p">(</span><span class="n">overlaps</span><span class="p">)]</span>
<span class="n">targets</span>
</code></pre></div>
<p>Let's plot the SNP allele ratio with the coverage ratio and see if the copy number status becomes more clear. Note that many targets contain no SNP, and that many SNPs are homozygous. The allele ratio of a homozygous SNP is 0 or 1 and unaffected by copy number alteration.</p>
<div class="highlight"><pre><span></span><code><span class="n">p1</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">()</span> <span class="o">+</span> <span class="nf">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2.5</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">coverage_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span> <span class="o">+</span> 
  <span class="nf">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">segments</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;green&#39;</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span>
               <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">,</span><span class="n">yend</span><span class="o">=</span><span class="n">coverage_ratio</span><span class="p">))</span>

<span class="n">p2</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">()</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">targets</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">target</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">allele_ratio</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">label</span><span class="p">),</span><span class="n">shape</span><span class="o">=</span><span class="m">21</span><span class="p">)</span>

<span class="n">p1</span> <span class="o">/</span> <span class="n">p2</span> <span class="o">+</span> <span class="nf">plot_layout</span><span class="p">(</span><span class="n">guides</span> <span class="o">=</span> <span class="s">&#39;collect&#39;</span><span class="p">)</span>
</code></pre></div>
<p>We can see that for most segments, particularly segments with a coverage ratio near 1, heterozygous SNPs have allele ratios near 0.5. This indicates that the average copy number may be 2. </p>
<blockquote>
<p>What do you think is the tumor cell content, and what has most likely happened to PTEN and BRCA2? </p>
</blockquote>
<h3 id="estimating-genome-wide-copy-number">Estimating genome-wide copy number<a class="headerlink" href="#estimating-genome-wide-copy-number" title="Permanent link">&para;</a></h3>
<p>Although the ploidy and purity of this example are relatively clear, that is not always the case. There are several <a href="https://academic.oup.com/nar/article/44/16/e131/2460163">methods</a> available that fits your data to potential ploidies and purities, and assigns the copy numbers corresponding to the best fit. Beware, this is somewhat prone to error, especially when the purity is below 50%, if there are few copy number alterations, or if there is some tumor cell heterogeneity. Also, reporting the total copy number of every segment does not always make sense. You should always review the result and curate if necessary. </p>
<h3 id="investigate-further">Investigate further<a class="headerlink" href="#investigate-further" title="Permanent link">&para;</a></h3>
<blockquote>
<p>Most copy number analysis tools use log-transformed coverage ratio (log-ratio). What may be the advantages of this?</p>
<p>Homozygous deletions are rarely larger than one, occationally a few, megabases. What appears to be the sizes of the deletions affecting PTEN and BRCA2?</p>
<p>Let's assume you suspect that this patient may have a TMPRSS2-ERG fusion. Is that supported by copy number data?</p>
<p>Is there a MYC amplification?</p>
<p>How many copies of AR would you say there are in each tumor cell?</p>
<p>Let's assume we have a segment with somatic copy-neutral loss of heterozogosity, i.e. one of the two homologous copies was lost and the other duplicated. What coverage ratio would we observe in this sample, and at what allele ratio(s) would the (germline-heterozygous) SNPs likely appear?</p>
</blockquote>

              
            </article>
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" data-md-state="hidden">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../structural_variation/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Structural Variants" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Structural Variants
            </div>
          </div>
        </a>
      
      
        
        <a href="../processing_of_rnaseq/" class="md-footer__link md-footer__link--next" aria-label="Next: RNAseq" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              RNAseq
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2021, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "navigation.sections", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest"], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.361d90f1.min.js"}</script>
    
    
      <script src="../../assets/javascripts/bundle.289a2a4b.min.js"></script>
      
    
  </body>
</html>