
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../real_life_examples_cmdline/">
      
      
        <link rel="next" href="../variant_calling/">
      
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>Bioinformatic processing of DNA data - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.342714a4.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      
  
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
  
  <style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20448%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M64%2032C28.7%2032%200%2060.7%200%2096v320c0%2035.3%2028.7%2064%2064%2064h224V368c0-26.5%2021.5-48%2048-48h112V96c0-35.3-28.7-64-64-64zm384%20320H336c-8.8%200-16%207.2-16%2016v112l32-32%2064-64z%22/%3E%3C/svg%3E');--md-admonition-icon--abstract:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20448%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M96%200C43%200%200%2043%200%2096v320c0%2053%2043%2096%2096%2096h320c17.7%200%2032-14.3%2032-32s-14.3-32-32-32v-64c17.7%200%2032-14.3%2032-32V32c0-17.7-14.3-32-32-32H96m0%20384h256v64H96c-17.7%200-32-14.3-32-32s14.3-32%2032-32m32-240c0-8.8%207.2-16%2016-16h192c8.8%200%2016%207.2%2016%2016s-7.2%2016-16%2016H144c-8.8%200-16-7.2-16-16m16%2048h192c8.8%200%2016%207.2%2016%2016s-7.2%2016-16%2016H144c-8.8%200-16-7.2-16-16s7.2-16%2016-16%22/%3E%3C/svg%3E');--md-admonition-icon--info:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20512%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M256%20512a256%20256%200%201%200%200-512%20256%20256%200%201%200%200%20512m-40-176h24v-64h-24c-13.3%200-24-10.7-24-24s10.7-24%2024-24h48c13.3%200%2024%2010.7%2024%2024v88h8c13.3%200%2024%2010.7%2024%2024s-10.7%2024-24%2024h-80c-13.3%200-24-10.7-24-24s10.7-24%2024-24m40-208a32%2032%200%201%201%200%2064%2032%2032%200%201%201%200-64%22/%3E%3C/svg%3E');--md-admonition-icon--tip:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20512%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M480%2032c0-12.9-7.8-24.6-19.8-29.6S434.5.2%20425.3%209.3L381.7%2053c-48%2048-113.1%2075-181%2075H64c-35.3%200-64%2028.7-64%2064v96c0%2035.3%2028.7%2064%2064%2064v128c0%2017.7%2014.3%2032%2032%2032h64c17.7%200%2032-14.3%2032-32V352h8.7c67.9%200%20133%2027%20181%2075l43.6%2043.6c9.2%209.2%2022.9%2011.9%2034.9%206.9s19.8-16.6%2019.8-29.6V300.3c18.6-8.8%2032-32.5%2032-60.4s-13.4-51.6-32-60.4zm-64%2076.7v262.6C357.2%20317.8%20280.5%20288%20200.7%20288H192v-96h8.7c79.8%200%20156.5-29.8%20215.3-83.3%22/%3E%3C/svg%3E');--md-admonition-icon--success:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20448%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M438.6%20105.4c12.5%2012.5%2012.5%2032.8%200%2045.3l-256%20256c-12.5%2012.5-32.8%2012.5-45.3%200l-128-128c-12.5-12.5-12.5-32.8%200-45.3s32.8-12.5%2045.3%200L160%20338.7l233.4-233.3c12.5-12.5%2032.8-12.5%2045.3%200z%22/%3E%3C/svg%3E');--md-admonition-icon--question:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20512%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M256%20512a256%20256%200%201%200%200-512%20256%20256%200%201%200%200%20512m-86.2-346.7c7.9-22.3%2029.1-37.3%2052.8-37.3h58.3c34.9%200%2063.1%2028.3%2063.1%2063.1%200%2022.6-12.1%2043.5-31.7%2054.8L280%20264.4c-.2%2013-10.9%2023.6-24%2023.6-13.3%200-24-10.7-24-24v-13.5c0-8.6%204.6-16.5%2012.1-20.8l44.3-25.4c4.7-2.7%207.6-7.7%207.6-13.1%200-8.4-6.8-15.1-15.1-15.1h-58.3c-3.4%200-6.4%202.1-7.5%205.3l-.4%201.2c-4.4%2012.5-18.2%2019-30.6%2014.6s-19-18.2-14.6-30.6l.4-1.2zM224%20352a32%2032%200%201%201%2064%200%2032%2032%200%201%201-64%200%22/%3E%3C/svg%3E');--md-admonition-icon--warning:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20512%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M256%2032c14.2%200%2027.3%207.5%2034.5%2019.8l216%20368c7.3%2012.4%207.3%2027.7.2%2040.1S486.3%20480%20472%20480H40c-14.3%200-27.6-7.7-34.7-20.1s-7-27.8.2-40.1l216-368C228.7%2039.5%20241.8%2032%20256%2032m0%20128c-13.3%200-24%2010.7-24%2024v112c0%2013.3%2010.7%2024%2024%2024s24-10.7%2024-24V184c0-13.3-10.7-24-24-24m32%20224a32%2032%200%201%200-64%200%2032%2032%200%201%200%2064%200%22/%3E%3C/svg%3E');--md-admonition-icon--failure:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20512%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M459.1%2052.4%20442.6%206.5c-1.9-3.9-6.1-6.5-10.5-6.5s-8.5%202.6-10.4%206.5l-16.5%2045.9-46%2016.8c-4.3%201.6-7.3%205.9-7.2%2010.4%200%204.5%203%208.7%207.2%2010.2l45.7%2016.8%2016.8%2045.8c1.5%204.4%205.8%207.5%2010.4%207.5s8.9-3.1%2010.4-7.5l16.5-45.8%2045.7-16.8c4.2-1.5%207.2-5.7%207.2-10.2%200-4.6-3-8.9-7.2-10.4zm-132.4%2053c-12.5-12.5-32.8-12.5-45.3%200l-2.9%202.9c-22-8-45.8-12.3-70.5-12.3C93.1%2096%200%20189.1%200%20304s93.1%20208%20208%20208%20208-93.1%20208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5%2012.5-32.8%200-45.3l-80-80zM200%20192c-57.4%200-104%2046.6-104%20104v8c0%208.8-7.2%2016-16%2016s-16-7.2-16-16v-8c0-75.1%2060.9-136%20136-136h8c8.8%200%2016%207.2%2016%2016s-7.2%2016-16%2016z%22/%3E%3C/svg%3E');--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20512%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M416%20398.9c58.5-41.1%2096-104.1%2096-174.9C512%20100.3%20397.4%200%20256%200S0%20100.3%200%20224c0%2070.7%2037.5%20133.8%2096%20174.9V464c0%2026.5%2021.5%2048%2048%2048h48v-48c0-8.8%207.2-16%2016-16s16%207.2%2016%2016v48h64v-48c0-8.8%207.2-16%2016-16s16%207.2%2016%2016v48h48c26.5%200%2048-21.5%2048-48v-65.1M96%20256a64%2064%200%201%201%20128%200%2064%2064%200%201%201-128%200m256-64a64%2064%200%201%201%200%20128%2064%2064%200%201%201%200-128%22/%3E%3C/svg%3E');--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20640%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M320%200c17.7%200%2032%2014.3%2032%2032v64h120c39.8%200%2072%2032.2%2072%2072v272c0%2039.8-32.2%2072-72%2072H168c-39.8%200-72-32.2-72-72V168c0-39.8%2032.2-72%2072-72h120V32c0-17.7%2014.3-32%2032-32M208%20384c-8.8%200-16%207.2-16%2016s7.2%2016%2016%2016h32c8.8%200%2016-7.2%2016-16s-7.2-16-16-16zm96%200c-8.8%200-16%207.2-16%2016s7.2%2016%2016%2016h32c8.8%200%2016-7.2%2016-16s-7.2-16-16-16zm96%200c-8.8%200-16%207.2-16%2016s7.2%2016%2016%2016h32c8.8%200%2016-7.2%2016-16s-7.2-16-16-16zM264%20256a40%2040%200%201%200-80%200%2040%2040%200%201%200%2080%200m152%2040a40%2040%200%201%200%200-80%2040%2040%200%201%200%200%2080M48%20224h16v192H48c-26.5%200-48-21.5-48-48v-96c0-26.5%2021.5-48%2048-48m544%200c26.5%200%2048%2021.5%2048%2048v96c0%2026.5-21.5%2048-48%2048h-16V224z%22/%3E%3C/svg%3E');--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20448%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M288%200H128c-17.7%200-32%2014.3-32%2032s14.3%2032%2032%2032v132.8c0%2011.8-3.3%2023.5-9.5%2033.5L10.3%20406.2C3.6%20417.2%200%20429.7%200%20442.6%200%20480.9%2031.1%20512%2069.4%20512h309.2c38.3%200%2069.4-31.1%2069.4-69.4%200-12.8-3.6-25.4-10.3-36.4L329.5%20230.4c-6.2-10.1-9.5-21.7-9.5-33.5V64c17.7%200%2032-14.3%2032-32S337.7%200%20320%200zm-96%20196.8V64h64v132.8c0%2023.7%206.6%2046.9%2019%2067.1l34.5%2056.1h-171l34.5-56.1c12.4-20.2%2019-43.4%2019-67.1%22/%3E%3C/svg%3E');--md-admonition-icon--quote:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20448%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M0%20216C0%20149.7%2053.7%2096%20120%2096h8c17.7%200%2032%2014.3%2032%2032s-14.3%2032-32%2032h-8c-30.9%200-56%2025.1-56%2056v8h64c35.3%200%2064%2028.7%2064%2064v64c0%2035.3-28.7%2064-64%2064H64c-35.3%200-64-28.7-64-64V216m256%200c0-66.3%2053.7-120%20120-120h8c17.7%200%2032%2014.3%2032%2032s-14.3%2032-32%2032h-8c-30.9%200-56%2025.1-56%2056v8h64c35.3%200%2064%2028.7%2064%2064v64c0%2035.3-28.7%2064-64%2064h-64c-35.3%200-64-28.7-64-64V216%22/%3E%3C/svg%3E');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bioinformatic-processing-of-dna-data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Bioinformatic processing of DNA data
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Course Details
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lectures
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Lab Session
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Lab Session
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_1" >
        
          
          <label class="md-nav__link" for="__nav_4_1" id="__nav_4_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exercise 1
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_1">
            <span class="md-nav__icon md-icon"></span>
            Exercise 1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../workspace_KI/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Workspace - KI-HPC
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../exercise_1_reference-genome_annotation-files_rawdata/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Reference Genome, Annotations and Raw Data
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Introduction_to_IGV/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction to IGV
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_examples_cmdline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Command Line Usage in Real Life
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_2" checked>
        
          
          <label class="md-nav__link" for="__nav_4_2" id="__nav_4_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exercise 2
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4_2">
            <span class="md-nav__icon md-icon"></span>
            Exercise 2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Processing of DNA sequencing data
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Processing of DNA sequencing data
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#running-bwa" class="md-nav__link">
    <span class="md-ellipsis">
      Running BWA
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#converting-sam-to-bam-format" class="md-nav__link">
    <span class="md-ellipsis">
      Converting SAM to BAM format
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sort-the-bam-files" class="md-nav__link">
    <span class="md-ellipsis">
      Sort the BAM files
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mark-duplicates" class="md-nav__link">
    <span class="md-ellipsis">
      Mark duplicates
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#position-sort-the-bam-files" class="md-nav__link">
    <span class="md-ellipsis">
      Position sort the BAM files
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-a-bam-index-for-use-with-gatk-igv-and-other-tools" class="md-nav__link">
    <span class="md-ellipsis">
      Create a BAM index for use with GATK, IGV, and other tools
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#indel-realignment" class="md-nav__link">
    <span class="md-ellipsis">
      Indel realignment
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#perform-base-quality-score-recalibration-bqsr" class="md-nav__link">
    <span class="md-ellipsis">
      Perform Base Quality Score Recalibration (BQSR)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Perform Base Quality Score Recalibration (BQSR)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#calculate-bqsr-table" class="md-nav__link">
    <span class="md-ellipsis">
      Calculate BQSR Table
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#apply-bqsr" class="md-nav__link">
    <span class="md-ellipsis">
      Apply BQSR
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#clean-up-the-sambam-files-we-no-longer-need" class="md-nav__link">
    <span class="md-ellipsis">
      Clean up the SAM/BAM files we no longer need
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-the-samtools-flagstat-tool" class="md-nav__link">
    <span class="md-ellipsis">
      Run the samtools flagstat tool
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-various-picard-collectmetrics-tools" class="md-nav__link">
    <span class="md-ellipsis">
      Run various picard CollectMetrics tools
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      Run FastQC
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-multiqc-to-produce-a-final-qc-report" class="md-nav__link">
    <span class="md-ellipsis">
      Run MultiQC to produce a final QC report
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Variant Calling
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
        
          
          <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exercise 3
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_3">
            <span class="md-nav__icon md-icon"></span>
            Exercise 3
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../calling_structural_variation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Structural Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Instructions_copy_number_lab/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Copy Number Variations
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
        
          
          <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exercise 4
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_4">
            <span class="md-nav__icon md-icon"></span>
            Exercise 4
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Bioinformatic_processing_of_RNA_data/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RNAseq
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
        
          
          <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exercise 5
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_5">
            <span class="md-nav__icon md-icon"></span>
            Exercise 5
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../exercise_5/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Variant Interpretation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_6" >
        
          
          <label class="md-nav__link" for="__nav_4_6" id="__nav_4_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exercise 6
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_6">
            <span class="md-nav__icon md-icon"></span>
            Exercise 6
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Instructions_project/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Project
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#running-bwa" class="md-nav__link">
    <span class="md-ellipsis">
      Running BWA
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#converting-sam-to-bam-format" class="md-nav__link">
    <span class="md-ellipsis">
      Converting SAM to BAM format
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sort-the-bam-files" class="md-nav__link">
    <span class="md-ellipsis">
      Sort the BAM files
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mark-duplicates" class="md-nav__link">
    <span class="md-ellipsis">
      Mark duplicates
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#position-sort-the-bam-files" class="md-nav__link">
    <span class="md-ellipsis">
      Position sort the BAM files
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-a-bam-index-for-use-with-gatk-igv-and-other-tools" class="md-nav__link">
    <span class="md-ellipsis">
      Create a BAM index for use with GATK, IGV, and other tools
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#indel-realignment" class="md-nav__link">
    <span class="md-ellipsis">
      Indel realignment
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#perform-base-quality-score-recalibration-bqsr" class="md-nav__link">
    <span class="md-ellipsis">
      Perform Base Quality Score Recalibration (BQSR)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Perform Base Quality Score Recalibration (BQSR)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#calculate-bqsr-table" class="md-nav__link">
    <span class="md-ellipsis">
      Calculate BQSR Table
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#apply-bqsr" class="md-nav__link">
    <span class="md-ellipsis">
      Apply BQSR
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#clean-up-the-sambam-files-we-no-longer-need" class="md-nav__link">
    <span class="md-ellipsis">
      Clean up the SAM/BAM files we no longer need
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-the-samtools-flagstat-tool" class="md-nav__link">
    <span class="md-ellipsis">
      Run the samtools flagstat tool
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-various-picard-collectmetrics-tools" class="md-nav__link">
    <span class="md-ellipsis">
      Run various picard CollectMetrics tools
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      Run FastQC
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-multiqc-to-produce-a-final-qc-report" class="md-nav__link">
    <span class="md-ellipsis">
      Run MultiQC to produce a final QC report
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="bioinformatic-processing-of-dna-data">Bioinformatic processing of DNA data<a class="headerlink" href="#bioinformatic-processing-of-dna-data" title="Permanent link">&para;</a></h1>
<p>In this exercise set, we will start by aligning the DNA data we inspected on Monday as FASTQ files. We will also inspect some additional quality metrics from the aligned data. After alignment, we will proceed to germline and somatic variant calling in the second part of the exercise set, in ´variant_calling.md´.</p>
<h2 id="running-bwa">Running BWA<a class="headerlink" href="#running-bwa" title="Permanent link">&para;</a></h2>
<p>To align our DNA data to a reference genome, we will be using the <a href="https://bio-bwa.sourceforge.net/">Burrows-Wheeler Aligner (BWA)</a>. BWA is a software package for mapping low-divergent sequences against a large reference genome, such as the human genome.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1"># Go to your student folder and make a folder for alignments</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>mkdir<span class="w"> </span>dna_alignments
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a><span class="c1"># First, align the normal sample</span>
</span><span id="__span-0-6"><a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a><span class="c1"># Remember to replace YOUR_KI_ID below with your ID</span>
</span><span id="__span-0-7"><a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a><span class="c1"># Runtime: ~ 10 min</span>
</span><span id="__span-0-8"><a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">2</span><span class="w"> </span>-Y<span class="w"> </span>-R<span class="w"> </span><span class="s2">&quot;@RG\tID:Exome_Norm\tPL:ILLUMINA\tPU:C1TD1ACXX-CGATGT.7\tLB:exome_norm_lib1\tSM:HCC1395BL_DNA&quot;</span><span class="w"> </span>-o<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm.sam<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/fastq/Exome_Norm/Exome_Norm_R1.fastq.gz<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/fastq/Exome_Norm/Exome_Norm_R2.fastq.gz
</span><span id="__span-0-9"><a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>
</span><span id="__span-0-10"><a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a><span class="c1"># Second, align the tumor sample</span>
</span><span id="__span-0-11"><a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a><span class="c1"># Remember to replace YOUR_KI_ID below with your ID</span>
</span><span id="__span-0-12"><a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a><span class="c1"># Runtime: ~ 15 min</span>
</span><span id="__span-0-13"><a id="__codelineno-0-13" name="__codelineno-0-13" href="#__codelineno-0-13"></a>bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">2</span><span class="w"> </span>-Y<span class="w"> </span>-R<span class="w"> </span><span class="s2">&quot;@RG\tID:Exome_Tumor\tPL:ILLUMINA\tPU:C1TD1ACXX-ATCACG.7\tLB:exome_tumor_lib1\tSM:HCC1395_DNA&quot;</span><span class="w"> </span>-o<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor.sam<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/fastq/Exome_Tumor/Exome_Tumor_R1.fastq.gz<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/fastq/Exome_Tumor/Exome_Tumor_R2.fastq.gz
</span></code></pre></div>
<h2 id="converting-sam-to-bam-format">Converting SAM to BAM format<a class="headerlink" href="#converting-sam-to-bam-format" title="Permanent link">&para;</a></h2>
<p>The output of the BWA alignment is a SAM file, which is a human-readable alignment file. Detailed information on the SAM format can be found <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">HERE</a>. Generally, we want to convert our alignment files into the BAM format, which contains the same information as the SAM file but is binary, therefore taking up less space on our servers and computers. We perform this conversion from SAM to BAM below.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="c1"># Make sure you are in the folder with the SAM files</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>samtools<span class="w"> </span>view<span class="w"> </span>-@<span class="w"> </span><span class="m">2</span><span class="w"> </span>-h<span class="w"> </span>-b<span class="w"> </span>-o<span class="w"> </span>Exome_Norm.bam<span class="w"> </span>Exome_Norm.sam
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>samtools<span class="w"> </span>view<span class="w"> </span>-@<span class="w"> </span><span class="m">2</span><span class="w"> </span>-h<span class="w"> </span>-b<span class="w"> </span>-o<span class="w"> </span>Exome_Tumor.bam<span class="w"> </span>Exome_Tumor.sam
</span><span id="__span-1-6"><a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>
</span><span id="__span-1-7"><a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a><span class="c1"># Have a look at the BAM file header and try to figure out what type of information it contains:</span>
</span><span id="__span-1-8"><a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a>samtools<span class="w"> </span>view<span class="w"> </span>-H<span class="w"> </span>Exome_Tumor.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>less<span class="w"> </span>-SN
</span><span id="__span-1-9"><a id="__codelineno-1-9" name="__codelineno-1-9" href="#__codelineno-1-9"></a>
</span><span id="__span-1-10"><a id="__codelineno-1-10" name="__codelineno-1-10" href="#__codelineno-1-10"></a><span class="c1"># Have a look at the BAM file data and try to figure out what type of information is in the BAM file:</span>
</span><span id="__span-1-11"><a id="__codelineno-1-11" name="__codelineno-1-11" href="#__codelineno-1-11"></a>samtools<span class="w"> </span>view<span class="w"> </span>Exome_Tumor.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>less<span class="w"> </span>-SN<span class="w"> </span>
</span></code></pre></div>
<h2 id="sort-the-bam-files">Sort the BAM files<a class="headerlink" href="#sort-the-bam-files" title="Permanent link">&para;</a></h2>
<p>Below, we sort our BAM files by query or read name. Reads that have the same query name are derived from the same template, so this ensures that read pairs are grouped together in the sorted BAM file. This is needed for duplicate marking in the next section.
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a><span class="c1"># Runtime: ~ 4 min</span>
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>picard<span class="w"> </span>SortSam<span class="w"> </span><span class="nv">I</span><span class="o">=</span>Exome_Norm.bam<span class="w"> </span><span class="nv">O</span><span class="o">=</span>Exome_Norm_namesorted.bam<span class="w"> </span><span class="nv">SO</span><span class="o">=</span>queryname
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>picard<span class="w"> </span>SortSam<span class="w"> </span><span class="nv">I</span><span class="o">=</span>Exome_Tumor.bam<span class="w"> </span><span class="nv">O</span><span class="o">=</span>Exome_Tumor_namesorted.bam<span class="w"> </span><span class="nv">SO</span><span class="o">=</span>queryname
</span></code></pre></div></p>
<h2 id="mark-duplicates">Mark duplicates<a class="headerlink" href="#mark-duplicates" title="Permanent link">&para;</a></h2>
<p>During library preparation, PCR is performed. Removing the PCR duplicates is necessary for substantial removal of PCR artefacts. Picard MarkDuplicates uses the names of the sequencing reads and alignment positions to identify PCR duplicates. </p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>picard<span class="w"> </span>MarkDuplicates<span class="w"> </span>-I<span class="w"> </span>Exome_Norm_namesorted.bam<span class="w"> </span>-O<span class="w"> </span>Exome_Norm_namesorted_mrkdup.bam<span class="w"> </span>-ASSUME_SORT_ORDER<span class="w"> </span>queryname<span class="w"> </span>-METRICS_FILE<span class="w"> </span>Exome_Norm_mrkdup_metrics.txt<span class="w"> </span>-QUIET<span class="w"> </span><span class="nb">true</span><span class="w"> </span>-COMPRESSION_LEVEL<span class="w"> </span><span class="m">0</span><span class="w"> </span>-VALIDATION_STRINGENCY<span class="w"> </span>LENIENT
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>picard<span class="w"> </span>MarkDuplicates<span class="w"> </span>-I<span class="w"> </span>Exome_Tumor_namesorted.bam<span class="w"> </span>-O<span class="w"> </span>Exome_Tumor_namesorted_mrkdup.bam<span class="w"> </span>-ASSUME_SORT_ORDER<span class="w"> </span>queryname<span class="w"> </span>-METRICS_FILE<span class="w"> </span>Exome_Tumor_mrkdup_metrics.txt<span class="w"> </span>-QUIET<span class="w"> </span><span class="nb">true</span><span class="w"> </span>-COMPRESSION_LEVEL<span class="w"> </span><span class="m">0</span><span class="w"> </span>-VALIDATION_STRINGENCY<span class="w"> </span>LENIENT
</span></code></pre></div>
<h2 id="position-sort-the-bam-files">Position sort the BAM files<a class="headerlink" href="#position-sort-the-bam-files" title="Permanent link">&para;</a></h2>
<p>For the subsequent steps of our DNA analysis, a position-sorted BAM file is required. Therefore, we will sort the BAM files by coordinate.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>picard<span class="w"> </span>SortSam<span class="w"> </span>-I<span class="w"> </span>Exome_Norm_namesorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>Exome_Norm_sorted_mrkdup.bam<span class="w"> </span>-SO<span class="w"> </span>coordinate
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a>picard<span class="w"> </span>SortSam<span class="w"> </span>-I<span class="w"> </span>Exome_Tumor_namesorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>Exome_Tumor_sorted_mrkdup.bam<span class="w"> </span>-SO<span class="w"> </span>coordinate
</span></code></pre></div>
<h2 id="create-a-bam-index-for-use-with-gatk-igv-and-other-tools">Create a BAM index for use with GATK, IGV, and other tools<a class="headerlink" href="#create-a-bam-index-for-use-with-gatk-igv-and-other-tools" title="Permanent link">&para;</a></h2>
<p>In order to efficiently load and search a BAM file, downstream applications typically require an index file. We will create this file next.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>picard<span class="w"> </span>BuildBamIndex<span class="w"> </span>-I<span class="w"> </span>Exome_Norm_sorted_mrkdup.bam
</span><span id="__span-5-4"><a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a>picard<span class="w"> </span>BuildBamIndex<span class="w"> </span>-I<span class="w"> </span>Exome_Tumor_sorted_mrkdup.bam
</span></code></pre></div>
<h2 id="indel-realignment">Indel realignment<a class="headerlink" href="#indel-realignment" title="Permanent link">&para;</a></h2>
<p>Depending on the downstream approach, a process called indel realignment may be needed. During alignment of the DNA data, each individual read is aligned to the reference genome individually. This may lead to misaligned reads in e.g. repetitive regions or due to somatic/germline variation. Documentation for the indel realignment step is <a href="https://github.com/broadinstitute/gatk-docs/blob/master/gatk3-tutorials/(howto)_Perform_local_realignment_around_indels.md">here</a>. In this course, HaplotypeCaller/MuTect2 will be applied to identify germline/somatic variation. These two variant callers do realignment internally and therefore we do not need to perform it here.</p>
<h2 id="perform-base-quality-score-recalibration-bqsr">Perform Base Quality Score Recalibration (BQSR)<a class="headerlink" href="#perform-base-quality-score-recalibration-bqsr" title="Permanent link">&para;</a></h2>
<p>In a nutshell, Base Quality Score Recalibration (BQSR) is a data pre-processing step that detects systematic errors made by the sequencing machine when it estimates the accuracy of each base call. This is the next step of our processing of the DNA data.</p>
<h3 id="calculate-bqsr-table">Calculate BQSR Table<a class="headerlink" href="#calculate-bqsr-table" title="Permanent link">&para;</a></h3>
<p>First, we calculate the BQSR table.</p>
<p>NOTE: For exome data, we should usually use the <code>--intervals</code> (<code>-L</code>) option in the commands below. This option excludes off-target sequences and sequences that may be poorly mapped, which have a higher error rate. Including them could lead to a skewed model and bad recalibration. <strong>We do not use this option in this exercise for simplicity, but in a real analysis it should be included.</strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>BaseRecalibrator<span class="w"> </span>-R<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa<span class="w"> </span>-I<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.table<span class="w"> </span>--known-sites<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/gatk/Homo_sapiens_assembly38.dbsnp138.vcf.gz<span class="w"> </span>--known-sites<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/gatk/Homo_sapiens_assembly38.known_indels.vcf.gz<span class="w"> </span>--known-sites<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/gatk/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz<span class="w"> </span>--preserve-qscores-less-than<span class="w"> </span><span class="m">6</span><span class="w"> </span>--disable-bam-index-caching
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>BaseRecalibrator<span class="w"> </span>-R<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa<span class="w"> </span>-I<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup_bqsr.table<span class="w"> </span>--known-sites<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/gatk/Homo_sapiens_assembly38.dbsnp138.vcf.gz<span class="w"> </span>--known-sites<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/gatk/Homo_sapiens_assembly38.known_indels.vcf.gz<span class="w"> </span>--known-sites<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/gatk/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz<span class="w"> </span>--preserve-qscores-less-than<span class="w"> </span><span class="m">6</span><span class="w"> </span>--disable-bam-index-caching
</span></code></pre></div>
<h3 id="apply-bqsr">Apply BQSR<a class="headerlink" href="#apply-bqsr" title="Permanent link">&para;</a></h3>
<p>Now, apply the BQSR table to the BAM files.
<div class="language-bash highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>ApplyBQSR<span class="w"> </span>-R<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa<span class="w"> </span>-I<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.bam<span class="w"> </span>--bqsr-recal-file<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.table<span class="w"> </span>--preserve-qscores-less-than<span class="w"> </span><span class="m">6</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">10</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">20</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">30</span>
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>ApplyBQSR<span class="w"> </span>-R<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa<span class="w"> </span>-I<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup.bam<span class="w"> </span>-O<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="w"> </span>--bqsr-recal-file<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup_bqsr.table<span class="w"> </span>--preserve-qscores-less-than<span class="w"> </span><span class="m">6</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">10</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">20</span><span class="w"> </span>--static-quantized-quals<span class="w"> </span><span class="m">30</span>
</span></code></pre></div></p>
<p>Create an index for these new BAM files.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>picard<span class="w"> </span>BuildBamIndex<span class="w"> </span><span class="nv">I</span><span class="o">=</span>Exome_Norm_sorted_mrkdup_bqsr.bam
</span><span id="__span-8-4"><a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>picard<span class="w"> </span>BuildBamIndex<span class="w"> </span><span class="nv">I</span><span class="o">=</span>Exome_Tumor_sorted_mrkdup_bqsr.bam
</span></code></pre></div>
<h2 id="clean-up-the-sambam-files-we-no-longer-need">Clean up the SAM/BAM files we no longer need<a class="headerlink" href="#clean-up-the-sambam-files-we-no-longer-need" title="Permanent link">&para;</a></h2>
<p>Keep the final sorted, duplicate-marked, BQSR BAM/bai/table files and the mrkdup.txt files. Delete everything else.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>mkdir<span class="w"> </span>final
</span><span id="__span-9-3"><a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a>mv<span class="w"> </span>*_sorted_mrkdup_bqsr.*<span class="w"> </span>final/
</span><span id="__span-9-4"><a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a>mv<span class="w"> </span>*.txt<span class="w"> </span>final/
</span><span id="__span-9-5"><a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a>
</span><span id="__span-9-6"><a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a>rm<span class="w"> </span>*.sam
</span><span id="__span-9-7"><a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a>rm<span class="w"> </span>*.bam
</span><span id="__span-9-8"><a id="__codelineno-9-8" name="__codelineno-9-8" href="#__codelineno-9-8"></a>rm<span class="w"> </span>*.bai
</span><span id="__span-9-9"><a id="__codelineno-9-9" name="__codelineno-9-9" href="#__codelineno-9-9"></a>
</span><span id="__span-9-10"><a id="__codelineno-9-10" name="__codelineno-9-10" href="#__codelineno-9-10"></a>mv<span class="w"> </span>final/*<span class="w"> </span>.
</span><span id="__span-9-11"><a id="__codelineno-9-11" name="__codelineno-9-11" href="#__codelineno-9-11"></a>rmdir<span class="w"> </span>final/
</span></code></pre></div>
<p>Next, we will look at some quality control metrics from the BAM files.</p>
<h2 id="run-the-samtools-flagstat-tool">Run the samtools flagstat tool<a class="headerlink" href="#run-the-samtools-flagstat-tool" title="Permanent link">&para;</a></h2>
<p><code>samtools flagstat</code> provides QC-metrics after aligning the reads, e.g. the fraction of read-pairs that map to different chromosomes (which should be limited for a good quality alignment).</p>
<p><div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a>
</span><span id="__span-10-3"><a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>samtools<span class="w"> </span>flagstat<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.bam<span class="w"> </span>&gt;<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_flagstat.txt
</span><span id="__span-10-4"><a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a>samtools<span class="w"> </span>flagstat<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="w"> </span>&gt;<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_flagstat.txt
</span></code></pre></div>
<strong>Question 1:</strong> Look inside the <code>Exome_Tumor_flagstat.txt</code> file and refer to the <code>samtools flagstat</code> documentation <a href="https://www.htslib.org/doc/samtools-flagstat.html">here</a>. 
* How many total reads are in the file?
* How many reads are marked as duplicates?
* What percentage of reads had a mate mapped to a different chromosome?</p>
<h2 id="run-various-picard-collectmetrics-tools">Run various picard CollectMetrics tools<a class="headerlink" href="#run-various-picard-collectmetrics-tools" title="Permanent link">&para;</a></h2>
<p>Picard is a widely used quality control tool in genomics and it is named after a famous <a href="https://en.wikipedia.org/wiki/Jean-Luc_Picard">Star Trek Captain</a>.</p>
<p>Descriptions of the outputs of picard tools can be found <a href="http://broadinstitute.github.io/picard/">here</a>. </p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="c1"># Create interval files for our exome BED files we created during Exercise Set 1</span>
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/exome
</span><span id="__span-11-3"><a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>picard<span class="w"> </span>BedToIntervalList<span class="w"> </span><span class="nv">I</span><span class="o">=</span>exome_regions.bed<span class="w"> </span><span class="nv">O</span><span class="o">=</span>exome_regions.bed.interval_list<span class="w"> </span><span class="nv">SD</span><span class="o">=</span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.dict
</span><span id="__span-11-4"><a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a>picard<span class="w"> </span>BedToIntervalList<span class="w"> </span><span class="nv">I</span><span class="o">=</span>probe_regions.bed<span class="w"> </span><span class="nv">O</span><span class="o">=</span>probe_regions.bed.interval_list<span class="w"> </span><span class="nv">SD</span><span class="o">=</span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.dict
</span><span id="__span-11-5"><a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>
</span><span id="__span-11-6"><a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a><span class="c1"># Then proceed to run Picard tools</span>
</span><span id="__span-11-7"><a id="__codelineno-11-7" name="__codelineno-11-7" href="#__codelineno-11-7"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-11-8"><a id="__codelineno-11-8" name="__codelineno-11-8" href="#__codelineno-11-8"></a>
</span><span id="__span-11-9"><a id="__codelineno-11-9" name="__codelineno-11-9" href="#__codelineno-11-9"></a><span class="c1"># Picard CollectInsertSizeMetrics</span>
</span><span id="__span-11-10"><a id="__codelineno-11-10" name="__codelineno-11-10" href="#__codelineno-11-10"></a>picard<span class="w"> </span>CollectInsertSizeMetrics<span class="w"> </span>-I<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_insert_size_metrics.txt<span class="w"> </span>-H<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_insert_size_metrics.pdf
</span><span id="__span-11-11"><a id="__codelineno-11-11" name="__codelineno-11-11" href="#__codelineno-11-11"></a>picard<span class="w"> </span>CollectInsertSizeMetrics<span class="w"> </span>-I<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_insert_size_metrics.txt<span class="w"> </span>-H<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_insert_size_metrics.pdf
</span><span id="__span-11-12"><a id="__codelineno-11-12" name="__codelineno-11-12" href="#__codelineno-11-12"></a>
</span><span id="__span-11-13"><a id="__codelineno-11-13" name="__codelineno-11-13" href="#__codelineno-11-13"></a><span class="c1"># Picard CollectAlignmentSummaryMetrics</span>
</span><span id="__span-11-14"><a id="__codelineno-11-14" name="__codelineno-11-14" href="#__codelineno-11-14"></a>picard<span class="w"> </span>CollectAlignmentSummaryMetrics<span class="w"> </span>-I<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_alignment_metrics.txt<span class="w"> </span>-R<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa
</span><span id="__span-11-15"><a id="__codelineno-11-15" name="__codelineno-11-15" href="#__codelineno-11-15"></a>picard<span class="w"> </span>CollectAlignmentSummaryMetrics<span class="w"> </span>-I<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_exome_alignment_metrics.txt<span class="w"> </span>-R<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa
</span><span id="__span-11-16"><a id="__codelineno-11-16" name="__codelineno-11-16" href="#__codelineno-11-16"></a>
</span><span id="__span-11-17"><a id="__codelineno-11-17" name="__codelineno-11-17" href="#__codelineno-11-17"></a><span class="c1">#Picard CollectHsMetrics</span>
</span><span id="__span-11-18"><a id="__codelineno-11-18" name="__codelineno-11-18" href="#__codelineno-11-18"></a><span class="c1">#Exome only</span>
</span><span id="__span-11-19"><a id="__codelineno-11-19" name="__codelineno-11-19" href="#__codelineno-11-19"></a>picard<span class="w"> </span>CollectHsMetrics<span class="w"> </span>-I<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_hs_metrics.txt<span class="w"> </span>-R<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa<span class="w"> </span>-BI<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/exome/probe_regions.bed.interval_list<span class="w"> </span>-TI<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/exome/exome_regions.bed.interval_list
</span><span id="__span-11-20"><a id="__codelineno-11-20" name="__codelineno-11-20" href="#__codelineno-11-20"></a>picard<span class="w"> </span>CollectHsMetrics<span class="w"> </span>-I<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="w"> </span>-O<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_hs_metrics.txt<span class="w"> </span>-R<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa<span class="w"> </span>-BI<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/exome/probe_regions.bed.interval_list<span class="w"> </span>-TI<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/references/exome/exome_regions.bed.interval_list
</span></code></pre></div>
<h2 id="run-fastqc">Run FastQC<a class="headerlink" href="#run-fastqc" title="Permanent link">&para;</a></h2>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-12-2"><a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a>fastqc<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>Exome_Norm_sorted_mrkdup_bqsr.bam
</span><span id="__span-12-3"><a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a>fastqc<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>Exome_Tumor_sorted_mrkdup_bqsr.bam
</span><span id="__span-12-4"><a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a>tree
</span></code></pre></div>
<h2 id="run-multiqc-to-produce-a-final-qc-report">Run MultiQC to produce a final QC report<a class="headerlink" href="#run-multiqc-to-produce-a-final-qc-report" title="Permanent link">&para;</a></h2>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="nb">cd</span><span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-13-2"><a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>mkdir<span class="w"> </span>post_align_qc
</span><span id="__span-13-3"><a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a><span class="nb">cd</span><span class="w"> </span>post_align_qc
</span><span id="__span-13-4"><a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a>multiqc<span class="w"> </span>/nfs/course/students/YOUR_KI_ID/data/dna_alignments
</span><span id="__span-13-5"><a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a>tree
</span><span id="__span-13-6"><a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a>
</span><span id="__span-13-7"><a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a><span class="c1"># Download the MultiQC output to your local computer and open the .html in your favorite browser</span>
</span><span id="__span-13-8"><a id="__codelineno-13-8" name="__codelineno-13-8" href="#__codelineno-13-8"></a>scp<span class="w"> </span>YOUR_KI_ID@c8cancergen02.ki.se:/nfs/course/students/YOUR_KI_ID/data/dna_alignments/post_align_qc/multiqc_report.html<span class="w"> </span>.
</span></code></pre></div>
<p><strong>Question 2:</strong> Look through the MultiQC report and answer the following:
* Look at the Picard Alignment Summary. What percentage of reads in the Tumor and Normal were unaligned? Do you think this is a good quality alignment result?
* Look at the Picard Mark Duplicates plot. What was the most common type of duplicate that was marked? (<em>FYI- optical duplicates are reads that result from a single amplification cluster, incorrectly detected as multiple clusters by the optical sensor of the sequencing instrument.</em>)</p>







  
  






                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2021, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "navigation.sections", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest"], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>