
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Project documentation with Markdown.">
      
      
      
      
      
      <link rel="icon" href="../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.1.6">
    
    
      
        <title>HUMAN GENOME ANNOTATION FILES - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.ded33207.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.a0c5b2b5.min.css">
      
      
  
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
  
  <style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M64 32C28.7 32 0 60.7 0 96v320c0 35.3 28.7 64 64 64h224V368c0-26.5 21.5-48 48-48h112V96c0-35.3-28.7-64-64-64H64zm384 320H336c-8.8 0-16 7.2-16 16v112l32-32 64-64 32-32z"/></svg>');--md-admonition-icon--abstract:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M96 0C43 0 0 43 0 96v320c0 53 43 96 96 96h320c17.7 0 32-14.3 32-32s-14.3-32-32-32v-64c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H96zm0 384h256v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"/></svg>');--md-admonition-icon--info:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-40-176h24v-64h-24c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24h-80c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"/></svg>');--md-admonition-icon--tip:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M480 32c0-12.9-7.8-24.6-19.8-29.6S434.5.2 425.3 9.3L381.7 53c-48 48-113.1 75-181 75H64c-35.3 0-64 28.7-64 64v96c0 35.3 28.7 64 64 64v128c0 17.7 14.3 32 32 32h64c17.7 0 32-14.3 32-32V352h8.7c67.9 0 133 27 181 75l43.6 43.6c9.2 9.2 22.9 11.9 34.9 6.9s19.8-16.6 19.8-29.6V300.4c18.6-8.8 32-32.5 32-60.4s-13.4-51.6-32-60.4V32zm-64 76.7v262.6C357.2 317.8 280.5 288 200.7 288H192v-96h8.7c79.8 0 156.5-29.8 215.3-83.3z"/></svg>');--md-admonition-icon--success:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7l233.4-233.3c12.5-12.5 32.8-12.5 45.3 0z"/></svg>');--md-admonition-icon--question:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-86.2-346.7c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1 0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6-13.3 0-24-10.7-24-24v-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1 0-8.4-6.8-15.1-15.1-15.1h-58.3c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1-64 0z"/></svg>');--md-admonition-icon--warning:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7.2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8.2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24v112c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0-64 0 32 32 0 1 0 64 0z"/></svg>');--md-admonition-icon--failure:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.1 52.4 442.6 6.5c-1.9-3.9-6.1-6.5-10.5-6.5s-8.5 2.6-10.4 6.5l-16.5 45.9-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4 0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2 0-4.6-3-8.9-7.2-10.4l-45.6-16.8zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9c-22-8-45.8-12.3-70.5-12.3C93.1 96 0 189.1 0 304s93.1 208 208 208 208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg>');--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M416 398.9c58.5-41.1 96-104.1 96-174.9C512 100.3 397.4 0 256 0S0 100.3 0 224c0 70.7 37.5 133.8 96 174.9V464c0 26.5 21.5 48 48 48h48v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h64v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h48c26.5 0 48-21.5 48-48v-65.1zM96 256a64 64 0 1 1 128 0 64 64 0 1 1-128 0zm256-64a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"/></svg>');--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M320 0c17.7 0 32 14.3 32 32v64h120c39.8 0 72 32.2 72 72v272c0 39.8-32.2 72-72 72H168c-39.8 0-72-32.2-72-72V168c0-39.8 32.2-72 72-72h120V32c0-17.7 14.3-32 32-32zM208 384c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zM264 256a40 40 0 1 0-80 0 40 40 0 1 0 80 0zm152 40a40 40 0 1 0 0-80 40 40 0 1 0 0 80zM48 224h16v192H48c-26.5 0-48-21.5-48-48v-96c0-26.5 21.5-48 48-48zm544 0c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48h-16V224h16z"/></svg>');--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M288 0H128c-17.7 0-32 14.3-32 32s14.3 32 32 32v132.8c0 11.8-3.3 23.5-9.5 33.5L10.3 406.2C3.6 417.2 0 429.7 0 442.6 0 480.9 31.1 512 69.4 512h309.2c38.3 0 69.4-31.1 69.4-69.4 0-12.8-3.6-25.4-10.3-36.4L329.5 230.4c-6.2-10.1-9.5-21.7-9.5-33.5V64c17.7 0 32-14.3 32-32S337.7 0 320 0h-32zm-96 196.8V64h64v132.8c0 23.7 6.6 46.9 19 67.1l34.5 56.1h-171l34.5-56.1c12.4-20.2 19-43.4 19-67.1z"/></svg>');--md-admonition-icon--quote:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 216C0 149.7 53.7 96 120 96h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V216zm256 0c0-66.3 53.7-120 120-120h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64h-64c-35.3 0-64-28.7-64-64V216z"/></svg>');}</style>


    
    
    
      
    
    
      <link rel="stylesheet" href="../assets/css/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#human-genome-annotation-files" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              HUMAN GENOME ANNOTATION FILES
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href=".." class="md-tabs__link">
      Course Details
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../schedule/" class="md-tabs__link">
      Schedule
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../aws/aws_ec2_instances/" class="md-tabs__link">
      AWS Instances Login
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../lectures/lectures/" class="md-tabs__link">
      Lectures
    </a>
  </li>

      
        
  
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../lab_session/basic_unix/" class="md-tabs__link">
        Lab Session
      </a>
    </li>
  

  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_5_1" id="__nav_5_1_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/ggplot2/" class="md-nav__link">
        ggplot tutorial
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_4" >
      
      
      
        <label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/structural_variation/" class="md-nav__link">
        Structural Variants
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/analysis_of_cnv/" class="md-nav__link">
        Copy Number Analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
          Day 5
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          Day 5
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/joint_curation/" class="md-nav__link">
        Joint Curation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/genomic_alteration_interpretation_exe/" class="md-nav__link">
        Genomic Alterations Interpretation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#files-for-annotation-of-variation-in-the-human-genome" class="md-nav__link">
    Files for annotation of variation in the human genome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#index-the-variation-files" class="md-nav__link">
    Index the variation files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interval-files-and-coordinates-for-the-exome-sequencing-assay" class="md-nav__link">
    Interval files and coordinates for the exome sequencing assay
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#calculate-the-size-of-the-seqcap-v3-exome" class="md-nav__link">
    Calculate the size of the SeqCap v3 exome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-an-inverval-list-for-the-exome-bed-files" class="md-nav__link">
    Create an inverval list for the exome bed files
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="human-genome-annotation-files">HUMAN GENOME ANNOTATION FILES<a class="headerlink" href="#human-genome-annotation-files" title="Permanent link">&para;</a></h1>
<h4 id="files-for-annotation-of-variation-in-the-human-genome">Files for annotation of variation in the human genome<a class="headerlink" href="#files-for-annotation-of-variation-in-the-human-genome" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/references/
mkdir<span class="w"> </span>-p<span class="w"> </span>gatk
<span class="nb">cd</span><span class="w"> </span>gatk

<span class="c1">#In case gsutil needs to be installed</span>
conda<span class="w"> </span>install<span class="w"> </span>gsutil

<span class="c1"># SNP calibration call sets - dbsnp, hapmap, omni, and 1000G</span>
<span class="c1"># Runtime: &lt; 2min</span>
gsutil<span class="w"> </span>cp<span class="w"> </span>gs://genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.dbsnp138.vcf<span class="w"> </span>.
<span class="c1"># Runtime: ~ 2min</span>
bgzip<span class="w"> </span>--threads<span class="w"> </span><span class="m">8</span><span class="w"> </span>Homo_sapiens_assembly38.dbsnp138.vcf
gsutil<span class="w"> </span>cp<span class="w"> </span>gs://genomics-public-data/resources/broad/hg38/v0/hapmap_3.3.hg38.vcf.gz<span class="w"> </span>.
gsutil<span class="w"> </span>cp<span class="w"> </span>gs://genomics-public-data/resources/broad/hg38/v0/1000G_omni2.5.hg38.vcf.gz<span class="w"> </span>.
gsutil<span class="w"> </span>cp<span class="w"> </span>gs://genomics-public-data/resources/broad/hg38/v0/1000G_phase1.snps.high_confidence.hg38.vcf.gz<span class="w"> </span>.

<span class="c1"># Indel calibration call sets - dbsnp, Mills</span>
gsutil<span class="w"> </span>cp<span class="w"> </span>gs://genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.known_indels.vcf.gz<span class="w"> </span>.
gsutil<span class="w"> </span>cp<span class="w"> </span>gs://genomics-public-data/resources/broad/hg38/v0/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz<span class="w"> </span>.

<span class="c1"># Interval lists that can be used to parallelize certain GATK tasks</span>
gsutil<span class="w"> </span>cp<span class="w"> </span>gs://genomics-public-data/resources/broad/hg38/v0/wgs_calling_regions.hg38.interval_list<span class="w"> </span>.
gsutil<span class="w"> </span>cp<span class="w"> </span>-r<span class="w"> </span>gs://genomics-public-data/resources/broad/hg38/v0/scattered_calling_intervals/<span class="w"> </span>.

<span class="c1"># list the files we just downloaded</span>
ls<span class="w"> </span>-lh
</code></pre></div>
<h4 id="index-the-variation-files">Index the variation files<a class="headerlink" href="#index-the-variation-files" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/references/gatk/

<span class="c1">#SNP calibration call sets - dbsnp, hapmap, omni, and 1000G</span>
<span class="c1"># Runtime: ~ 4min</span>
gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>IndexFeatureFile<span class="w"> </span>-I<span class="w"> </span>~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.dbsnp138.vcf.gz
gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>IndexFeatureFile<span class="w"> </span>-I<span class="w"> </span>~/workspace/inputs/references/gatk/hapmap_3.3.hg38.vcf.gz
gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>IndexFeatureFile<span class="w"> </span>-I<span class="w"> </span>~/workspace/inputs/references/gatk/1000G_omni2.5.hg38.vcf.gz
<span class="c1"># Runtime: ~ 3min</span>
gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>IndexFeatureFile<span class="w"> </span>-I<span class="w"> </span>~/workspace/inputs/references/gatk/1000G_phase1.snps.high_confidence.hg38.vcf.gz

<span class="c1">#Indel calibration call sets - dbsnp, Mills</span>
gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>IndexFeatureFile<span class="w"> </span>-I<span class="w"> </span>~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.known_indels.vcf.gz
gatk<span class="w"> </span>--java-options<span class="w"> </span><span class="s1">&#39;-Xmx12g&#39;</span><span class="w"> </span>IndexFeatureFile<span class="w"> </span>-I<span class="w"> </span>~/workspace/inputs/references/gatk/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz
</code></pre></div>
<h3 id="interval-files-and-coordinates-for-the-exome-sequencing-assay">Interval files and coordinates for the exome sequencing assay<a class="headerlink" href="#interval-files-and-coordinates-for-the-exome-sequencing-assay" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1"># change directories</span>
mkdir<span class="w"> </span>-p<span class="w"> </span>~/workspace/inputs/references/exome
<span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/references/exome

<span class="c1"># download the files</span>
wget<span class="w"> </span>-c<span class="w"> </span>https://sequencing.roche.com/content/dam/rochesequence/worldwide/shared-designs/SeqCapEZ_Exome_v3.0_Design_Annotation_files.zip
unzip<span class="w"> </span>SeqCapEZ_Exome_v3.0_Design_Annotation_files.zip

<span class="c1"># remove the zip</span>
rm<span class="w"> </span>-f<span class="w"> </span>SeqCapEZ_Exome_v3.0_Design_Annotation_files.zip

<span class="c1"># Lift-over of the Roche coordinates from hg19 to the hg38 assembly.</span>
<span class="c1"># download the software</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/liftOver
chmod<span class="w"> </span>+x<span class="w"> </span>liftOver

<span class="c1"># change to the appropriate directory</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/references/exome

<span class="c1"># download the chain file</span>
wget<span class="w"> </span>-c<span class="w"> </span>http://hgdownload.cse.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz

<span class="c1"># run liftover</span>
liftOver<span class="w"> </span>SeqCapEZ_Exome_v3.0_Design_Annotation_files/SeqCap_EZ_Exome_v3_hg19_primary_targets.bed<span class="w">  </span>hg19ToHg38.over.chain.gz<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_primary_targets.bed<span class="w"> </span>unMapped.bed

liftOver<span class="w"> </span>SeqCapEZ_Exome_v3.0_Design_Annotation_files/SeqCap_EZ_Exome_v3_hg19_capture_targets.bed<span class="w">  </span>hg19ToHg38.over.chain.gz<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_capture_targets.bed<span class="w"> </span>unMapped1.bed

<span class="c1"># create a version in standard bed format (chr, start, stop)</span>
cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span>-3<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_primary_targets.bed<span class="w"> </span>&gt;<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.bed

cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span>-3<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_capture_targets.bed<span class="w"> </span>&gt;<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.bed

<span class="c1"># take a quick look at the format of these files</span>
head<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.bed
head<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.bed
</code></pre></div>
<h3 id="calculate-the-size-of-the-seqcap-v3-exome">Calculate the size of the SeqCap v3 exome<a class="headerlink" href="#calculate-the-size-of-the-seqcap-v3-exome" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1">#This can be done in many ways - give it a try yourself before trying the code below and compare results</span>

<span class="c1"># first sort the bed files and store the sorted versions</span>
bedtools<span class="w"> </span>sort<span class="w"> </span>-i<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.bed<span class="w"> </span>&gt;<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.bed
bedtools<span class="w"> </span>sort<span class="w"> </span>-i<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.bed<span class="w"> </span>&gt;<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.bed

<span class="c1"># now merge the bed files to collapse any overlapping regions so they are not double counted.</span>
bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.bed<span class="w"> </span>&gt;<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.bed

bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.bed<span class="w"> </span>&gt;<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.merge.bed

<span class="c1"># finally use a Perl one liner to determine the size of the files in Mb</span>
<span class="nv">FILES</span><span class="o">=(</span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.merge.bed<span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.bed<span class="o">)</span>
<span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">FILES</span><span class="p">[0]</span><span class="si">}</span>

<span class="k">for</span><span class="w"> </span>FILE<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="si">${</span><span class="nv">FILES</span><span class="p">[@]</span><span class="si">}</span>
<span class="k">do</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;--------------------------------------------------------&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="nv">$FILE</span>
<span class="c1">#With merge</span>
cat<span class="w"> </span><span class="nv">$FILE</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sortBed<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span><span class="p">|</span><span class="w"> </span>mergeBed<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span><span class="p">|</span><span class="w"> </span>perl<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;$counter=0; while(&lt;&gt;){chomp; @array = split &quot;\t&quot;, $_; $counter = $counter + ($array[2] - $array[1]);}; print $counter/1000000, &quot;\n&quot;;&#39;</span>
<span class="k">done</span>
<span class="c1"># note that the size of the space targeted by the exome reagent is ~63 Mbp. Does that sound reasonable?</span>

<span class="c1"># now create a subset of these bed files</span>
grep<span class="w"> </span>-w<span class="w"> </span>-P<span class="w"> </span><span class="s2">&quot;^chr6|^chr17&quot;</span><span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.bed<span class="w"> </span>&gt;<span class="w"> </span>exome_regions.bed
<span class="c1">#When creating files, make a habit to investigate the output to avoid downstream confusion</span>
head<span class="w"> </span>-n<span class="w"> </span><span class="m">10</span><span class="w"> </span>exome_regions.bed

grep<span class="w"> </span>-w<span class="w"> </span>-P<span class="w"> </span><span class="s2">&quot;^chr6|^chr17&quot;</span><span class="w"> </span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.merge.bed<span class="w"> </span>&gt;<span class="w"> </span>probe_regions.bed
head<span class="w"> </span>-n<span class="w"> </span><span class="m">10</span><span class="w"> </span>probe_regions.bed

<span class="c1"># clean up intermediate files</span>
<span class="c1">#rm -fr SeqCapEZ_Exome_v3.0_Design_Annotation_files/ SeqCap_EZ_Exome_v3_hg38_primary_targets.bed SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.bed SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.bed unMapped.bed</span>
</code></pre></div>
<h3 id="create-an-inverval-list-for-the-exome-bed-files">Create an inverval list for the exome bed files<a class="headerlink" href="#create-an-inverval-list-for-the-exome-bed-files" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1"># first for the complete exome and probe bed file</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/references/
mkdir<span class="w"> </span>temp
<span class="nb">cd</span><span class="w"> </span>temp
wget<span class="w"> </span>http://genomedata.org/pmbio-workshop/references/genome/all/ref_genome.dict
<span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/references/exome
java<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>BedToIntervalList<span class="w"> </span><span class="nv">I</span><span class="o">=</span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.bed<span class="w"> </span><span class="nv">O</span><span class="o">=</span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.interval_list<span class="w"> </span><span class="nv">SD</span><span class="o">=</span>~/workspace/inputs/references/temp/ref_genome.dict
java<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$PICARD</span><span class="w"> </span>BedToIntervalList<span class="w"> </span><span class="nv">I</span><span class="o">=</span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.merge.bed<span class="w"> </span><span class="nv">O</span><span class="o">=</span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.merge.interval_list<span class="w"> </span><span class="nv">SD</span><span class="o">=</span>~/workspace/inputs/references/temp/ref_genome.dict
rm<span class="w"> </span>-fr<span class="w"> </span>~/workspace/inputs/references/temp/

<span class="c1"># next for our subset exome and probe regions file</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/inputs/references/exome
java<span class="w"> </span>-jar<span class="w"> </span>/usr/local/bin/picard.jar<span class="w"> </span>BedToIntervalList<span class="w"> </span><span class="nv">I</span><span class="o">=</span>exome_regions.bed<span class="w"> </span><span class="nv">O</span><span class="o">=</span>exome_regions.bed.interval_list<span class="w"> </span><span class="nv">SD</span><span class="o">=</span>~/workspace/inputs/references/genome/ref_genome.dict
java<span class="w"> </span>-jar<span class="w"> </span>/usr/local/bin/picard.jar<span class="w"> </span>BedToIntervalList<span class="w"> </span><span class="nv">I</span><span class="o">=</span>probe_regions.bed<span class="w"> </span><span class="nv">O</span><span class="o">=</span>probe_regions.bed.interval_list<span class="w"> </span><span class="nv">SD</span><span class="o">=</span>~/workspace/inputs/references/genome/ref_genome.dict
</code></pre></div>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2023, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "content.code.annotate", "content.tabs.link", "search.highlight", "search.share", "search.suggest"], "search": "../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.51198bba.min.js"></script>
      
    
  </body>
</html>