
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Project documentation with Markdown.">
      
      
      
      
      
      <link rel="icon" href="../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.1.6">
    
    
      
        <title>INSTALLATION NOTES - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.ded33207.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.a0c5b2b5.min.css">
      
      
  
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
  
  <style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M64 32C28.7 32 0 60.7 0 96v320c0 35.3 28.7 64 64 64h224V368c0-26.5 21.5-48 48-48h112V96c0-35.3-28.7-64-64-64H64zm384 320H336c-8.8 0-16 7.2-16 16v112l32-32 64-64 32-32z"/></svg>');--md-admonition-icon--abstract:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M96 0C43 0 0 43 0 96v320c0 53 43 96 96 96h320c17.7 0 32-14.3 32-32s-14.3-32-32-32v-64c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H96zm0 384h256v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"/></svg>');--md-admonition-icon--info:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-40-176h24v-64h-24c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24h-80c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"/></svg>');--md-admonition-icon--tip:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M480 32c0-12.9-7.8-24.6-19.8-29.6S434.5.2 425.3 9.3L381.7 53c-48 48-113.1 75-181 75H64c-35.3 0-64 28.7-64 64v96c0 35.3 28.7 64 64 64v128c0 17.7 14.3 32 32 32h64c17.7 0 32-14.3 32-32V352h8.7c67.9 0 133 27 181 75l43.6 43.6c9.2 9.2 22.9 11.9 34.9 6.9s19.8-16.6 19.8-29.6V300.4c18.6-8.8 32-32.5 32-60.4s-13.4-51.6-32-60.4V32zm-64 76.7v262.6C357.2 317.8 280.5 288 200.7 288H192v-96h8.7c79.8 0 156.5-29.8 215.3-83.3z"/></svg>');--md-admonition-icon--success:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7l233.4-233.3c12.5-12.5 32.8-12.5 45.3 0z"/></svg>');--md-admonition-icon--question:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-86.2-346.7c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1 0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6-13.3 0-24-10.7-24-24v-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1 0-8.4-6.8-15.1-15.1-15.1h-58.3c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1-64 0z"/></svg>');--md-admonition-icon--warning:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7.2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8.2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24v112c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0-64 0 32 32 0 1 0 64 0z"/></svg>');--md-admonition-icon--failure:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.1 52.4 442.6 6.5c-1.9-3.9-6.1-6.5-10.5-6.5s-8.5 2.6-10.4 6.5l-16.5 45.9-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4 0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2 0-4.6-3-8.9-7.2-10.4l-45.6-16.8zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9c-22-8-45.8-12.3-70.5-12.3C93.1 96 0 189.1 0 304s93.1 208 208 208 208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg>');--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M416 398.9c58.5-41.1 96-104.1 96-174.9C512 100.3 397.4 0 256 0S0 100.3 0 224c0 70.7 37.5 133.8 96 174.9V464c0 26.5 21.5 48 48 48h48v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h64v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h48c26.5 0 48-21.5 48-48v-65.1zM96 256a64 64 0 1 1 128 0 64 64 0 1 1-128 0zm256-64a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"/></svg>');--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M320 0c17.7 0 32 14.3 32 32v64h120c39.8 0 72 32.2 72 72v272c0 39.8-32.2 72-72 72H168c-39.8 0-72-32.2-72-72V168c0-39.8 32.2-72 72-72h120V32c0-17.7 14.3-32 32-32zM208 384c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zM264 256a40 40 0 1 0-80 0 40 40 0 1 0 80 0zm152 40a40 40 0 1 0 0-80 40 40 0 1 0 0 80zM48 224h16v192H48c-26.5 0-48-21.5-48-48v-96c0-26.5 21.5-48 48-48zm544 0c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48h-16V224h16z"/></svg>');--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M288 0H128c-17.7 0-32 14.3-32 32s14.3 32 32 32v132.8c0 11.8-3.3 23.5-9.5 33.5L10.3 406.2C3.6 417.2 0 429.7 0 442.6 0 480.9 31.1 512 69.4 512h309.2c38.3 0 69.4-31.1 69.4-69.4 0-12.8-3.6-25.4-10.3-36.4L329.5 230.4c-6.2-10.1-9.5-21.7-9.5-33.5V64c17.7 0 32-14.3 32-32S337.7 0 320 0h-32zm-96 196.8V64h64v132.8c0 23.7 6.6 46.9 19 67.1l34.5 56.1h-171l34.5-56.1c12.4-20.2 19-43.4 19-67.1z"/></svg>');--md-admonition-icon--quote:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 216C0 149.7 53.7 96 120 96h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V216zm256 0c0-66.3 53.7-120 120-120h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64h-64c-35.3 0-64-28.7-64-64V216z"/></svg>');}</style>


    
    
    
      
    
    
      <link rel="stylesheet" href="../assets/css/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#installation-notes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              INSTALLATION NOTES
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href=".." class="md-tabs__link">
      Course Details
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../schedule/" class="md-tabs__link">
      Schedule
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../aws/aws_ec2_instances/" class="md-tabs__link">
      AWS Instances Login
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../lectures/lectures/" class="md-tabs__link">
      Lectures
    </a>
  </li>

      
        
  
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../lab_session/basic_unix/" class="md-tabs__link">
        Lab Session
      </a>
    </li>
  

  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_5_1" id="__nav_5_1_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/ggplot2/" class="md-nav__link">
        ggplot tutorial
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_4" >
      
      
      
        <label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/structural_variation/" class="md-nav__link">
        Structural Variants
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/analysis_of_cnv/" class="md-nav__link">
        Copy Number Analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
          Day 5
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          Day 5
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/joint_curation/" class="md-nav__link">
        Joint Curation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../lab_session/genomic_alteration_interpretation_exe/" class="md-nav__link">
        Genomic Alterations Interpretation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#prepare-for-installation" class="md-nav__link">
    Prepare for installation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-samtools" class="md-nav__link">
    Install Samtools
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-picard" class="md-nav__link">
    Install PICARD
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-bwa" class="md-nav__link">
    Install BWA
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-gatk-4" class="md-nav__link">
    Install GATK 4
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-vep-934" class="md-nav__link">
    Install VEP 93.4
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-varscan" class="md-nav__link">
    Install Varscan
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-bcftools" class="md-nav__link">
    Install BCFtools
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-strelka" class="md-nav__link">
    Install Strelka
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-sambamba" class="md-nav__link">
    Install Sambamba
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-hisat2" class="md-nav__link">
    Install HISAT2
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-stringtie" class="md-nav__link">
    Install StringTie
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-gffcompare" class="md-nav__link">
    Install Gffcompare
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-r" class="md-nav__link">
    Install R
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-copycat" class="md-nav__link">
    Install copyCat
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-cnvnator" class="md-nav__link">
    Install CNVnator
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-cnvkit" class="md-nav__link">
    Install CNVkit
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-kallisto" class="md-nav__link">
    Install Kallisto
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-pizzly" class="md-nav__link">
    Install Pizzly
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#manta" class="md-nav__link">
    Manta
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mosdepth" class="md-nav__link">
    mosdepth
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bam-readcount" class="md-nav__link">
    bam-readcount
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vt" class="md-nav__link">
    vt
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vcf-annotation-tools" class="md-nav__link">
    vcf-annotation-tools
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#install-seqtk" class="md-nav__link">
    Install seqtk
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#activate-the-tools-environment" class="md-nav__link">
    Activate the tools environment
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="installation-notes">INSTALLATION NOTES<a class="headerlink" href="#installation-notes" title="Permanent link">&para;</a></h1>
<p>This workshop requires a large number of different bioinformatics tools. The instructions for installing these tools exist here. Note that depending on the operating system and environment, some additional dependencies would likely be needed. If you are using the AWS instance built for this course these dependencies have already been installed. The remainder of this section will assume that you are on the AWS instance, however these instructions should work on any ubuntu distribution with the required dependencies.</p>
<h2 id="prepare-for-installation">Prepare for installation<a class="headerlink" href="#prepare-for-installation" title="Permanent link">&para;</a></h2>
<p>For this workshop we will be using the workspace folder to store results, executables, and input files. To start we must choose a single directory for installing tools, typically in linux, user compiled tools are installed in /usr/local/bin however backups of the tools we will be using have already been installed there. In this tutorial we will install tools in ~/workspace/bin. Lets go ahead and make a bin directory in ~/workspace to get started.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># make a bin directory</span>
mkdir<span class="w"> </span>-p<span class="w"> </span>~/workspace/bin
</code></pre></div>
<h2 id="install-samtools">Install Samtools<a class="headerlink" href="#install-samtools" title="Permanent link">&para;</a></h2>
<p>Samtools is a software package based in C which provies utilities for manipulating alignment files (SAM/BAM/CRAM). It is open source, available on github, and is under an MIT license. Let’s go ahead and download the source code from github to our bin directory and extract it with tar. Next we need to cd into our extracted samtools source code and configure the software. Running ./configure will make sure all dependencies are available and will also let the software know where it should install to. After that we will need to run make to actually build the software. Finally we can run make install which will copy the built software and the underlying libraries, documentation, etc. to their final locations. We can check the installation and print out the help message by providing the full path to the executable.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># change to bin directory</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin

<span class="c1"># download and extract the source code</span>
wget<span class="w"> </span>https://github.com/samtools/samtools/releases/download/1.14/samtools-1.14.tar.bz2
tar<span class="w"> </span>--bzip2<span class="w"> </span>-xvf<span class="w"> </span>samtools-1.14.tar.bz2

<span class="c1"># configure and compile</span>
<span class="nb">cd</span><span class="w"> </span>samtools-1.14/
./configure<span class="w"> </span>--prefix<span class="o">=</span>/home/ubuntu/workspace/bin/samtools-1.14/
make
make<span class="w"> </span>install
ln<span class="w"> </span>-s<span class="w"> </span>/home/ubuntu/workspace/bin/samtools-1.14/bin/samtools<span class="w"> </span>/home/ubuntu/workspace/bin/samtools

<span class="c1"># check instalation</span>
~/workspace/bin/samtools<span class="w"> </span>--help
</code></pre></div>
<h2 id="install-picard">Install PICARD<a class="headerlink" href="#install-picard" title="Permanent link">&para;</a></h2>
<p>PICARD is a set of java based tools developed by the Broad institute. It is very useful for manipulating next generation sequencing data and is available under an open source MIT license. The version of Picard we will be using requires java 8 which has already been installed. All we need to do is download the jar file which is a package file used to distribute java code. We can do this with wget. To run the software, we simply need to call java with the -jar option and provide the jar file.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># change to the bin directory and download the jar file</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>https://github.com/broadinstitute/picard/releases/download/2.26.6/picard.jar

<span class="c1"># check the installation</span>
java<span class="w"> </span>-jar<span class="w"> </span>~/workspace/bin/picard.jar<span class="w"> </span>-h
</code></pre></div>
<h2 id="install-bwa">Install BWA<a class="headerlink" href="#install-bwa" title="Permanent link">&para;</a></h2>
<p>BWA is a popular DNA alignment tool used for mapping sequences to a reference genome. It is available under an open source GPLv3 license. To install BWA, we first need to download and extract the source code. Unlike with samtools theres no ./configure file so we can just run make to build the software. We can then make a symlink with ln -s which is just a reference to another file. In this case we will make a symlink so the executable in ~/workspace/bin/bwa-0.7.17/bwa and be found in ~/workspace/bin/bwa.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># change to the bin folder, download, and extract the source code</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>https://sourceforge.net/projects/bio-bwa/files/bwa-0.7.17.tar.bz2
tar<span class="w"> </span>--bzip2<span class="w"> </span>-xvf<span class="w"> </span>bwa-0.7.17.tar.bz2


<span class="c1"># build the software</span>
<span class="nb">cd</span><span class="w">  </span>bwa-0.7.17
make

<span class="c1"># make symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/bwa-0.7.17/bwa<span class="w"> </span>~/workspace/bin/bwa

<span class="c1"># check the installation</span>
~/workspace/bin/bwa
</code></pre></div>
<h2 id="install-gatk-4">Install GATK 4<a class="headerlink" href="#install-gatk-4" title="Permanent link">&para;</a></h2>
<p>GATK is a toolkit developed by the broad institute focused primarily on variant discovery and genotyping. It is open source, hosted on github, and available under a BSD 3-clause license. First let’s download and unzip GATK from github. The creators of GATK recommend running GATK through conda which is a package, environment, and dependency management software, in essence conda basically creates a virtual environment from which to run software. The next step then is to tell conda to create a virtual environment for GATK by using the yaml file included within GATK as the instructions for creating the virtual environment. We do this with the command conda env create, we also use the -p option to specify where this environment should be stored. We will also make a symlink so the executable downloaded is available directly from our bin folder. To run GATK we must first start up the virtual environment with the command source activate, we can then run the program by providing the path to the executable. To exit the virtual environment run the command source deactivate.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and unzip</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>https://github.com/broadinstitute/gatk/releases/download/4.2.3.0/gatk-4.2.3.0.zip
unzip<span class="w"> </span>gatk-4.2.3.0.zip

<span class="c1"># make sure ubuntu user can create their own conda environments</span>
sudo<span class="w"> </span>chown<span class="w"> </span>-R<span class="w"> </span>ubuntu:ubuntu<span class="w"> </span>/home/ubuntu/.conda

<span class="c1"># create conda environment for gatk</span>
<span class="nb">cd</span><span class="w"> </span>gatk-4.2.3.0/
conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>gatkcondaenv.yml<span class="w"> </span>-p<span class="w"> </span>~/workspace/bin/conda/gatk

<span class="c1"># make symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/gatk-4.2.3.0/gatk<span class="w"> </span>~/workspace/bin/gatk

<span class="c1"># test installation</span>
conda<span class="w"> </span>activate<span class="w"> </span>~/workspace/bin/conda/gatk
~/workspace/bin/gatk

<span class="c1"># to exit the virtual environment</span>
conda<span class="w"> </span>deactivate
</code></pre></div>
<h2 id="install-vep-934">Install VEP 93.4<a class="headerlink" href="#install-vep-934" title="Permanent link">&para;</a></h2>
<p>VEP is a variant annotation tool developed by ensembl and written in perl. By default VEP will perform annotations by making web-based API queries however it is much faster to have a local copy of cache and fasta files. The AWS AMI image we’re using already has these files for hg38 in the directory ~/workspace/vep_cache as they can take a bit of time to download. To get an idea of what it’s like to install these we will install a vep_cache for petromyzon_marinus, a much smaller genome. To start we need to download vep from github using wget and unzip VEP. From there we can use the INSTALL.pl script vep provides to install the software which will ask a series of questions listed below. We also make a symlink when the installer completes.</p>
<p>Note that the following assumes the existence of a particular version of Perl. We had to install Perl 5.22.0 since this is the last version supported by VEP and the version that comes with Ubuntu 18.04 is newer than this.</p>
<p>When prompted by the install step below use these answers:</p>
<p>Do you wish to exit so you can get updates (y) or continue (n): n [ENTER]
Do you want to continue installing the API (y/n)? y [ENTER]
Do you want to install any cache files (y/n)? y [ENTER] 147 [ENTER]
Do you want to install any FASTA files (y/n)? y [ENTER] 42 [ENTER]
Do you want to install any plugins (y/n)? n [ENTER]</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and unzip vep</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>https://github.com/Ensembl/ensembl-vep/archive/refs/tags/release/104.3.zip
unzip<span class="w"> </span><span class="m">104</span>.3.zip

<span class="c1"># run the INSTALL.pl script provided by VEP</span>
<span class="nb">cd</span><span class="w"> </span>ensembl-vep-release-104.3/
/usr/local/bin/perl-5.22.0/perl<span class="w"> </span>-MCPAN<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;install DBI&#39;</span>
/usr/local/bin/perl-5.22.0/perl<span class="w"> </span>INSTALL.pl<span class="w"> </span>--CACHEDIR<span class="w"> </span>~/workspace/vep_cache
<span class="c1">#1. Do you wish to exit so you can get updates (y) or continue (n): n [ENTER]</span>
<span class="c1">#2. Do you want to continue installing the API (y/n)? y [ENTER] (if asked)</span>
<span class="c1">#3. Do you want to install any cache files (y/n)? y [ENTER] 147 [ENTER]</span>
<span class="c1">#4. Do you want to install any FASTA files (y/n)? y [ENTER] 42 [ENTER]</span>
<span class="c1">#5. Do you want to install any plugins (y/n)? n [ENTER]</span>

<span class="c1"># make a symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/ensembl-vep-release-104.3/vep<span class="w"> </span>~/workspace/bin/vep

<span class="c1"># test the Installation</span>
~/workspace/bin/vep<span class="w"> </span>--help
</code></pre></div>
<h2 id="install-varscan">Install Varscan<a class="headerlink" href="#install-varscan" title="Permanent link">&para;</a></h2>
<p>Varscan is a java program designed to call variants in sequencing data. It was developed at the Genome Institute at Washington University and is hosted on github. To use Varscan we simply need to download the distributed jar file into our~/workspace/bin. As with the other java programs which have already been installed in this section we can invoke Varscan via java -jar</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Install Varscan</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
curl<span class="w"> </span>-L<span class="w"> </span>-k<span class="w"> </span>-o<span class="w"> </span>VarScan.v2.4.2.jar<span class="w"> </span>https://github.com/dkoboldt/varscan/releases/download/2.4.2/VarScan.v2.4.2.jar
java<span class="w"> </span>-jar<span class="w"> </span>~/workspace/bin/VarScan.v2.4.2.jar
</code></pre></div>
<h2 id="install-bcftools">Install BCFtools<a class="headerlink" href="#install-bcftools" title="Permanent link">&para;</a></h2>
<p>BCFtools is an open source program for variant calling and manipulating files in Variant Call Format (VCF) or Binary Variant Call Format (BCF). To install we first need to download and extract the source code with curl and tar respectively. We can then call make to build the program and make install to copy the program to the desired directory.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/workspace/bin
curl<span class="w"> </span>-L<span class="w"> </span>-k<span class="w"> </span>-o<span class="w"> </span>bcftools-1.14.tar.bz2<span class="w"> </span>https://github.com/samtools/bcftools/releases/download/1.14/bcftools-1.14.tar.bz2
tar<span class="w"> </span>--bzip2<span class="w"> </span>-xvf<span class="w"> </span>bcftools-1.14.tar.bz2

<span class="c1">#install the software</span>
<span class="nb">cd</span><span class="w"> </span>bcftools-1.14
make<span class="w"> </span>-j
make<span class="w"> </span><span class="nv">prefix</span><span class="o">=</span>~/workspace/bin/bcftools-1.14<span class="w"> </span>install
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/bcftools-1.14/bin/bcftools<span class="w"> </span>~/workspace/bin/bcftools

<span class="c1"># test installation</span>
~/workspace/bin/bcftools<span class="w"> </span>-h
</code></pre></div>
<h2 id="install-strelka">Install Strelka<a class="headerlink" href="#install-strelka" title="Permanent link">&para;</a></h2>
<p>Strekla is a germline and somatic variant caller developed by illumina and available under an open source GPLv3 license. The binary distribution for strelka is already built and hosted on github so to install all we have to do is download and extract the software. It is important to note that strelka is built on python 2 and won’t work for python 3. The AMI we’re using contains both python versions so we just have to make sure we invoke strelka with python2, you can view the python versions on the AMI with python2 --version and python3 --version.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and extract</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin

conda<span class="w"> </span>create<span class="w"> </span>--name<span class="w"> </span>strelka-env<span class="w"> </span><span class="nv">python</span><span class="o">=</span><span class="m">2</span>.7

curl<span class="w"> </span>-L<span class="w"> </span>-k<span class="w"> </span>-o<span class="w"> </span>strelka-2.9.10.centos6_x86_64.tar.bz2<span class="w"> </span>https://github.com/Illumina/strelka/releases/download/v2.9.10/strelka-2.9.10.centos6_x86_64.tar.bz2
tar<span class="w"> </span>--bz2<span class="w"> </span>-xvf<span class="w"> </span>strelka-2.9.10.centos6_x86_64.tar.bz2

<span class="c1"># test installation</span>
python2<span class="w"> </span>~/workspace/bin/strelka-2.9.10.centos6_x86_64/bin/configureStrelkaSomaticWorkflow.py<span class="w"> </span>-h
</code></pre></div>
<h2 id="install-sambamba">Install Sambamba<a class="headerlink" href="#install-sambamba" title="Permanent link">&para;</a></h2>
<p>Sambamba is a high performance alternative to samtools and provides a subset of samtools functionality. It is up to 6x faster for duplicate read marking and 4x faster for viewing alignment files. To install sambamba we can just download the binary distribution and extract it. From there we just make a symlink to make using it a bit more intuitive.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and extract</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
curl<span class="w"> </span>-L<span class="w"> </span>-k<span class="w"> </span>-o<span class="w"> </span>sambamba_v0.6.4_linux.tar.bz2<span class="w"> </span>https://github.com/lomereiter/sambamba/releases/download/v0.6.4/sambamba_v0.6.4_linux.tar.bz2
tar<span class="w"> </span>--bzip2<span class="w"> </span>-xvf<span class="w"> </span>sambamba_v0.6.4_linux.tar.bz2

<span class="c1"># create symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/sambamba_v0.6.4<span class="w"> </span>~/workspace/bin/sambamba

<span class="c1"># test installation</span>
~/workspace/bin/sambamba
</code></pre></div>
<h2 id="install-hisat2">Install HISAT2<a class="headerlink" href="#install-hisat2" title="Permanent link">&para;</a></h2>
<p>HISAT2 is a graph based alignment algorithm devoloped at Johns Hopkins University. It is heavily used in the bioinformatics community for RNAseq based alignments. To Install we will need to download and extract the binary executable. We then make a symlink to put it with the other executables we’ve installed.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and extract</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/downloads/hisat2-2.1.0-Linux_x86_64.zip
unzip<span class="w"> </span>hisat2-2.1.0-Linux_x86_64.zip

<span class="c1"># create symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/hisat2-2.1.0/hisat2<span class="w"> </span>~/workspace/bin/hisat2

<span class="c1"># test installation</span>
~/workspace/bin/hisat2<span class="w"> </span>--help
</code></pre></div>
<h2 id="install-stringtie">Install StringTie<a class="headerlink" href="#install-stringtie" title="Permanent link">&para;</a></h2>
<p>StringTie is a software program to perform transcript assembly and quantification of RNAseq data. The binary distributions are available so to install we can just download this distribution and extract it. Like with our other programs we also make a symlink to make it easier to find.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and extract</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>http://ccb.jhu.edu/software/stringtie/dl/stringtie-2.2.0.Linux_x86_64.tar.gz
tar<span class="w"> </span>-xzvf<span class="w"> </span>stringtie-2.2.0.Linux_x86_64.tar.gz

<span class="c1"># make symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/stringtie-2.2.0.Linux_x86_64/stringtie<span class="w"> </span>~/workspace/bin/stringtie

<span class="c1"># test installation</span>
~/workspace/bin/stringtie<span class="w"> </span>-h
</code></pre></div>
<h2 id="install-gffcompare">Install Gffcompare<a class="headerlink" href="#install-gffcompare" title="Permanent link">&para;</a></h2>
<p>Gffcompare is a program that is used to perform operations on general feature format (GFF) and general transfer format (GTF) files. It has a binary distribution compatible with the linux we’re using so we will just download, extract, and make a symlink.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and extract</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>http://ccb.jhu.edu/software/stringtie/dl/gffcompare-0.9.8.Linux_x86_64.tar.gz
tar<span class="w"> </span>-xzvf<span class="w"> </span>gffcompare-0.9.8.Linux_x86_64.tar.gz

<span class="c1"># make symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/gffcompare-0.9.8.Linux_x86_64/gffcompare<span class="w"> </span>~/workspace/bin/gffcompare

<span class="c1"># check Installation</span>
~/workspace/bin/gffcompare
</code></pre></div>
<h2 id="install-r">Install R<a class="headerlink" href="#install-r" title="Permanent link">&para;</a></h2>
<p>R is a feature rich interpretive programming language originally released in 1995. It is heavily used in the bioinformatics community largely due to numerous R libraries available on bioconductor. It takes a several minutes to compile so we’ll use one which has already been setup. If we were to install R, we first would need to download and extract the source code. Next we’d configure the installation with --with-x=no which tells R to install without X11, a windowing system for displays. We’d also specify --prefix which is where the R framework will go, this includes the additional R libraries we’ll download later. From there we’d do make and make install to build the software and copy the files to their proper location and create symlinks for the executables. Finally we’d install the devtools and Biocmanager packages from the command line to make installing additional packages easier. We’ve commented out the code below, however it is exactly what was run to set up the R we will be using, except the installation location.</p>
<div class="highlight"><pre><span></span><code><span class="c1">## download and extract</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>https://cran.r-project.org/src/base/R-3/R-3.5.1.tar.gz
tar<span class="w"> </span>-zxvf<span class="w"> </span>R-3.5.1.tar.gz

<span class="c1">## configure the installation, build the code</span>
<span class="nb">cd</span><span class="w"> </span>R-3.5.1
./configure<span class="w"> </span>--prefix<span class="o">=</span>/home/ubuntu/workspace/bin<span class="w"> </span>--with-x<span class="o">=</span>no
make
make<span class="w"> </span>install

<span class="c1">## make symlinks</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/R-3.5.1/bin/Rscript<span class="w"> </span>~/workspace/bin/Rscript
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/lib64/R/bin/R<span class="w"> </span>~/workspace/bin/R

<span class="c1">## test installation</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
~/workspace/bin/Rscript<span class="w"> </span>--version

<span class="c1">## install additional packages</span>
~/workspace/bin/R<span class="w"> </span>--vanilla<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;install.packages(c(&quot;devtools&quot;, &quot;BiocManager&quot;, &quot;dplyr&quot;, &quot;tidyr&quot;, &quot;ggplot2&quot;), repos=&quot;http://cran.us.r-project.org&quot;)&#39;</span>
</code></pre></div>
<h2 id="install-copycat">Install copyCat<a class="headerlink" href="#install-copycat" title="Permanent link">&para;</a></h2>
<p>copyCat is an R library for detecting copy number aberrations in sequencing data. The library is only available on github so we will have to use the BiocManager library to install a few of the underlying package dependencies. If installing a package from cran or bioconductor these dependencies would be automatically installed. After these dependencies are installed we can use the devtools package to install copycat directory from its github repository.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Install R Library dependencies</span>
~/workspace/bin/R<span class="w"> </span>--vanilla<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;BiocManager::install(c(&quot;IRanges&quot;, &quot;DNAcopy&quot;))&#39;</span>

<span class="c1"># install copyCat</span>
~/workspace/bin/R<span class="w"> </span>--vanilla<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;devtools::install_github(&quot;chrisamiller/copycat&quot;)&#39;</span>
</code></pre></div>
<h2 id="install-cnvnator">Install CNVnator<a class="headerlink" href="#install-cnvnator" title="Permanent link">&para;</a></h2>
<p>CNVnator is a depth based copy number caller. It is open source and available on github under a creative common public license (CCPL). To install we first download and extract the source code. CNVnator relies on a specific version of samtools which is distributed with CNVnator, so our first step is to run make on that samtools. To finish the installation process we can then run make in CNVnator’s main source directory.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and decompress</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin

<span class="c1">#download and install dependency package &quot;root&quot; from Cern (https://root.cern/install/). </span>

curl<span class="w"> </span>-OL<span class="w"> </span>https://root.cern/download/root_v6.20.08.Linux-ubuntu20-x86_64-gcc9.3.tar.gz
tar<span class="w"> </span>-xvzf<span class="w"> </span>root_v6.20.08.Linux-ubuntu20-x86_64-gcc9.3.tar.gz
<span class="nb">source</span><span class="w"> </span>root/bin/thisroot.sh


wget<span class="w"> </span>https://github.com/abyzovlab/CNVnator/releases/download/v0.3.3/CNVnator_v0.3.3.zip
unzip<span class="w"> </span>CNVnator_v0.3.3.zip

<span class="c1"># make the samtools dependency distributed with CNVnator</span>
<span class="nb">cd</span><span class="w"> </span>CNVnator_v0.3.3/src/samtools
make

<span class="c1"># make CNVnator</span>
<span class="nb">cd</span><span class="w"> </span>../
make

<span class="c1"># make a symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/CNVnator_v0.3.3/src/cnvnator<span class="w"> </span>~/workspace/bin/cnvnator

<span class="c1"># test installation</span>
~/workspace/bin/cnvnator
</code></pre></div>
<h2 id="install-cnvkit">Install CNVkit<a class="headerlink" href="#install-cnvkit" title="Permanent link">&para;</a></h2>
<p>CNVkit is a python based copy number caller designed for use with hybrid capture. To install we can download and extract the package. We then must use conda to set up the environment to run cnvkit. This process, while straight forward, takes some time so we’ve commented out the installation instructions for this tool and will use the conda environment that has already been set up.</p>
<div class="highlight"><pre><span></span><code><span class="c1">## download and unzip</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>https://github.com/etal/cnvkit/archive/refs/tags/v0.9.9.zip
unzip<span class="w"> </span>v0.9.9.zip

<span class="c1">## add conda channels</span>
conda<span class="w"> </span>config<span class="w"> </span>--add<span class="w"> </span>channels<span class="w"> </span>defaults
conda<span class="w"> </span>config<span class="w"> </span>--add<span class="w"> </span>channels<span class="w"> </span>conda-forge
conda<span class="w"> </span>config<span class="w"> </span>--add<span class="w"> </span>channels<span class="w"> </span>bioconda

<span class="c1">## create conda environment</span>
conda<span class="w"> </span>create<span class="w"> </span>-n<span class="w"> </span>cnvkit<span class="w"> </span><span class="nv">python</span><span class="o">=</span><span class="m">3</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/cnvkit-0.9.9/cnvkit.py<span class="w"> </span>~/workspace/bin/cnvkit.py

<span class="c1"># test installation</span>
<span class="nb">source</span><span class="w"> </span>activate<span class="w"> </span>cnvkit
<span class="c1">#install all dependencies</span>
~/workspace/bin/cnvkit.py<span class="w"> </span>--help

<span class="c1"># to exit the virtual environment</span>
<span class="nb">source</span><span class="w"> </span>deactivate
</code></pre></div>
<h2 id="install-kallisto">Install Kallisto<a class="headerlink" href="#install-kallisto" title="Permanent link">&para;</a></h2>
<p>Kallisto is a kmer-based alignment algorithm used for quantifying transcripts in RNAseq data. Kallisto has a binary distribution available so to use the program we only have to download and extract the software from github.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and extract</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>https://github.com/pachterlab/kallisto/releases/download/v0.46.2/kallisto_linux-v0.46.2.tar.gz
tar<span class="w"> </span>-zxvf<span class="w"> </span>kallisto_linux-v0.46.2.tar.gz
mv<span class="w"> </span>kallisto<span class="w"> </span>kallisto_linux-v0.46.2

<span class="c1"># make symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/kallisto_linux-v0.46.2/kallisto<span class="w"> </span>~/workspace/bin/kallisto

<span class="c1"># test installation</span>
~/workspace/bin/kallisto
</code></pre></div>
<h2 id="install-pizzly">Install Pizzly<a class="headerlink" href="#install-pizzly" title="Permanent link">&para;</a></h2>
<p>Pizzly is a fusion detection algorithm which uses output from Kallisto. Pizzly has a binary distribution so we can download and extract that from github to get started.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and extract</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
mkdir<span class="w"> </span>pizzly-v0.37.3
<span class="nb">cd</span><span class="w"> </span>pizzly-v0.37.3
wget<span class="w"> </span>https://github.com/pmelsted/pizzly/releases/download/v0.37.3/pizzly_linux.tar.gz
tar<span class="w"> </span>-zxvf<span class="w"> </span>pizzly_linux.tar.gz

<span class="c1"># make symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/pizzly-v0.37.3/pizzly<span class="w"> </span>~/workspace/bin/pizzly

<span class="c1"># test executable</span>
~/workspace/bin/pizzly<span class="w"> </span>--help
</code></pre></div>
<h2 id="manta">Manta<a class="headerlink" href="#manta" title="Permanent link">&para;</a></h2>
<p>Manta is a structural variant caller developed by Illumina and available on gitub under the GPL_v3 license. It uses paired-end sequencing reads to build a breakend association graph to identify structural varaints.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># download and extract</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
wget<span class="w"> </span>https://github.com/Illumina/manta/releases/download/v1.6.0/manta-1.6.0.centos6_x86_64.tar.bz2
tar<span class="w"> </span>--bzip2<span class="w"> </span>-xvf<span class="w"> </span>manta-1.6.0.centos6_x86_64.tar.bz2

<span class="c1">#we can use strelka-env for this also</span>

conda<span class="w"> </span>activate<span class="w"> </span>strelka-env

<span class="c1"># test installation</span>
python2<span class="w"> </span>~/workspace/bin/manta-1.6.0.centos6_x86_64/bin/configManta.py<span class="w"> </span>--help

conda<span class="w"> </span>deactivate
</code></pre></div>
<h2 id="mosdepth">mosdepth<a class="headerlink" href="#mosdepth" title="Permanent link">&para;</a></h2>
<p>mosdepth is a program for determining depth in sequencing data. The easiest way to install mosdepth is through bioconda a channel for the conda package manager. The AMI already has conda setup to install to /usr/local/bin/miniconda and so we’ve already installed mosdepth for you. However below are the commands used during the installation.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># add the bioconda channel</span>
conda<span class="w"> </span>config<span class="w"> </span>--add<span class="w"> </span>channels<span class="w"> </span>bioconda

<span class="c1"># install mosdepth with the conda package manager</span>
conda<span class="w"> </span>install<span class="w"> </span>mosdepth
</code></pre></div>
<h2 id="bam-readcount">bam-readcount<a class="headerlink" href="#bam-readcount" title="Permanent link">&para;</a></h2>
<p>bam-readcount is a program for determing read support for individual variants (SNVs and Indels only). We are going to point this local install of bam-readcount to use the samtools installation we completed above. Samtools is a dependency of bam-readcount. This tool uses Cmake to create its makefile, so compiling from source has an extra step here. Instead of using an official release from github we are cloning the latest code from the master branch. In general this practice should be avoided and you should use an official release instead.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># install bam-readcount</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
git<span class="w"> </span>clone<span class="w"> </span>https://github.com/genome/bam-readcount.git
mv<span class="w"> </span>bam-readcount<span class="w"> </span>bam-readcount-latest
<span class="nb">cd</span><span class="w"> </span>bam-readcount-latest
<span class="nb">export</span><span class="w"> </span><span class="nv">SAMTOOLS_ROOT</span><span class="o">=</span>/home/ubuntu//workspace/bin/samtools-1.14
cmake<span class="w"> </span>-Wno-dev<span class="w"> </span>/home/ubuntu/workspace/bin/bam-readcount-latest
make

<span class="c1"># create symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>~/workspace/bin/bam-readcount-latest/bin/bam-readcount<span class="w"> </span>~/workspace/bin/bam-readcount

<span class="c1"># test installation</span>
~/workspace/bin/bam-readcount
</code></pre></div>
<h2 id="vt">vt<a class="headerlink" href="#vt" title="Permanent link">&para;</a></h2>
<p>vt is a variant tool set that discovers short variants from Next Generation Sequencing data. We will use this for the purpose of splitting multi-allelic variants.</p>
<div class="highlight"><pre><span></span><code><span class="c1">#install vt</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
conda<span class="w"> </span>install<span class="w"> </span>-c<span class="w"> </span>bioconda<span class="w"> </span>vt

<span class="c1"># create symlink</span>
ln<span class="w"> </span>-s<span class="w"> </span>/home/ubuntu/miniconda3/bin/vt<span class="w"> </span>~/workspace/bin/vt

<span class="c1"># test installation</span>
~/workspace/bin/vt
</code></pre></div>
<h2 id="vcf-annotation-tools">vcf-annotation-tools<a class="headerlink" href="#vcf-annotation-tools" title="Permanent link">&para;</a></h2>
<p>VCF Annotation Tools is a python package that includes several tools to annotate VCF files with data from other tools. We will be using this for the purpose of adding bam readcounts to the vcf files.</p>
<div class="highlight"><pre><span></span><code><span class="c1">#install vcf-annotation-tools</span>
pip<span class="w"> </span>install<span class="w"> </span>vcf-annotation-tools

<span class="c1">#testing Installation</span>
vcf-readcount-annotator<span class="w"> </span>-h
</code></pre></div>
<h2 id="install-seqtk">Install seqtk<a class="headerlink" href="#install-seqtk" title="Permanent link">&para;</a></h2>
<p>Seqtk is a lighweight tool for processing FASTQ and FASTA files. We will use seqtk to subset RNA-seq fastq files to more quickly run the fusion alignment module.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Download</span>
<span class="nb">cd</span><span class="w"> </span>~/workspace/bin
git<span class="w"> </span>clone<span class="w"> </span>https://github.com/lh3/seqtk.git<span class="w"> </span>seqtk.v1

<span class="c1"># Install</span>
<span class="nb">cd</span><span class="w"> </span>seqtk.v1
make
<span class="c1"># (Ignore warning message)</span>
make<span class="w"> </span>install

<span class="c1"># Check install</span>
ln<span class="w"> </span>-s<span class="w"> </span>/usr/local/bin/seqtk<span class="w"> </span>~/workspace/bin/seqtk
~/workspace/bin/seqtk
</code></pre></div>
<h2 id="activate-the-tools-environment">Activate the tools environment<a class="headerlink" href="#activate-the-tools-environment" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">source</span><span class="w"> </span>.bashrc
</code></pre></div>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2023, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "content.code.annotate", "content.tabs.link", "search.highlight", "search.share", "search.suggest"], "search": "../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.51198bba.min.js"></script>
      
    
  </body>
</html>