
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Project documentation with Markdown.">
      
      
      
      
        <link rel="prev" href="../R_for_bioinformatics/">
      
      
        <link rel="next" href="../ggplot2/">
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.1.6">
    
    
      
        <title>Command Line Usage in Real Life - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.ded33207.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.a0c5b2b5.min.css">
      
      
  
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
    
    
  
  
  <style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M64 32C28.7 32 0 60.7 0 96v320c0 35.3 28.7 64 64 64h224V368c0-26.5 21.5-48 48-48h112V96c0-35.3-28.7-64-64-64H64zm384 320H336c-8.8 0-16 7.2-16 16v112l32-32 64-64 32-32z"/></svg>');--md-admonition-icon--abstract:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M96 0C43 0 0 43 0 96v320c0 53 43 96 96 96h320c17.7 0 32-14.3 32-32s-14.3-32-32-32v-64c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H96zm0 384h256v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"/></svg>');--md-admonition-icon--info:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-40-176h24v-64h-24c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24h-80c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"/></svg>');--md-admonition-icon--tip:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M480 32c0-12.9-7.8-24.6-19.8-29.6S434.5.2 425.3 9.3L381.7 53c-48 48-113.1 75-181 75H64c-35.3 0-64 28.7-64 64v96c0 35.3 28.7 64 64 64v128c0 17.7 14.3 32 32 32h64c17.7 0 32-14.3 32-32V352h8.7c67.9 0 133 27 181 75l43.6 43.6c9.2 9.2 22.9 11.9 34.9 6.9s19.8-16.6 19.8-29.6V300.4c18.6-8.8 32-32.5 32-60.4s-13.4-51.6-32-60.4V32zm-64 76.7v262.6C357.2 317.8 280.5 288 200.7 288H192v-96h8.7c79.8 0 156.5-29.8 215.3-83.3z"/></svg>');--md-admonition-icon--success:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7l233.4-233.3c12.5-12.5 32.8-12.5 45.3 0z"/></svg>');--md-admonition-icon--question:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-86.2-346.7c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1 0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6-13.3 0-24-10.7-24-24v-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1 0-8.4-6.8-15.1-15.1-15.1h-58.3c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1-64 0z"/></svg>');--md-admonition-icon--warning:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7.2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8.2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24v112c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0-64 0 32 32 0 1 0 64 0z"/></svg>');--md-admonition-icon--failure:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.1 52.4 442.6 6.5c-1.9-3.9-6.1-6.5-10.5-6.5s-8.5 2.6-10.4 6.5l-16.5 45.9-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4 0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2 0-4.6-3-8.9-7.2-10.4l-45.6-16.8zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9c-22-8-45.8-12.3-70.5-12.3C93.1 96 0 189.1 0 304s93.1 208 208 208 208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg>');--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M416 398.9c58.5-41.1 96-104.1 96-174.9C512 100.3 397.4 0 256 0S0 100.3 0 224c0 70.7 37.5 133.8 96 174.9V464c0 26.5 21.5 48 48 48h48v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h64v-48c0-8.8 7.2-16 16-16s16 7.2 16 16v48h48c26.5 0 48-21.5 48-48v-65.1zM96 256a64 64 0 1 1 128 0 64 64 0 1 1-128 0zm256-64a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"/></svg>');--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M320 0c17.7 0 32 14.3 32 32v64h120c39.8 0 72 32.2 72 72v272c0 39.8-32.2 72-72 72H168c-39.8 0-72-32.2-72-72V168c0-39.8 32.2-72 72-72h120V32c0-17.7 14.3-32 32-32zM208 384c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16h-32zM264 256a40 40 0 1 0-80 0 40 40 0 1 0 80 0zm152 40a40 40 0 1 0 0-80 40 40 0 1 0 0 80zM48 224h16v192H48c-26.5 0-48-21.5-48-48v-96c0-26.5 21.5-48 48-48zm544 0c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48h-16V224h16z"/></svg>');--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M288 0H128c-17.7 0-32 14.3-32 32s14.3 32 32 32v132.8c0 11.8-3.3 23.5-9.5 33.5L10.3 406.2C3.6 417.2 0 429.7 0 442.6 0 480.9 31.1 512 69.4 512h309.2c38.3 0 69.4-31.1 69.4-69.4 0-12.8-3.6-25.4-10.3-36.4L329.5 230.4c-6.2-10.1-9.5-21.7-9.5-33.5V64c17.7 0 32-14.3 32-32S337.7 0 320 0h-32zm-96 196.8V64h64v132.8c0 23.7 6.6 46.9 19 67.1l34.5 56.1h-171l34.5-56.1c12.4-20.2 19-43.4 19-67.1z"/></svg>');--md-admonition-icon--quote:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 216C0 149.7 53.7 96 120 96h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V216zm256 0c0-66.3 53.7-120 120-120h8c17.7 0 32 14.3 32 32s-14.3 32-32 32h-8c-30.9 0-56 25.1-56 56v8h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64h-64c-35.3 0-64-28.7-64-64V216z"/></svg>');}</style>


    
    
    
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#real-life-example-of-how-to-use-command-line-features-vt-2023" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Command Line Usage in Real Life
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      Course Details
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../schedule/" class="md-tabs__link">
      Schedule
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../aws/aws_ec2_instances/" class="md-tabs__link">
      AWS Instances Login
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../lectures/lectures/" class="md-tabs__link">
      Lectures
    </a>
  </li>

      
        
  
  
    
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../basic_unix/" class="md-tabs__link md-tabs__link--active">
        Lab Session
      </a>
    </li>
  

  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/KI_digital_logotyp_negativ_RGB.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_1" checked>
      
      
      
        <label class="md-nav__link" for="__nav_5_1" id="__nav_5_1_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_1_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Command Line Usage in Real Life
      </a>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ggplot2/" class="md-nav__link">
        ggplot tutorial
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_4" >
      
      
      
        <label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../structural_variation/" class="md-nav__link">
        Structural Variants
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../analysis_of_cnv/" class="md-nav__link">
        Copy Number Analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
          Day 5
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          Day 5
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../joint_curation/" class="md-nav__link">
        Joint Curation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../genomic_alteration_interpretation_exe/" class="md-nav__link">
        Genomic Alterations Interpretation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="real-life-example-of-how-to-use-command-line-features-vt-2023">Real life example of how to use command line features - vt 2023<a class="headerlink" href="#real-life-example-of-how-to-use-command-line-features-vt-2023" title="Permanent link">&para;</a></h1>
<p>This is an example of how to use various UNIX/bash/command line features for finding files, assigning variables, parsing and modifying text strings etc. The purpose in the example is to find files containing quality control metrics, and creating a table of certain variables from these files. </p>
<p>This is not to be run as part of the lab, just a demonstration of a real life example when multiple various command line features are utilized.</p>
<p>First, move to the directory of interest and find the files:</p>
<p><div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/Courses/cancer<span class="se">\ </span>bioinfo<span class="se">\ </span>course<span class="se">\ </span>v2/cmd_line_demo<span class="w">  </span><span class="c1"># move to project directory</span>
<span class="nb">pwd</span><span class="w">  </span><span class="c1"># check present working directory</span>
ls<span class="w"> </span>-lhtr<span class="w">  </span><span class="c1"># list files with detailed info (-l), human readable size (-h), in time order (-t), reveresed order with latest last (-r)</span>
find<span class="w"> </span>.<span class="w"> </span>-name<span class="w"> </span><span class="s2">&quot;*hsmetrics.txt&quot;</span><span class="w">  </span><span class="c1"># find the files in this directory that ends with &quot;txt&quot;</span>
find<span class="w"> </span>.<span class="w"> </span>-name<span class="w"> </span><span class="s2">&quot;*hsmetrics.txt&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>wc<span class="w"> </span>-l<span class="w">  </span><span class="c1"># count the number of found files</span>
<span class="nv">files</span><span class="o">=(</span><span class="k">$(</span>find<span class="w"> </span>.<span class="w"> </span>-name<span class="w"> </span><span class="s2">&quot;*hsmetrics.txt&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sort<span class="k">)</span><span class="o">)</span><span class="w">  </span><span class="c1"># assign a sorted list of the files to an array</span>
<span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">files</span><span class="p">[@]</span><span class="si">}</span><span class="w">  </span><span class="c1"># print all items of the array</span>
<span class="nb">echo</span><span class="w"> </span><span class="si">${#</span><span class="nv">files</span><span class="p">[@]</span><span class="si">}</span><span class="w">  </span><span class="c1"># print number of items of the array</span>
</code></pre></div>
Output : </p>
<div class="highlight"><pre><span></span><code>## /Users/rebber/Courses/cancer bioinfo course v2/cmd_line_demo
## total 136
## -rw-r--r--  1 rebber  staff   7.0K Apr 14 12:01 PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.slurm.log
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   7.8K Apr 14 12:01 PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.slurm.log
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   8.1K Apr 14 12:01 PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.slurm.log
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   5.3K Apr 14 12:01 PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.slurm.log
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## ./PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## ./PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## ./PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## ./PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
##        4
## ./PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt ./PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt ./PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt ./PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## 4
</code></pre></div>
<p>Now we have the files of interest in the array <code>$files</code>.
We can use this array for running further operations on the files.</p>
<p>Inspect the files:
<div class="highlight"><pre><span></span><code>ls<span class="w"> </span>-lh<span class="w"> </span><span class="si">${</span><span class="nv">files</span><span class="p">[@]</span><span class="si">}</span><span class="w">  </span><span class="c1"># list files with detailed view and human readable sizes</span>
wc<span class="w"> </span>-l<span class="w"> </span><span class="si">${</span><span class="nv">files</span><span class="p">[@]</span><span class="si">}</span><span class="w">  </span><span class="c1"># count number of rows of all files</span>
less<span class="w"> </span>-SN<span class="w"> </span><span class="si">${</span><span class="nv">files</span><span class="p">[@]</span><span class="si">}</span><span class="w">  </span><span class="c1"># open files in simple file viewer, with lines chopped (-S) and numbered (-N), next file can be accessed with :n, more options available in help page by pressing &quot;h&quot;</span>
head<span class="w"> </span>-n10<span class="w"> </span><span class="si">${</span><span class="nv">files</span><span class="p">[0]</span><span class="si">}</span><span class="w">  </span><span class="c1"># view 10 first lines of first file</span>
tail<span class="w"> </span>-n3<span class="w"> </span><span class="si">${</span><span class="nv">files</span><span class="p">[0]</span><span class="si">}</span><span class="w">  </span><span class="c1"># view 3 last lines of first file</span>
</code></pre></div></p>
<p>Output :
<div class="highlight"><pre><span></span><code>## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 ./PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 ./PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 ./PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 ./PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
##      213 ./PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
##      213 ./PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
##      213 ./PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
##      213 ./PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
##      852 total
## ## htsjdk.samtools.metrics.StringHeader
## # CollectHsMetrics BAIT_INTERVALS=[/nfs/PROBIO/new_design_eval/targets/all_baits_twist.interval_list] BAIT_SET_NAME=PB TARGET_INTERVALS=[/nfs/PROBIO/new_design_eval/targets/all_customized_targets_twist.interval_list] INPUT=/dev/stdin OUTPUT=metrics/realignment_cust_targets/HsMetrics/MarkDuplicates/PB/PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt METRIC_ACCUMULATION_LEVEL=[ALL_READS] TMP_DIR=[/scratch/tmp/rebber] REFERENCE_SEQUENCE=/nfs/PROBIO/autoseq-genome/genome/human_g1k_v37_decoy.fasta    NEAR_DISTANCE=250 MINIMUM_MAPPING_QUALITY=20 MINIMUM_BASE_QUALITY=20 CLIP_OVERLAPPING_READS=true INCLUDE_INDELS=false COVERAGE_CAP=200 SAMPLE_SIZE=10000 ALLELE_FRACTION=[0.001, 0.005, 0.01, 0.02, 0.05, 0.1, 0.2, 0.3, 0.5] VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false
## ## htsjdk.samtools.metrics.StringHeader
## # Started on: Tue May 26 17:12:11 CEST 2020
## 
## ## METRICS CLASS picard.analysis.directed.HsMetrics
## BAIT_SET BAIT_TERRITORY  BAIT_DESIGN_EFFICIENCY  ON_BAIT_BASES   NEAR_BAIT_BASES OFF_BAIT_BASES  PCT_SELECTED_BASES  PCT_OFF_BAIT    ON_BAIT_VS_SELECTED MEAN_BAIT_COVERAGE  PCT_USABLE_BASES_ON_BAIT    PCT_USABLE_BASES_ON_TARGET  FOLD_ENRICHMENT HS_LIBRARY_SIZE HS_PENALTY_10X  HS_PENALTY_20X  HS_PENALTY_30X  HS_PENALTY_40X  HS_PENALTY_50X  HS_PENALTY_100X TARGET_TERRITORY    GENOME_SIZE TOTAL_READS PF_READS    PF_BASES    PF_UNIQUE_READS PF_UQ_READS_ALIGNED PF_BASES_ALIGNED    PF_UQ_BASES_ALIGNED ON_TARGET_BASES PCT_PF_READS    PCT_PF_UQ_READS PCT_PF_UQ_READS_ALIGNED MEAN_TARGET_COVERAGE    MEDIAN_TARGET_COVERAGE  MAX_TARGET_COVERAGE ZERO_CVG_TARGETS_PCT    PCT_EXC_DUPE    PCT_EXC_ADAPTER PCT_EXC_MAPQ    PCT_EXC_BASEQ   PCT_EXC_OVERLAP PCT_EXC_OFF_TARGET  FOLD_80_BASE_PENALTY    PCT_TARGET_BASES_1X PCT_TARGET_BASES_2X PCT_TARGET_BASES_10X    PCT_TARGET_BASES_20X    PCT_TARGET_BASES_30X    PCT_TARGET_BASES_40X    PCT_TARGET_BASES_50X    PCT_TARGET_BASES_100X   AT_DROPOUT  GC_DROPOUT  HET_SNP_SENSITIVITY HET_SNP_Q   SAMPLE  LIBRARY READ_GROUP
## PB   290831  0.91208 682291515   195630134   1507472292  0.368041    0.631959    0.777167    2346.006839 0.281894    0.169875    3085.649544     0   0   0   0   0   0   265261  3137454505  16795512    16795512    2420380851  16795512    16679404    2385393941  2385393941  411162960   1   1   0.993087    1550.031705 200 7219    0   0   0   0.101408    0.004983    0.298104    0.423367    7.750159    0.999612    0.999585    0.999582    0.999529    0.999457    0.999416    0.999359    0.99928 7.694876    0.179254    0.999597    34          
## 
## ## HISTOGRAM java.lang.Integer
## 199  0   0
## 200  265028  0
</code></pre></div></p>
<p>Extract sample names from file names:
<div class="highlight"><pre><span></span><code>ls<span class="w"> </span>-lh<span class="w"> </span><span class="si">${</span><span class="nv">files</span><span class="p">[@]</span><span class="si">}</span>

<span class="c1"># Method 1: all at once</span>
<span class="c1"># print content of array with line break as separator to get one item on each line</span>
<span class="c1"># use cut to get the second field of each line when separator is &quot;/&quot;</span>
<span class="c1"># use cut to get the first field of each line when separator is &quot;.&quot;</span>
<span class="nb">printf</span><span class="w"> </span><span class="s1">&#39;%s\n&#39;</span><span class="w"> </span><span class="si">${</span><span class="nv">files</span><span class="p">[@]</span><span class="si">}</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">2</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;/&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;.&quot;</span>

<span class="c1"># Method 2: one at a time in for loop</span>
<span class="c1"># loop over each of the items in the $files array</span>
<span class="c1"># get the basename for the file, i.e. just the file name not full path </span>
<span class="c1"># use sed to remove everything matching pattern &quot;.PB.hsmetrics.txt&quot; (actually replace it with nothing)</span>
<span class="k">for</span><span class="w"> </span>f<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="si">${</span><span class="nv">files</span><span class="p">[@]</span><span class="si">}</span><span class="p">;</span><span class="w"> </span><span class="k">do</span>
<span class="w">    </span>basename<span class="w"> </span><span class="nv">$f</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span><span class="s1">&#39;s/.PB.hsmetrics.txt//g&#39;</span>
<span class="k">done</span>
</code></pre></div>
Output:
<div class="highlight"><pre><span></span><code>## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 ./PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 ./PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 ./PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 ./PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## PB-P-HD1-CFDNA-1811-KH20190925-PB20190927
## PB-P-HD2-CFDNA-1811-KH20190925-PB20190927
## PB-P-HD3-CFDNA-1811-KH20190925-PB20190927
## PB-P-HD6-CFDNA-1811-KH20190925-PB20190927
## PB-P-HD1-CFDNA-1811-KH20190925-PB20190927
## PB-P-HD2-CFDNA-1811-KH20190925-PB20190927
## PB-P-HD3-CFDNA-1811-KH20190925-PB20190927
## PB-P-HD6-CFDNA-1811-KH20190925-PB20190927
</code></pre></div>
Other combinations of the tools in the two methods can also be used to get the same results. 
For most tasks there are many possible ways to solve them on the command line. Just use whatever tools you are familiar with, and when there is something you can't figure out how to do, google offers tons of knowledge and ideas.</p>
<p>Some more things that can be done with a for loop:
<div class="highlight"><pre><span></span><code><span class="k">for</span><span class="w"> </span>f<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="si">${</span><span class="nv">files</span><span class="p">[@]</span><span class="si">}</span><span class="p">;</span><span class="w"> </span><span class="k">do</span>
<span class="w">    </span><span class="c1"># assign output of commands to a variable with the $() construction:</span>
<span class="w">    </span><span class="nv">samp</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$f</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span><span class="s1">&#39;s/.PB.hsmetrics.txt//g&#39;</span><span class="k">)</span>
<span class="w">    </span><span class="c1"># check conditions with if, here if $samp varaible matches some regex:</span>
<span class="w">    </span><span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="nv">$samp</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span>HD1<span class="w"> </span><span class="o">]]</span><span class="p">;</span><span class="w"> </span><span class="k">then</span>
<span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;This is HD1 sample:&quot;</span>
<span class="w">    </span><span class="k">elif</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="nv">$samp</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span>HD2<span class="w"> </span><span class="o">]]</span><span class="p">;</span><span class="w"> </span><span class="k">then</span>
<span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;This is HD2 sample:&quot;</span>
<span class="w">    </span><span class="k">else</span><span class="w"> </span>
<span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;This another sample:&quot;</span>
<span class="w">    </span><span class="k">fi</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="nv">$samp</span>
<span class="w">    </span><span class="c1"># search for the sample name in the file content with grep, -c gives the count of hits</span>
<span class="w">    </span><span class="nv">hits</span><span class="o">=</span><span class="k">$(</span>grep<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$samp</span><span class="s2">&quot;</span><span class="w"> </span><span class="nv">$f</span><span class="k">)</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;File </span><span class="nv">$f</span><span class="s2"> contains sample name </span><span class="nv">$samp</span><span class="s2"> </span><span class="nv">$hits</span><span class="s2"> times&quot;</span>
<span class="w">    </span><span class="nb">echo</span><span class="w">  </span><span class="c1"># empty line</span>
<span class="k">done</span>
</code></pre></div>
Output:
<div class="highlight"><pre><span></span><code>## This is HD1 sample:
## PB-P-HD1-CFDNA-1811-KH20190925-PB20190927
## File ./PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt contains sample name PB-P-HD1-CFDNA-1811-KH20190925-PB20190927 1 times
## 
## This is HD2 sample:
## PB-P-HD2-CFDNA-1811-KH20190925-PB20190927
## File ./PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt contains sample name PB-P-HD2-CFDNA-1811-KH20190925-PB20190927 1 times
## 
## This another sample:
## PB-P-HD3-CFDNA-1811-KH20190925-PB20190927
## File ./PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt contains sample name PB-P-HD3-CFDNA-1811-KH20190925-PB20190927 1 times
## 
## This another sample:
## PB-P-HD6-CFDNA-1811-KH20190925-PB20190927
## File ./PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt contains sample name PB-P-HD6-CFDNA-1811-KH20190925-PB20190927 1 times
</code></pre></div>
To create a table with the info from specific rows and columns we can use the awk program. In this case we will take columns MEAN_TARGET_COVERAGE (number 34) and FOLD_ENRICHMENT (number 13). The relevant data is on line 8, with the header on line 7. 
NR: row number in total; FNR: row number in file.
"&gt;" writes the output to a new file
column -t aligns the output columns nicely
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span>-v<span class="w"> </span><span class="nv">OFS</span><span class="o">=</span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;NR==7 {print &quot;file&quot;, $34, $13}; FNR==8 {print FILENAME, $34, $13}&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span><span class="si">${</span><span class="nv">files</span><span class="p">[@]</span><span class="si">}</span><span class="w"> </span>&gt;<span class="w"> </span>qc_summary.txt
ls<span class="w"> </span>-lhtr<span class="w"> </span>
cat<span class="w"> </span>qc_summary.txt
cat<span class="w"> </span>qc_summary.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>column<span class="w"> </span>-t
</code></pre></div>
Output:
<div class="highlight"><pre><span></span><code>## total 144
## -rw-r--r--  1 rebber  staff   7.0K Apr 14 12:01 PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.slurm.log
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   7.8K Apr 14 12:01 PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.slurm.log
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   8.1K Apr 14 12:01 PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.slurm.log
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   5.3K Apr 14 12:01 PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.slurm.log
## -rw-r--r--  1 rebber  staff   4.1K Apr 14 12:01 PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt
## -rw-r--r--  1 rebber  staff   382B Apr 14 17:56 qc_summary.txt
## file MEAN_TARGET_COVERAGE    FOLD_ENRICHMENT
## ./PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt 1550.031705 3085.649544
## ./PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt 2283.624973 3389.760665
## ./PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt 2599.327089 3688.749254
## ./PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt 1173.933741 4831.030898
## file                                                          MEAN_TARGET_COVERAGE  FOLD_ENRICHMENT
## ./PB-P-HD1-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt  1550.031705           3085.649544
## ./PB-P-HD2-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt  2283.624973           3389.760665
## ./PB-P-HD3-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt  2599.327089           3688.749254
## ./PB-P-HD6-CFDNA-1811-KH20190925-PB20190927.PB.hsmetrics.txt  1173.933741           4831.030898
</code></pre></div></p>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2023, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "content.code.annotate", "content.tabs.link", "search.highlight", "search.share", "search.suggest"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.51198bba.min.js"></script>
      
    
  </body>
</html>