
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.2.3, mkdocs-material-8.1.4">
    
    
      
        <title>Annotation - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bb3983ee.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.e6a45f82.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#annotation-files" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Annotation
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          Pre-course setup
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Pre-course setup" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Pre-course setup
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../installation/" class="md-nav__link">
        Installation
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" type="checkbox" id="__nav_4_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4_2">
          Download Files
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Download Files" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_2">
          <span class="md-nav__icon md-icon"></span>
          Download Files
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../references/" class="md-nav__link">
        Reference
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../annotations/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Lab Session" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_1" type="checkbox" id="__nav_5_1" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_1">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 1" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_2" type="checkbox" id="__nav_5_2" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5_2">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 2" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Annotation
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Annotation
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#files-for-annotation-of-variation-in-the-human-genome" class="md-nav__link">
    Files for annotation of variation in the human genome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#index-the-variation-files" class="md-nav__link">
    Index the variation files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interval-files-and-coordinates-for-the-exome-sequencing-assay" class="md-nav__link">
    Interval files and coordinates for the exome sequencing assay
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#calculate-the-size-of-the-seqcap-v3-exome" class="md-nav__link">
    Calculate the size of the SeqCap v3 exome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-an-inverval-list-for-the-exome-bed-files" class="md-nav__link">
    Create an inverval list for the exome bed files
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_3" type="checkbox" id="__nav_5_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_3">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 3" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_4" type="checkbox" id="__nav_5_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_4">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 4" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../analysis_of_cnv/" class="md-nav__link">
        Copy Number Variations
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#files-for-annotation-of-variation-in-the-human-genome" class="md-nav__link">
    Files for annotation of variation in the human genome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#index-the-variation-files" class="md-nav__link">
    Index the variation files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interval-files-and-coordinates-for-the-exome-sequencing-assay" class="md-nav__link">
    Interval files and coordinates for the exome sequencing assay
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#calculate-the-size-of-the-seqcap-v3-exome" class="md-nav__link">
    Calculate the size of the SeqCap v3 exome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-an-inverval-list-for-the-exome-bed-files" class="md-nav__link">
    Create an inverval list for the exome bed files
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                

<h1 id="annotation-files">Annotation files<a class="headerlink" href="#annotation-files" title="Permanent link">&para;</a></h1>
<h2 id="files-for-annotation-of-variation-in-the-human-genome">Files for annotation of variation in the human genome<a class="headerlink" href="#files-for-annotation-of-variation-in-the-human-genome" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="c1">#These are available from the Broad institute in Boston, who also provides the GATK software suite.</span>
<span class="nb">cd</span> ~/workspace/inputs/references/gatk

<span class="c1">#Download the following files:</span>
<span class="c1"># SNP calibration call sets - dbsnp, hapmap, omni, and 1000G</span>
<span class="c1"># Runtime: &lt; 2min</span>
gsutil cp gs://genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.dbsnp138.vcf .
<span class="c1"># Runtime: ~ 2min</span>
bgzip --threads <span class="m">8</span> Homo_sapiens_assembly38.dbsnp138.vcf
gsutil cp gs://genomics-public-data/resources/broad/hg38/v0/hapmap_3.3.hg38.vcf.gz .
gsutil cp gs://genomics-public-data/resources/broad/hg38/v0/1000G_omni2.5.hg38.vcf.gz .
gsutil cp gs://genomics-public-data/resources/broad/hg38/v0/1000G_phase1.snps.high_confidence.hg38.vcf.gz .

<span class="c1"># Have a loot what the files contain. </span>
<span class="c1"># Dont worry, we will look at VCF files later.</span>
gunzip -c 1000G_phase1.snps.high_confidence.hg38.vcf.gz <span class="p">|</span> less -SN

<span class="c1"># Indel calibration call sets - dbsnp, Mills</span>
gsutil cp gs://genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.known_indels.vcf.gz .
gsutil cp gs://genomics-public-data/resources/broad/hg38/v0/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz .

<span class="c1"># Interval lists that can be used to parallelize certain GATK tasks</span>
gsutil cp gs://genomics-public-data/resources/broad/hg38/v0/wgs_calling_regions.hg38.interval_list .
gsutil cp -r gs://genomics-public-data/resources/broad/hg38/v0/scattered_calling_intervals/ .

<span class="c1"># list the files we just downloaded</span>
ls -lh
</code></pre></div>
<h2 id="index-the-variation-files">Index the variation files<a class="headerlink" href="#index-the-variation-files" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/inputs/references/gatk/

<span class="c1">#SNP calibration call sets - dbsnp, hapmap, omni, and 1000G</span>
<span class="c1"># Runtime: ~ 4min</span>
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> IndexFeatureFile -I ~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.dbsnp138.vcf.gz
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> IndexFeatureFile -I ~/workspace/inputs/references/gatk/hapmap_3.3.hg38.vcf.gz
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> IndexFeatureFile -I ~/workspace/inputs/references/gatk/1000G_omni2.5.hg38.vcf.gz
<span class="c1"># Runtime: ~ 3min</span>
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> IndexFeatureFile -I ~/workspace/inputs/references/gatk/1000G_phase1.snps.high_confidence.hg38.vcf.gz

<span class="c1">#Indel calibration call sets - dbsnp, Mills</span>
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> IndexFeatureFile -I ~/workspace/inputs/references/gatk/Homo_sapiens_assembly38.known_indels.vcf.gz
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> IndexFeatureFile -I ~/workspace/inputs/references/gatk/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz
</code></pre></div>
<h2 id="interval-files-and-coordinates-for-the-exome-sequencing-assay">Interval files and coordinates for the exome sequencing assay<a class="headerlink" href="#interval-files-and-coordinates-for-the-exome-sequencing-assay" title="Permanent link">&para;</a></h2>
<p>These files contain the location of the exons in the human genome that were "targeted", that is for which targeting oligos were created (primary_targets). </p>
<p>The files also contain the location of the actual oligos (capture_targets).</p>
<div class="highlight"><pre><span></span><code><span class="c1"># change directories</span>
<span class="nb">cd</span> ~/workspace/inputs/references/exome

<span class="c1"># download the files</span>
wget -c https://sequencing.roche.com/content/dam/rochesequence/worldwide/shared-designs/SeqCapEZ_Exome_v3.0_Design_Annotation_files.zip
unzip SeqCapEZ_Exome_v3.0_Design_Annotation_files.zip

<span class="c1"># remove the zip</span>
rm -f SeqCapEZ_Exome_v3.0_Design_Annotation_files.zip

<span class="c1"># Lift-over of the Roche coordinates from hg19 to the hg38 assembly.</span>
<span class="c1"># the software is availble from USCS, downloadble here:</span>
<span class="c1">#wget http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/liftOver</span>
<span class="c1">#chmod +x liftOver</span>
<span class="c1"># the chain file</span>
wget -c http://hgdownload.cse.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz

<span class="c1"># run liftover</span>
liftOver SeqCapEZ_Exome_v3.0_Design_Annotation_files/SeqCap_EZ_Exome_v3_hg19_primary_targets.bed  hg19ToHg38.over.chain.gz SeqCap_EZ_Exome_v3_hg38_primary_targets.bed unMapped.bed

liftOver SeqCapEZ_Exome_v3.0_Design_Annotation_files/SeqCap_EZ_Exome_v3_hg19_capture_targets.bed  hg19ToHg38.over.chain.gz SeqCap_EZ_Exome_v3_hg38_capture_targets.bed unMapped1.bed

<span class="c1"># create a version in standard bed format (chr, start, stop)</span>
cut -f <span class="m">1</span>-3 SeqCap_EZ_Exome_v3_hg38_primary_targets.bed &gt; SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.bed

cut -f <span class="m">1</span>-3 SeqCap_EZ_Exome_v3_hg38_capture_targets.bed &gt; SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.bed

<span class="c1"># take a quick look at the format of these files</span>
head SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.bed
head SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.bed
</code></pre></div>
<h2 id="calculate-the-size-of-the-seqcap-v3-exome">Calculate the size of the SeqCap v3 exome<a class="headerlink" href="#calculate-the-size-of-the-seqcap-v3-exome" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="c1">#This can be done in many ways - give it a try yourself before trying the code below and compare results</span>

<span class="c1"># first sort the bed files and store the sorted versions</span>
bedtools sort -i SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.bed &gt; SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.bed
bedtools sort -i SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.bed &gt; SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.bed

<span class="c1"># now merge the bed files to collapse any overlapping regions so they are not double counted.</span>
bedtools merge -i SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.bed &gt; SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.bed

bedtools merge -i SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.bed &gt; SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.merge.bed

<span class="c1"># finally use a Perl one liner to determine the size of the files in Mb</span>
<span class="nv">FILES</span><span class="o">=(</span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.merge.bed SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.bed<span class="o">)</span>
<span class="nb">echo</span> <span class="si">${</span><span class="nv">FILES</span><span class="p">[0]</span><span class="si">}</span>

<span class="k">for</span> FILE <span class="k">in</span> <span class="si">${</span><span class="nv">FILES</span><span class="p">[@]</span><span class="si">}</span>
<span class="k">do</span>
<span class="nb">echo</span> <span class="s2">&quot;--------------------------------------------------------&quot;</span>
<span class="nb">echo</span> <span class="nv">$FILE</span>
<span class="c1">#With merge</span>
cat <span class="nv">$FILE</span> <span class="p">|</span> sortBed -i stdin <span class="p">|</span> mergeBed -i stdin <span class="p">|</span> perl -e <span class="s1">&#39;$counter=0; while(&lt;&gt;){chomp; @array = split &quot;\t&quot;, $_; $counter = $counter + ($array[2] - $array[1]);}; print $counter/1000000, &quot;\n&quot;;&#39;</span>
<span class="k">done</span>
<span class="c1"># note that the size of the space targeted by the exome reagent is ~63 Mbp. Does that sound reasonable?</span>

<span class="c1"># now create a subset of these bed files</span>
grep -w -P <span class="s2">&quot;^chr6|^chr17&quot;</span> SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.bed &gt; exome_regions.bed
<span class="c1">#When creating files, make a habit to investigate the output to avoid downstream confusion</span>
head -n <span class="m">10</span> exome_regions.bed

grep -w -P <span class="s2">&quot;^chr6|^chr17&quot;</span> SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.merge.bed &gt; probe_regions.bed
head -n <span class="m">10</span> probe_regions.bed

<span class="c1"># clean up intermediate files</span>
<span class="c1">#rm -fr SeqCapEZ_Exome_v3.0_Design_Annotation_files/ SeqCap_EZ_Exome_v3_hg38_primary_targets.bed SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.bed SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.bed unMapped.bed</span>
</code></pre></div>
<h2 id="create-an-inverval-list-for-the-exome-bed-files">Create an inverval list for the exome bed files<a class="headerlink" href="#create-an-inverval-list-for-the-exome-bed-files" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="c1"># first for the complete exome and probe bed file</span>
<span class="nb">cd</span> ~/workspace/inputs/references/
<span class="c1">#Commands below already run</span>
<span class="c1">#mkdir temp</span>
<span class="nb">cd</span> temp
<span class="c1">#Commands below already run</span>
<span class="c1">#wget http://genomedata.org/pmbio-workshop/references/genome/all/ref_genome.dict</span>
<span class="nb">cd</span> ~/workspace/inputs/references/exome
java -jar <span class="nv">$PICARD</span> BedToIntervalList <span class="nv">I</span><span class="o">=</span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.bed <span class="nv">O</span><span class="o">=</span>SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.interval_list <span class="nv">SD</span><span class="o">=</span>~/workspace/inputs/references/temp/ref_genome.dict
java -jar <span class="nv">$PICARD</span> BedToIntervalList <span class="nv">I</span><span class="o">=</span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.merge.bed <span class="nv">O</span><span class="o">=</span>SeqCap_EZ_Exome_v3_hg38_capture_targets.v2.sort.merge.interval_list <span class="nv">SD</span><span class="o">=</span>~/workspace/inputs/references/temp/ref_genome.dict
<span class="c1">#rm -fr ~/workspace/inputs/references/temp/</span>
<span class="c1">#Explore the interval-lists and what the files contain, scroll down until the coordinates are shown</span>
less -SN SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.sort.merge.interval_list

<span class="c1"># next for our subset exome and probe regions file</span>
<span class="nb">cd</span> ~/workspace/inputs/references/exome
java -jar /usr/local/bin/picard.jar BedToIntervalList <span class="nv">I</span><span class="o">=</span>exome_regions.bed <span class="nv">O</span><span class="o">=</span>exome_regions.bed.interval_list <span class="nv">SD</span><span class="o">=</span>~/workspace/inputs/references/genome/ref_genome.dict
java -jar /usr/local/bin/picard.jar BedToIntervalList <span class="nv">I</span><span class="o">=</span>probe_regions.bed <span class="nv">O</span><span class="o">=</span>probe_regions.bed.interval_list <span class="nv">SD</span><span class="o">=</span>~/workspace/inputs/references/genome/ref_genome.dict
</code></pre></div>

              
            </article>
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" data-md-state="hidden">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../human_genome/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Human Genome" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Human Genome
            </div>
          </div>
        </a>
      
      
        
        <a href="../processing_of_dna/" class="md-footer__link md-footer__link--next" aria-label="Next: Processing of DNA sequencing data" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Processing of DNA sequencing data
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2021, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "navigation.sections", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest"], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.361d90f1.min.js"}</script>
    
    
      <script src="../../assets/javascripts/bundle.289a2a4b.min.js"></script>
      
    
  </body>
</html>