
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.2.3, mkdocs-material-8.1.4">
    
    
      
        <title>RNAseq - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bb3983ee.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.e6a45f82.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bioinformatic-processing-of-rna-data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              RNAseq
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Lab Session" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_1" type="checkbox" id="__nav_5_1" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_1">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 1" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_2" type="checkbox" id="__nav_5_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_2">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 2" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_3" type="checkbox" id="__nav_5_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_3">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 3" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_4" type="checkbox" id="__nav_5_4" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5_4">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 4" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../structural_variation/" class="md-nav__link">
        Structural Variants
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../analysis_of_cnv/" class="md-nav__link">
        Copy Number Variations
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          RNAseq
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        RNAseq
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#adapter-trimming-of-the-fastq-files" class="md-nav__link">
    Adapter Trimming of the FASTQ files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-fastqc-to-check-the-quality-of-the-fastq-files-as-for-dna-data" class="md-nav__link">
    Run fastqc to check the quality of the fastq-files (as for DNA data)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#alignment" class="md-nav__link">
    Alignment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#merging-bams" class="md-nav__link">
    Merging BAMs
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gtf-general-transfer-format-and-creating-files-for-downstream-processing" class="md-nav__link">
    GTF (General Transfer Format) and creating files for downstream processing
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#post-alignment-qc" class="md-nav__link">
    Post-alignment QC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#indexing-bams" class="md-nav__link">
    Indexing BAMs
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-a-simplified-reference-only-stringtie-expression-approach" class="md-nav__link">
    Run a simplified "reference only" StringTie expression approach
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference-free-expression-analysis-with-kallisto" class="md-nav__link">
    Reference Free Expression Analysis with Kallisto
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generate-abundance-estimates-for-all-samples-using-kallisto" class="md-nav__link">
    Generate abundance estimates for all samples using Kallisto
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_5" type="checkbox" id="__nav_5_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_5">
          Day 5
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 5" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          Day 5
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../joint_curation/" class="md-nav__link">
        Joint Curation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../genomic_alteration_interpretation_exe/" class="md-nav__link">
        Genomic Alterations Interpretation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#adapter-trimming-of-the-fastq-files" class="md-nav__link">
    Adapter Trimming of the FASTQ files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-fastqc-to-check-the-quality-of-the-fastq-files-as-for-dna-data" class="md-nav__link">
    Run fastqc to check the quality of the fastq-files (as for DNA data)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#alignment" class="md-nav__link">
    Alignment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#merging-bams" class="md-nav__link">
    Merging BAMs
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gtf-general-transfer-format-and-creating-files-for-downstream-processing" class="md-nav__link">
    GTF (General Transfer Format) and creating files for downstream processing
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#post-alignment-qc" class="md-nav__link">
    Post-alignment QC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#indexing-bams" class="md-nav__link">
    Indexing BAMs
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run-a-simplified-reference-only-stringtie-expression-approach" class="md-nav__link">
    Run a simplified "reference only" StringTie expression approach
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference-free-expression-analysis-with-kallisto" class="md-nav__link">
    Reference Free Expression Analysis with Kallisto
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generate-abundance-estimates-for-all-samples-using-kallisto" class="md-nav__link">
    Generate abundance estimates for all samples using Kallisto
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                

<h1 id="bioinformatic-processing-of-rna-data">BIOINFORMATIC PROCESSING OF RNA DATA<a class="headerlink" href="#bioinformatic-processing-of-rna-data" title="Permanent link">&para;</a></h1>
<h2 id="adapter-trimming-of-the-fastq-files">Adapter Trimming of the FASTQ files<a class="headerlink" href="#adapter-trimming-of-the-fastq-files" title="Permanent link">&para;</a></h2>
<p>The purpose of adapter trimming is to remove sequences in our data that correspond to the Illumina sequence adapters.  The most common adapter trimming scenario is the removal of adapter sequences that occur at the end of read sequences. This happens when a DNA (or cDNA) fragment is shorter than the read length.  For example if we sequence our RNA-seq fragments to 150 base length and a fragment is only 140 bases long the read will end with 10 bases of adapter sequence. Since this adapter sequence does not correspond to the genome, it will not align. Too much adapter sequence can actually prevent reads from aligning at all. Adapter trimming may therefore sometime improve the overall alignment success rate for an RNA-seq data set.  Adapter trimming involves a simplistic alignment itself and therefore can be computationally expensive.</p>
<p><div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/inputs/references

<span class="c1">#Command already run</span>
<span class="c1">#wget -c http://genomedata.org/rnaseq-tutorial/illumina_multiplex.fa</span>

<span class="c1">#Have a look at the adapter sequences that we want to remove from the RNA data</span>
less -SN ~/workspace/inputs/references/illumina_multiplex.fa

<span class="c1">#Start with tumor RNAseq data</span>
<span class="nb">cd</span> ~/workspace/inputs/data/fastq/RNAseq_Tumor

<span class="c1">#Have a look to see what is in the folder</span>
ls -halt

<span class="c1">#pPerform trimming using flexbar</span>

<span class="c1">#Before using a new tool, make a habit of looking the tool up,</span>
<span class="c1"># google &quot;flexbar trimming adapters&quot; or something similar</span>

<span class="c1">#As this step takes approxmately 25 min, we will use nohup. </span>
<span class="c1">#Google &quot;nohup&quot; ...</span>
<span class="c1">#If running with nohup the command will continue even if the connection breaks to the aws server</span>

<span class="c1">#Also, notice that we are using the flexbar argument --threads 7 which means that we are using 7 cpu threads. To check how many you have at your disposal use the command:</span>
lscpu
<span class="c1">#We have 2 thread left which is good, not to hog all resources</span>
<span class="c1">#To check the available RAM use the command:</span>
cat /proc/meminfo

<span class="c1">#When trimming the adapters run two simultaneous jobs using 7 cores/job and send both to the background if connection breaks. Redirect the standard output to a custom log file</span>
<span class="c1">#nohup cmd &gt; custom-out.log &amp;</span>

nohup flexbar --adapter-min-overlap <span class="m">7</span> --adapter-trim-end RIGHT --adapters ~/workspace/inputs/references/illumina_multiplex.fa --pre-trim-left <span class="m">13</span> --max-uncalled <span class="m">300</span> --min-read-length <span class="m">25</span> --threads <span class="m">7</span> --zip-output GZ --reads RNAseq_Tumor_Lane1_R1.fastq.gz --reads2 RNAseq_Tumor_Lane1_R2.fastq.gz --target ~/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane1 &gt; RNAseq_Tumor_Lane1.log <span class="p">&amp;</span>

<span class="c1">#Start the second process</span>
nohup flexbar --adapter-min-overlap <span class="m">7</span> --adapter-trim-end RIGHT --adapters ~/workspace/inputs/references/illumina_multiplex.fa --pre-trim-left <span class="m">13</span> --max-uncalled <span class="m">300</span> --min-read-length <span class="m">25</span> --threads <span class="m">7</span> --zip-output GZ --reads RNAseq_Tumor_Lane2_R1.fastq.gz --reads2 RNAseq_Tumor_Lane2_R2.fastq.gz --target ~/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane2 &gt; RNAseq_Tumor_Lane2.log <span class="p">&amp;</span>

<span class="c1">#Use htop to monitor the processor load</span>
htop
<span class="c1">#quit htop by pressing &quot;q&quot;</span>

<span class="c1">#have a look at one of the log files</span>
less -SN RNAseq_Tumor_Lane1.log

<span class="c1">####### Wait for the processes to stop before continuing #######</span>

<span class="c1">#Start the </span>
<span class="c1">#Normal RNAseq data</span>
<span class="nb">cd</span> ~/workspace/inputs/data/fastq/RNAseq_Norm

nohup flexbar --adapter-min-overlap <span class="m">7</span> --adapter-trim-end RIGHT --adapters ~/workspace/inputs/references/illumina_multiplex.fa --pre-trim-left <span class="m">13</span> --max-uncalled <span class="m">300</span> --min-read-length <span class="m">25</span> --threads <span class="m">7</span> --zip-output GZ --reads RNAseq_Norm_Lane1_R1.fastq.gz --reads2 RNAseq_Norm_Lane1_R2.fastq.gz --target ~/workspace/inputs/data/fastq/RNAseq_Norm/RNAseq_Norm_Lane1 &gt; RNAseq_Normal_Lane1.log <span class="p">&amp;</span>

flexbar --adapter-min-overlap <span class="m">7</span> --adapter-trim-end RIGHT --adapters ~/workspace/inputs/references/illumina_multiplex.fa --pre-trim-left <span class="m">13</span> --max-uncalled <span class="m">300</span> --min-read-length <span class="m">25</span> --threads <span class="m">7</span> --zip-output GZ --reads RNAseq_Norm_Lane2_R1.fastq.gz --reads2 RNAseq_Norm_Lane2_R2.fastq.gz --target ~/workspace/inputs/data/fastq/RNAseq_Norm/RNAseq_Norm_Lane2 &gt; RNAseq_Normal_Lane2.log <span class="p">&amp;</span>

<span class="c1">####### Wait for the processes to stop #######</span>
</code></pre></div>
As you can see, nohup is very useful for sending processes to the backgroud that will continue if connection breaks or if you need to log out.
For repetition how to send stdout and stderr to a file while running nohup, have a look <a href="https://askubuntu.com/questions/420981/how-do-i-save-terminal-output-to-a-file">here</a>.</p>
<h2 id="run-fastqc-to-check-the-quality-of-the-fastq-files-as-for-dna-data">Run fastqc to check the quality of the fastq-files (as for DNA data)<a class="headerlink" href="#run-fastqc-to-check-the-quality-of-the-fastq-files-as-for-dna-data" title="Permanent link">&para;</a></h2>
<p><div class="highlight"><pre><span></span><code><span class="c1">#Run fastqc and compre the trimmed and non-trimmed file</span>
<span class="nb">cd</span> ~/workspace/inputs/data/fastq/RNAseq_Tumor
fastqc RNAseq_Tumor_Lane1_R1.fastq.gz
fastqc RNAseq_Tumor_Lane1_1.fastq.gz

<span class="c1"># Download the output to your computer</span>
scp -ri ~/course_EC2_01.pem ubuntu@AWS_ADDRESS_HERE:~/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane1_1_fastqc.html .
scp -ri ~/course_EC2_01.pem ubuntu@AWS_ADDRESS_HERE:~/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane1_R1_fastqc.html .
</code></pre></div>
Compare the two html files side by side:
- Any obivous difference?</p>
<div class="highlight"><pre><span></span><code><span class="c1">#Move the trimmed data</span>
mkdir -p ~/workspace/inputs/data/fastq/RNAseq_Tumor/trimmed
mkdir -p ~/workspace/inputs/data/fastq/RNAseq_Norm/trimmed
mv ~/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane1_1.fastq.gz ~/workspace/inputs/data/fastq/RNAseq_Tumor/trimmed/RNAseq_Tumor_Lane1_R1.fastq.gz
mv ~/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane1_2.fastq.gz ~/workspace/inputs/data/fastq/RNAseq_Tumor/trimmed/RNAseq_Tumor_Lane1_R2.fastq.gz
mv ~/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane2_1.fastq.gz ~/workspace/inputs/data/fastq/RNAseq_Tumor/trimmed/RNAseq_Tumor_Lane2_R1.fastq.gz
mv ~/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane2_2.fastq.gz ~/workspace/inputs/data/fastq/RNAseq_Tumor/trimmed/RNAseq_Tumor_Lane2_R2.fastq.gz

mv ~/workspace/inputs/data/fastq/RNAseq_Norm/RNAseq_Norm_Lane1_1.fastq.gz ~/workspace/inputs/data/fastq/RNAseq_Norm/trimmed/RNAseq_Norm_Lane1_R1.fastq.gz
mv ~/workspace/inputs/data/fastq/RNAseq_Norm/RNAseq_Norm_Lane1_2.fastq.gz ~/workspace/inputs/data/fastq/RNAseq_Norm/trimmed/RNAseq_Norm_Lane1_R2.fastq.gz
mv ~/workspace/inputs/data/fastq/RNAseq_Norm/RNAseq_Norm_Lane2_1.fastq.gz ~/workspace/inputs/data/fastq/RNAseq_Norm/trimmed/RNAseq_Norm_Lane2_R1.fastq.gz
mv ~/workspace/inputs/data/fastq/RNAseq_Norm/RNAseq_Norm_Lane2_2.fastq.gz ~/workspace/inputs/data/fastq/RNAseq_Norm/trimmed/RNAseq_Norm_Lane2_R2.fastq.gz

<span class="c1">#make a habit of always checking the result of any command</span>
ls -halt ~/workspace/inputs/data/fastq/RNAseq_Tumor/trimmed
ls -halt ~/workspace/inputs/data/fastq/RNAseq_Norm/trimmed
</code></pre></div>
<h2 id="alignment">Alignment<a class="headerlink" href="#alignment" title="Permanent link">&para;</a></h2>
<p>We will use the aligner <a href="https://ccb.jhu.edu/software/hisat2/index.shtml">HISAT2</a> to perform spliced alignments to our reference genome. For efficiency, the output of HISAT (SAM format) will be piped directly to another program called <a href="http://lomereiter.github.io/sambamba/">sambamba</a> to first convert to BAM format and then sort that BAM file. Before each command we will also create and assign a path for temporary directories.</p>
<div class="highlight"><pre><span></span><code>mkdir -p ~/workspace/rnaseq/alignments
<span class="nb">cd</span> ~/workspace/rnaseq/alignments

<span class="c1"># Align tumor data</span>
<span class="c1"># Runtime: ~15 min each run</span>
<span class="nv">TUMOR_DATA_1_TEMP</span><span class="o">=</span><span class="sb">`</span>mktemp -d ~/workspace/rnaseq/alignments/2895626107.XXXXXXXXXXXX<span class="sb">`</span>
nohup hisat2 -p <span class="m">7</span> --dta -x ~/workspace/inputs/references/transcriptome/ref_genome --rg-id <span class="m">2895626107</span> --rg PL:ILLUMINA --rg PU:H3MYFBBXX-GCCAAT.4 --rg LB:rna_tumor_lib1 --rg SM:HCC1395_RNA --rna-strandness RF -1 ~/workspace/inputs/data/fastq/RNAseq_Tumor/trimmed/RNAseq_Tumor_Lane1_R1.fastq.gz -2  ~/workspace/inputs/data/fastq/RNAseq_Tumor/trimmed/RNAseq_Tumor_Lane1_R2.fastq.gz <span class="p">|</span> sambamba view -S -f bam -l <span class="m">0</span> /dev/stdin <span class="p">|</span> sambamba sort -t <span class="m">8</span> -m 24G --tmpdir <span class="nv">$TUMOR_DATA_1_TEMP</span> -o ~/workspace/rnaseq/alignments/RNAseq_Tumor_Lane1.bam /dev/stdin <span class="p">&amp;</span>

<span class="nv">TUMOR_DATA_2_TEMP</span><span class="o">=</span><span class="sb">`</span>mktemp -d ~/workspace/rnaseq/alignments/2895626112.XXXXXXXXXXXX<span class="sb">`</span>
nohup hisat2 -p <span class="m">7</span> --dta -x ~/workspace/inputs/references/transcriptome/ref_genome --rg-id <span class="m">2895626112</span> --rg PL:ILLUMINA --rg PU:H3MYFBBXX-GCCAAT.5 --rg LB:rna_tumor_lib1 --rg SM:HCC1395_RNA --rna-strandness RF -1 ~/workspace/inputs/data/fastq/RNAseq_Tumor/trimmed/RNAseq_Tumor_Lane2_R1.fastq.gz -2  ~/workspace/inputs/data/fastq/RNAseq_Tumor/trimmed/RNAseq_Tumor_Lane2_R2.fastq.gz <span class="p">|</span> sambamba view -S -f bam -l <span class="m">0</span> /dev/stdin <span class="p">|</span> sambamba sort -t <span class="m">8</span> -m 24G --tmpdir <span class="nv">$TUMOR_DATA_2_TEMP</span> -o ~/workspace/rnaseq/alignments/RNAseq_Tumor_Lane2.bam /dev/stdin <span class="p">&amp;</span>

<span class="c1">#Wait for the processes to finish, monitor using htop</span>
rmdir <span class="nv">$TUMOR_DATA_2_TEMP</span> <span class="nv">$TUMOR_DATA_1_TEMP</span>

<span class="c1"># Align normal data</span>

<span class="nv">NORMAL_DATA_1_TEMP</span><span class="o">=</span><span class="sb">`</span>mktemp -d ~/workspace/rnaseq/alignments/2895625992.XXXXXXXXXXXX<span class="sb">`</span>
nohup hisat2 -p <span class="m">7</span> --dta -x ~/workspace/inputs/references/transcriptome/ref_genome --rg-id <span class="m">2895625992</span> --rg PL:ILLUMINA --rg PU:H3MYFBBXX-CTTGTA.4 --rg LB:rna_norm_lib1 --rg SM:HCC1395BL_RNA --rna-strandness RF -1 ~/workspace/inputs/data/fastq/RNAseq_Norm/trimmed/RNAseq_Norm_Lane1_R1.fastq.gz -2  ~/workspace/inputs/data/fastq/RNAseq_Norm/trimmed/RNAseq_Norm_Lane1_R2.fastq.gz <span class="p">|</span> sambamba view -S -f bam -l <span class="m">0</span> /dev/stdin <span class="p">|</span> sambamba sort -t <span class="m">8</span> -m 24G --tmpdir <span class="nv">$NORMAL_DATA_1_TEMP</span> -o ~/workspace/rnaseq/alignments/RNAseq_Norm_Lane1.bam /dev/stdin <span class="p">&amp;</span>

<span class="nv">NORMAL_DATA_2_TEMP</span><span class="o">=</span><span class="sb">`</span>mktemp -d ~/workspace/rnaseq/alignments/2895626097.XXXXXXXXXXXX<span class="sb">`</span>

nohup hisat2 -p <span class="m">7</span> --dta -x ~/workspace/inputs/references/transcriptome/ref_genome --rg-id <span class="m">2895626097</span> --rg PL:ILLUMINA --rg PU:H3MYFBBXX-CTTGTA.5 --rg LB:rna_norm_lib1 --rg SM:HCC1395BL_RNA --rna-strandness RF -1 ~/workspace/inputs/data/fastq/RNAseq_Norm/trimmed/RNAseq_Norm_Lane2_R1.fastq.gz -2  ~/workspace/inputs/data/fastq/RNAseq_Norm/trimmed/RNAseq_Norm_Lane2_R2.fastq.gz <span class="p">|</span> sambamba view -S -f bam -l <span class="m">0</span> /dev/stdin <span class="p">|</span> sambamba sort -t <span class="m">8</span> -m 24G --tmpdir <span class="nv">$NORMAL_DATA_2_TEMP</span> -o ~/workspace/rnaseq/alignments/RNAseq_Norm_Lane2.bam /dev/stdin <span class="p">&amp;</span>

<span class="c1">#Wait for the processes to finish, monitor using htop</span>
rmdir <span class="nv">$NORMAL_DATA_2_TEMP</span> <span class="nv">$NORMAL_DATA_1_TEMP</span>
</code></pre></div>
<h2 id="merging-bams">Merging BAMs<a class="headerlink" href="#merging-bams" title="Permanent link">&para;</a></h2>
<p>Since we have multiple BAMs of each sample that just represent additional data for the same sequence library, we should combine them into a single BAM for convenience before proceeding.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/rnaseq/alignments
<span class="c1">#Runtime: ~ 8m each merging command</span>
sambamba merge -t <span class="m">8</span> ~/workspace/rnaseq/alignments/RNAseq_Norm.bam ~/workspace/rnaseq/alignments/RNAseq_Norm_Lane1.bam ~/workspace/rnaseq/alignments/RNAseq_Norm_Lane2.bam

sambamba merge -t <span class="m">8</span> ~/workspace/rnaseq/alignments/RNAseq_Tumor.bam ~/workspace/rnaseq/alignments/RNAseq_Tumor_Lane1.bam ~/workspace/rnaseq/alignments/RNAseq_Tumor_Lane2.bam
</code></pre></div>
<h2 id="gtf-general-transfer-format-and-creating-files-for-downstream-processing">GTF (General Transfer Format) and creating files for downstream processing<a class="headerlink" href="#gtf-general-transfer-format-and-creating-files-for-downstream-processing" title="Permanent link">&para;</a></h2>
<p>We will encounter the GTF file format during the exercises below, described in detail <a href="https://www.ensembl.org/info/website/upload/gff.html#fields">here</a>.</p>
<p>The GTF format is use to describe genes and other features of DNA, RNA and proteins.</p>
<p>In the section below we will prepare files needed for RNAseq QC</p>
<p><strong>NOTE: for the sake of time we are only investigating genes on chr6 and chr17</strong></p>
<div class="highlight"><pre><span></span><code><span class="c1">#Go to this directory:</span>
<span class="nb">cd</span> ~/workspace/inputs/references/transcriptome

<span class="c1">#Have a peak into the transcriptome gtf file and try to make sense of it using the ensembl description above</span>
less -SN ref_transcriptome.gtf

<span class="c1">#check on which chromsomes from which we have transcripts</span>
cut -f1 ref_transcriptome.gtf <span class="p">|</span> sort <span class="p">|</span> uniq -c

<span class="c1"># Generating the necessary input files for picard CollectRnaSeqMetrics</span>
<span class="nb">cd</span> ~/workspace/inputs/references/transcriptome

grep -i rrna ref_transcriptome.gtf &gt; ref_ribosome.gtf

<span class="c1">#A bed file with Ribosomal RNA</span>
gff2bed &lt; ~/workspace/inputs/references/transcriptome/ref_ribosome.gtf &gt; ref_ribosome.bed

<span class="c1">#Check the ribosome bed file</span>
less -SN ref_ribosome.bed

java -jar <span class="nv">$PICARD</span> BedToIntervalList <span class="nv">I</span><span class="o">=</span>~/workspace/inputs/references/transcriptome/ref_ribosome.bed <span class="nv">O</span><span class="o">=</span>~/workspace/inputs/references/transcriptome/ref_ribosome.interval_list <span class="nv">SD</span><span class="o">=</span>~/workspace/inputs/references/genome/ref_genome.dict

gtfToGenePred -genePredExt ~/workspace/inputs/references/transcriptome/ref_transcriptome.gtf ~/workspace/inputs/references/transcriptome/ref_flat.txt

cat ref_flat.txt <span class="p">|</span> awk <span class="s1">&#39;{print $12&quot;\t&quot;$0}&#39;</span> <span class="p">|</span> cut -d<span class="s1">$&#39;\t&#39;</span> -f1-11 &gt; ref_flat_final.txt

mv ref_flat_final.txt ref_flat.txt
</code></pre></div>
<h2 id="post-alignment-qc">Post-alignment QC<a class="headerlink" href="#post-alignment-qc" title="Permanent link">&para;</a></h2>
<p><div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/rnaseq/alignments
<span class="c1">#Run samtools flagstat</span>
<span class="c1">#Runtime: ~2min, start and send to background</span>
nohup samtools flagstat ~/workspace/rnaseq/alignments/RNAseq_Norm.bam &gt; ~/workspace/rnaseq/alignments/RNAseq_Norm_flagstat.txt <span class="m">2</span>&gt; flagstat_RNA_N_bam.out <span class="p">&amp;</span>
nohup samtools flagstat ~/workspace/rnaseq/alignments/RNAseq_Tumor.bam &gt; ~/workspace/rnaseq/alignments/RNAseq_Tumor_flagstat.txt <span class="m">2</span>&gt; flagstat_RNA_T_bam.out <span class="p">&amp;</span>

<span class="c1">#Runtime: ~12 min, start and send to background</span>
nohup fastqc -t <span class="m">8</span> ~/workspace/rnaseq/alignments/RNAseq_Tumor.bam &gt; fastqc_RNA_T_bam.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>
nohup fastqc -t <span class="m">8</span> ~/workspace/rnaseq/alignments/RNAseq_Norm.bam &gt; fastqc_RNA_N_bam.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>

<span class="c1"># Runtime: 26min</span>
nohup java -jar <span class="nv">$PICARD</span> CollectRnaSeqMetrics <span class="nv">I</span><span class="o">=</span>~/workspace/rnaseq/alignments/RNAseq_Norm.bam <span class="nv">O</span><span class="o">=</span>~/workspace/rnaseq/alignments/RNAseq_Norm.RNA_Metrics <span class="nv">REF_FLAT</span><span class="o">=</span>~/workspace/inputs/references/transcriptome/ref_flat.txt <span class="nv">STRAND</span><span class="o">=</span>SECOND_READ_TRANSCRIPTION_STRAND <span class="nv">RIBOSOMAL_INTERVALS</span><span class="o">=</span>~/workspace/inputs/references/transcriptome/ref_ribosome.interval_list &gt; collectRnaSeqMetrics_N.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>
<span class="c1">#Follow the progress of the program by looking in the nohup output file</span>
less -SN collectRnaSeqMetrics_N.out

nohup java -jar <span class="nv">$PICARD</span> CollectRnaSeqMetrics <span class="nv">I</span><span class="o">=</span>~/workspace/rnaseq/alignments/RNAseq_Tumor.bam <span class="nv">O</span><span class="o">=</span>~/workspace/rnaseq/alignments/RNAseq_Tumor.RNA_Metrics <span class="nv">REF_FLAT</span><span class="o">=</span>~/workspace/inputs/references/transcriptome/ref_flat.txt <span class="nv">STRAND</span><span class="o">=</span>SECOND_READ_TRANSCRIPTION_STRAND <span class="nv">RIBOSOMAL_INTERVALS</span><span class="o">=</span>~/workspace/inputs/references/transcriptome/ref_ribosome.interval_list &gt; collectRnaSeqMetrics_T.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>

<span class="nb">cd</span> ~/workspace/rnaseq
<span class="c1">#mkdir post_align_qc</span>
<span class="nb">cd</span> post_align_qc
multiqc ~/workspace/rnaseq/alignments/

<span class="c1">#Finally, download multiqc files to your local computer</span>
<span class="c1"># Download MultiQC output to the local computer, open the .html in you favourite browser.</span>
scp -ri ~/course_EC2_01.pem ubuntu@AWS_ADDRESS_HERE:~/workspace/rnaseq/post_align_qc/multiqc* .
</code></pre></div>
Familiarize yourself with the RNAseq MultiQC data</p>
<h2 id="indexing-bams">Indexing BAMs<a class="headerlink" href="#indexing-bams" title="Permanent link">&para;</a></h2>
<p>In order to be able to view our BAM files in IGV, as usual we need to index them
<div class="highlight"><pre><span></span><code><span class="nb">cd</span>  ~/workspace/rnaseq/alignments/
samtools index RNAseq_Norm.bam
samtools index RNAseq_Tumor.bam
</code></pre></div></p>
<h2 id="run-a-simplified-reference-only-stringtie-expression-approach">Run a simplified "reference only" StringTie expression approach<a class="headerlink" href="#run-a-simplified-reference-only-stringtie-expression-approach" title="Permanent link">&para;</a></h2>
<ul>
<li>The StringTie developer's recommend to: <ul>
<li>perform reference guided transcript compilation (aka transcript assembly) on each individual sample.</li>
<li>merge transcript predictions from all samples into a single model of the transcriptome.</li>
<li>annotate this predicted transcriptome with known transcriptome information.</li>
<li>estimate abundance for each of the transcripts in this final transcriptome model in each sample.</li>
</ul>
</li>
</ul>
<p>In the final result we would have abundance for all the same transcripts across all samples. This includes a combination of predicted and known transcripts. </p>
<p>It is sometimes convenient to have a more simplified workflow where we only have values for known transcripts. This is particularly true in species where we already have comprehensive high quality transcriptome annotations and there is less of a focus on de novo transcript discovery.</p>
<p>The following workflow produces a "reference-only" transcriptome result in which we will perform abundance calculations on each lane of data individually.</p>
<p><div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/rnaseq
<span class="c1">#mkdir ref-only-expression</span>
<span class="nb">cd</span> ref-only-expression

nohup stringtie -p <span class="m">4</span> -e -G ~/workspace/inputs/references/transcriptome/ref_transcriptome.gtf -o ~/workspace/rnaseq/ref-only-expression/RNAseq_Tumor_Lane1/transcripts.gtf -A ~/workspace/rnaseq/ref-only-expression/RNAseq_Tumor_Lane1/gene_abundances.tsv ~/workspace/rnaseq/alignments/RNAseq_Tumor_Lane1.bam &gt; knowntranscripts_T1_L1.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>
nohup stringtie -p <span class="m">4</span> -e -G ~/workspace/inputs/references/transcriptome/ref_transcriptome.gtf -o ~/workspace/rnaseq/ref-only-expression/RNAseq_Tumor_Lane2/transcripts.gtf -A ~/workspace/rnaseq/ref-only-expression/RNAseq_Tumor_Lane2/gene_abundances.tsv ~/workspace/rnaseq/alignments/RNAseq_Tumor_Lane2.bam &gt; knowntranscripts_T1_L2.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>
nohup stringtie -p <span class="m">4</span> -e -G ~/workspace/inputs/references/transcriptome/ref_transcriptome.gtf -o ~/workspace/rnaseq/ref-only-expression/RNAseq_Norm_Lane1/transcripts.gtf -A ~/workspace/rnaseq/ref-only-expression/RNAseq_Norm_Lane1/gene_abundances.tsv ~/workspace/rnaseq/alignments/RNAseq_Norm_Lane1.bam &gt; knowntranscripts_N1_L1.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>
nohup stringtie -p <span class="m">4</span> -e -G ~/workspace/inputs/references/transcriptome/ref_transcriptome.gtf -o ~/workspace/rnaseq/ref-only-expression/RNAseq_Norm_Lane2/transcripts.gtf -A ~/workspace/rnaseq/ref-only-expression/RNAseq_Norm_Lane2/gene_abundances.tsv ~/workspace/rnaseq/alignments/RNAseq_Norm_Lane2.bam &gt; knowntranscripts_T2_L2.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>

<span class="c1">#Wait to continue submitting jobs until the scripts have finished</span>
</code></pre></div>
Create a tidy expression matrix files for the StringTie results. This will be done at both the gene and transcript level and also will take into account the various expression measures produced: coverage, FPKM, and TPM.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/rnaseq/ref-only-expression

<span class="c1">#Already run</span>
<span class="c1">#wget https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/stringtie_expression_matrix.pl</span>
<span class="c1">#chmod +x stringtie_expression_matrix.pl</span>

./stringtie_expression_matrix.pl --expression_metric<span class="o">=</span>TPM --result_dirs<span class="o">=</span><span class="s1">&#39;RNAseq_Norm_Lane1,RNAseq_Norm_Lane2,RNAseq_Tumor_Lane1,RNAseq_Tumor_Lane2&#39;</span> --transcript_matrix_file<span class="o">=</span>transcript_tpm_all_samples.tsv --gene_matrix_file<span class="o">=</span>gene_tpm_all_samples.tsv

./stringtie_expression_matrix.pl --expression_metric<span class="o">=</span>FPKM --result_dirs<span class="o">=</span><span class="s1">&#39;RNAseq_Norm_Lane1,RNAseq_Norm_Lane2,RNAseq_Tumor_Lane1,RNAseq_Tumor_Lane2&#39;</span> --transcript_matrix_file<span class="o">=</span>transcript_fpkm_all_samples.tsv --gene_matrix_file<span class="o">=</span>gene_fpkm_all_samples.tsv

./stringtie_expression_matrix.pl --expression_metric<span class="o">=</span>Coverage --result_dirs<span class="o">=</span><span class="s1">&#39;RNAseq_Norm_Lane1,RNAseq_Norm_Lane2,RNAseq_Tumor_Lane1,RNAseq_Tumor_Lane2&#39;</span> --transcript_matrix_file<span class="o">=</span>transcript_coverage_all_samples.tsv --gene_matrix_file<span class="o">=</span>gene_coverage_all_samples.tsv

<span class="c1">#Have a look at the output files</span>
head gene_coverage_all_samples.tsv transcript_coverage_all_samples.tsv gene_fpkm_all_samples.tsv transcript_fpkm_all_samples.tsv gene_tpm_all_samples.tsv transcript_tpm_all_samples.tsv
</code></pre></div>
<h2 id="reference-free-expression-analysis-with-kallisto">Reference Free Expression Analysis with Kallisto<a class="headerlink" href="#reference-free-expression-analysis-with-kallisto" title="Permanent link">&para;</a></h2>
<p>Remember that in previous sections we have been using reference genome fasta sequences for the reference for alignment and subsequent steps. However, Kallisto works directly on target cDNA/transcript sequences. </p>
<p>We have for stringtie used transcript annotations for genes on our subset of chromosomes (i.e. chr6 and chr17). The transcript models were downloaded from Ensembl in GTF format. This GTF contains a description of the coordinates of exons that make up each transcript but it does not contain the transcript sequences themselves which Kallisto is using. There are many places we could obtain such transcript sequences. For example, we could have download them directly in Fasta format from the Ensembl FTP site (or from UCSC or NCBI).</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/rnaseq/
<span class="c1">#mkdir kallisto</span>
<span class="nb">cd</span> kallisto

<span class="c1"># first check that the GTF and Fasta file are present for Kallisto</span>
head ~/workspace/inputs/references/transcriptome/ref_transcriptome.gtf
head ~/workspace/inputs/references/transcriptome/kallisto/ref_transcriptome_clean.fa

<span class="c1">#Each transcript and its sequence</span>
head ~/workspace/inputs/references/transcriptome/kallisto/ref_transcriptome_clean.fa

<span class="c1"># now check for the kallisto index is there</span>
ls -halt ~/workspace/inputs/references/transcriptome/kallisto/ref_transcriptome_kallisto_index

<span class="c1"># create a list of all transcript IDs for later use:</span>
<span class="nb">cd</span> ~/workspace/rnaseq/kallisto/
cat ~/workspace/inputs/references/transcriptome/kallisto/ref_transcriptome_clean.fa <span class="p">|</span> grep <span class="s2">&quot;&gt;&quot;</span> <span class="p">|</span> perl -ne <span class="s1">&#39;$_ =~ s/\&gt;//; print $_&#39;</span> <span class="p">|</span> sort <span class="p">|</span> uniq &gt; transcript_id_list.txt

head -n <span class="m">10</span> transcript_id_list.txt
</code></pre></div>
<h2 id="generate-abundance-estimates-for-all-samples-using-kallisto">Generate abundance estimates for all samples using Kallisto<a class="headerlink" href="#generate-abundance-estimates-for-all-samples-using-kallisto" title="Permanent link">&para;</a></h2>
<p>As we did with StringTie we will generate transcript abundances for each of our demonstration samples using Kallisto. Here we are treating the two lanes for each sample as if they were independent samples.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/rnaseq/kallisto/
<span class="c1">#mkdir quants</span>
<span class="nb">cd</span> quants

nohup kallisto quant --index<span class="o">=</span>/home/ubuntu/workspace/inputs/references/transcriptome/kallisto/ref_transcriptome_kallisto_index --output-dir<span class="o">=</span>RNAseq_Norm_Lane1 --threads<span class="o">=</span><span class="m">8</span> --plaintext /home/ubuntu/workspace/inputs/data/fastq/RNAseq_Norm/RNAseq_Norm_Lane1_R1.fastq.gz /home/ubuntu/workspace/inputs/data/fastq/RNAseq_Norm/RNAseq_Norm_Lane1_R2.fastq.gz &gt; kallisto_quant_N_L1.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>

nohup kallisto quant --index<span class="o">=</span>/home/ubuntu/workspace/inputs/references/transcriptome/kallisto/ref_transcriptome_kallisto_index --output-dir<span class="o">=</span>RNAseq_Norm_Lane2 --threads<span class="o">=</span><span class="m">8</span> --plaintext /home/ubuntu/workspace/inputs/data/fastq/RNAseq_Norm/RNAseq_Norm_Lane2_R1.fastq.gz /home/ubuntu/workspace/inputs/data/fastq/RNAseq_Norm/RNAseq_Norm_Lane2_R2.fastq.gz &gt; kallisto_quant_N_L2.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>

nohup kallisto quant --index<span class="o">=</span>/home/ubuntu/workspace/inputs/references/transcriptome/kallisto/ref_transcriptome_kallisto_index --output-dir<span class="o">=</span>RNAseq_Tumor_Lane1 --threads<span class="o">=</span><span class="m">8</span> --plaintext /home/ubuntu/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane1_R1.fastq.gz /home/ubuntu/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane1_R2.fastq.gz &gt; kallisto_quant_T_L1.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>

nohup kallisto quant --index<span class="o">=</span>/home/ubuntu/workspace/inputs/references/transcriptome/kallisto/ref_transcriptome_kallisto_index --output-dir<span class="o">=</span>RNAseq_Tumor_Lane2 --threads<span class="o">=</span><span class="m">8</span> --plaintext /home/ubuntu/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane2_R1.fastq.gz /home/ubuntu/workspace/inputs/data/fastq/RNAseq_Tumor/RNAseq_Tumor_Lane2_R2.fastq.gz &gt; kallisto_quant_T_L2.out <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>

<span class="c1">#check nohup output</span>
less -SN kallisto_quant_T_L2.out

<span class="c1">#Create a single TSV file that has the TPM abundance estimates for all six samples.</span>
<span class="c1">#First check the contets of an abudance output file from Kallisto</span>
less -SN ./RNAseq_Norm_Lane1/abundance.tsv

<span class="c1">#Merge all files</span>
paste */abundance.tsv <span class="p">|</span> cut -f <span class="m">1</span>,2,5,10,15,20 &gt; transcript_tpms_all_samples.tsv
ls -1 */abundance.tsv <span class="p">|</span> perl -ne <span class="s1">&#39;chomp $_; if ($_ =~ /(\S+)\/abundance\.tsv/){print &quot;\t$1&quot;}&#39;</span> <span class="p">|</span> perl -ne <span class="s1">&#39;print &quot;target_id\tlength$_\n&quot;&#39;</span> &gt; header.tsv

cat header.tsv transcript_tpms_all_samples.tsv <span class="p">|</span> grep -v <span class="s2">&quot;tpm&quot;</span> &gt; transcript_tpms_all_samples.tsv2

mv transcript_tpms_all_samples.tsv2 transcript_tpms_all_samples.tsv

rm -f header.tsv

head transcript_tpms_all_samples.tsv

<span class="c1">#First create a gene version of the Kallisto TPM matrix, we will simply sum the TPM values for transcripts of the same gene.</span>

<span class="c1"># Command already run</span>
<span class="c1">#wget https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/kallisto_gene_matrix.pl</span>
<span class="c1">#chmod +x kallisto_gene_matrix.pl</span>

./kallisto_gene_matrix.pl --gtf_file<span class="o">=</span>/home/ubuntu/workspace/inputs/references/transcriptome/ref_transcriptome.gtf --kallisto_transcript_matrix_in<span class="o">=</span>transcript_tpms_all_samples.tsv --kallisto_transcript_matrix_out<span class="o">=</span>gene_tpms_all_samples.tsv

less -SN gene_tpms_all_samples.tsv
</code></pre></div>
<h1 id="compare-expression-values-between-kallisto-and-stringtie">Compare expression values between Kallisto and StringTie<a class="headerlink" href="#compare-expression-values-between-kallisto-and-stringtie" title="Permanent link">&para;</a></h1>
<p>To compare the two approaches we can use the expression value for each Ensembl transcript.</p>
<p>To do this comparison, we need to gather the expression estimates for each of our replicates from each approach. </p>
<p>First - download the files to your local machine</p>
<div class="highlight"><pre><span></span><code><span class="c1">#Run these commands on your local machine</span>

scp -ri ~/course_EC2_01.pem ubuntu@AWS_ADDRESS_HERE:/home/ubuntu/workspace/rnaseq/kallisto/transcript_tpms_all_samples.tsv .

mv transcript_tpms_all_samples.tsv kallisto_transcript_tpms_all_samples.tsv

scp -ri ~/course_EC2_01.pem ubuntu@AWS_ADDRESS_HERE:/home/ubuntu/workspace/rnaseq/ref-only-expression/transcript_tpm_all_samples.tsv .
mv transcript_tpm_all_samples.tsv stringtie_transcript_tpms_all_samples.tsv

<span class="c1">#Remember, one gene can have many transcripts. The ensembl gene ID for TP53 is ENSG00000141510. Check which ensembl transcripts that exist in the reference gtf.</span>

<span class="c1">#RUN ON THE AWS SERVER</span>
grep ENST ~/workspace/inputs/references/transcriptome/ref_transcriptome.gtf <span class="p">|</span> grep ENSG00000141510 <span class="p">|</span> perl -nle <span class="s1">&#39;@a = split /\t/; $_ =~ /transcript_id\s&quot;(\S*)&quot;;/g; print $1;&#39;</span> <span class="p">|</span> sort <span class="p">|</span> uniq

<span class="c1"># start R</span>

<span class="c1"># set the working directory to where the files were downloaded</span>
setwd<span class="o">(</span><span class="s2">&quot;DIR GOES HERE&quot;</span><span class="o">)</span>

<span class="c1"># load libraries</span>
library<span class="o">(</span>ggplot2<span class="o">)</span>
library<span class="o">(</span>reshape2<span class="o">)</span>

<span class="c1"># read in data</span>
kallisto_transcript_tpm &lt;- read.delim<span class="o">(</span><span class="s2">&quot;kallisto_transcript_tpms_all_samples.tsv&quot;</span><span class="o">)</span>
head<span class="o">(</span>kallisto_transcript_tpm<span class="o">)</span>

stringtie_transcript_tpm &lt;- read.delim<span class="o">(</span><span class="s2">&quot;stringtie_transcript_tpms_all_samples.tsv&quot;</span><span class="o">)</span>
head<span class="o">(</span>stringtie_transcript_tpm<span class="o">)</span>

<span class="c1"># minor reformatting</span>
kallisto_transcript_tpm &lt;- kallisto_transcript_tpm<span class="o">[</span>,-2<span class="o">]</span>
kallisto_transcript_tpm &lt;- melt<span class="o">(</span>kallisto_transcript_tpm, id.vars<span class="o">=</span>c<span class="o">(</span><span class="s2">&quot;target_id&quot;</span><span class="o">))</span>
head<span class="o">(</span>kallisto_transcript_tpm<span class="o">)</span>

stringtie_transcript_tpm &lt;- melt<span class="o">(</span>stringtie_transcript_tpm, id.vars<span class="o">=</span>c<span class="o">(</span><span class="s2">&quot;Transcript_ID&quot;</span><span class="o">))</span>
head<span class="o">(</span>stringtie_transcript_tpm<span class="o">)</span>

<span class="c1"># merge the data</span>
kallisto_stringtie_tpm &lt;- merge<span class="o">(</span>kallisto_transcript_tpm, stringtie_transcript_tpm, by.x<span class="o">=</span>c<span class="o">(</span><span class="s2">&quot;target_id&quot;</span>, <span class="s2">&quot;variable&quot;</span><span class="o">)</span>, by.y<span class="o">=</span>c<span class="o">(</span><span class="s2">&quot;Transcript_ID&quot;</span>, <span class="s2">&quot;variable&quot;</span><span class="o">)</span>, <span class="nv">suffixes</span><span class="o">=</span>c<span class="o">(</span><span class="s2">&quot;.kallisto&quot;</span>, <span class="s2">&quot;.stringtie&quot;</span><span class="o">))</span>

<span class="c1">#The numeric vector is read as a character vector, correcting ..</span>
kallisto_stringtie_tpm<span class="nv">$value</span>.stringtie <span class="o">=</span> as.numeric<span class="o">(</span>kallisto_stringtie_tpm<span class="nv">$value</span>.stringtie<span class="o">)</span>
str<span class="o">(</span>kallisto_stringtie_tpm<span class="o">)</span>

<span class="c1">#TP53 transcript ID vector from the transcript GTF</span>
<span class="nv">TP53</span> <span class="o">=</span> c<span class="o">(</span><span class="s2">&quot;ENST00000269305&quot;</span>, <span class="s2">&quot;ENST00000359597&quot;</span>, <span class="s2">&quot;ENST00000413465&quot;</span>, <span class="s2">&quot;ENST00000420246&quot;</span>, <span class="s2">&quot;ENST00000445888&quot;</span>, <span class="s2">&quot;ENST00000455263&quot;</span>, <span class="s2">&quot;ENST00000503591&quot;</span>, <span class="s2">&quot;ENST00000504290&quot;</span>, <span class="s2">&quot;ENST00000504937&quot;</span>, <span class="s2">&quot;ENST00000505014&quot;</span>, <span class="s2">&quot;ENST00000508793&quot;</span>, <span class="s2">&quot;ENST00000509690&quot;</span>, <span class="s2">&quot;ENST00000510385&quot;</span>, <span class="s2">&quot;ENST00000514944&quot;</span>, <span class="s2">&quot;ENST00000574684&quot;</span>, <span class="s2">&quot;ENST00000576024&quot;</span>, <span class="s2">&quot;ENST00000604348&quot;</span>, <span class="s2">&quot;ENST00000610292&quot;</span>, <span class="s2">&quot;ENST00000610538&quot;</span>, <span class="s2">&quot;ENST00000610623&quot;</span>, <span class="s2">&quot;ENST00000615910&quot;</span>, <span class="s2">&quot;ENST00000617185&quot;</span>, <span class="s2">&quot;ENST00000618944&quot;</span>, <span class="s2">&quot;ENST00000619186&quot;</span>, <span class="s2">&quot;ENST00000619485&quot;</span>, <span class="s2">&quot;ENST00000620739&quot;</span>, <span class="s2">&quot;ENST00000622645&quot;</span>, <span class="s2">&quot;ENST00000635293&quot;</span><span class="o">)</span>

<span class="nv">idx</span> <span class="o">=</span> match<span class="o">(</span>kallisto_stringtie_tpm<span class="nv">$target_id</span>, TP53<span class="o">)</span>
idx.1 <span class="o">=</span> which<span class="o">(</span>is.na<span class="o">(</span>idx<span class="o">)</span> <span class="o">==</span> FALSE<span class="o">)</span>
idx.2 <span class="o">=</span> idx<span class="o">[</span>idx.1<span class="o">]</span>
TP53<span class="o">[</span>idx.2<span class="o">]</span> <span class="o">==</span> kallisto_stringtie_tpm<span class="nv">$target_id</span><span class="o">[</span>idx.1<span class="o">]</span>
<span class="c1">#Check TP53 expression values</span>
kallisto_stringtie_tpm<span class="o">[</span>idx.2,<span class="o">]</span>
<span class="c1">#To plot TP53, make a new column</span>
kallisto_stringtie_tpm<span class="nv">$TP53</span> <span class="o">=</span> <span class="s2">&quot;Other transcript&quot;</span>
kallisto_stringtie_tpm<span class="nv">$TP53</span><span class="o">[</span>idx.2<span class="o">]</span> <span class="o">=</span> <span class="s2">&quot;TP53&quot;</span>

kallisto_stringtie_tpm<span class="nv">$TP53</span> <span class="o">=</span> factor<span class="o">(</span>kallisto_stringtie_tpm<span class="nv">$TP53</span>, <span class="nv">levels</span> <span class="o">=</span> c<span class="o">(</span><span class="s2">&quot;Other transcript&quot;</span>, <span class="s2">&quot;TP53&quot;</span><span class="o">))</span>
str<span class="o">(</span>kallisto_stringtie_tpm<span class="o">)</span>

<span class="c1">#To remove 0/NA values</span>
kallisto_stringtie_tpm<span class="nv">$value</span>.stringtie<span class="o">[</span>which<span class="o">(</span>is.na<span class="o">(</span>kallisto_stringtie_tpm<span class="nv">$value</span>.stringtie<span class="o">)==</span>TRUE<span class="o">)]</span> <span class="o">=</span> <span class="m">0</span>.001
<span class="c1">#kallisto_stringtie_tpm$value.kallisto[which(is.na(kallisto_stringtie_tpm$value.kallisto)==TRUE)] = 0.001</span>
kallisto_stringtie_tpm<span class="nv">$value</span>.stringtie <span class="o">=</span> kallisto_stringtie_tpm<span class="nv">$value</span>.stringtie + <span class="m">0</span>.1
kallisto_stringtie_tpm<span class="nv">$value</span>.kallisto <span class="o">=</span> kallisto_stringtie_tpm<span class="nv">$value</span>.kallisto + <span class="m">0</span>.1
<span class="c1">########### plot the result ######################</span>

<span class="c1">#pdf(file=&quot;transcript_stringtie_v_kallisto.pdf&quot;, height=8, width=8)</span>
ggplot<span class="o">(</span><span class="nv">data</span><span class="o">=</span>kallisto_stringtie_tpm, aes<span class="o">(</span><span class="nv">x</span><span class="o">=</span>log2<span class="o">(</span>value.kallisto<span class="o">)</span>, <span class="nv">y</span><span class="o">=</span>log2<span class="o">(</span>value.stringtie<span class="o">)</span>, <span class="nv">colour</span><span class="o">=</span>TP53<span class="o">))</span> +
geom_point<span class="o">(</span><span class="nv">alpha</span><span class="o">=</span><span class="m">1</span><span class="o">)</span> +
scale_colour_manual<span class="o">(</span><span class="nv">values</span> <span class="o">=</span> c<span class="o">(</span><span class="s2">&quot;forestgreen&quot;</span>, <span class="s2">&quot;firebrick&quot;</span>, <span class="s2">&quot;dodgerblue&quot;</span><span class="o">)</span>, <span class="nv">name</span> <span class="o">=</span> <span class="s2">&quot;Transcripts&quot;</span>, <span class="nv">drop</span><span class="o">=</span>TRUE<span class="o">)</span> +
geom_point<span class="o">(</span><span class="nv">data</span> <span class="o">=</span> subset<span class="o">(</span>kallisto_stringtie_tpm, <span class="nv">TP53</span> <span class="o">==</span> <span class="s2">&quot;TP53&quot;</span><span class="o">)</span>, aes<span class="o">(</span><span class="nv">x</span><span class="o">=</span>log2<span class="o">(</span>value.kallisto<span class="o">)</span>, <span class="nv">y</span><span class="o">=</span>log2<span class="o">(</span>value.stringtie<span class="o">)</span>, <span class="nv">colour</span><span class="o">=</span>TP53<span class="o">)</span>, inherit.aes<span class="o">=</span>FALSE<span class="o">)</span>+
facet_wrap<span class="o">(</span>~variable<span class="o">)</span>+
theme_bw<span class="o">()</span>

<span class="c1">#To plot density disbributions</span>
<span class="nv">df</span> <span class="o">=</span> rbind<span class="o">(</span>data.frame<span class="o">(</span><span class="nv">variable</span> <span class="o">=</span> kallisto_stringtie_tpm<span class="nv">$variable</span>, <span class="nv">value</span> <span class="o">=</span>  kallisto_stringtie_tpm<span class="nv">$value</span>.kallisto, <span class="nb">type</span> <span class="o">=</span> <span class="s2">&quot;kallisto&quot;</span><span class="o">)</span>, data.frame<span class="o">(</span><span class="nv">variable</span> <span class="o">=</span> kallisto_stringtie_tpm<span class="nv">$variable</span>, <span class="nv">value</span> <span class="o">=</span>  kallisto_stringtie_tpm<span class="nv">$value</span>.stringtie, <span class="nb">type</span> <span class="o">=</span> <span class="s2">&quot;stringtie&quot;</span><span class="o">))</span>
df<span class="nv">$type</span> <span class="o">=</span> factor<span class="o">(</span>df<span class="nv">$type</span>, <span class="nv">levels</span> <span class="o">=</span>c<span class="o">(</span><span class="s2">&quot;kallisto&quot;</span>,<span class="s2">&quot;stringtie&quot;</span><span class="o">))</span>
head<span class="o">(</span>df<span class="o">)</span>

ggplot<span class="o">(</span><span class="nv">data</span><span class="o">=</span>df, aes<span class="o">(</span><span class="nv">x</span><span class="o">=</span>log2<span class="o">(</span>value<span class="o">)))</span> +
geom_density<span class="o">()</span> +
facet_wrap<span class="o">(</span>type~variable, <span class="nv">ncol</span> <span class="o">=</span> <span class="m">4</span><span class="o">)</span>+
ggtitle<span class="o">(</span><span class="s2">&quot;Density distributions&quot;</span><span class="o">)</span>+
xlab<span class="o">(</span><span class="s2">&quot;Log2 expression value&quot;</span><span class="o">)</span>+
theme<span class="o">(</span>legend.position <span class="o">=</span> <span class="s2">&quot;none&quot;</span><span class="o">)</span>
<span class="c1">#dev.off()</span>
</code></pre></div>
<h1 id="gene-fusion-detection">Gene Fusion detection<a class="headerlink" href="#gene-fusion-detection" title="Permanent link">&para;</a></h1>
<h1 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h1>
<p>In addition to providing information about gene expression, RNA-seq data can be used to discover transcripts which result from chromosomal translocations. Translocations and their resultant chimeric (AKA fusion) transcripts are important driver mutations in many cancers. A <a href="https://www.ncbi.nlm.nih.gov/pubmed/27485475">variety of specialized alignment and filtering strategies</a> have been developed to identify fusion transcripts from RNA, but these programs suffer from low specificity (i.e. many false-positives) and poor correlation across methods.</p>
<p>This tutorial uses the <a href="https://pachterlab.github.io/kallisto/about">kallisto</a> and <a href="https://github.com/pmelsted/pizzly">pizzly</a> tools for fusion detection from RNA-seq data. Kallisto quantifies transcript abundance through <a href="http://tinyheero.github.io/2015/09/02/pseudoalignments-kallisto.html">pseudoalignment</a>. Pizzly aligns reads which kallisto has flagged as potentially spanning fusion junctions. Running the tutorial requires RNA fastq files, a reference transcriptome, and a gene annotation file- see below.</p>
<p>Some of the tools used to detection gene fusions from RNAseq are Arriba, EricScript, FusionCatcher, Fusion-Inspector, fusion-report, Pizzly, Squid, Star-Fusion.</p>
<p>All these tools have their own merits and demerits. They are often used in combinations or all together. Hence, we need to assess which one is suitable for our data.</p>
<p>We will use Pizzly in this exercise.</p>
<h1 id="setup">Setup<a class="headerlink" href="#setup" title="Permanent link">&para;</a></h1>
<p><strong>Prerequisites- This module assumes you have completed the initial setup process for the course including:</strong>
- installed Conda package manager, R, pizzly, and kallisto.
    - have fastq sequence files of normal and tumor RNA at ~/workspace/inputs/data/fastq/chr6_and_chr17/. </p>
<p><strong>Additional setup:</strong></p>
<p><strong><em>Important: pizzly will not work with the most recent Ensembl human GTF annotation file. Download the version 87 GTF as shown in the below code block. We will subset the reference transcriptome fasta file. Fasta splitting programs which do not preserve the full Ensembl header such as gffread or gtf_to_fasta will not work with pizzly.</em></strong></p>
<ul>
<li>Download Ensembl GTF and fasta and parse to include only chromosomes 6 and 17 (<strong>15 min</strong>): </li>
</ul>
<p><div class="highlight"><pre><span></span><code><span class="c1"># Get files from source</span>
<span class="c1">#Already run</span>
<span class="c1">#mkdir -p ~/workspace/inputs/reference/fusion</span>
<span class="c1">#cd ~/workspace/inputs/reference/fusion/</span>
<span class="c1">#wget ftp://ftp.ensembl.org/pub/release-94/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz</span>
<span class="c1">#gunzip Homo_sapiens.GRCh38.cdna.all.fa.gz       </span>
<span class="c1">#wget ftp://ftp.ensembl.org/pub/release-87/gtf/homo_sapiens/Homo_sapiens.GRCh38.87.gtf.gz</span>
<span class="c1">#gunzip -k Homo_sapiens.GRCh38.87.gtf.gz</span>

<span class="nb">cd</span> ~/workspace/inputs/reference/fusion/
ls -halt

<span class="c1"># Get annotations for only chromosomes 6 and 17</span>
cat Homo_sapiens.GRCh38.87.gtf <span class="p">|</span> grep --color<span class="o">=</span>never -w <span class="s2">&quot;^6&quot;</span> &gt; chr617.gtf
cat Homo_sapiens.GRCh38.87.gtf <span class="p">|</span> grep --color<span class="o">=</span>never -w <span class="s2">&quot;^17&quot;</span>&gt;&gt; chr617.gtf 
head chr617.gtf
tail chr617.gtf

<span class="c1">#### CODE BELOW ALREDY RUN - TAKES &gt;2h</span>
<span class="c1"># Parse transcriptome fasta, preserving full ensembl headers</span>
<span class="c1">#  Setup directory</span>
<span class="c1">#mkdir -p ~/workspace/inputs/reference/fusion/per-feature</span>
<span class="c1">#cd ~/workspace/inputs/reference/fusion/per-feature</span>
<span class="c1">#  Split fasta at each instance of a sequence header and write to new file</span>
<span class="c1">#csplit -s -z ../Homo_sapiens.GRCh38.cdna.all.fa &#39;/&gt;/&#39; &#39;{*}&#39;</span>
<span class="c1">#  If from chromosomes 6 or 17, rename files using the columns of the original ensemble header</span>
<span class="c1">#####  (This step takes about 120 minutes. You can proceed with the next section in ~/workspace/inputs/data/fastq/chr6_and_chr17)</span>

<span class="c1">#Run with Nohup in case connection is lost</span>
<span class="c1">#for f in xx*; do awk -F &quot;:&quot; &#39;NR==1 &amp;&amp; $3==&quot;6&quot; || NR==1 &amp;&amp; $3==&quot;17&quot;{print $2 &quot;.&quot; $3 &quot;.&quot; $4 &quot;.&quot; $5}&#39; $f | xargs -I{} mv $f {}.fa; done</span>
<span class="c1">#  Concatenate features from chromsomes 6 and 17 to a new reference fasta  </span>
<span class="c1">#cd ~/workspace/inputs/reference/fusion</span>
<span class="c1">#cat ./per-feature/GRCh38.6.*.fa ./per-feature/GRCh38.17.*.fa &gt; chr617.fa</span>
<span class="c1">#rm -rf per-feature</span>
</code></pre></div>
To get one read pair each for normal and tumor, merge the chr6_and_chr17 only RNA-seq fastqs.</p>
<div class="highlight"><pre><span></span><code>mkdir -p ~/workspace/inputs/data/fastq/chr6_and_chr17/fusion
<span class="nb">cd</span> ~/workspace/inputs/data/fastq/chr6_and_chr17/fusion
cat ../../RNAseq_Norm/RNAseq_Norm_Lane1_R1.fastq.gz ../../RNAseq_Norm/RNAseq_Norm_Lane2_R1.fastq.gz &gt; RNAseq_Norm_R1.fastq.gz
cat ../../RNAseq_Norm/RNAseq_Norm_Lane1_R2.fastq.gz ../../RNAseq_Norm/RNAseq_Norm_Lane2_R2.fastq.gz &gt; RNAseq_Norm_R2.fastq.gz
cat ../../RNAseq_Tumor/RNAseq_Tumor_Lane1_R1.fastq.gz ../../RNAseq_Tumor/RNAseq_Tumor_Lane2_R1.fastq.gz &gt; RNAseq_Tumor_R1.fastq.gz
cat ../../RNAseq_Tumor/RNAseq_Tumor_Lane1_R2.fastq.gz ../../RNAseq_Tumor/RNAseq_Tumor_Lane2_R2.fastq.gz &gt; RNAseq_Tumor_R2.fastq.gz
</code></pre></div>
<ul>
<li>Subsample fastqs to allow fusion alignment to run quickly (<strong>5 min</strong>):
<div class="highlight"><pre><span></span><code><span class="c1"># Use seqtk to take subsamples of the 30% of the fastq read pairs  </span>
seqtk sample -s100 RNAseq_Norm_R1.fastq.gz <span class="m">0</span>.3 &gt; subRNAseq_Norm_R1.fastq.gz
seqtk sample -s100 RNAseq_Norm_R2.fastq.gz <span class="m">0</span>.3 &gt; subRNAseq_Norm_R2.fastq.gz
seqtk sample -s100 RNAseq_Tumor_R1.fastq.gz <span class="m">0</span>.3 &gt; subRNAseq_Tumor_R1.fastq.gz
seqtk sample -s100 RNAseq_Tumor_R2.fastq.gz <span class="m">0</span>.3 &gt; subRNAseq_Tumor_R2.fastq.gz
</code></pre></div></li>
</ul>
<h2 id="run-fusion-alignment">Run Fusion Alignment<a class="headerlink" href="#run-fusion-alignment" title="Permanent link">&para;</a></h2>
<ul>
<li>Create kallisto index:</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1">#mkdir -p ~/workspace/rnaseq/fusion</span>
<span class="nb">cd</span> ~/workspace/rnaseq/fusion
kallisto index -i index.617.idx -k <span class="m">31</span> --make-unique ~/workspace/inputs/reference/fusion/chr617.fa
</code></pre></div>
<h1 id="call-fusions">Call fusions<a class="headerlink" href="#call-fusions" title="Permanent link">&para;</a></h1>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/rnaseq/fusion
<span class="c1">#Quantify potential fusions (**15 min**):</span>
nohup kallisto quant -i index.617.idx --fusion -o kquant-norm617 ~/workspace/inputs/data/fastq/chr6_and_chr17/fusion/subRNAseq_Norm_R1.fastq.gz ~/workspace/inputs/data/fastq/chr6_and_chr17/fusion/subRNAseq_Norm_R2.fastq.gz <span class="m">2</span>&gt; kallisto_quant_N_bam.out <span class="p">&amp;</span>

nohup kallisto quant -i index.617.idx --fusion -o kquant-tumor617 ~/workspace/inputs/data/fastq/chr6_and_chr17/fusion/subRNAseq_Tumor_R1.fastq.gz ~/workspace/inputs/data/fastq/chr6_and_chr17/fusion/subRNAseq_Tumor_R2.fastq.gz <span class="m">2</span>&gt; kallisto_quant_T_bam.out <span class="p">&amp;</span>

<span class="c1">#Call fusions with pizzly: **120 mins**</span>
<span class="nb">cd</span> ~/workspace/rnaseq/fusion
nohup pizzly -k <span class="m">31</span> --gtf ~/workspace/inputs/reference/fusion/chr617.gtf --cache index-norm617.cache.txt --align-score <span class="m">2</span> --insert-size <span class="m">400</span> --fasta ~/workspace/inputs/reference/fusion/chr617.fa --output norm-fuse617 kquant-norm617/fusion.txt <span class="m">2</span>&gt; pizzly_N_bam.out <span class="p">&amp;</span>

nohup pizzly -k <span class="m">31</span> --gtf ~/workspace/inputs/reference/fusion/chr617.gtf --cache index-tumor617.cache.txt --align-score <span class="m">2</span> --insert-size <span class="m">400</span> --fasta ~/workspace/inputs/reference/fusion/chr617.fa --output tumor-fuse617 kquant-tumor617/fusion.txt <span class="m">2</span>&gt; pizzly_T_bam.out <span class="p">&amp;</span>
</code></pre></div>
<ul>
<li>If using 30% of reads with the above process, expect about 13,000 retained transcripts from normal and about 3,000 retained transcripts from tumor. See next section to investigate the output of the pizzly fusion calling.</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1">#To go the fusion directory</span>
<span class="nb">cd</span> ~/workspace/rnaseq/fusion

<span class="c1"># ALREADY RUN: Get R scripts for later use</span>
<span class="c1">#wget https://raw.githubusercontent.com/griffithlab/pmbio.org/master/assets/course_scripts/mod.grolar.R</span>
<span class="c1">#wget https://raw.githubusercontent.com/griffithlab/pmbio.org/master/assets/course_scripts/import_Pizzly.R</span>
scp -ri ~/course_EC2_01.pem ubuntu@ec2-34-203-226-9.compute-1.amazonaws.com:~/workspace/rnaseq/fusion/*.json .
scp -ri ~/course_EC2_01.pem ubuntu@ec2-34-203-226-9.compute-1.amazonaws.com:~/workspace/rnaseq/fusion/mod20220117.grolar.R .

<span class="c1">#list files</span>
ls -halt

<span class="c1"># Open R on AWS and set working directory </span>
R
setwd<span class="o">(</span><span class="s2">&quot;~/workspace/rnaseq/fusion/&quot;</span><span class="o">)</span>
<span class="c1">#Check the path</span>
getwd<span class="o">()</span>

<span class="c1"># Load required packages</span>
library<span class="o">(</span>cowplot<span class="o">)</span>
library<span class="o">(</span>jsonlite<span class="o">)</span>
library<span class="o">(</span>dplyr<span class="o">)</span>
library<span class="o">(</span>EnsDb.Hsapiens.v86<span class="o">)</span>
library<span class="o">(</span>ggplot2<span class="o">)</span>
library<span class="o">(</span>chimeraviz<span class="o">)</span>

<span class="c1"># Change to your own output location</span>
setwd<span class="o">(</span><span class="s2">&quot;~&quot;</span><span class="o">)</span>

<span class="c1"># Load data</span>
<span class="nv">suffix</span> <span class="o">=</span> <span class="s2">&quot;617.json&quot;</span>
print<span class="o">(</span>suffix<span class="o">)</span>
<span class="nv">JSON_files</span> <span class="o">=</span> list.files<span class="o">(</span><span class="nv">path</span> <span class="o">=</span> <span class="s2">&quot;~&quot;</span>, <span class="nv">pattern</span> <span class="o">=</span> paste0<span class="o">(</span><span class="s2">&quot;*&quot;</span>,suffix<span class="o">))</span>
print<span class="o">(</span>JSON_files<span class="o">)</span>
<span class="nv">Ids</span> <span class="o">=</span> gsub<span class="o">(</span>suffix, <span class="s2">&quot;&quot;</span>, JSON_files<span class="o">)</span>
print<span class="o">(</span>Ids<span class="o">)</span>

<span class="c1"># Assuming you have used GRCh38 gene models</span>
edb &lt;- EnsDb.Hsapiens.v86
listColumns<span class="o">(</span>edb<span class="o">)</span>
supportedFilters<span class="o">(</span>edb<span class="o">)</span>

<span class="c1"># Load the functoin https://github.com/MattBashton/grolar/blob/master/grolar.R</span>
source<span class="o">(</span><span class="s2">&quot;mod20220117.grolar.R&quot;</span><span class="o">)</span>

<span class="c1"># Suffix which apears after sample id in output file name</span>
<span class="c1"># Use above funciton on all output files</span>
lapply<span class="o">(</span>Ids, <span class="k">function</span><span class="o">(</span>x<span class="o">)</span> GetFusionz_and_namez<span class="o">(</span>x, <span class="nv">suffix</span> <span class="o">=</span> <span class="s2">&quot;617.json&quot;</span><span class="o">)</span> <span class="o">)</span>

<span class="c1"># This function will flatten out the JSON giving you a list of gene A and gene B sorted by splitcount then paircount</span>
<span class="c1"># Each fusion now has a unique identifier, sequence positions, and distance values for genes from the same chromosome:</span>

<span class="c1"># Read the flattened files</span>
<span class="nv">normal</span><span class="o">=</span>read.table<span class="o">(</span><span class="s2">&quot;~/norm-fuse_fusions_filt_sorted.txt&quot;</span>, <span class="nv">header</span><span class="o">=</span>T<span class="o">)</span>
<span class="nv">tumor</span><span class="o">=</span>read.table<span class="o">(</span><span class="s2">&quot;~/tumor-fuse_fusions_filt_sorted.txt&quot;</span>, <span class="nv">header</span><span class="o">=</span>T<span class="o">)</span>

<span class="c1">#Investigate the first five rows</span>
head<span class="o">(</span>normal, <span class="m">5</span><span class="o">)</span>
head<span class="o">(</span>tumor, <span class="m">5</span><span class="o">)</span>

<span class="c1"># Common filtering tasks for fusion output include removing fusions from the tumor sample which are present in the normal, and removing fusions for which the is little support by pair and split read counts:</span>

normal<span class="nv">$genepr</span><span class="o">=</span>paste0<span class="o">(</span>normal<span class="nv">$geneA</span>.name,<span class="s2">&quot;.&quot;</span>,normal<span class="nv">$geneB</span>.name<span class="o">)</span>
tumor<span class="nv">$genepr</span><span class="o">=</span>paste0<span class="o">(</span>tumor<span class="nv">$geneA</span>.name,<span class="s2">&quot;.&quot;</span>,tumor<span class="nv">$geneB</span>.name<span class="o">)</span>
<span class="nv">uniqueTumor</span><span class="o">=</span>subset<span class="o">(</span>tumor, !<span class="o">(</span>tumor<span class="nv">$genepr</span> %in% normal<span class="nv">$genepr</span><span class="o">))</span>
nrow<span class="o">(</span>uniqueTumor<span class="o">)==</span>nrow<span class="o">(</span>tumor<span class="o">)</span>
<span class="o">[</span><span class="m">1</span><span class="o">]</span> FALSE
nrow<span class="o">(</span>tumor<span class="o">)</span>-nrow<span class="o">(</span>uniqueTumor<span class="o">)</span>
<span class="o">[</span><span class="m">1</span><span class="o">]</span> <span class="m">2</span>
<span class="c1"># There are two fusions (or at least fusion gene pairs) from the normal sample which are also present in the tumor. </span>
<span class="c1"># Examine the output of- </span>
<span class="nv">shared_src_tumor</span><span class="o">=</span>subset<span class="o">(</span>tumor, <span class="o">(</span>tumor<span class="nv">$genepr</span> %in% normal<span class="nv">$genepr</span><span class="o">))</span>
<span class="nv">shared_src_normal</span><span class="o">=</span>subset<span class="o">(</span>normal, <span class="o">(</span>normal<span class="nv">$genepr</span> %in% tumor<span class="nv">$genepr</span><span class="o">))</span>
shared_src_tumor
shared_src_normal

<span class="c1"># Filtering by counts:</span>

<span class="c1"># Merge normal and tumor data</span>
normal<span class="nv">$sample</span><span class="o">=</span><span class="s2">&quot;normal&quot;</span>
tumor<span class="nv">$sample</span><span class="o">=</span><span class="s2">&quot;tumor&quot;</span>
<span class="nv">allfusions</span><span class="o">=</span>rbind<span class="o">(</span>normal,tumor<span class="o">)</span>
<span class="c1"># Compare counts of paired and split reads</span>
tapply<span class="o">(</span>allfusions<span class="nv">$paircount</span>, allfusions<span class="nv">$sample</span>, summary<span class="o">)</span>
tapply<span class="o">(</span>allfusions<span class="nv">$splitcount</span>, allfusions<span class="nv">$sample</span>, summary<span class="o">)</span>

<span class="c1"># As a density plot</span>
<span class="nv">p1</span><span class="o">=</span>ggplot<span class="o">(</span>allfusions, aes<span class="o">(</span>paircount, <span class="nv">fill</span><span class="o">=</span>sample<span class="o">))</span>+
geom_density<span class="o">(</span><span class="nv">alpha</span><span class="o">=</span>.4<span class="o">)</span>+
geom_vline<span class="o">(</span><span class="nv">xintercept</span><span class="o">=</span><span class="m">2</span><span class="o">)</span>+
coord_fixed<span class="o">(</span><span class="nv">ratio</span><span class="o">=</span><span class="m">15</span><span class="o">)</span>

<span class="nv">p2</span><span class="o">=</span>ggplot<span class="o">(</span>allfusions, aes<span class="o">(</span>splitcount, <span class="nv">fill</span><span class="o">=</span>sample<span class="o">))</span>+
geom_density<span class="o">(</span><span class="nv">alpha</span><span class="o">=</span>.4<span class="o">)</span>+coord_cartesian<span class="o">(</span><span class="nv">ylim</span><span class="o">=</span> c<span class="o">(</span><span class="m">0</span>,.2<span class="o">))</span>+
geom_vline<span class="o">(</span><span class="nv">xintercept</span><span class="o">=</span><span class="m">5</span><span class="o">)</span>+
coord_fixed<span class="o">(</span><span class="nv">ratio</span><span class="o">=</span><span class="m">200</span><span class="o">)</span>

plot_grid<span class="o">(</span>p1,p2, <span class="nv">ncol</span> <span class="o">=</span> <span class="m">2</span>, <span class="nv">rel_heights</span> <span class="o">=</span> c<span class="o">(</span><span class="m">1</span>,1<span class="o">))</span>

nrow<span class="o">(</span>allfusions<span class="o">)</span>
<span class="nv">allfusions</span><span class="o">=</span>allfusions<span class="o">[</span>which<span class="o">(</span>allfusions<span class="nv">$paircount</span> &gt;<span class="o">=</span> <span class="m">2</span> <span class="p">&amp;</span> allfusions<span class="nv">$splitcount</span> &gt;<span class="o">=</span> <span class="m">5</span><span class="o">)</span>,<span class="o">]</span>
nrow<span class="o">(</span>allfusions<span class="o">)</span>
<span class="c1">#write.table(allfusions, &quot;allfusions.txt&quot;)</span>
write.table<span class="o">(</span>allfusions, <span class="nv">file</span><span class="o">=</span><span class="s2">&quot;allfusions.txt&quot;</span>,row.names<span class="o">=</span>F, col.names<span class="o">=</span>T, <span class="nv">quote</span><span class="o">=</span>F, <span class="nv">sep</span><span class="o">=</span><span class="s2">&quot;\t&quot;</span><span class="o">)</span>

<span class="c1">#Chimeraviz is an R package for visualizing fusions from RNA-seq data. The chimeraviz package has import functions built in for a variety of fusion-finder programs, but not for pizzly. We will have to load our own import function that you downloaded above:</span>

<span class="c1"># Enter R, install and load chimeraviz </span>
R
source<span class="o">(</span><span class="s2">&quot;https://bioconductor.org/biocLite.R&quot;</span><span class="o">)</span>
biocLite<span class="o">(</span><span class="s2">&quot;chimeraviz&quot;</span><span class="o">)</span>
<span class="c1"># (if asked to update old packages, you can ignore- Update all/some/none? [a/s/n]:)</span>
library<span class="o">(</span>chimeraviz<span class="o">)</span>

<span class="c1"># Use the pizzly importer script to import fusion data</span>
source<span class="o">(</span><span class="s2">&quot;./import_Pizzly.R&quot;</span><span class="o">)</span>
<span class="c1">#  You can view the function by calling it without variables</span>
<span class="c1">#importPizzly</span>
<span class="nv">fusions</span> <span class="o">=</span> importPizzly<span class="o">(</span><span class="s2">&quot;./allfusions.txt&quot;</span>,<span class="s2">&quot;hg38&quot;</span><span class="o">)</span>
plot_circle<span class="o">(</span>fusions<span class="o">)</span>
</code></pre></div>

              
            </article>
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" data-md-state="hidden">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../analysis_of_cnv/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Copy Number Variations" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Copy Number Variations
            </div>
          </div>
        </a>
      
      
        
        <a href="../joint_curation/" class="md-footer__link md-footer__link--next" aria-label="Next: Joint Curation" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Joint Curation
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2021, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "navigation.sections", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest"], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.361d90f1.min.js"}</script>
    
    
      <script src="../../assets/javascripts/bundle.289a2a4b.min.js"></script>
      
    
  </body>
</html>