
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.2.3, mkdocs-material-8.1.4">
    
    
      
        <title>Variant Calling - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bb3983ee.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.e6a45f82.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#calling-germline-and-somatic-mutations" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Variant Calling
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../aws/aws_ec2_instances/" class="md-nav__link">
        AWS Instances Login
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Lab Session" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_1" type="checkbox" id="__nav_5_1" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_1">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 1" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_2" type="checkbox" id="__nav_5_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_2">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 2" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../human_genome/" class="md-nav__link">
        Human Genome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_3" type="checkbox" id="__nav_5_3" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5_3">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 3" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Variant Calling
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Variant Calling
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#germline-variants" class="md-nav__link">
    Germline variants
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#explore-the-ouput-files" class="md-nav__link">
    Explore the ouput files
  </a>
  
    <nav class="md-nav" aria-label="Explore the ouput files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#realigned-bam-file" class="md-nav__link">
    Realigned bam file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vcf-variant-call-format-file" class="md-nav__link">
    VCF (variant call format) file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-germline-variants" class="md-nav__link">
    Filtering germline variants
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#separate-snps-from-indels-in-the-variant-call-set" class="md-nav__link">
    Separate SNPs from Indels in the variant call set
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#apply-filters-to-the-snp-and-indel-call-sets" class="md-nav__link">
    Apply filters to the SNP and Indel call sets
  </a>
  
    <nav class="md-nav" aria-label="Apply filters to the SNP and Indel call sets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merge-filtered-snp-and-indel-vcfs-back-together" class="md-nav__link">
    Merge filtered SNP and INDEL vcfs back together
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-pass-variants-only" class="md-nav__link">
    Extract PASS variants only
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-annotation-of-filtered-variants" class="md-nav__link">
    Perform annotation of filtered variants
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#somatic-variants" class="md-nav__link">
    Somatic variants
  </a>
  
    <nav class="md-nav" aria-label="Somatic variants">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#varscan" class="md-nav__link">
    Varscan
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#strelka" class="md-nav__link">
    Strelka
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mutect2" class="md-nav__link">
    Mutect2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#exome-data-commands" class="md-nav__link">
    Exome data commands
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#now-we-are-going-to-merge-the-variants-detected-from-all-three-variant-callers" class="md-nav__link">
    Now we are going to merge the variants detected from all three variant callers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#left-align-and-trim" class="md-nav__link">
    Left Align and Trim
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#we-will-split-multi-allelic-variants-into-multiple-records" class="md-nav__link">
    We will split multi-allelic variants into multiple records
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#basic-filtering-on-somatic-variants" class="md-nav__link">
    Basic Filtering on Somatic Variants
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#annotation-with-vep" class="md-nav__link">
    Annotation with VEP.
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#inspecting-variants-in-igv" class="md-nav__link">
    Inspecting variants in IGV
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_4" type="checkbox" id="__nav_5_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_4">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 4" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../structural_variation/" class="md-nav__link">
        Structural Variants
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../analysis_of_cnv/" class="md-nav__link">
        Copy Number Variations
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_5" type="checkbox" id="__nav_5_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_5">
          Day 5
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 5" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          Day 5
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../joint_curation/" class="md-nav__link">
        Joint Curation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../genomic_alteration_interpretation_exe/" class="md-nav__link">
        Genomic Alterations Interpretation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#germline-variants" class="md-nav__link">
    Germline variants
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#explore-the-ouput-files" class="md-nav__link">
    Explore the ouput files
  </a>
  
    <nav class="md-nav" aria-label="Explore the ouput files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#realigned-bam-file" class="md-nav__link">
    Realigned bam file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vcf-variant-call-format-file" class="md-nav__link">
    VCF (variant call format) file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-germline-variants" class="md-nav__link">
    Filtering germline variants
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#separate-snps-from-indels-in-the-variant-call-set" class="md-nav__link">
    Separate SNPs from Indels in the variant call set
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#apply-filters-to-the-snp-and-indel-call-sets" class="md-nav__link">
    Apply filters to the SNP and Indel call sets
  </a>
  
    <nav class="md-nav" aria-label="Apply filters to the SNP and Indel call sets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#merge-filtered-snp-and-indel-vcfs-back-together" class="md-nav__link">
    Merge filtered SNP and INDEL vcfs back together
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-pass-variants-only" class="md-nav__link">
    Extract PASS variants only
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-annotation-of-filtered-variants" class="md-nav__link">
    Perform annotation of filtered variants
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#somatic-variants" class="md-nav__link">
    Somatic variants
  </a>
  
    <nav class="md-nav" aria-label="Somatic variants">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#varscan" class="md-nav__link">
    Varscan
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#strelka" class="md-nav__link">
    Strelka
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mutect2" class="md-nav__link">
    Mutect2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#exome-data-commands" class="md-nav__link">
    Exome data commands
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#now-we-are-going-to-merge-the-variants-detected-from-all-three-variant-callers" class="md-nav__link">
    Now we are going to merge the variants detected from all three variant callers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#left-align-and-trim" class="md-nav__link">
    Left Align and Trim
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#we-will-split-multi-allelic-variants-into-multiple-records" class="md-nav__link">
    We will split multi-allelic variants into multiple records
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#basic-filtering-on-somatic-variants" class="md-nav__link">
    Basic Filtering on Somatic Variants
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#annotation-with-vep" class="md-nav__link">
    Annotation with VEP.
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#inspecting-variants-in-igv" class="md-nav__link">
    Inspecting variants in IGV
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                

<h1 id="calling-germline-and-somatic-mutations">Calling germline- and somatic mutations<a class="headerlink" href="#calling-germline-and-somatic-mutations" title="Permanent link">&para;</a></h1>
<h2 id="germline-variants">Germline variants<a class="headerlink" href="#germline-variants" title="Permanent link">&para;</a></h2>
<p>We will start by calling germline variants. For this purpose we will use the <a href="https://gatk.broadinstitute.org/hc/en-us/articles/4414586765723-HaplotypeCaller">GATK HaplotypeCaller</a> from the Broad Institute. We will use this tool on the germline DNA bam files that were processed yesterday. </p>
<p>Active forum and tutorials for the GATK software suit is available <a href="https://gatk.broadinstitute.org/hc/en-us/community/topics">here</a>.</p>
<p>The HaplotypeCaller can be run in single- or joint sample mode. The joint sample mode gives improved accuracy, especially in samples with low coverage. For simplicity, we will only use the single sample mode in this course. </p>
<p>When haplotype caller detects variation it performes local realignment of the reads which improves variant accuracy, especially in e.g. repetitive regions. </p>
<div class="highlight"><pre><span></span><code><span class="c1">#Make sure that $GATK_REGIONS is set correctly</span>
<span class="nb">echo</span> <span class="nv">$GATK_REGIONS</span>

<span class="c1">#Create working dir for germline results if not already created</span>
mkdir -p ~/workspace/germline
<span class="nb">cd</span> ~/workspace/germline

<span class="c1">#The argument --java-options &#39;-Xmx12g&#39; tells GATK to use 12GB of memory</span>
<span class="c1">#To list the haplotype caller arguments run </span>
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> HaplotypeCaller --help

<span class="c1">#Call variants for exome data</span>
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> HaplotypeCaller <span class="se">\</span>
    -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    -I ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam <span class="se">\</span>
    -O ~/workspace/germline/Exome_Norm_HC_calls.vcf <span class="se">\</span>
    --bam-output ~/workspace/germline/Exome_Norm_HC_out.bam <span class="nv">$GATK_REGIONS</span>
</code></pre></div>
<h2 id="explore-the-ouput-files">Explore the ouput files<a class="headerlink" href="#explore-the-ouput-files" title="Permanent link">&para;</a></h2>
<h3 id="realigned-bam-file">Realigned bam file<a class="headerlink" href="#realigned-bam-file" title="Permanent link">&para;</a></h3>
<p>Navigate to the folder where you want to place the BAMs
e.g. ~/course_bams</p>
<div class="highlight"><pre><span></span><code><span class="c1">#Copy the haplotype caller realigned bam-file to your local computer </span>
scp -ri ~/PEM_KEY_ID.pem ubuntu@AWS_ADDRESS_HERE:~/workspace/germline/Exome_Norm_HC_out.ba* .
</code></pre></div>
<p>Launch IGV (remember <strong>hg38</strong>)
- Open the bam files
    - Exome_Norm_sorted_mrkdup_bqsr.bam (intput file to haplotype caller).
    - Exome_Norm_HC_out.bam (realigned bam file by haplotype caller).
- Navigate to chr17:76286974-76287203.
    - Color alignments by: no color.
    - Enable ruler in the menue bar.
    - In the first base of the indel: Sort aligments according to base.
    - Select squished view.
    - Zoom out.
- Try to answer the following:
    - What characterises misaligned variant-supporting reads in non-realigned bam file?
    - Are there any false positive variants take in the non-realigned bam file?
- <strong><em>OBS</em></strong>: Leave IGV open, we will use it in the below section.</p>
<p><img alt="" src="https://i.imgur.com/yQS7rie.png" /></p>
<h3 id="vcf-variant-call-format-file">VCF (variant call format) file<a class="headerlink" href="#vcf-variant-call-format-file" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1">#Make sure you hare in the same directory as the output files</span>
<span class="nb">cd</span> ~/workspace/germline

<span class="c1">#List files</span>
ls -halt

<span class="c1">#Open the VCF file</span>
less -SN Exome_Norm_HC_calls.vcf
</code></pre></div>
<p>The VCF file format is quite complex, the specification can be found <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf">here</a>.</p>
<p>Try to do the following:
- Identify the VCF header section.
- Identify the data column header.
- Find the variant on chr17 pos 106742 in the VCF file.
    - Lead, type /106742 and press ENTER
    - What is the read depth in this position?
    - Is it a high quality variant?
    - Go go IGV in this position
        - Select expanded view.
        - Color alignments by read strand.
        - Sort alignments by base.
        - Does the visual impression in IGV support the quality of the variant?
- Identify a way to quickly separate SNPs and INDELs in the VCF.
- Find the variant on chr17 pos 1787008 in the VCF file
    - Is this a high quality variant?
    - Navgiate in IGV to the position.
    - Try to explain the information in the VCF to what is shown in IGV, is it correct?</p>
<h3 id="filtering-germline-variants">Filtering germline variants<a class="headerlink" href="#filtering-germline-variants" title="Permanent link">&para;</a></h3>
<p>As noted above, the raw output of GATK HaplotypeCaller will include many variants with varying degrees of quality. For various reasons we might wish to further filter these to a higher confidence set of variants. The hard-filtering approached applied here is futher described in detail at the <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890471-Hard-filtering-germline-short-variants">GATK webside</a>. </p>
<h3 id="separate-snps-from-indels-in-the-variant-call-set">Separate SNPs from Indels in the variant call set<a class="headerlink" href="#separate-snps-from-indels-in-the-variant-call-set" title="Permanent link">&para;</a></h3>
<p>First, we will separate out the SNPs and Indels from the VCF into new separate VCFs. Note that the variant type (SNP, INDEL, MIXED, etc) is not stored explicitly in the vcf but instead inferred from the genotypes. We will use a versatile GATK tool called <code>SelectVariants</code>. This command can be used for all kinds of simple filtering or subsetting purposes. We will run it twice to select by variant type, once for SNPs, and then again for Indels, to produce two new VCFs.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/germline/
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> SelectVariants <span class="se">\</span>
    -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    -V ~/workspace/germline/Exome_Norm_HC_calls.vcf -select-type SNP <span class="se">\</span>
    -O ~/workspace/germline/Exome_Norm_HC_calls.snps.vcf
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> SelectVariants <span class="se">\</span>
    -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    -V ~/workspace/germline/Exome_Norm_HC_calls.vcf -select-type INDEL <span class="se">\</span>
    -O ~/workspace/germline/Exome_Norm_HC_calls.indels.vcf
</code></pre></div>
<h3 id="apply-filters-to-the-snp-and-indel-call-sets">Apply filters to the SNP and Indel call sets<a class="headerlink" href="#apply-filters-to-the-snp-and-indel-call-sets" title="Permanent link">&para;</a></h3>
<p>Next, we will perform so-called hard-filtering by applying a number of cutoffs. Multiple filters can be combined arbitrarily. Each can be given its own name so that you can later determine which one or more filters a variant fails. Visit the <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890471-Hard-filtering-germline-short-variants">GATK documentation on hard-filtering</a> to learn more about the following hard filtering options. Notice that different filters and cutoffs are recommended for SNVs and Indels. This is why we first split them into separate files.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/germline/
gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> VariantFiltration <span class="se">\</span>
    -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    -V ~/workspace/germline/Exome_Norm_HC_calls.snps.vcf <span class="se">\</span>
    --filter-expression <span class="s2">&quot;QD &lt; 2.0&quot;</span> --filter-name <span class="s2">&quot;QD_lt_2&quot;</span> <span class="se">\</span>
    --filter-expression <span class="s2">&quot;FS &gt; 60.0&quot;</span> --filter-name <span class="s2">&quot;FS_gt_60&quot;</span> <span class="se">\</span>
    --filter-expression <span class="s2">&quot;MQ &lt; 40.0&quot;</span> --filter-name <span class="s2">&quot;MQ_lt_40&quot;</span> <span class="se">\</span>
    --filter-expression <span class="s2">&quot;MQRankSum &lt; -12.5&quot;</span> --filter-name <span class="s2">&quot;MQRS_lt_n12.5&quot;</span> <span class="se">\</span>
    --filter-expression <span class="s2">&quot;ReadPosRankSum &lt; -8.0&quot;</span> --filter-name <span class="s2">&quot;RPRS_lt_n8&quot;</span> <span class="se">\</span>
    --filter-expression <span class="s2">&quot;SOR &gt; 3.0&quot;</span> --filter-name <span class="s2">&quot;SOR_gt_3&quot;</span> <span class="se">\</span>
    -O ~/workspace/germline/Exome_Norm_HC_calls.snps.filtered.vcf

gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> VariantFiltration <span class="se">\</span>
    -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    -V ~/workspace/germline/Exome_Norm_HC_calls.indels.vcf <span class="se">\</span>
    --filter-expression <span class="s2">&quot;QD &lt; 2.0&quot;</span> --filter-name <span class="s2">&quot;QD_lt_2&quot;</span> <span class="se">\</span>
    --filter-expression <span class="s2">&quot;FS &gt; 200.0&quot;</span> --filter-name <span class="s2">&quot;FS_gt_200&quot;</span> <span class="se">\</span>
    --filter-expression <span class="s2">&quot;ReadPosRankSum &lt; -20.0&quot;</span> --filter-name <span class="s2">&quot;RPRS_lt_n20&quot;</span> <span class="se">\</span>
    --filter-expression <span class="s2">&quot;SOR &gt; 10.0&quot;</span> --filter-name <span class="s2">&quot;SOR_gt_10&quot;</span> <span class="se">\</span>
    -O ~/workspace/germline/Exome_Norm_HC_calls.indels.filtered.vcf
</code></pre></div>
<p>Notice that warnings are given with regards to MQRankSum and ReadPosRankSum, these are only calculated if the site is called as heterozygous. E.g. for this site (inspect in IGV and in the VCF file):
- chr17:1067361</p>
<p>This was discussed at the <a href="https://sites.google.com/a/broadinstitute.org/legacy-gatk-forum-discussions/2017-01-18-2016-08-11/8323-Why-do-MQRankSum-and-ReadPosRankSum-not-appear-in-some-vcf-file-entries">GATK forum</a>.</p>
<p>The variants are now marked as PASS or or <em>filter-name</em>. Use <code>grep -v</code> and <code>head</code> to skip past all the VCF header lines and view the first few records. </p>
<p><div class="highlight"><pre><span></span><code><span class="c1">#inspect the 10 first variants that did not pass</span>
grep -v <span class="s2">&quot;##&quot;</span> Exome_Norm_HC_calls.snps.filtered.vcf <span class="p">|</span> grep -vP <span class="s2">&quot;\tPASS\t&quot;</span> <span class="p">|</span> head -10
<span class="c1">#inspect the 10 first variants that did pass</span>
grep -v <span class="s2">&quot;##&quot;</span> Exome_Norm_HC_calls.snps.filtered.vcf <span class="p">|</span> grep -P <span class="s2">&quot;\tPASS\t&quot;</span> <span class="p">|</span> head -10
</code></pre></div>
Try to do the following:
- Count the number of variants that failed and passed filtering.</p>
<h4 id="merge-filtered-snp-and-indel-vcfs-back-together">Merge filtered SNP and INDEL vcfs back together<a class="headerlink" href="#merge-filtered-snp-and-indel-vcfs-back-together" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> MergeVcfs <span class="se">\</span>
    -I ~/workspace/germline/Exome_Norm_HC_calls.snps.filtered.vcf <span class="se">\</span>
    -I ~/workspace/germline/Exome_Norm_HC_calls.indels.filtered.vcf <span class="se">\</span>
    -O ~/workspace/germline/Exome_Norm_HC_calls.filtered.vcf

<span class="c1">#Nbr of variants in the SNP file</span>
grep -v <span class="s2">&quot;##&quot;</span> Exome_Norm_HC_calls.snps.filtered.vcf <span class="p">|</span> wc -l

<span class="c1">#Nbr of variants in the INDEL file</span>
grep -v <span class="s2">&quot;##&quot;</span> Exome_Norm_HC_calls.indels.filtered.vcf <span class="p">|</span> wc -l

<span class="c1">#Nbr of variants in the merged file</span>
grep -v <span class="s2">&quot;##&quot;</span> Exome_Norm_HC_calls.filtered.vcf <span class="p">|</span> wc -l
</code></pre></div>
<h4 id="extract-pass-variants-only">Extract PASS variants only<a class="headerlink" href="#extract-pass-variants-only" title="Permanent link">&para;</a></h4>
<p>It would also be convenient to have a vcf with only passing variants. For this, we can go back the <code>GATK SelectVariants</code> tool. This will be run much as above except with the <code>--exlude-filtered</code> option instead of <code>-select-type</code>.</p>
<div class="highlight"><pre><span></span><code>gatk --java-options <span class="s1">&#39;-Xmx12g&#39;</span> SelectVariants <span class="se">\</span>
    -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    -V ~/workspace/germline/Exome_Norm_HC_calls.filtered.vcf <span class="se">\</span>
    -O ~/workspace/germline/Exome_Norm_HC_calls.filtered.PASS.vcf <span class="se">\</span>
    --exclude-filtered
</code></pre></div>
<h3 id="perform-annotation-of-filtered-variants">Perform annotation of filtered variants<a class="headerlink" href="#perform-annotation-of-filtered-variants" title="Permanent link">&para;</a></h3>
<p>Now that we have high-confidence, filtered variants, we want to start understanding which of these variants might be clinically or biologically relevant. Ensembl's Variant Effect Predictor (VEP) annotation software is a powerful tool for annotating variants with a great deal of biological features. This includes such information as protein consequence (non-coding or coding), population frequencies, links to external databases, various scores designed to estimate the importance of individual variants on protein function, and much more.</p>
<p>Note, the first time you run VEP it will create a fasta index for the reference genome in VEP cache. Therefore, it will take longer to run that first time but should speed up substantially for subsequent runs on files with similar numbers of variants.</p>
<div class="highlight"><pre><span></span><code><span class="c1">#VEP annotate hard-filtered exome results</span>

<span class="c1">####### Already done code below</span>
<span class="c1">#cd ~/workspace/vep_cache/</span>
<span class="c1">#wget https://github.com/Ensembl/VEP_plugins/archive/refs/heads/release/104.zip</span>
<span class="c1">#unzip 104.zip </span>
<span class="c1">#ls -halt ~/workspace/vep_cache/</span>
<span class="c1">#Already done</span>
<span class="c1">#mv VEP_plugins-release-104 Plugins</span>
<span class="c1">#ls -halt ~/workspace/vep_cache/Plugins</span>

<span class="c1">#get annotation fasta for homo_sapiens </span>
<span class="c1">#cd ~/workspace/vep_cache/homo_sapiens</span>
<span class="c1">#Already done</span>
<span class="c1">#wget ftp://ftp.ensembl.org/pub/release-104/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz</span>

<span class="c1">####### Start here</span>
<span class="nb">cd</span> ~/workspace/germline
<span class="c1">#Remember, we are using the REF ~/workspace/inputs/references/genome/ref_genome.fa which only contains chr6 and chr17.</span>
nohup vep --cache --dir_cache ~/workspace/vep_cache <span class="se">\</span>
    --dir_plugins ~/workspace/vep_cache/Plugins <span class="se">\</span>
    --fasta ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    --fork <span class="m">8</span> --assembly<span class="o">=</span>GRCh38 --offline --vcf <span class="se">\</span>
    -i ~/workspace/germline/Exome_Norm_HC_calls.filtered.PASS.vcf <span class="se">\</span>
    -o ~/workspace/germline/Exome_Norm_HC_calls.filtered.PASS.vep.vcf <span class="se">\</span>
    --check_existing --total_length --allele_number --no_escape --everything <span class="se">\</span>
    --use_given_ref --force_overwrite --coding_only <span class="p">&amp;</span>

<span class="c1">#Open the VEP vcf, row 39 and 40, contains information about the VEP annotation.</span>
less -SN Exome_Norm_HC_calls.filtered.PASS.vep.vcf
<span class="c1">#Or just grep the VEP info.</span>
grep <span class="s2">&quot;##&quot;</span> Exome_Norm_HC_calls.filtered.PASS.vep.vcf <span class="p">|</span> grep <span class="s2">&quot;VEP&quot;</span> <span class="p">|</span> less -SN
<span class="c1">#Copy the html VEP summary to your local computer</span>
scp -ri ~/PEM_KEY_ID.pem ubuntu@AWS_ADDRESS_HERE:~/workspace/germline/Exome_Norm_HC_calls.filtered.PASS.vep.vcf_summary.html .
</code></pre></div>
<p>Open the HTML file.</p>
<p>Answer the following:
- How many variants were processed by VEP?
- What is the most common kind of consequence?
- Why are only variants the coding regions detected?
- Open the Exome_Norm_HC_out.bam and Exome_Norm_sorted_mrkdup_bqsr.bam in IGV 
    - Find a missense variant and examine it in IGV, does the variant look real?
    - E.g. take chr17 744946
        - Find the variant in the VEP VCF as well as IGV.
- VEP assigns each variant a consequence types, as described <a href="https://m.ensembl.org/info/genome/variation/prediction/predicted_data.html">here</a>.
    - What is the difference between splice region variants and splice donor/acceptor variants?
    - The --pick option is used, as described <a href="https://m.ensembl.org/info/docs/tools/vep/script/vep_other.html#pick">here</a>, do you think it makes sense to use? </p>
<h2 id="somatic-variants">Somatic variants<a class="headerlink" href="#somatic-variants" title="Permanent link">&para;</a></h2>
<p>We will run multiple variant callers and merging the callset.</p>
<h3 id="varscan">Varscan<a class="headerlink" href="#varscan" title="Permanent link">&para;</a></h3>
<p>First off is <a href="http://varscan.sourceforge.net/">VARSCAN</a>, that employs a robust heuristic/statistic approach to call variants that meet desired thresholds for read depth, base quality, variant allele frequency, and statistical significance.</p>
<p>As seen below varscan uses the mpileup command from samtools. What the mpileup command does can be explored <a href="https://www.htslib.org/doc/samtools-mpileup.html">here</a></p>
<div class="highlight"><pre><span></span><code>mkdir -p ~/workspace/somatic/varscan
<span class="nb">cd</span> ~/workspace/somatic/varscan

<span class="c1">#Have a look at the input data to the varscan caller</span>
samtools mpileup -l ~/workspace/inputs/references/exome/exome_regions.bed <span class="se">\</span>
    --no-BAQ -f ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam <span class="se">\</span>
    ~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam <span class="p">|</span> less -SN

<span class="c1"># Run varscan</span>
java -Xmx12g -jar ~/workspace/bin/VarScan.v2.4.2.jar somatic <span class="se">\</span>
    &lt;<span class="o">(</span>samtools mpileup -l ~/workspace/inputs/references/exome/exome_regions.bed <span class="se">\</span>
        --no-BAQ -f ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
        ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam <span class="se">\</span>
        ~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam<span class="o">)</span> <span class="se">\</span>
    ~/workspace/somatic/varscan/exome --min-var-freq <span class="m">0</span>.05 --mpileup <span class="m">1</span> --output-vcf

ls -halt

java -Xmx12g -jar ~/workspace/bin/VarScan.v2.4.2.jar processSomatic <span class="se">\</span>
    exome.snp.vcf exome.snp --min-tumor-freq <span class="m">0</span>.05 --max-normal-freq <span class="m">0</span>.01

java -Xmx12g -jar ~/workspace/bin/VarScan.v2.4.2.jar processSomatic <span class="se">\</span>
    exome.indel.vcf exome.indel --min-tumor-freq <span class="m">0</span>.05 --max-normal-freq <span class="m">0</span>.01

find ~/workspace/somatic/varscan -name <span class="s1">&#39;*.vcf&#39;</span> -exec bgzip -f <span class="o">{}</span> <span class="se">\;</span>
find ~/workspace/somatic/varscan -name <span class="s1">&#39;*.vcf.gz&#39;</span> -exec tabix -f <span class="o">{}</span> <span class="se">\;</span>

gatk VariantFiltration -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    -V exome.snp.Somatic.vcf.gz --mask exome.snp.Somatic.hc.vcf.gz <span class="se">\</span>
    --mask-name <span class="s2">&quot;processSomatic&quot;</span> --filter-not-in-mask -O exome.snp.Somatic.hc.filter.vcf.gz

gatk VariantFiltration -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    -V exome.indel.Somatic.vcf.gz --mask exome.indel.Somatic.hc.vcf.gz <span class="se">\</span>
    --mask-name <span class="s2">&quot;processSomatic&quot;</span> --filter-not-in-mask -O exome.indel.Somatic.hc.filter.vcf.gz

bcftools concat -a -o exome.vcf.gz -O z exome.snp.Somatic.hc.filter.vcf.gz exome.indel.Somatic.hc.filter.vcf.gz
tabix -f ~/workspace/somatic/varscan/exome.vcf.gz
</code></pre></div>
<h3 id="strelka">Strelka<a class="headerlink" href="#strelka" title="Permanent link">&para;</a></h3>
<p>Now we are going to run the second variant caller, <a href="https://github.com/Illumina/strelka/blob/master/docs/userGuide/README.md">STRELKA</a>. Strelka calls germline and somatic small variants from mapped sequencing reads and is optimized for rapid clinical analysis of germline variation in small cohorts and somatic variation in tumor/normal sample pairs. Both germline and somatic callers include a final empirical variant rescoring step using a random forest model to reflect numerous features indicative of call reliability which may not be represented in the core variant calling probability model.</p>
<div class="highlight"><pre><span></span><code>mkdir -p ~/workspace/somatic/strelka/exome
<span class="nb">cd</span> ~

<span class="nb">source</span> activate strelka-env
~/workspace/bin/strelka-2.9.10.centos6_x86_64/bin/configureStrelkaSomaticWorkflow.py <span class="se">\</span>
    --normalBam<span class="o">=</span>workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam <span class="se">\</span>
    --tumorBam<span class="o">=</span>workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam <span class="se">\</span>
    --referenceFasta<span class="o">=</span>workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    --exome --runDir<span class="o">=</span>workspace/somatic/strelka/exome

<span class="c1">#Please specify according to the number of cpus available or how many you would like to allocate to this job. In this case, four were given.</span>
<span class="c1"># Runtime: ~ 3min</span>
python2 ~/workspace/somatic/strelka/exome/runWorkflow.py -m <span class="nb">local</span> -j <span class="m">8</span>

conda deactivate

<span class="nb">cd</span> ~/workspace/somatic/strelka/exome/results/variants
zcat somatic.snvs.vcf.gz <span class="p">|</span> <span class="se">\</span>
    awk <span class="s1">&#39;{if(/^##/) print; else if(/^#/) print &quot;##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=\&quot;Genotype\&quot;&gt;\n&quot;$0; else print $1&quot;\t&quot;$2&quot;\t&quot;$3&quot;\t&quot;$4&quot;\t&quot;$5&quot;\t&quot;$6&quot;\t&quot;$7&quot;\t&quot;$8&quot;\tGT:&quot;$9&quot;\t./.:&quot;$10&quot;\t./.:&quot;$11;}&#39;</span> - <span class="se">\</span>
    &gt; somatic.snvs.gt.vcf
zcat somatic.indels.vcf.gz <span class="p">|</span> <span class="se">\</span>
    awk <span class="s1">&#39;{if(/^##/) print; else if(/^#/) print &quot;##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=\&quot;Genotype\&quot;&gt;\n&quot;$0; else print $1&quot;\t&quot;$2&quot;\t&quot;$3&quot;\t&quot;$4&quot;\t&quot;$5&quot;\t&quot;$6&quot;\t&quot;$7&quot;\t&quot;$8&quot;\tGT:&quot;$9&quot;\t./.:&quot;$10&quot;\t./.:&quot;$11;}&#39;</span> - <span class="se">\</span>
    &gt; somatic.indels.gt.vcf
find ~/workspace/somatic/strelka/exome/results/variants/ -name <span class="s2">&quot;*.vcf&quot;</span> -exec bgzip -f <span class="o">{}</span> <span class="se">\;</span>
find ~/workspace/somatic/strelka/exome/results/variants/ -name <span class="s2">&quot;*.vcf.gz&quot;</span> -exec tabix -f <span class="o">{}</span> <span class="se">\;</span>

bcftools concat -a -o exome.vcf.gz -O z somatic.snvs.gt.vcf.gz somatic.indels.gt.vcf.gz

tabix exome.vcf.gz
</code></pre></div>
<h3 id="mutect2">Mutect2<a class="headerlink" href="#mutect2" title="Permanent link">&para;</a></h3>
<p>Last up is <a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.php">MuTect2</a>. MuTect2 is a somatic SNP and indel caller that combines the DREAM challenge-winning somatic genotyping engine of the original MuTect (Cibulskis et al., 2013) with the assembly-based machinery of HaplotypeCaller.</p>
<h3 id="exome-data-commands">Exome data commands<a class="headerlink" href="#exome-data-commands" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1">#Obtaining germline resource from GATK</span>
<span class="nb">cd</span> ~/workspace/inputs/references
<span class="c1">#Already done</span>
<span class="c1">#gsutil cp gs://gatk-best-practices/somatic-hg38/af-only-gnomad.hg38.vcf.gz .</span>
<span class="c1">#gsutil cp gs://gatk-best-practices/somatic-hg38/af-only-gnomad.hg38.vcf.gz.tbi .</span>

mkdir -p ~/workspace/somatic/mutect
<span class="nb">cd</span> ~/workspace/somatic/mutect

<span class="c1">#Creating a panel of normals</span>
<span class="c1"># Runtime: ~ 17min</span>
nohup gatk --java-options <span class="s2">&quot;-Xmx12G&quot;</span> Mutect2 <span class="se">\</span>
    -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    -I ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam <span class="se">\</span>
    -tumor-sample HCC1395BL_DNA -O Exome_Norm_PON.vcf.gz <span class="p">&amp;</span>

<span class="c1">##### WAIT FOR THIS NOHUP STEP TO FINISH BEFORE CONTINUING #####</span>
<span class="c1">##### WAIT FOR THIS NOHUP STEP TO FINISH BEFORE CONTINUING #####</span>
<span class="c1">##### WAIT FOR THIS NOHUP STEP TO FINISH BEFORE CONTINUING #####</span>

<span class="c1">#Running Mutect2 Using latest version of GATK</span>
<span class="c1"># Runtime: ~20m</span>
nohup gatk --java-options <span class="s2">&quot;-Xmx12G&quot;</span> Mutect2 <span class="se">\</span>
    -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    -I ~/workspace/align/Exome_Tumor_sorted_mrkdup_bqsr.bam <span class="se">\</span>
    -tumor HCC1395_DNA -I ~/workspace/align/Exome_Norm_sorted_mrkdup_bqsr.bam <span class="se">\</span>
    -normal HCC1395BL_DNA --germline-resource ~/workspace/inputs/references/af-only-gnomad.hg38.vcf.gz <span class="se">\</span>
    --af-of-alleles-not-in-resource <span class="m">0</span>.00003125 --panel-of-normals <span class="se">\</span>
    ~/workspace/somatic/mutect/Exome_Norm_PON.vcf.gz -O ~/workspace/somatic/mutect/exome.vcf.gz <span class="se">\</span>
    -L chr6 -L chr17 <span class="p">&amp;</span>

<span class="c1">##### WAIT FOR NOHUP FOR THIS STEP TO FINISH BEFORE CONTINUING #####</span>
<span class="c1">##### WAIT FOR NOHUP FOR THIS STEP TO FINISH BEFORE CONTINUING #####</span>
<span class="c1">##### WAIT FOR NOHUP FOR THIS STEP TO FINISH BEFORE CONTINUING #####</span>


<span class="c1"># Filtering mutect variants</span>
gatk --java-options <span class="s2">&quot;-Xmx12G&quot;</span> FilterMutectCalls -V ~/workspace/somatic/mutect/exome.vcf.gz <span class="se">\</span>
    -O ~/workspace/somatic/mutect/exome_filtered.vcf.gz <span class="se">\</span>
    -R ~/workspace/inputs/references/genome/ref_genome.fa

<span class="c1">#Running mutect2 using gatk version 3.6</span>
<span class="c1">#java -Xmx12g -jar /usr/local/bin/GenomeAnalysisTK.jar -T MuTect2 --disable_auto_index_creation_and_locking_when_reading_rods -R ~/workspace/data/raw_data/references/ref_genome.fa -I:tumor ~/workspace/data/DNA_alignments/chr6+chr17/final/Exome_Tumor_sorted_mrkdup_bqsr.bam -I:Normal ~/workspace/data/DNA_alignments/chr6+chr17/final/Exome_Norm_sorted_mrkdup_bqsr.bam --dbsnp ~/workspace/data/raw_data/references/Homo_sapiens_assembly38.dbsnp138.vcf.gz --cosmic ~/workspace/data/raw_data/references/Cosmic_v79.dictsorted.vcf.gz -o ~/workspace/data/results/somatic/mutect/exome.vcf.gz -L ~/workspace/data/results/inputs/SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.interval_list</span>

<span class="nb">echo</span> ~/workspace/somatic/mutect/exome_filtered.vcf.gz &gt; ~/workspace/somatic/mutect/exome_vcf.fof
bcftools concat --allow-overlaps --remove-duplicates <span class="se">\</span>
    --file-list ~/workspace/somatic/mutect/exome_vcf.fof --output-type z <span class="se">\</span>
    --output ~/workspace/somatic/mutect/mutect_exome_filtered.vcf.gz

tabix ~/workspace/somatic/mutect/mutect_exome_filtered.vcf.gz
</code></pre></div>
<h3 id="now-we-are-going-to-merge-the-variants-detected-from-all-three-variant-callers">Now we are going to merge the variants detected from all three variant callers<a class="headerlink" href="#now-we-are-going-to-merge-the-variants-detected-from-all-three-variant-callers" title="Permanent link">&para;</a></h3>
<p>The reason for merging is that the variant callers are working differently internally to identify somatic alterations, with different strengths and weaknesses. With outputs from all three algorithms, we can now merge the variants to generate a comprehensive list of detected variants:</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Unzip the vcf.gz files before combining Variants</span>
<span class="nb">cd</span> ~/workspace/somatic
ls -halt ~/workspace/somatic/varscan/exome.vcf.gz
ls -halt ~/workspace/somatic/strelka/exome/results/variants/exome.vcf.gz
ls -halt ~/workspace/somatic/mutect/mutect_exome_filtered.vcf.gz

gunzip ~/workspace/somatic/varscan/exome.vcf.gz
gunzip ~/workspace/somatic/strelka/exome/results/variants/exome.vcf.gz
gunzip ~/workspace/somatic/mutect/mutect_exome_filtered.vcf.gz

<span class="c1">#Need to change header sample names in vcf file produced by mutect2 in order to combine variants with those from other algorithms</span>
sed -i <span class="s1">&#39;s/HCC1395BL_DNA/NORMAL/&#39;</span> ~/workspace/somatic/mutect/mutect_exome_filtered.vcf
sed -i <span class="s1">&#39;s/HCC1395_DNA/TUMOR/&#39;</span> ~/workspace/somatic/mutect/mutect_exome_filtered.vcf

<span class="c1"># (UNIQUIFY command) java -Xmx4g -jar /usr/local/bin/GenomeAnalysisTK.jar -T CombineVariants -R ~/workspace/data/raw_data/references/ref_genome.fa -genotypeMergeOptions UNIQUIFY --variant:varscan ~/workspace/data/results/somatic/varscan/exome.vcf --variant:strelka ~/workspace/data/results/somatic/strelka/exome/results/variants/exome.vcf --variant:mutect ~/workspace/data/results/somatic/mutect/new_gatk_files/exome.vcf -o ~/workspace/data/results/somatic/exome.unique.vcf.gz</span>
<span class="c1">#java -Xmx24g -jar ~/workspace/bin/GenomeAnalysisTK.jar -T  CombineVariants -R ~/workspace/inputs/references/genome/ref_genome.fa -genotypeMergeOptions PRIORITIZE --rod_priority_list mutect,varscan,strelka --variant:varscan ~/workspace/somatic/varscan/exome.vcf --variant:strelka ~/workspace/somatic/strelka/exome/results/variants/exome.vcf --variant:mutect ~/workspace/somatic/mutect/exome.vcf -o ~/workspace/somatic/exome.merged.vcf.gz</span>

java -Xmx12g -jar ~/workspace/bin/picard.jar MergeVcfs <span class="se">\</span>
    -I ~/workspace/somatic/varscan/exome.vcf <span class="se">\</span>
    -I ~/workspace/somatic/strelka/exome/results/variants/exome.vcf <span class="se">\</span>
    -I ~/workspace/somatic/mutect/mutect_exome_filtered.vcf <span class="se">\</span>
    -O ~/workspace/somatic/exome.merged.vcf

bgzip -c ~/workspace/somatic/exome.merged.vcf &gt; ~/workspace/somatic/exome.merged.vcf.gz
tabix -p vcf ~/workspace/somatic/exome.merged.vcf.gz
</code></pre></div>
<h3 id="left-align-and-trim">Left Align and Trim<a class="headerlink" href="#left-align-and-trim" title="Permanent link">&para;</a></h3>
<p>The reason for left align the variants and trim then is explained <a href="https://genome.sph.umich.edu/wiki/Variant_Normalization#Left_alignment">here</a>.
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/somatic/

gatk --java-options <span class="s2">&quot;-Xmx12G&quot;</span> LeftAlignAndTrimVariants <span class="se">\</span>
    -V ~/workspace/somatic/exome.merged.vcf.gz <span class="se">\</span>
    -O exome.merged.leftalignandtrim.vcf <span class="se">\</span>
    -R ~/workspace/inputs/references/genome/ref_genome.fa
</code></pre></div>
Note that when running on chromosome 6 and 17 merged variants file, this gave 0 variants aligned.</p>
<h3 id="we-will-split-multi-allelic-variants-into-multiple-records">We will split multi-allelic variants into multiple records<a class="headerlink" href="#we-will-split-multi-allelic-variants-into-multiple-records" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/somatic/

vt decompose -s ~/workspace/somatic/exome.merged.leftalignandtrim.vcf <span class="se">\</span>
    -o ~/workspace/somatic/exome.merged.leftalignandtrim.decomposed.vcf
</code></pre></div>
<h3 id="basic-filtering-on-somatic-variants">Basic Filtering on Somatic Variants<a class="headerlink" href="#basic-filtering-on-somatic-variants" title="Permanent link">&para;</a></h3>
<p>First, let's do a basic filtering for <code>PASS</code> only variants on our merged and normalized vcf file:
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/somatic
gatk --java-options <span class="s2">&quot;-Xmx12G&quot;</span> SelectVariants -R ~/workspace/inputs/references/genome/ref_genome.fa <span class="se">\</span>
    --exclude-filtered -V ~/workspace/somatic/exome.merged.leftalignandtrim.decomposed.vcf <span class="se">\</span>
    -O ~/workspace/somatic/exome.merged.norm.pass_only.vcf
</code></pre></div></p>
<h3 id="annotation-with-vep">Annotation with VEP.<a class="headerlink" href="#annotation-with-vep" title="Permanent link">&para;</a></h3>
<p>Again, we will use VEP to annotate the somatic variants as we did for the germline variants. </p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/somatic
<span class="c1"># Runtime: ~4min</span>

<span class="c1">#ssh -o ServerAliveInterval=300 -i course_EC2_01.pem ubuntu@ec2-52-23-206-90.compute-1.amazonaws.com</span>
<span class="c1">#source .bashrc</span>
<span class="c1">#cd workspace/somatic/</span>
nohup vep --cache --dir_cache ~/workspace/vep_cache <span class="se">\</span>
    --dir_plugins ~/workspace/vep_cache/Plugins <span class="se">\</span>
    --fasta ~/workspace/inputs/references/genome/ref_genome.fa --fork <span class="m">8</span> <span class="se">\</span>
    --assembly<span class="o">=</span>GRCh38 --offline --vcf -i ~/workspace/somatic/exome.merged.norm.pass_only.vcf <span class="se">\</span>
    -o ~/workspace/somatic/exome.merged.norm.annotated.vcf <span class="se">\</span>
    --check_existing --total_length --allele_number  --no_escape --everything <span class="se">\</span>
    --use_given_ref --force_overwrite <span class="p">&amp;</span>


<span class="c1">#Copy the html VEP summary to your local computer</span>
scp -ri ~/PEM_KEY_ID.pem ubuntu@AWS_ADDRESS_HERE:~/workspace/somatic/exome.merged.norm.annotated.vcf_summary.html .
</code></pre></div>
<p>Open the HTML file.</p>
<p>Answer the following:
- How many variants were processed by VEP?
- Any diffrence in the distribution of variants vs. the germline variants? 
- The variant categories e.g. intron_variant and regulatory_region_variant should be approached carefully, why?</p>
<h2 id="inspecting-variants-in-igv">Inspecting variants in IGV<a class="headerlink" href="#inspecting-variants-in-igv" title="Permanent link">&para;</a></h2>
<ul>
<li>
<p>Download the procdessed DNA bam files to your local machine if you have not done so yet, see the session "Introduction to IGV"</p>
</li>
<li>
<p>Open the Exome_Tumor_sorted_mrkdup_bqsr.bam and Exome_Norm_sorted_mrkdup_bqsr.bam in IGV.</p>
</li>
<li>
<p>Keep the VCF file open in a terminal window on AWS</p>
</li>
<li>Try to find a relevant variant in TP53 in the VCF file on AWS
<div class="highlight"><pre><span></span><code><span class="c1">#Can be done in many ways, this is just an example.</span>
<span class="nb">cd</span> ~/workspace/somatic
grep <span class="s2">&quot;TP53&quot;</span> exome.merged.norm.annotated.vcf <span class="p">|</span> less -SN
<span class="c1">#### Or </span>
less -SN exome.merged.norm.annotated.vcf
<span class="c1">#In the less window type /TP53</span>
</code></pre></div></li>
<li>Why is the TP53 variant in three rows in the VCF file? <ul>
<li>lead: how many somatic variant callers were run?</li>
</ul>
</li>
<li>
<p>Go to the position (chr17 7675088) on IGV in your local machine
<img alt="" src="https://i.imgur.com/TQ8KKZ8.png" /></p>
<ul>
<li>Is it present in the normal DNA?<ul>
<li>Do you still think it is valid? How come the variant can be present in the germline DNA?</li>
</ul>
</li>
</ul>
</li>
<li>
<p>Do the same thing for a stop_gained and a frameshift variant
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/somatic
grep <span class="s2">&quot;stop_gained&quot;</span> exome.merged.norm.annotated.vcf <span class="p">|</span> less -SN
</code></pre></div></p>
</li>
<li>The top variant is seen only once, this is just identified correctly by one caller because it changes two bases in a row<ul>
<li>open chr6 1930220 in IGV</li>
</ul>
</li>
</ul>
<p><img alt="" src="https://i.imgur.com/DbVcGjS.png" /></p>
<ul>
<li>let us convert the vcf-file to a MAF file format<ul>
<li>note all information is not kept but it simplifies looking at the variants
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/somatic
perl ~/workspace/bin/mskcc-vcf2maf/vcf2maf.pl --input-vcf ~/workspace/somatic/exome.merged.norm.annotated.vcf --output-maf ~/workspace/somatic/exome.merged.norm.annotated.maf  --tumor-id TUMOR --normal-id NORMAL --inhibit-vep --ref-fasta ~/workspace/inputs/references/genome/ref_genome.fa

less -SN ~/workspace/somatic/exome.merged.norm.annotated.maf

<span class="c1">#Count the number of variant types</span>
cut -f9 ~/workspace/somatic/exome.merged.norm.annotated.maf <span class="p">|</span> sort <span class="p">|</span> uniq -c
<span class="c1">#As you see the variant nomenclature is not the same in VCF and MAF</span>

grep Nonsense_Mutation ~/workspace/somatic/exome.merged.norm.annotated.maf <span class="p">|</span> less -SN 
</code></pre></div></li>
</ul>
</li>
<li>
<p>Have a look at the nosense variant in CCDC40 in IGV</p>
<ul>
<li>chr17 80050160</li>
<li>What is the VAF?</li>
<li>What is the VAF of the TP53 variant?</li>
<li>Reflections?<ul>
<li>Lead: clonality</li>
</ul>
</li>
</ul>
</li>
<li>
<p>Can you find a variant in BRCA1? </p>
<ul>
<li>What is the impact? Is it relevant?</li>
</ul>
</li>
<li>
<p>There was a BRCA 1 variant in the germline that we did not dicsuss earlier
<div class="highlight"><pre><span></span><code><span class="c1">#Let us grep &quot;BRCA1&quot; and send the output to another grep command and take the &quot;HIGH&quot; impact variants</span>
grep BRCA1 ~/workspace/germline/Exome_Norm_HC_calls.filtered.PASS.vep.vcf <span class="p">|</span> grep <span class="s2">&quot;HIGH&quot;</span> <span class="p">|</span> less -SN 
</code></pre></div></p>
</li>
<li>let us inspect this variant in IGV, why is one allele missing in the tumor?<ul>
<li>This is the variant location: chr17   43057078 
<img alt="" src="https://i.imgur.com/PXKeG7C.png" /></li>
</ul>
</li>
</ul>

              
            </article>
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" data-md-state="hidden">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../intro_igv/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Introduction to IGV" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Introduction to IGV
            </div>
          </div>
        </a>
      
      
        
        <a href="../structural_variation/" class="md-footer__link md-footer__link--next" aria-label="Next: Structural Variants" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Structural Variants
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2021, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "navigation.sections", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest"], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.361d90f1.min.js"}</script>
    
    
      <script src="../../assets/javascripts/bundle.289a2a4b.min.js"></script>
      
    
  </body>
</html>