---
title: Calling germline and somatic mutations

---

# Calling germline and somatic mutations

## Germline variant calling
We will start by calling germline variants. For this purpose we will use the [GATK HaplotypeCaller](https://gatk.broadinstitute.org/hc/en-us/articles/4414586765723-HaplotypeCaller) from the Broad Institute. We will use this tool on the germline DNA BAM files that we processed earlier in the day. 

Active forum and tutorials for the GATK software suite are available [here](https://gatk.broadinstitute.org/hc/en-us/community/topics).

HaplotypeCaller can be run in single- or joint sample mode. The joint sample mode gives improved accuracy, especially in samples with low coverage. For simplicity, we will only use the single sample mode in this course. 

When HaplotypeCaller detects variation in the genome, it performs local realignment of the reads in the regions with detected variation. This improves variant accuracy, especially in e.g., repetitive regions. 

```bash
#Make sure that $GATK_REGIONS variable is set correctly
export GATK_REGIONS='-L chr6 -L chr17'
echo $GATK_REGIONS

#Create working directory for germline results
mkdir -p /nfs/course/students/YOUR_KI_ID/germline
cd /nfs/course/students/YOUR_KI_ID/germline

#Call variants from exome data
#The argument --java-options '-Xmx12g' tells GATK to use 12GB of memory
gatk --java-options '-Xmx12g' HaplotypeCaller \
    -R /nfs/course/students/YOUR_KI_ID/references/genome/ref_genome.fa \
    -I /nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.bam \
    -O /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.vcf \
    --bam-output /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_out.bam $GATK_REGIONS
```

## Exploring the HaplotypeCaller output files

Next, we will take a look at the results from HaplotypeCaller in IGV. To do this, we need to copy some files to our local computer to open in IGV.

### Copying and opening the realigned BAM file in IGV

Navigate to the folder where you want to place the BAM files on your local computer, e.g. ~/course_bams.

```bash
#Copy the HaplotypeCaller realigned BAM file to your local computer 
scp -r YOUR_KI_ID@c8cancergen02.ki.se:/nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_out.ba* .
```

Next, launch IGV on your computer and remember to select the **hg38** reference genome. Follow the steps below:
- Open the BAM files in IGV (File > Load from file):
    - Exome_Norm_sorted_mrkdup_bqsr.bam (input file to HaplotypeCaller).
    - Exome_Norm_HC_out.bam (realigned BAM file generated by HaplotypeCaller).
- Navigate to chr17:76286974-76287203.
    - Color alignments by: no color.
    - Enable ruler in the menu bar.
    - In the first base of the indel: Sort alignments according to base.
    - Select squished view.
    - Zoom out.

Answer the following questions:
- What characterizes misaligned variant-supporting reads in the non-realigned BAM file (i.e. what features do they have in IGV)?
- Are there any false positive variants in the non-realigned BAM file?


![](https://i.imgur.com/yQS7rie.png)


### Inspecting the HaplotypeCaller VCF (variant call format) file

Next, let's take a look at the VCF file produced by HaplotypeCaller on the server.

```bash
#Make sure you hare in the same directory as the output files
cd /nfs/course/students/YOUR_KI_ID/germline

#List files
ls -halt

#Open the VCF file
less -SN Exome_Norm_HC_calls.vcf
```

The VCF file format is quite complex, but the specifications can be found [here](https://samtools.github.io/hts-specs/VCFv4.2.pdf).

After opening the VCF file as shown above, do the following:
- Identify the VCF header section.
- Identify the data column header.
- Find the variant on chr17 pos 106742 in the VCF file using the following method:
    - With the file open using ´less´, type ´/106742´ and press ENTER. This will scroll the file to the position where the variant with coordinate 106742 is located. 
    
Answer the following questions:
- What is the read depth at this position?

- Is this a high quality variant?

Next, go back to IGV and the BAM files you have opened there. Go to the variant position chr17:106742 and do the following by right clicking on the reads:
- Select expanded view.
- Color alignments by read strand.
- Sort alignments by base.

Answer the following question:
- Does the visual impression in IGV support the quality of the variant?

Finally, find the variant on chr17 pos 1787008 in the VCF file and answer the following questions:
- Is this a high quality variant?

- Navigate in IGV to the variant position. Does the information in IGV match the VCF file?


### Filtering germline variants

The raw output of GATK HaplotypeCaller will include many variants with varying degrees of quality. For various reasons we might wish to further filter these to a higher confidence set of variants. We will use a hard-filtering approach that is further described in detail on the [GATK webside](https://gatk.broadinstitute.org/hc/en-us/articles/360035890471-Hard-filtering-germline-short-variants). 

### Separate SNPs from indels in the variant call set

First, we will separate out the SNPs and indels from the VCF into new separate VCFs. Note that the variant type (SNP, INDEL, MIXED, etc) is not stored explicitly in the VCF but instead inferred from the genotypes. We will use a versatile GATK tool called `SelectVariants`. This command can be used for all kinds of simple filtering or subsetting purposes. We will run it twice to select variants from the VCF files, once for SNPs, and then again for indels, ultimately producing two new VCFs.

```bash
cd /nfs/course/students/YOUR_KI_ID/germline/
gatk --java-options '-Xmx12g' SelectVariants \
    -R /nfs/course/inputs/references/genome/ref_genome.fa \
    -V /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.vcf -select-type SNP \
    -O /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.snps.vcf
gatk --java-options '-Xmx12g' SelectVariants \
    -R /nfs/course/inputs/references/genome/ref_genome.fa \
    -V /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.vcf -select-type INDEL \
    -O /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.indels.vcf
```

### Apply filters to the SNP and indel call sets

Next, we will perform so-called hard filtering by applying a number of cutoffs to the variants. As you will see, each can be given its own name so that you can later determine which filters a variant has failed. Visit the [GATK documentation on hard-filtering](https://gatk.broadinstitute.org/hc/en-us/articles/360035890471-Hard-filtering-germline-short-variants) to learn more about the following hard filtering options. Notice that different filters and cutoffs are recommended for SNVs and indels. This is why we first split them into separate files.

```bash
cd /nfs/course/students/YOUR_KI_ID/germline/
gatk --java-options '-Xmx12g' VariantFiltration \
    -R /nfs/course/inputs/references/genome/ref_genome.fa \
    -V /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.snps.vcf \
    --filter-expression "QD < 2.0" --filter-name "QD_lt_2" \
    --filter-expression "FS > 60.0" --filter-name "FS_gt_60" \
    --filter-expression "MQ < 40.0" --filter-name "MQ_lt_40" \
    --filter-expression "MQRankSum < -12.5" --filter-name "MQRS_lt_n12.5" \
    --filter-expression "ReadPosRankSum < -8.0" --filter-name "RPRS_lt_n8" \
    --filter-expression "SOR > 3.0" --filter-name "SOR_gt_3" \
    -O /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.snps.filtered.vcf

gatk --java-options '-Xmx12g' VariantFiltration \
    -R /nfs/course/inputs/references/genome/ref_genome.fa \
    -V /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.indels.vcf \
    --filter-expression "QD < 2.0" --filter-name "QD_lt_2" \
    --filter-expression "FS > 200.0" --filter-name "FS_gt_200" \
    --filter-expression "ReadPosRankSum < -20.0" --filter-name "RPRS_lt_n20" \
    --filter-expression "SOR > 10.0" --filter-name "SOR_gt_10" \
    -O /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.indels.filtered.vcf
```

Notice that warnings are given with regards to MQRankSum and ReadPosRankSum. These are only calculated if the site is called as heterozygous, e.g. for this site (inspect in IGV and in the VCF file): chr17:1067361.nThis was discussed at the [GATK forum](https://sites.google.com/a/broadinstitute.org/legacy-gatk-forum-discussions/2017-01-18-2016-08-11/8323-Why-do-MQRankSum-and-ReadPosRankSum-not-appear-in-some-vcf-file-entries).

The variants are now marked as PASS or *filter-name*. Use `grep -v` and `head` to skip past all the VCF header lines and view the first few records. 

```bash
#inspect the 10 first variants that did not pass
grep -v "##" Exome_Norm_HC_calls.snps.filtered.vcf | grep -vP "\tPASS\t" | head -10
#inspect the 10 first variants that did pass
grep -v "##" Exome_Norm_HC_calls.snps.filtered.vcf | grep -P "\tPASS\t" | head -10
```
Answer the following question:
- How many varinats have failed and passed filtering?


#### Merge the filtered SNP and indel VCFs back together

```bash
gatk --java-options '-Xmx12g' MergeVcfs \
    -I /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.snps.filtered.vcf \
    -I /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.indels.filtered.vcf \
    -O /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.filtered.vcf

#Number of variants in the SNP file
grep -v "##" Exome_Norm_HC_calls.snps.filtered.vcf | wc -l

#Number of variants in the indel file
grep -v "##" Exome_Norm_HC_calls.indels.filtered.vcf | wc -l

#Number of variants in the merged file
grep -v "##" Exome_Norm_HC_calls.filtered.vcf | wc -l
```

#### Extract PASS variants only

It would also be convenient to have a VCF with only passing variants. For this, we can go back the `GATK SelectVariants` tool. This will be run much as above except with the `--exlude-filtered` option instead of `-select-type`.

```bash
gatk --java-options '-Xmx12g' SelectVariants \
    -R /nfs/course/inputs/references/genome/ref_genome.fa \
    -V /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.filtered.vcf \
    -O /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.filtered.PASS.vcf \
    --exclude-filtered
```

### Perform annotation of the filtered variants
Now that we have high-confidence, filtered variants, we want to start understanding which of these variants might be clinically or biologically relevant. Ensembl's Variant Effect Predictor (VEP) annotation software is a powerful tool for annotating variants with a great deal of biological features. This includes information such as protein consequence (non-coding or coding), population frequencies, links to external databases, various scores designed to estimate the importance of individual variants on protein function, and much more.

Note, the first time you run VEP it will create a FASTA index for the reference genome in the VEP cache. Therefore, it will take longer to run the first time but should speed up substantially for subsequent runs on files with similar numbers of variants.


```bash
#VEP annotate hard-filtered exome results

####### Already run code below for your reference (no need to run these commands)
#cd /nfs/course/inputs/references/vep/
#wget https://github.com/Ensembl/VEP_plugins/archive/refs/heads/release/104.zip
#unzip 104.zip 
#ls -halt /nfs/course/inputs/references/vep/
#mv VEP_plugins-release-104 Plugins
#ls -halt /nfs/course/inputs/references/vep/Plugins

#Get annotation fasta for homo_sapiens 
#cd /nfs/course/inputs/references/vep/homo_sapiens
#wget ftp://ftp.ensembl.org/pub/release-104/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz

####### Start running commands from here

cd /nfs/course/students/YOUR_KI_ID/germline
#Remember, we are using the reference genome file /nfs/course/inputs/references/genome/ref_genome.fa that only contains chr6 and chr17.

source activate ensembl-vep-113  # activate a separate environment for vep
# Runtime ~1 min
nohup vep --cache --dir_cache /nfs/course/inputs/references/vep \
    --fasta /nfs/course/inputs/references/genome/ref_genome.fa \
    --fork 4 --assembly=GRCh38 --offline --vcf \
    -i Exome_Norm_HC_calls.filtered.PASS.vcf \
    -o Exome_Norm_HC_calls.filtered.PASS.vep.vcf \
    --check_existing --total_length --allele_number --no_escape --everything \
    --use_given_ref --force_overwrite --coding_only &

#Open the VEP VCF file: lines 39-41 contain information about the VEP annotation.
less -SN Exome_Norm_HC_calls.filtered.PASS.vep.vcf
#Or just grep the VEP info.
grep "##" Exome_Norm_HC_calls.filtered.PASS.vep.vcf | grep "VEP" | less -SN
```

Copy the HTML VEP summary to your local computer, run locally and first navigate to your preferred folder:
```bash
scp -r YOUR_KI_ID@c8cancergen02.ki.se:/nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.filtered.PASS.vep.vcf_summary.html .
```

Open the HTML VEP summary file on your local computer.

Answer the following questions:
- How many variants were processed by VEP?

- Why are only variants the coding regions detected?

- VEP assigns each variant a consequence types, as described [here](https://m.ensembl.org/info/genome/variation/prediction/predicted_data.html).
    - What is the most common kind of variant consequence in the HTML summary file?
    
    - What is the difference between splice region variants and splice donor/acceptor variants?
    
Let's also inspect one of the germline variants in IGV, a BRCA1 variant.
```bash
#Let us grep "BRCA1" and send the output to another grep command and take the "HIGH" impact variants
grep BRCA1 /nfs/course/students/YOUR_KI_ID/germline/Exome_Norm_HC_calls.filtered.PASS.vep.vcf | grep "HIGH" | less -SN 
```
Copy the file to your local computer to open it in IGV.

Answer the following question:
- In IGV, for the BRCA1 variant found above at the location chr17:43057078, why is one allele missing in the tumor?

![](https://i.imgur.com/PXKeG7C.png)
    


## Somatic variant calling

We have now finished calling germline variants and will proceed to call somatic variants from our data. We will run two variant callers, SAGE and Mutect2, and then merge the results to form a final callset.

### SAGE

We will first run [SAGE](https://github.com/hartwigmedical/hmftools/blob/master/sage/README.md). SAGE stands for Somatic Alternations in GEnome and is a precise and highly sensitive somatic SNV, MNV and small INDEL caller.

```bash
#Obtaining HMFtools resources
#Already done, no need to run these commands, only shown for demonstration purpose
#mkdir /nfs/course/inputs/references/sage
#cd /nfs/course/inputs/references/sage
#wget https://storage.googleapis.com/hmf-public/HMFtools-Resources/pipeline/oncoanalyser/2.0/38/hmf_pipeline_resources.38_v2.0.0--3.tar.gz
#tar -xzf hmf_pipeline_resources.38_v2.0.0--3.tar.gz


mkdir -p /nfs/course/students/YOUR_KI_ID/somatic/sage
cd /nfs/course/students/YOUR_KI_ID/somatic/sage

#Files
# Normal: /nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.bam
# Tumor: /nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup_bqsr.bam

# activate the bioinfo tools environment
source activate bioinfo-tools

java -Xms4G -Xmx25G -cp /nfs/course/bin/sage_v3.4.4.jar com.hartwig.hmftools.sage.SageApplication \
    -threads 4 -tumor exome_tumor_lib1 \
    -tumor_bam /nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup_bqsr.bam \
    -reference exome_norm_lib1 \
    -reference_bam /nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.bam \
    -ref_genome_version 38 -ref_genome /nfs/course/inputs/references/genome/ref_genome.fa \
    -hotspots /nfs/course/inputs/references/sage/hmf_pipeline_resources.38_v2.0--3/dna/variants/KnownHotspots.somatic.38.vcf.gz \
    -panel_bed /nfs/course/inputs/references/sage/hmf_pipeline_resources.38_v2.0--3/dna/variants/ActionableCodingPanel.38.bed.gz \
    -high_confidence_bed /nfs/course/inputs/references/sage/hmf_pipeline_resources.38_v2.0--3/dna/variants/HG001_GRCh38_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-X_v.3.3.2_highconf_nosomaticdel_noCENorHET7.bed.gz \
    -ensembl_data_dir /nfs/course/inputs/references/sage/hmf_pipeline_resources.38_v2.0--3/common/ensembl_data \
    -hotspot_min_tumor_qual 150 -hotspot_min_tumor_vaf 0.0002 -panel_min_tumor_qual 250 \
    -panel_min_tumor_vaf 0.0002 -panel_only -hard_min_tumor_vaf 0.002 \
    -min_map_quality 20 -min_avg_base_qual 30 \
    -output_vcf exome.sage.vcf.gz
ls -lhtr
less -SN exome.sage.vcf.gz  # look at the produced vcf
```


### Mutect2

Next we will run [MuTect2](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.php). MuTect2 is a somatic SNP and indel caller that combines the DREAM challenge-winning somatic genotyping engine of the original MuTect (Cibulskis et al., 2013) with the assembly-based machinery of HaplotypeCaller.

### Running Mutect2 for exome data
```bash
#Obtaining germline resource from GATK
cd /nfs/course/inputs/references
#Already done, no need to run the two lines below
#gsutil cp gs://gatk-best-practices/somatic-hg38/af-only-gnomad.hg38.vcf.gz .
#gsutil cp gs://gatk-best-practices/somatic-hg38/af-only-gnomad.hg38.vcf.gz.tbi .

mkdir -p /nfs/course/students/YOUR_KI_ID/somatic/mutect
cd /nfs/course/students/YOUR_KI_ID/somatic/mutect

#Creating a panel of normals
# Runtime: ~ 13 min
nohup gatk --java-options "-Xmx12G" Mutect2 \
    -R /nfs/course/inputs/references/genome/ref_genome.fa \
    -I /nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.bam \
    -tumor-sample HCC1395BL_DNA -O Exome_Norm_PON.vcf.gz &

##### WAIT FOR THIS NOHUP STEP TO FINISH BEFORE CONTINUING #####

#Running Mutect2 using GATK
# Runtime: ~18m
nohup gatk --java-options "-Xmx12G" Mutect2 \
    -R /nfs/course/inputs/references/genome/ref_genome.fa \
    -I /nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Tumor_sorted_mrkdup_bqsr.bam \
    -tumor HCC1395_DNA -I /nfs/course/students/YOUR_KI_ID/data/dna_alignments/Exome_Norm_sorted_mrkdup_bqsr.bam \
    -normal HCC1395BL_DNA --germline-resource /nfs/course/inputs/references/af-only-gnomad.hg38.vcf.gz \
    --af-of-alleles-not-in-resource 0.00003125 --panel-of-normals Exome_Norm_PON.vcf.gz \
    -O exome.mutect.vcf.gz -L chr6 -L chr17 &

##### WAIT FOR THIS NOHUP STEP TO FINISH BEFORE CONTINUING #####

# Filtering the Mutect variants
gatk --java-options "-Xmx12G" FilterMutectCalls -V exome.mutect.vcf.gz \
    -O exome_filtered.mutect.vcf.gz \
    -R /nfs/course/inputs/references/genome/ref_genome.fa

echo exome_filtered.mutect.vcf.gz > exome_vcf.mutect.fof
bcftools concat --allow-overlaps --remove-duplicates \
    --file-list exome_vcf.mutect.fof --output-type z \
    --output mutect_exome_filtered.vcf.gz

# index the vcf
tabix mutect_exome_filtered.vcf.gz
```

### Now we are going to merge the variants detected from the two variant callers

The reason for merging is that the variant callers are working differently internally to identify somatic alterations, with different strengths and weaknesses. With outputs from two algorithms, we can now merge the variants to generate a comprehensive list of detected variants.

**Edit the commands below to merge the SAGE and Mutect2 variants**

```bash
# Unzip the vcf.gz files before combining Variants
cd /nfs/course/students/YOUR_KI_ID/somatic
ls -halt sage/exome.sage.vcf.gz
ls -halt mutect/mutect_exome_filtered.vcf.gz

gunzip sage/exome.sage.vcf.gz
gunzip mutect/mutect_exome_filtered.vcf.gz
ls -halt sage/exome.sage.vcf
ls -halt mutect/mutect_exome_filtered.vcf

#Need to change header sample names in vcf files in order to combine variants from different algorithms
#lets set something easily understandable, i.e. "TUMOR" and "NORMAL"
grep "^#CHROM" mutect/mutect_exome_filtered.vcf sage/exome.sage.vcf  # checkthe header lines where the sample names are the two last columns
sed -i 's/HCC1395BL_DNA/NORMAL/' mutect/mutect_exome_filtered.vcf
sed -i 's/HCC1395_DNA/TUMOR/' mutect/mutect_exome_filtered.vcf
sed -i 's/exome_norm_lib1/NORMAL/' sage/exome.sage.vcf
sed -i 's/exome_tumor_lib1/TUMOR/' sage/exome.sage.vcf
grep "^#CHROM" mutect/mutect_exome_filtered.vcf sage/exome.sage.vcf  # checkthe header lines where the sample names are the two last columns

# merge
gatk --java-options "-Xmx12G" MergeVcfs \
    -I sage/exome.sage.vcf \
    -I /nfs/course/students/YOUR_KI_ID/somatic/mutect/mutect_exome_filtered.vcf \
    -O exome.merged.vcf

# compress the vcf
bgzip -c exome.merged.vcf > exome.merged.vcf.gz
tabix -p vcf exome.merged.vcf.gz
```


### Left Align and Trim
The reason for left aligning the variants and trimming them is explained [here](
https://genome.sph.umich.edu/wiki/Variant_Normalization#Left_alignment).
```bash
cd /nfs/course/students/YOUR_KI_ID/somatic/

gatk --java-options "-Xmx12G" LeftAlignAndTrimVariants \
    -V exome.merged.vcf.gz \
    -O exome.merged.leftalignandtrim.vcf \
    -R /nfs/course/inputs/references/genome/ref_genome.fa
```


### Split multi-allelic variants into multiple records

```bash
cd /nfs/course/students/YOUR_KI_ID/somatic/

vt decompose -s exome.merged.leftalignandtrim.vcf \
    -o exome.merged.leftalignandtrim.decomposed.vcf
```


### Basic Filtering on Somatic Variants

First, let's do a basic filtering for `PASS` only variants on our merged and normalized vcf file:
```bash
cd /nfs/course/students/YOUR_KI_ID/somatic
gatk --java-options "-Xmx12G" SelectVariants -R /nfs/course/inputs/references/genome/ref_genome.fa \
    --exclude-filtered -V exome.merged.leftalignandtrim.decomposed.vcf \
    -O exome.merged.norm.pass_only.vcf
```


### Annotation with VEP 
Again, we will use VEP to annotate the somatic variants as we did for the germline variants. 

```bash
cd /nfs/course/students/YOUR_KI_ID/somatic
# Runtime: ~1min

source activate ensembl-vep-113  # activate a separate environment for vep
nohup vep --cache --dir_cache /nfs/course/inputs/references/vep \
    --fasta /nfs/course/inputs/references/genome/ref_genome.fa --fork 4 \
    --assembly=GRCh38 --offline --vcf -i exome.merged.norm.pass_only.vcf \
    -o exome.merged.norm.annotated.vcf \
    --check_existing --total_length --allele_number  --no_escape --everything \
    --use_given_ref --force_overwrite &
```

Copy the HTML VEP summary to your local computer, run locally and first navigate to your preferred folder:
```bash
scp -r YOUR_KI_ID@c8cancergen02.ki.se:/nfs/course/students/YOUR_KI_ID/somatic/exome.merged.norm.annotated.vcf_summary.html .
```

Open the HTML file.

Answer the following questions:

- How many variants were processed by VEP?

- Is there any difference in the distribution of somatic variants vs the germline variants? 

- The variant categories e.g. intron_variant and regulatory_region_variant should be approached carefully- why?


## Inspecting the somatic variants in IGV
Download the processed DNA BAM files to your local machine if you have not done so yet. Open the Exome_Tumor_sorted_mrkdup_bqsr.bam and Exome_Norm_sorted_mrkdup_bqsr.bam in IGV.

Below, we will try to find a relevant variant in TP53 in the VCF file on the server to inspect further.
```bash
#Can be done in many ways, this is just an example.
cd /nfs/course/students/YOUR_KI_ID/somatic
grep "TP53" exome.merged.norm.annotated.vcf | less -SN
#### Or 
less -SN exome.merged.norm.annotated.vcf
#In the less window type /TP53

# to skip the very long INFO column, and align header and columns
grep -P "#CHROM|TP53" exome.merged.norm.annotated.vcf | cut -f 1-7,9-11 | column -t | less -SN
```

Answer the following questions:

- Why is the TP53 variant on two rows in the VCF file? (Hint: how many somatic variant callers were run?)

- What is the VAF of the TP53 variant? (Hint: look in the "FORMAT" column which field corresponds to "AF" and then check this field in the correct sample column)

- Download the vepped vcf to your local machine and go to the position (chr17:7675088) on IGV in your local machine. Is the TP53 variant present in the normal DNA? Do you still think this is a valid variant? How come the variant can be present in the germline DNA but be called by a somatic variant caller?

![](https://i.imgur.com/TQ8KKZ8.png)

Let's convert the VCF file to the MAF file format. In this conversion, all information from the VCF file is not kept but it simplifies looking at the variants.
```bash
cd /nfs/course/students/YOUR_KI_ID/somatic
perl /nfs/course/bin/vcf2maf-1.6.22/vcf2maf.pl \
    --input-vcf exome.merged.norm.annotated.vcf --output-maf exome.merged.norm.annotated.maf \
    --tumor-id TUMOR --normal-id NORMAL --inhibit-vep --ref-fasta /nfs/course/inputs/references/genome/ref_genome.fa

less -SN exome.merged.norm.annotated.maf

#Count the number of variant types
cut -f9 exome.merged.norm.annotated.maf | sort | uniq -c
#As you see the variant nomenclature is not the same in VCF and MAF

grep Nonsense_Mutation exome.merged.norm.annotated.maf | less -SN 
```
Have a look at the nonsense variant in CCDC40 in IGV, at the position chr17:80050160. Then answer the following questions:

- What is the VAF?

- Is the variant clonal (present in all cancer cells) or subclonal (present in only a subset of the cancer cells)?

- Can you find a somatic variant in BRCA1? What is the impact? Is it relevant?




