---
hide:
  - navigation
---

# Welcome to Clinical Cancer Genomics Course

<p class="head-p1">This course aims to provide an introduction to cancer genomics and to support to obtain practical knowledge regarding how to apply state of the art methodology to interrogate the cancer genome in a routine clinical setting or a clinical trial setting.
    
The course will include lectures covering the technology advancements that have enabled high-throughput analysis of cancer genomes and the knowledge that can be obtained by applying these technologies. This encompasses both laboratory sample processing and downstream bioinformatics. Lectures will be held in the mornings with computer-based exercises in the afternoon. </p>

<p class="head-p1">The main objective of the course is to facilitate that students get an understanding of basic theory and obtain practical knowledge that will enable course participants to apply the covered methodologies in their own research- or clinical laboratory.</p>

<p class="head-p1">
    The setup and practical exercises are a modified version of an <a href="https://github.com/griffithlab/pmbio.org" target="_blank">existing course</a> generously shared under an <a href="https://github.com/griffithlab/pmbio.org/blob/master/LICENSE" target="_blank">MIT</a> and <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="_blank">Creative Commons</a> license.
    This is the first version of the Clinical Cancer Genomics course, later updated iterations of the course will be shared according to the same principle. 
</p>

-------------------------------

## Learning Outcomes 

At the end of this course the student will be able to:

* Show a basic insight into the cancer genome.
* Understand how the cancer genome can be interrogated through tissues and liquid biopsies.
* Understand how to apply technology to obtain relevant information from the cancer genome.
* Understand the file formats used in high throughput sequencing.
* use the command line and running bioinformatic tools.
* Understand the constituents of a bioinformatics pipeline for processing Illumina sequencing data and to run such a pipeline.
* Perform quality control on DNA- and RNA sequencing data for cancer sequencing purposes.
* Call somatic and germline variation.
* Curate somatic and germline variation for a clinical setting.
* Annotate somatic and germline variation.
* Visualise data in R.
* Use online resources such as genome browsers and portals for variant annotation.

-------------------------------

## Contents of the course 

An introduction to the cancer genome and mutational processes in cancer. 

* Overview of disease heterogeneity â€“ the concept of cancer subtypes. 
* The clinical impact of analysing the cancer genome. 
* The concept of personalized therapy by tumour profiling. 
* Intra-patient tumour heterogeneity. 
    * How to enable cancer genomics through tissues and liquid biopsies 
    * How to apply to high-throughput methodology to interrogate the cancer genome. 
    * Illumina sequencing file formats. 
* Bioinformatics pipelines. 
* Processing of DNA and RNA sequencing data. 
* QC of both DNA and RNA sequencing data 
* Calling somatic and germline variation: 
    * Point mutations and indels.  
    * Copy-number alterations. 
    * Structural variation. 
* File formats for variant calling. 
* Annotating somatic and germline variation. 
* How to curate somatic- and germline variation for clinical use.
--------------------------------------

## Literature and other teaching material

Recommended reading before the course:

1. <a href="https://doi.org/10.1038/s41576-021-00338-8" target="_blank"> Clinical cancer genomic profiling </a>
2. <a href="https://doi.org/10.1038/s41436-018-0278-z" target="_blank"> Standard operating procedure for somatic variant refinement of sequencing data with paired tumor and normal samples </a>

**Note:**

1. Lectures in the morning with computer exercises in the afternoon.
2. 100% attendance is recommended, due to that each session is exclusive and cannot be compensated for later on. The student will be asked to review the issue presented in case of absence in a session.
3. Each computer exercise addresses one or multiple learning outcomes. Each student will hand in a written report form each computer exercise. All intended learning outcomes need to be achieved in order to pass the exam.

## Additional Information 

:fontawesome-regular-calendar: <span class="add-info">2023-04-24 to 2023-04-28</span></br>
:fontawesome-regular-clock: <span class="add-info">Monday - Friday ( 09:00 - 17:00 ) </span></br>
:fontawesome-regular-envelope: <span class="add-info"> johan.lindberg@ki.se</span>