
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="../../assets/img/favicon.ico">
      <meta name="generator" content="mkdocs-1.2.3, mkdocs-material-8.1.4">
    
    
      
        <title>Human Genome - Clinical Cancer Genomics Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bb3983ee.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.e6a45f82.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/css/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#human-genome-reference-files" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-header__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Clinical Cancer Genomics Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Human Genome
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Clinical Cancer Genomics Course" class="md-nav__button md-logo" aria-label="Clinical Cancer Genomics Course" data-md-component="logo">
      
  <img src="../../assets/img/logo.svg" alt="logo">

    </a>
    Clinical Cancer Genomics Course
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Course Details
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../schedule/" class="md-nav__link">
        Schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../lectures/lectures/" class="md-nav__link">
        Lectures
      </a>
    </li>
  

    
      
      
      

  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          Pre-course setup
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Pre-course setup" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Pre-course setup
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../installation/" class="md-nav__link">
        Installation
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" type="checkbox" id="__nav_4_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4_2">
          Download Files
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Download Files" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_2">
          <span class="md-nav__icon md-icon"></span>
          Download Files
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../references/" class="md-nav__link">
        Reference
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../annotations/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Lab Session
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Lab Session" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Lab Session
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_1" type="checkbox" id="__nav_5_1" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_1">
          Day 1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 1" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basic_unix/" class="md-nav__link">
        Basic Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../basics_awk/" class="md-nav__link">
        AWK
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../aws_env/" class="md-nav__link">
        Workspace - AWS
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../R_for_bioinformatics/" class="md-nav__link">
        R for Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../real_life_example_cli/" class="md-nav__link">
        Command Line Usage in Real Life
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_2" type="checkbox" id="__nav_5_2" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5_2">
          Day 2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 2" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Human Genome
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Human Genome
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#split-the-long-fasta-by-chromosome" class="md-nav__link">
    Split the long fasta by chromosome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#index-the-fasta-files" class="md-nav__link">
    Index the fasta files
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../annotation_files/" class="md-nav__link">
        Annotation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_dna/" class="md-nav__link">
        Processing of DNA sequencing data
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../intro_igv/" class="md-nav__link">
        Introduction to IGV
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_3" type="checkbox" id="__nav_5_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_3">
          Day 3
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 3" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling/" class="md-nav__link">
        Variant Calling
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_4" type="checkbox" id="__nav_5_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_4">
          Day 4
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Day 4" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../analysis_of_cnv/" class="md-nav__link">
        Copy Number Variations
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../processing_of_rnaseq/" class="md-nav__link">
        RNAseq
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#split-the-long-fasta-by-chromosome" class="md-nav__link">
    Split the long fasta by chromosome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#index-the-fasta-files" class="md-nav__link">
    Index the fasta files
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                

<h1 id="human-genome-reference-files">Human genome reference files<a class="headerlink" href="#human-genome-reference-files" title="Permanent link">&para;</a></h1>
<p>The human genome is work in progress. Different versions exist (assemblies) and is often a source of confusion in genomics. Quick use of published data is a bit more challenging if a differnt genome build was applied than what is used in your lab. </p>
<p>Also, there may even be differences in the <em>latest</em> versions, e.g. the GRCh37 and the hg19 assemblies from NCBI and USCS, respectively, had different mitochondrial genome. </p>
<p>In genomics, the reference genome offers a scaffold upon which new data can be mapped, which is a much more efficient way rather than building a genome from scratch.</p>
<p>Avery nice overview of the human genome versions can be found <a href="http://genome.ucsc.edu/FAQ/FAQreleases.html">here</a>.</p>
<p>During this course version GRCh38, from the Genome Reference Consortium, with modifications fromt he 1000 genomes consortium will be used. It includes extra decoy and HLA sequences. The files are already available on AWS but can be downloaded from <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/GRCh38_reference_genome/">here</a>.</p>
<p>As several of the analysis steps wold take many hours we will use a smaller version of the reference genome which contains chr6 and chr17. These were choosen because the datat that will be analysed contain interesting varaiants on these chromosomes. 
- Connect to your AWS instance</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Remember, source the settings, paths and variables file</span>
<span class="nb">source</span> .bashrc

<span class="c1"># Check the CHRS variable.</span>
<span class="nb">echo</span> <span class="nv">$CHRS</span>

<span class="c1"># This variable should return &quot;chr6_and_chr17&quot;, if not, reach of for assistance.</span>

<span class="c1"># Go to the directory for the reference genome files</span>
<span class="nb">cd</span> ~/workspace/inputs/references/genome

<span class="c1"># List the files</span>
ls -halt

<span class="c1"># Check what the reference genome looks like. When you have the reference genome open just type &quot;1000&quot; and press enter. You will then jump 1000 lines.</span>
less -SN ref_genome.fa

<span class="c1">#less -SN will be used during the course. Check the man page. Give it some time to understand how it it structured, it is very helpful to be able to read the help-pages fast when working on the command line.</span>
man less

<span class="c1"># Check the chromosome headers in the fasta file.</span>
cat ref_genome.fa <span class="p">|</span> grep -P <span class="s2">&quot;^&gt;&quot;</span>

<span class="c1"># View the first 10 lines of this file. Note the header line starting with `&gt;`. </span>
<span class="c1"># Why does the sequence look like this?</span>
<span class="nb">cd</span> ~/workspace/inputs/references/genome
head -n <span class="m">10</span> ref_genome.fa

<span class="c1"># How many lines and characters are in this file?</span>
wc ref_genome.fa

<span class="c1"># How long are to two chromosomes combined (in bases and Mbp)? Use grep to skip the header lines for each chromosome.</span>
grep -v <span class="s2">&quot;&gt;&quot;</span> ref_genome.fa <span class="p">|</span> wc

<span class="c1"># How long does that command take to run?</span>
<span class="nb">time</span> grep -v <span class="s2">&quot;&gt;&quot;</span> ref_genome.fa <span class="p">|</span> wc

<span class="c1"># View 10 lines from the end of the filefile</span>
tail -n <span class="m">10</span> ref_genome.fa

<span class="c1"># What is the count of each base in the entire reference genome file (skipping the header lines for each sequence)?</span>
<span class="c1"># Runtime: ~30s</span>
cat ref_genome.fa <span class="p">|</span> grep -v <span class="s2">&quot;&gt;&quot;</span> <span class="p">|</span> perl -ne <span class="s1">&#39;chomp $_; $bases{$_}++ for split //; if (eof){print &quot;$_ $bases{$_}\n&quot; for sort keys %bases}&#39;</span>
<span class="c1"># What does each of these bases refer to? What are the &quot;unexpected bases&quot;?</span>
<span class="c1"># Google &quot;IUPAC nucleic acid codes&quot;</span>
</code></pre></div>
<h2 id="split-the-long-fasta-by-chromosome">Split the long fasta by chromosome<a class="headerlink" href="#split-the-long-fasta-by-chromosome" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/inputs/references/genome

<span class="c1"># Split.</span>
faSplit byname ref_genome.fa ./ref_genome_split/

<span class="c1"># View contents.</span>
tree
</code></pre></div>
<h2 id="index-the-fasta-files">Index the fasta files<a class="headerlink" href="#index-the-fasta-files" title="Permanent link">&para;</a></h2>
<p>Indexing is widely used in bioinformatics workflows to improve performance. Typically it is applied to large data files with many records to improve the ability of tools to rapidly access specific locations of the file. For example, if we have alignments against the entire genome and we want to visualize those alignments for a single gene on chr17 in a genome viewer such as IGV, we don’t want the viewer to have to scan through the entire file. Indexing allows us to jump right to the correct place in the file and pull out just the information we need without reading much of the file.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># first remove the .fai and .dict files that were downloaded. Do not remove the .fa file though!</span>
<span class="nb">cd</span> ~/workspace/inputs/references/genome
ls -halt
rm -f ref_genome.fa.fai ref_genome.dict

<span class="c1"># Use samtools to create a fasta index file.</span>
samtools faidx ref_genome.fa

<span class="c1"># View the contents of the index file.</span>
head ref_genome.fa.fai
<span class="c1"># What information does the index fail store?</span>
<span class="c1"># google &quot;fasta index fai format&quot; or something similar</span>

<span class="c1"># Use picard to create a dictionary file.</span>
java -jar <span class="nv">$PICARD</span> CreateSequenceDictionary -R ref_genome.fa -O ref_genome.dict

<span class="c1"># View the content of the dictionary file.</span>
head ref_genome.dict
<span class="c1"># less can also be applied</span>

<span class="c1">#Also index the split chromosomes.</span>
samtools faidx ./ref_genome_split/chr6.fa
samtools faidx ./ref_genome_split/chr17.fa

<span class="c1"># Create reference index for the genome to use BWA</span>
bwa index ref_genome.fa
</code></pre></div>
<h1 id="human-genome-transcriptome-reference-files">Human genome transcriptome reference files<a class="headerlink" href="#human-genome-transcriptome-reference-files" title="Permanent link">&para;</a></h1>
<div class="highlight"><pre><span></span><code><span class="c1"># Make sure CHRS environment variable is set.</span>
<span class="nb">echo</span> <span class="nv">$CHRS</span>
<span class="nb">cd</span> ~/workspace/inputs/references/transcriptome

<span class="c1"># List files in the directory</span>
ls -halt

<span class="c1"># Take a look at the contents of the gtf file. Press &#39;q&#39; to exit the &#39;less&#39; display. </span>
<span class="c1"># When the file is displayed usin &quot;less&quot; type &quot;/&quot; and then a something you want to highlight e.g. &quot;chr6&quot;. This is very useful when searching for specific infomration</span>
less -SN ref_transcriptome.gtf
</code></pre></div>
<h2 id="explore-the-contents-of-the-transcriptome-reference-files">Explore the contents of the transcriptome reference files<a class="headerlink" href="#explore-the-contents-of-the-transcriptome-reference-files" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="c1">#How many chromsomes are represented?</span>
cut -f1 ref_transcriptome.gtf <span class="p">|</span> sort <span class="p">|</span> uniq -c

<span class="c1"># How many unique gene IDs are in the .gtf file?</span>
<span class="c1"># We can use a perl command-line command to find out:</span>
perl -ne <span class="s1">&#39;if ($_ =~ /(gene_id\s\&quot;ENSG\w+\&quot;)/){print &quot;$1\n&quot;}&#39;</span> ref_transcriptome.gtf <span class="p">|</span> sort <span class="p">|</span> uniq <span class="p">|</span> wc -l

<span class="c1"># what are all the feature types listed in the third column of the GTF?</span>
<span class="c1"># how does the following command (3 commands piped together) answer that question?</span>
cut -f <span class="m">3</span> ref_transcriptome.gtf <span class="p">|</span> sort <span class="p">|</span> uniq -c
</code></pre></div>
<p>Below we will prepare files for the bioinformatic tools (HISAT/Kallisto) used to analyse RNA-sequencing data.</p>
<h2 id="create-a-reference-index-for-transcriptome-with-hisat-for-splice-rna-alignments-to-the-genome">Create a reference index for transcriptome with HISAT for splice RNA alignments to the genome<a class="headerlink" href="#create-a-reference-index-for-transcriptome-with-hisat-for-splice-rna-alignments-to-the-genome" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/inputs/references/transcriptome

<span class="c1"># Create a database of observed splice sites represented in our reference transcriptome GTF</span>
~/workspace/bin/hisat2-2.1.0/hisat2_extract_splice_sites.py ref_transcriptome.gtf &gt; splicesites.tsv
head splicesites.tsv

<span class="c1"># Create a database of exon regions in our reference transcriptome GTF</span>
~/workspace/bin/hisat2-2.1.0/hisat2_extract_exons.py ref_transcriptome.gtf &gt; exons.tsv
head exons.tsv

<span class="c1"># build the reference genome index for HISAT and supply the exon and splice site information extracted in the previous steps</span>
<span class="c1"># specify to use 8 threads with the `-p 8` option</span>
<span class="c1"># run time for this index is ~5 minutes</span>
~/workspace/bin/hisat2-2.1.0/hisat2-build -p <span class="m">8</span> --ss splicesites.tsv --exon exons.tsv ~/workspace/inputs/references/genome/ref_genome.fa ref_genome
</code></pre></div>
<h2 id="create-a-reference-transcriptome-index-for-use-with-kallisto">Create a reference transcriptome index for use with Kallisto<a class="headerlink" href="#create-a-reference-transcriptome-index-for-use-with-kallisto" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/workspace/inputs/references/transcriptome
mkdir kallisto
<span class="nb">cd</span> kallisto

<span class="c1"># tidy up the headers to just include the ensembl transcript ids</span>
cat ../ref_transcriptome.fa <span class="p">|</span> perl -ne <span class="s1">&#39;if ($_ =~ /\d+\s+(ENST\d+)/){print &quot;&gt;$1\n&quot;}else{print $_}&#39;</span> &gt; ref_transcriptome_clean.fa

<span class="c1"># run time for this index is ~30 seconds</span>
kallisto index --index<span class="o">=</span>ref_transcriptome_kallisto_index ref_transcriptome_clean.fa
</code></pre></div>

              
            </article>
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" data-md-state="hidden">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../real_life_example_cli/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Command Line Usage in Real Life" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Command Line Usage in Real Life
            </div>
          </div>
        </a>
      
      
        
        <a href="../annotation_files/" class="md-footer__link md-footer__link--next" aria-label="Next: Annotation" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Annotation
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2021, Johan Lindberg et. al., MEB, Karolinska Institutet, All rights reserved.
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "navigation.sections", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest"], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.361d90f1.min.js"}</script>
    
    
      <script src="../../assets/javascripts/bundle.289a2a4b.min.js"></script>
      
    
  </body>
</html>